// akeyless/dist/model/AccountCustomFieldCreate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AccountCustomFieldCreate model module.
 * @module model/AccountCustomFieldCreate
 * @version 5.0.14
 */
var AccountCustomFieldCreate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AccountCustomFieldCreate</code>.
   * accountCustomFieldCreate is a command that creates a new custom field in the account settings
   * @alias module:model/AccountCustomFieldCreate
   * @param name {String} The name of the custom field
   * @param object {String} The object to create the custom field
   * @param objectType {String} The object type to create the custom field [e.g. STATIC_SECRET,DYNAMIC_SECRET,ROTATED_SECRET]
   */
  function AccountCustomFieldCreate(name, object, objectType) {
    _classCallCheck(this, AccountCustomFieldCreate);

    AccountCustomFieldCreate.initialize(this, name, object, objectType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AccountCustomFieldCreate, null, [{
    key: "initialize",
    value: function initialize(obj, name, object, objectType) {
      obj['json'] = false;
      obj['name'] = name;
      obj['object'] = object || 'ITEM';
      obj['object-type'] = objectType;
      obj['required'] = false;
    }
    /**
     * Constructs a <code>AccountCustomFieldCreate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AccountCustomFieldCreate} obj Optional instance to populate.
     * @return {module:model/AccountCustomFieldCreate} The populated <code>AccountCustomFieldCreate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccountCustomFieldCreate();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }

        if (data.hasOwnProperty('object-type')) {
          obj['object-type'] = _ApiClient["default"].convertToType(data['object-type'], 'String');
        }

        if (data.hasOwnProperty('required')) {
          obj['required'] = _ApiClient["default"].convertToType(data['required'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AccountCustomFieldCreate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AccountCustomFieldCreate</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(AccountCustomFieldCreate.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['object'] && !(typeof data['object'] === 'string' || data['object'] instanceof String)) {
        throw new Error("Expected the field `object` to be a primitive type in the JSON string but got " + data['object']);
      } // ensure the json data is a string


      if (data['object-type'] && !(typeof data['object-type'] === 'string' || data['object-type'] instanceof String)) {
        throw new Error("Expected the field `object-type` to be a primitive type in the JSON string but got " + data['object-type']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return AccountCustomFieldCreate;
}();

AccountCustomFieldCreate.RequiredProperties = ["name", "object", "object-type"];
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

AccountCustomFieldCreate.prototype['json'] = false;
/**
 * The name of the custom field
 * @member {String} name
 */

AccountCustomFieldCreate.prototype['name'] = undefined;
/**
 * The object to create the custom field
 * @member {String} object
 * @default 'ITEM'
 */

AccountCustomFieldCreate.prototype['object'] = 'ITEM';
/**
 * The object type to create the custom field [e.g. STATIC_SECRET,DYNAMIC_SECRET,ROTATED_SECRET]
 * @member {String} object-type
 */

AccountCustomFieldCreate.prototype['object-type'] = undefined;
/**
 * Specify whether the custom field is mandatory
 * @member {Boolean} required
 * @default false
 */

AccountCustomFieldCreate.prototype['required'] = false;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AccountCustomFieldCreate.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AccountCustomFieldCreate.prototype['uid-token'] = undefined;
var _default = AccountCustomFieldCreate;
exports["default"] = _default;
