// akeyless/dist/model/AccountCustomFieldList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AccountCustomFieldList model module.
 * @module model/AccountCustomFieldList
 * @version 5.0.14
 */
var AccountCustomFieldList = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AccountCustomFieldList</code>.
   * accountCustomFieldList is a command that retrieves the account custom fields
   * @alias module:model/AccountCustomFieldList
   */
  function AccountCustomFieldList() {
    _classCallCheck(this, AccountCustomFieldList);

    AccountCustomFieldList.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AccountCustomFieldList, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
    }
    /**
     * Constructs a <code>AccountCustomFieldList</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AccountCustomFieldList} obj Optional instance to populate.
     * @return {module:model/AccountCustomFieldList} The populated <code>AccountCustomFieldList</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccountCustomFieldList();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }

        if (data.hasOwnProperty('object-type')) {
          obj['object-type'] = _ApiClient["default"].convertToType(data['object-type'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AccountCustomFieldList</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AccountCustomFieldList</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['object'] && !(typeof data['object'] === 'string' || data['object'] instanceof String)) {
        throw new Error("Expected the field `object` to be a primitive type in the JSON string but got " + data['object']);
      } // ensure the json data is a string


      if (data['object-type'] && !(typeof data['object-type'] === 'string' || data['object-type'] instanceof String)) {
        throw new Error("Expected the field `object-type` to be a primitive type in the JSON string but got " + data['object-type']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return AccountCustomFieldList;
}();
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */


AccountCustomFieldList.prototype['json'] = false;
/**
 * The object to filter by the custom fields in: body
 * @member {String} object
 */

AccountCustomFieldList.prototype['object'] = undefined;
/**
 * The object type to filter by the custom fields in: body
 * @member {String} object-type
 */

AccountCustomFieldList.prototype['object-type'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AccountCustomFieldList.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AccountCustomFieldList.prototype['uid-token'] = undefined;
var _default = AccountCustomFieldList;
exports["default"] = _default;
