// akeyless/dist/model/AccountCustomFieldGetOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _NullTime = _interopRequireDefault(require("./NullTime"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AccountCustomFieldGetOutput model module.
 * @module model/AccountCustomFieldGetOutput
 * @version 5.0.14
 */
var AccountCustomFieldGetOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AccountCustomFieldGetOutput</code>.
   * @alias module:model/AccountCustomFieldGetOutput
   */
  function AccountCustomFieldGetOutput() {
    _classCallCheck(this, AccountCustomFieldGetOutput);

    AccountCustomFieldGetOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AccountCustomFieldGetOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AccountCustomFieldGetOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AccountCustomFieldGetOutput} obj Optional instance to populate.
     * @return {module:model/AccountCustomFieldGetOutput} The populated <code>AccountCustomFieldGetOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AccountCustomFieldGetOutput();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('deletion_date')) {
          obj['deletion_date'] = _NullTime["default"].constructFromObject(data['deletion_date']);
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }

        if (data.hasOwnProperty('object_type')) {
          obj['object_type'] = _ApiClient["default"].convertToType(data['object_type'], 'String');
        }

        if (data.hasOwnProperty('required')) {
          obj['required'] = _ApiClient["default"].convertToType(data['required'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AccountCustomFieldGetOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AccountCustomFieldGetOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['account_id'] && !(typeof data['account_id'] === 'string' || data['account_id'] instanceof String)) {
        throw new Error("Expected the field `account_id` to be a primitive type in the JSON string but got " + data['account_id']);
      } // validate the optional field `deletion_date`


      if (data['deletion_date']) {
        // data not null
        _NullTime["default"].validateJSON(data['deletion_date']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['object'] && !(typeof data['object'] === 'string' || data['object'] instanceof String)) {
        throw new Error("Expected the field `object` to be a primitive type in the JSON string but got " + data['object']);
      } // ensure the json data is a string


      if (data['object_type'] && !(typeof data['object_type'] === 'string' || data['object_type'] instanceof String)) {
        throw new Error("Expected the field `object_type` to be a primitive type in the JSON string but got " + data['object_type']);
      }

      return true;
    }
  }]);

  return AccountCustomFieldGetOutput;
}();
/**
 * @member {String} account_id
 */


AccountCustomFieldGetOutput.prototype['account_id'] = undefined;
/**
 * @member {Date} creation_date
 */

AccountCustomFieldGetOutput.prototype['creation_date'] = undefined;
/**
 * @member {module:model/NullTime} deletion_date
 */

AccountCustomFieldGetOutput.prototype['deletion_date'] = undefined;
/**
 * @member {Number} id
 */

AccountCustomFieldGetOutput.prototype['id'] = undefined;
/**
 * @member {Date} modification_date
 */

AccountCustomFieldGetOutput.prototype['modification_date'] = undefined;
/**
 * @member {String} name
 */

AccountCustomFieldGetOutput.prototype['name'] = undefined;
/**
 * @member {String} object
 */

AccountCustomFieldGetOutput.prototype['object'] = undefined;
/**
 * @member {String} object_type
 */

AccountCustomFieldGetOutput.prototype['object_type'] = undefined;
/**
 * @member {Boolean} required
 */

AccountCustomFieldGetOutput.prototype['required'] = undefined;
var _default = AccountCustomFieldGetOutput;
exports["default"] = _default;
