// akeyless/dist/model/AcmeAccount.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AcmeAccount model module.
 * @module model/AcmeAccount
 * @version 5.0.1
 */
var AcmeAccount = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AcmeAccount</code>.
   * AcmeAccount is copied without the jwk as it seems like it has issues with sdk
   * @alias module:model/AcmeAccount
   */
  function AcmeAccount() {
    _classCallCheck(this, AcmeAccount);

    AcmeAccount.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AcmeAccount, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AcmeAccount</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AcmeAccount} obj Optional instance to populate.
     * @return {module:model/AcmeAccount} The populated <code>AcmeAccount</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AcmeAccount();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('key_digest')) {
          obj['key_digest'] = _ApiClient["default"].convertToType(data['key_digest'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AcmeAccount</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AcmeAccount</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['account_id'] && !(typeof data['account_id'] === 'string' || data['account_id'] instanceof String)) {
        throw new Error("Expected the field `account_id` to be a primitive type in the JSON string but got " + data['account_id']);
      } // ensure the json data is a string


      if (data['key_digest'] && !(typeof data['key_digest'] === 'string' || data['key_digest'] instanceof String)) {
        throw new Error("Expected the field `key_digest` to be a primitive type in the JSON string but got " + data['key_digest']);
      } // ensure the json data is a string


      if (data['status'] && !(typeof data['status'] === 'string' || data['status'] instanceof String)) {
        throw new Error("Expected the field `status` to be a primitive type in the JSON string but got " + data['status']);
      }

      return true;
    }
  }]);

  return AcmeAccount;
}();
/**
 * AccountId is the ACME account id, not Akeyless account id
 * @member {String} account_id
 */


AcmeAccount.prototype['account_id'] = undefined;
/**
 * @member {String} key_digest
 */

AcmeAccount.prototype['key_digest'] = undefined;
/**
 * @member {String} status
 */

AcmeAccount.prototype['status'] = undefined;
var _default = AcmeAccount;
exports["default"] = _default;
