// akeyless/dist/model/AiInsightsConfigPart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AiInsightsConfigPart model module.
 * @module model/AiInsightsConfigPart
 * @version 5.0.14
 */
var AiInsightsConfigPart = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AiInsightsConfigPart</code>.
   * @alias module:model/AiInsightsConfigPart
   */
  function AiInsightsConfigPart() {
    _classCallCheck(this, AiInsightsConfigPart);

    AiInsightsConfigPart.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AiInsightsConfigPart, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AiInsightsConfigPart</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AiInsightsConfigPart} obj Optional instance to populate.
     * @return {module:model/AiInsightsConfigPart} The populated <code>AiInsightsConfigPart</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AiInsightsConfigPart();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'Boolean');
        }

        if (data.hasOwnProperty('model')) {
          obj['model'] = _ApiClient["default"].convertToType(data['model'], 'String');
        }

        if (data.hasOwnProperty('target_id')) {
          obj['target_id'] = _ApiClient["default"].convertToType(data['target_id'], 'Number');
        }

        if (data.hasOwnProperty('target_name')) {
          obj['target_name'] = _ApiClient["default"].convertToType(data['target_name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AiInsightsConfigPart</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AiInsightsConfigPart</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['model'] && !(typeof data['model'] === 'string' || data['model'] instanceof String)) {
        throw new Error("Expected the field `model` to be a primitive type in the JSON string but got " + data['model']);
      } // ensure the json data is a string


      if (data['target_name'] && !(typeof data['target_name'] === 'string' || data['target_name'] instanceof String)) {
        throw new Error("Expected the field `target_name` to be a primitive type in the JSON string but got " + data['target_name']);
      }

      return true;
    }
  }]);

  return AiInsightsConfigPart;
}();
/**
 * @member {Boolean} enable
 */


AiInsightsConfigPart.prototype['enable'] = undefined;
/**
 * @member {String} model
 */

AiInsightsConfigPart.prototype['model'] = undefined;
/**
 * @member {Number} target_id
 */

AiInsightsConfigPart.prototype['target_id'] = undefined;
/**
 * @member {String} target_name
 */

AiInsightsConfigPart.prototype['target_name'] = undefined;
var _default = AiInsightsConfigPart;
exports["default"] = _default;
