// akeyless/dist/model/AkeylessGatewayConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdminsConfigPart = _interopRequireDefault(require("./AdminsConfigPart"));

var _CFConfigPart = _interopRequireDefault(require("./CFConfigPart"));

var _CaCertificatesConfigPart = _interopRequireDefault(require("./CaCertificatesConfigPart"));

var _CacheConfigPart = _interopRequireDefault(require("./CacheConfigPart"));

var _DefaultConfigPart = _interopRequireDefault(require("./DefaultConfigPart"));

var _GatewayMessageQueueInfo = _interopRequireDefault(require("./GatewayMessageQueueInfo"));

var _GeneralConfigPart = _interopRequireDefault(require("./GeneralConfigPart"));

var _K8SAuthsConfigPart = _interopRequireDefault(require("./K8SAuthsConfigPart"));

var _KMIPConfigPart = _interopRequireDefault(require("./KMIPConfigPart"));

var _KerberosConfigPart = _interopRequireDefault(require("./KerberosConfigPart"));

var _LdapConfigPart = _interopRequireDefault(require("./LdapConfigPart"));

var _LeadershipConfigPart = _interopRequireDefault(require("./LeadershipConfigPart"));

var _LogForwardingConfigPart = _interopRequireDefault(require("./LogForwardingConfigPart"));

var _MigrationsConfigPart = _interopRequireDefault(require("./MigrationsConfigPart"));

var _ProducersConfigPart = _interopRequireDefault(require("./ProducersConfigPart"));

var _RotatorsConfigPart = _interopRequireDefault(require("./RotatorsConfigPart"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AkeylessGatewayConfig model module.
 * @module model/AkeylessGatewayConfig
 * @version 5.0.6
 */
var AkeylessGatewayConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AkeylessGatewayConfig</code>.
   * @alias module:model/AkeylessGatewayConfig
   */
  function AkeylessGatewayConfig() {
    _classCallCheck(this, AkeylessGatewayConfig);

    AkeylessGatewayConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AkeylessGatewayConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AkeylessGatewayConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AkeylessGatewayConfig} obj Optional instance to populate.
     * @return {module:model/AkeylessGatewayConfig} The populated <code>AkeylessGatewayConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AkeylessGatewayConfig();

        if (data.hasOwnProperty('admins')) {
          obj['admins'] = _AdminsConfigPart["default"].constructFromObject(data['admins']);
        }

        if (data.hasOwnProperty('ca_certificates')) {
          obj['ca_certificates'] = _CaCertificatesConfigPart["default"].constructFromObject(data['ca_certificates']);
        }

        if (data.hasOwnProperty('cache')) {
          obj['cache'] = _CacheConfigPart["default"].constructFromObject(data['cache']);
        }

        if (data.hasOwnProperty('cf')) {
          obj['cf'] = _CFConfigPart["default"].constructFromObject(data['cf']);
        }

        if (data.hasOwnProperty('config_protection_key_name')) {
          obj['config_protection_key_name'] = _ApiClient["default"].convertToType(data['config_protection_key_name'], 'String');
        }

        if (data.hasOwnProperty('general')) {
          obj['general'] = _GeneralConfigPart["default"].constructFromObject(data['general']);
        }

        if (data.hasOwnProperty('k8s_auths')) {
          obj['k8s_auths'] = _K8SAuthsConfigPart["default"].constructFromObject(data['k8s_auths']);
        }

        if (data.hasOwnProperty('kerberos')) {
          obj['kerberos'] = _KerberosConfigPart["default"].constructFromObject(data['kerberos']);
        }

        if (data.hasOwnProperty('kmip_clients')) {
          obj['kmip_clients'] = _KMIPConfigPart["default"].constructFromObject(data['kmip_clients']);
        }

        if (data.hasOwnProperty('ldap')) {
          obj['ldap'] = _LdapConfigPart["default"].constructFromObject(data['ldap']);
        }

        if (data.hasOwnProperty('leadership')) {
          obj['leadership'] = _LeadershipConfigPart["default"].constructFromObject(data['leadership']);
        }

        if (data.hasOwnProperty('log_forwarding')) {
          obj['log_forwarding'] = _LogForwardingConfigPart["default"].constructFromObject(data['log_forwarding']);
        }

        if (data.hasOwnProperty('message_queue_info')) {
          obj['message_queue_info'] = _GatewayMessageQueueInfo["default"].constructFromObject(data['message_queue_info']);
        }

        if (data.hasOwnProperty('migrations')) {
          obj['migrations'] = _MigrationsConfigPart["default"].constructFromObject(data['migrations']);
        }

        if (data.hasOwnProperty('producers')) {
          obj['producers'] = _ProducersConfigPart["default"].constructFromObject(data['producers']);
        }

        if (data.hasOwnProperty('rotators')) {
          obj['rotators'] = _RotatorsConfigPart["default"].constructFromObject(data['rotators']);
        }

        if (data.hasOwnProperty('saml')) {
          obj['saml'] = _DefaultConfigPart["default"].constructFromObject(data['saml']);
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AkeylessGatewayConfig</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AkeylessGatewayConfig</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `admins`
      if (data['admins']) {
        // data not null
        _AdminsConfigPart["default"].validateJSON(data['admins']);
      } // validate the optional field `ca_certificates`


      if (data['ca_certificates']) {
        // data not null
        _CaCertificatesConfigPart["default"].validateJSON(data['ca_certificates']);
      } // validate the optional field `cache`


      if (data['cache']) {
        // data not null
        _CacheConfigPart["default"].validateJSON(data['cache']);
      } // validate the optional field `cf`


      if (data['cf']) {
        // data not null
        _CFConfigPart["default"].validateJSON(data['cf']);
      } // ensure the json data is a string


      if (data['config_protection_key_name'] && !(typeof data['config_protection_key_name'] === 'string' || data['config_protection_key_name'] instanceof String)) {
        throw new Error("Expected the field `config_protection_key_name` to be a primitive type in the JSON string but got " + data['config_protection_key_name']);
      } // validate the optional field `general`


      if (data['general']) {
        // data not null
        _GeneralConfigPart["default"].validateJSON(data['general']);
      } // validate the optional field `k8s_auths`


      if (data['k8s_auths']) {
        // data not null
        _K8SAuthsConfigPart["default"].validateJSON(data['k8s_auths']);
      } // validate the optional field `kerberos`


      if (data['kerberos']) {
        // data not null
        _KerberosConfigPart["default"].validateJSON(data['kerberos']);
      } // validate the optional field `kmip_clients`


      if (data['kmip_clients']) {
        // data not null
        _KMIPConfigPart["default"].validateJSON(data['kmip_clients']);
      } // validate the optional field `ldap`


      if (data['ldap']) {
        // data not null
        _LdapConfigPart["default"].validateJSON(data['ldap']);
      } // validate the optional field `leadership`


      if (data['leadership']) {
        // data not null
        _LeadershipConfigPart["default"].validateJSON(data['leadership']);
      } // validate the optional field `log_forwarding`


      if (data['log_forwarding']) {
        // data not null
        _LogForwardingConfigPart["default"].validateJSON(data['log_forwarding']);
      } // validate the optional field `message_queue_info`


      if (data['message_queue_info']) {
        // data not null
        _GatewayMessageQueueInfo["default"].validateJSON(data['message_queue_info']);
      } // validate the optional field `migrations`


      if (data['migrations']) {
        // data not null
        _MigrationsConfigPart["default"].validateJSON(data['migrations']);
      } // validate the optional field `producers`


      if (data['producers']) {
        // data not null
        _ProducersConfigPart["default"].validateJSON(data['producers']);
      } // validate the optional field `rotators`


      if (data['rotators']) {
        // data not null
        _RotatorsConfigPart["default"].validateJSON(data['rotators']);
      } // validate the optional field `saml`


      if (data['saml']) {
        // data not null
        _DefaultConfigPart["default"].validateJSON(data['saml']);
      }

      return true;
    }
  }]);

  return AkeylessGatewayConfig;
}();
/**
 * @member {module:model/AdminsConfigPart} admins
 */


AkeylessGatewayConfig.prototype['admins'] = undefined;
/**
 * @member {module:model/CaCertificatesConfigPart} ca_certificates
 */

AkeylessGatewayConfig.prototype['ca_certificates'] = undefined;
/**
 * @member {module:model/CacheConfigPart} cache
 */

AkeylessGatewayConfig.prototype['cache'] = undefined;
/**
 * @member {module:model/CFConfigPart} cf
 */

AkeylessGatewayConfig.prototype['cf'] = undefined;
/**
 * @member {String} config_protection_key_name
 */

AkeylessGatewayConfig.prototype['config_protection_key_name'] = undefined;
/**
 * @member {module:model/GeneralConfigPart} general
 */

AkeylessGatewayConfig.prototype['general'] = undefined;
/**
 * @member {module:model/K8SAuthsConfigPart} k8s_auths
 */

AkeylessGatewayConfig.prototype['k8s_auths'] = undefined;
/**
 * @member {module:model/KerberosConfigPart} kerberos
 */

AkeylessGatewayConfig.prototype['kerberos'] = undefined;
/**
 * @member {module:model/KMIPConfigPart} kmip_clients
 */

AkeylessGatewayConfig.prototype['kmip_clients'] = undefined;
/**
 * @member {module:model/LdapConfigPart} ldap
 */

AkeylessGatewayConfig.prototype['ldap'] = undefined;
/**
 * @member {module:model/LeadershipConfigPart} leadership
 */

AkeylessGatewayConfig.prototype['leadership'] = undefined;
/**
 * @member {module:model/LogForwardingConfigPart} log_forwarding
 */

AkeylessGatewayConfig.prototype['log_forwarding'] = undefined;
/**
 * @member {module:model/GatewayMessageQueueInfo} message_queue_info
 */

AkeylessGatewayConfig.prototype['message_queue_info'] = undefined;
/**
 * @member {module:model/MigrationsConfigPart} migrations
 */

AkeylessGatewayConfig.prototype['migrations'] = undefined;
/**
 * @member {module:model/ProducersConfigPart} producers
 */

AkeylessGatewayConfig.prototype['producers'] = undefined;
/**
 * @member {module:model/RotatorsConfigPart} rotators
 */

AkeylessGatewayConfig.prototype['rotators'] = undefined;
/**
 * @member {module:model/DefaultConfigPart} saml
 */

AkeylessGatewayConfig.prototype['saml'] = undefined;
/**
 * @member {Number} version
 */

AkeylessGatewayConfig.prototype['version'] = undefined;
var _default = AkeylessGatewayConfig;
exports["default"] = _default;
