// akeyless/dist/model/AliasDetailsOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AliasDetailsOutput model module.
 * @module model/AliasDetailsOutput
 * @version 5.0.1
 */
var AliasDetailsOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AliasDetailsOutput</code>.
   * @alias module:model/AliasDetailsOutput
   */
  function AliasDetailsOutput() {
    _classCallCheck(this, AliasDetailsOutput);

    AliasDetailsOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AliasDetailsOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AliasDetailsOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AliasDetailsOutput} obj Optional instance to populate.
     * @return {module:model/AliasDetailsOutput} The populated <code>AliasDetailsOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AliasDetailsOutput();

        if (data.hasOwnProperty('access_type')) {
          obj['access_type'] = _ApiClient["default"].convertToType(data['access_type'], 'String');
        }

        if (data.hasOwnProperty('tenant_tag')) {
          obj['tenant_tag'] = _ApiClient["default"].convertToType(data['tenant_tag'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AliasDetailsOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AliasDetailsOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_type'] && !(typeof data['access_type'] === 'string' || data['access_type'] instanceof String)) {
        throw new Error("Expected the field `access_type` to be a primitive type in the JSON string but got " + data['access_type']);
      } // ensure the json data is a string


      if (data['tenant_tag'] && !(typeof data['tenant_tag'] === 'string' || data['tenant_tag'] instanceof String)) {
        throw new Error("Expected the field `tenant_tag` to be a primitive type in the JSON string but got " + data['tenant_tag']);
      }

      return true;
    }
  }]);

  return AliasDetailsOutput;
}();
/**
 * @member {String} access_type
 */


AliasDetailsOutput.prototype['access_type'] = undefined;
/**
 * @member {String} tenant_tag
 */

AliasDetailsOutput.prototype['tenant_tag'] = undefined;
var _default = AliasDetailsOutput;
exports["default"] = _default;
