// akeyless/dist/model/AllAnalyticsData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateAnalyticAggregation = _interopRequireDefault(require("./CertificateAnalyticAggregation"));

var _ClientsUsageReport = _interopRequireDefault(require("./ClientsUsageReport"));

var _UsageReportSummary = _interopRequireDefault(require("./UsageReportSummary"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AllAnalyticsData model module.
 * @module model/AllAnalyticsData
 * @version 4.2.2
 */
var AllAnalyticsData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllAnalyticsData</code>.
   * @alias module:model/AllAnalyticsData
   */
  function AllAnalyticsData() {
    _classCallCheck(this, AllAnalyticsData);

    AllAnalyticsData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AllAnalyticsData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllAnalyticsData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllAnalyticsData} obj Optional instance to populate.
     * @return {module:model/AllAnalyticsData} The populated <code>AllAnalyticsData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllAnalyticsData();

        if (data.hasOwnProperty('analytics_data')) {
          obj['analytics_data'] = _ApiClient["default"].convertToType(data['analytics_data'], {
            'String': [['String']]
          });
        }

        if (data.hasOwnProperty('certificates_expiry_data')) {
          obj['certificates_expiry_data'] = _CertificateAnalyticAggregation["default"].constructFromObject(data['certificates_expiry_data']);
        }

        if (data.hasOwnProperty('clients_usage_reports')) {
          obj['clients_usage_reports'] = _ApiClient["default"].convertToType(data['clients_usage_reports'], {
            'String': _ClientsUsageReport["default"]
          });
        }

        if (data.hasOwnProperty('date_updated')) {
          obj['date_updated'] = _ApiClient["default"].convertToType(data['date_updated'], 'Number');
        }

        if (data.hasOwnProperty('usage_reports')) {
          obj['usage_reports'] = _ApiClient["default"].convertToType(data['usage_reports'], {
            'String': _UsageReportSummary["default"]
          });
        }
      }

      return obj;
    }
  }]);

  return AllAnalyticsData;
}();
/**
 * @member {Object.<String, Array.<Array.<String>>>} analytics_data
 */


AllAnalyticsData.prototype['analytics_data'] = undefined;
/**
 * @member {module:model/CertificateAnalyticAggregation} certificates_expiry_data
 */

AllAnalyticsData.prototype['certificates_expiry_data'] = undefined;
/**
 * @member {Object.<String, module:model/ClientsUsageReport>} clients_usage_reports
 */

AllAnalyticsData.prototype['clients_usage_reports'] = undefined;
/**
 * @member {Number} date_updated
 */

AllAnalyticsData.prototype['date_updated'] = undefined;
/**
 * @member {Object.<String, module:model/UsageReportSummary>} usage_reports
 */

AllAnalyticsData.prototype['usage_reports'] = undefined;
var _default = AllAnalyticsData;
exports["default"] = _default;
