// akeyless/dist/model/AllowedIpSettings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AllowedIpSettings model module.
 * @module model/AllowedIpSettings
 * @version 5.0.14
 */
var AllowedIpSettings = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllowedIpSettings</code>.
   * @alias module:model/AllowedIpSettings
   */
  function AllowedIpSettings() {
    _classCallCheck(this, AllowedIpSettings);

    AllowedIpSettings.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AllowedIpSettings, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllowedIpSettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllowedIpSettings} obj Optional instance to populate.
     * @return {module:model/AllowedIpSettings} The populated <code>AllowedIpSettings</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllowedIpSettings();

        if (data.hasOwnProperty('cidr_whitelist')) {
          obj['cidr_whitelist'] = _ApiClient["default"].convertToType(data['cidr_whitelist'], 'String');
        }

        if (data.hasOwnProperty('lock')) {
          obj['lock'] = _ApiClient["default"].convertToType(data['lock'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AllowedIpSettings</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AllowedIpSettings</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['cidr_whitelist'] && !(typeof data['cidr_whitelist'] === 'string' || data['cidr_whitelist'] instanceof String)) {
        throw new Error("Expected the field `cidr_whitelist` to be a primitive type in the JSON string but got " + data['cidr_whitelist']);
      }

      return true;
    }
  }]);

  return AllowedIpSettings;
}();
/**
 * @member {String} cidr_whitelist
 */


AllowedIpSettings.prototype['cidr_whitelist'] = undefined;
/**
 * @member {Boolean} lock
 */

AllowedIpSettings.prototype['lock'] = undefined;
var _default = AllowedIpSettings;
exports["default"] = _default;
