// akeyless/dist/model/AllowedClientType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AllowedClientType model module.
 * @module model/AllowedClientType
 * @version 5.0.20
 */
var AllowedClientType = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AllowedClientType</code>.
   * @alias module:model/AllowedClientType
   */
  function AllowedClientType() {
    _classCallCheck(this, AllowedClientType);

    AllowedClientType.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AllowedClientType, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AllowedClientType</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AllowedClientType} obj Optional instance to populate.
     * @return {module:model/AllowedClientType} The populated <code>AllowedClientType</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AllowedClientType();

        if (data.hasOwnProperty('client_types')) {
          obj['client_types'] = _ApiClient["default"].convertToType(data['client_types'], ['String']);
        }

        if (data.hasOwnProperty('lock')) {
          obj['lock'] = _ApiClient["default"].convertToType(data['lock'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AllowedClientType</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AllowedClientType</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['client_types'])) {
        throw new Error("Expected the field `client_types` to be an array in the JSON data but got " + data['client_types']);
      }

      return true;
    }
  }]);

  return AllowedClientType;
}();
/**
 * @member {Array.<String>} client_types
 */


AllowedClientType.prototype['client_types'] = undefined;
/**
 * @member {Boolean} lock
 */

AllowedClientType.prototype['lock'] = undefined;
var _default = AllowedClientType;
exports["default"] = _default;
