// akeyless/dist/model/ArtifactoryTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArtifactoryTargetDetails model module.
 * @module model/ArtifactoryTargetDetails
 * @version 5.0.14
 */
var ArtifactoryTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ArtifactoryTargetDetails</code>.
   * @alias module:model/ArtifactoryTargetDetails
   */
  function ArtifactoryTargetDetails() {
    _classCallCheck(this, ArtifactoryTargetDetails);

    ArtifactoryTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArtifactoryTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ArtifactoryTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArtifactoryTargetDetails} obj Optional instance to populate.
     * @return {module:model/ArtifactoryTargetDetails} The populated <code>ArtifactoryTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArtifactoryTargetDetails();

        if (data.hasOwnProperty('artifactory_admin_apikey')) {
          obj['artifactory_admin_apikey'] = _ApiClient["default"].convertToType(data['artifactory_admin_apikey'], 'String');
        }

        if (data.hasOwnProperty('artifactory_admin_username')) {
          obj['artifactory_admin_username'] = _ApiClient["default"].convertToType(data['artifactory_admin_username'], 'String');
        }

        if (data.hasOwnProperty('artifactory_base_url')) {
          obj['artifactory_base_url'] = _ApiClient["default"].convertToType(data['artifactory_base_url'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ArtifactoryTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ArtifactoryTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['artifactory_admin_apikey'] && !(typeof data['artifactory_admin_apikey'] === 'string' || data['artifactory_admin_apikey'] instanceof String)) {
        throw new Error("Expected the field `artifactory_admin_apikey` to be a primitive type in the JSON string but got " + data['artifactory_admin_apikey']);
      } // ensure the json data is a string


      if (data['artifactory_admin_username'] && !(typeof data['artifactory_admin_username'] === 'string' || data['artifactory_admin_username'] instanceof String)) {
        throw new Error("Expected the field `artifactory_admin_username` to be a primitive type in the JSON string but got " + data['artifactory_admin_username']);
      } // ensure the json data is a string


      if (data['artifactory_base_url'] && !(typeof data['artifactory_base_url'] === 'string' || data['artifactory_base_url'] instanceof String)) {
        throw new Error("Expected the field `artifactory_base_url` to be a primitive type in the JSON string but got " + data['artifactory_base_url']);
      }

      return true;
    }
  }]);

  return ArtifactoryTargetDetails;
}();
/**
 * @member {String} artifactory_admin_apikey
 */


ArtifactoryTargetDetails.prototype['artifactory_admin_apikey'] = undefined;
/**
 * @member {String} artifactory_admin_username
 */

ArtifactoryTargetDetails.prototype['artifactory_admin_username'] = undefined;
/**
 * @member {String} artifactory_base_url
 */

ArtifactoryTargetDetails.prototype['artifactory_base_url'] = undefined;
var _default = ArtifactoryTargetDetails;
exports["default"] = _default;
