// akeyless/dist/model/AssocRoleAuthMethod.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AssocRoleAuthMethod model module.
 * @module model/AssocRoleAuthMethod
 * @version 5.0.8
 */
var AssocRoleAuthMethod = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AssocRoleAuthMethod</code>.
   * assocRoleAuthMethod is a command that creates an association between role and auth method.
   * @alias module:model/AssocRoleAuthMethod
   * @param amName {String} The auth method to associate
   * @param roleName {String} The role to associate
   */
  function AssocRoleAuthMethod(amName, roleName) {
    _classCallCheck(this, AssocRoleAuthMethod);

    AssocRoleAuthMethod.initialize(this, amName, roleName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AssocRoleAuthMethod, null, [{
    key: "initialize",
    value: function initialize(obj, amName, roleName) {
      obj['am-name'] = amName;
      obj['case-sensitive'] = 'true';
      obj['json'] = false;
      obj['role-name'] = roleName;
    }
    /**
     * Constructs a <code>AssocRoleAuthMethod</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AssocRoleAuthMethod} obj Optional instance to populate.
     * @return {module:model/AssocRoleAuthMethod} The populated <code>AssocRoleAuthMethod</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AssocRoleAuthMethod();

        if (data.hasOwnProperty('am-name')) {
          obj['am-name'] = _ApiClient["default"].convertToType(data['am-name'], 'String');
        }

        if (data.hasOwnProperty('case-sensitive')) {
          obj['case-sensitive'] = _ApiClient["default"].convertToType(data['case-sensitive'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('role-name')) {
          obj['role-name'] = _ApiClient["default"].convertToType(data['role-name'], 'String');
        }

        if (data.hasOwnProperty('sub-claims')) {
          obj['sub-claims'] = _ApiClient["default"].convertToType(data['sub-claims'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AssocRoleAuthMethod</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AssocRoleAuthMethod</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(AssocRoleAuthMethod.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['am-name'] && !(typeof data['am-name'] === 'string' || data['am-name'] instanceof String)) {
        throw new Error("Expected the field `am-name` to be a primitive type in the JSON string but got " + data['am-name']);
      } // ensure the json data is a string


      if (data['case-sensitive'] && !(typeof data['case-sensitive'] === 'string' || data['case-sensitive'] instanceof String)) {
        throw new Error("Expected the field `case-sensitive` to be a primitive type in the JSON string but got " + data['case-sensitive']);
      } // ensure the json data is a string


      if (data['role-name'] && !(typeof data['role-name'] === 'string' || data['role-name'] instanceof String)) {
        throw new Error("Expected the field `role-name` to be a primitive type in the JSON string but got " + data['role-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return AssocRoleAuthMethod;
}();

AssocRoleAuthMethod.RequiredProperties = ["am-name", "role-name"];
/**
 * The auth method to associate
 * @member {String} am-name
 */

AssocRoleAuthMethod.prototype['am-name'] = undefined;
/**
 * Treat sub claims as case-sensitive [true/false]
 * @member {String} case-sensitive
 * @default 'true'
 */

AssocRoleAuthMethod.prototype['case-sensitive'] = 'true';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

AssocRoleAuthMethod.prototype['json'] = false;
/**
 * The role to associate
 * @member {String} role-name
 */

AssocRoleAuthMethod.prototype['role-name'] = undefined;
/**
 * key/val of sub claims, e.g group=admins,developers
 * @member {Object.<String, String>} sub-claims
 */

AssocRoleAuthMethod.prototype['sub-claims'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AssocRoleAuthMethod.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AssocRoleAuthMethod.prototype['uid-token'] = undefined;
var _default = AssocRoleAuthMethod;
exports["default"] = _default;
