// akeyless/dist/model/AttributeTypeAndValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AttributeTypeAndValue model module.
 * @module model/AttributeTypeAndValue
 * @version 5.0.0
 */
var AttributeTypeAndValue = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AttributeTypeAndValue</code>.
   * AttributeTypeAndValue mirrors the ASN.1 structure of the same name in RFC 5280, Section 4.1.2.4.
   * @alias module:model/AttributeTypeAndValue
   */
  function AttributeTypeAndValue() {
    _classCallCheck(this, AttributeTypeAndValue);

    AttributeTypeAndValue.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AttributeTypeAndValue, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AttributeTypeAndValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AttributeTypeAndValue} obj Optional instance to populate.
     * @return {module:model/AttributeTypeAndValue} The populated <code>AttributeTypeAndValue</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AttributeTypeAndValue();

        if (data.hasOwnProperty('Type')) {
          obj['Type'] = _ApiClient["default"].convertToType(data['Type'], ['Number']);
        }

        if (data.hasOwnProperty('Value')) {
          obj['Value'] = _ApiClient["default"].convertToType(data['Value'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AttributeTypeAndValue</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AttributeTypeAndValue</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['Type'])) {
        throw new Error("Expected the field `Type` to be an array in the JSON data but got " + data['Type']);
      } // ensure the json data is a string


      if (data['Value'] && !(typeof data['Value'] === 'string' || data['Value'] instanceof String)) {
        throw new Error("Expected the field `Value` to be a primitive type in the JSON string but got " + data['Value']);
      }

      return true;
    }
  }]);

  return AttributeTypeAndValue;
}();
/**
 * @member {Array.<Number>} Type
 */


AttributeTypeAndValue.prototype['Type'] = undefined;
/**
 * @member {String} Value
 */

AttributeTypeAndValue.prototype['Value'] = undefined;
var _default = AttributeTypeAndValue;
exports["default"] = _default;
