// akeyless/dist/model/AuthMethodUpdateCert.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethodUpdateCert model module.
 * @module model/AuthMethodUpdateCert
 * @version 5.0.5
 */
var AuthMethodUpdateCert = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethodUpdateCert</code>.
   * authMethodUpdateCert is a command that updates a new auth method that will be able to authenticate using a client certificate
   * @alias module:model/AuthMethodUpdateCert
   * @param name {String} Auth Method name
   * @param uniqueIdentifier {String} A unique identifier (ID) value should be configured, such as common_name or organizational_unit Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
   */
  function AuthMethodUpdateCert(name, uniqueIdentifier) {
    _classCallCheck(this, AuthMethodUpdateCert);

    AuthMethodUpdateCert.initialize(this, name, uniqueIdentifier);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethodUpdateCert, null, [{
    key: "initialize",
    value: function initialize(obj, name, uniqueIdentifier) {
      obj['access-expires'] = 0;
      obj['json'] = false;
      obj['jwt-ttl'] = 0;
      obj['name'] = name;
      obj['unique-identifier'] = uniqueIdentifier;
    }
    /**
     * Constructs a <code>AuthMethodUpdateCert</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethodUpdateCert} obj Optional instance to populate.
     * @return {module:model/AuthMethodUpdateCert} The populated <code>AuthMethodUpdateCert</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethodUpdateCert();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('allowed-cors')) {
          obj['allowed-cors'] = _ApiClient["default"].convertToType(data['allowed-cors'], 'String');
        }

        if (data.hasOwnProperty('audit-logs-claims')) {
          obj['audit-logs-claims'] = _ApiClient["default"].convertToType(data['audit-logs-claims'], ['String']);
        }

        if (data.hasOwnProperty('bound-common-names')) {
          obj['bound-common-names'] = _ApiClient["default"].convertToType(data['bound-common-names'], ['String']);
        }

        if (data.hasOwnProperty('bound-dns-sans')) {
          obj['bound-dns-sans'] = _ApiClient["default"].convertToType(data['bound-dns-sans'], ['String']);
        }

        if (data.hasOwnProperty('bound-email-sans')) {
          obj['bound-email-sans'] = _ApiClient["default"].convertToType(data['bound-email-sans'], ['String']);
        }

        if (data.hasOwnProperty('bound-extensions')) {
          obj['bound-extensions'] = _ApiClient["default"].convertToType(data['bound-extensions'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('bound-organizational-units')) {
          obj['bound-organizational-units'] = _ApiClient["default"].convertToType(data['bound-organizational-units'], ['String']);
        }

        if (data.hasOwnProperty('bound-uri-sans')) {
          obj['bound-uri-sans'] = _ApiClient["default"].convertToType(data['bound-uri-sans'], ['String']);
        }

        if (data.hasOwnProperty('certificate-data')) {
          obj['certificate-data'] = _ApiClient["default"].convertToType(data['certificate-data'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('expiration-event-in')) {
          obj['expiration-event-in'] = _ApiClient["default"].convertToType(data['expiration-event-in'], ['String']);
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('product-type')) {
          obj['product-type'] = _ApiClient["default"].convertToType(data['product-type'], ['String']);
        }

        if (data.hasOwnProperty('revoked-cert-ids')) {
          obj['revoked-cert-ids'] = _ApiClient["default"].convertToType(data['revoked-cert-ids'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('unique-identifier')) {
          obj['unique-identifier'] = _ApiClient["default"].convertToType(data['unique-identifier'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AuthMethodUpdateCert</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AuthMethodUpdateCert</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(AuthMethodUpdateCert.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['allowed-cors'] && !(typeof data['allowed-cors'] === 'string' || data['allowed-cors'] instanceof String)) {
        throw new Error("Expected the field `allowed-cors` to be a primitive type in the JSON string but got " + data['allowed-cors']);
      } // ensure the json data is an array


      if (!Array.isArray(data['audit-logs-claims'])) {
        throw new Error("Expected the field `audit-logs-claims` to be an array in the JSON data but got " + data['audit-logs-claims']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-common-names'])) {
        throw new Error("Expected the field `bound-common-names` to be an array in the JSON data but got " + data['bound-common-names']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-dns-sans'])) {
        throw new Error("Expected the field `bound-dns-sans` to be an array in the JSON data but got " + data['bound-dns-sans']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-email-sans'])) {
        throw new Error("Expected the field `bound-email-sans` to be an array in the JSON data but got " + data['bound-email-sans']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-extensions'])) {
        throw new Error("Expected the field `bound-extensions` to be an array in the JSON data but got " + data['bound-extensions']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-ips'])) {
        throw new Error("Expected the field `bound-ips` to be an array in the JSON data but got " + data['bound-ips']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-organizational-units'])) {
        throw new Error("Expected the field `bound-organizational-units` to be an array in the JSON data but got " + data['bound-organizational-units']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-uri-sans'])) {
        throw new Error("Expected the field `bound-uri-sans` to be an array in the JSON data but got " + data['bound-uri-sans']);
      } // ensure the json data is a string


      if (data['certificate-data'] && !(typeof data['certificate-data'] === 'string' || data['certificate-data'] instanceof String)) {
        throw new Error("Expected the field `certificate-data` to be a primitive type in the JSON string but got " + data['certificate-data']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is an array


      if (!Array.isArray(data['expiration-event-in'])) {
        throw new Error("Expected the field `expiration-event-in` to be an array in the JSON data but got " + data['expiration-event-in']);
      } // ensure the json data is an array


      if (!Array.isArray(data['gw-bound-ips'])) {
        throw new Error("Expected the field `gw-bound-ips` to be an array in the JSON data but got " + data['gw-bound-ips']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is an array


      if (!Array.isArray(data['product-type'])) {
        throw new Error("Expected the field `product-type` to be an array in the JSON data but got " + data['product-type']);
      } // ensure the json data is an array


      if (!Array.isArray(data['revoked-cert-ids'])) {
        throw new Error("Expected the field `revoked-cert-ids` to be an array in the JSON data but got " + data['revoked-cert-ids']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['unique-identifier'] && !(typeof data['unique-identifier'] === 'string' || data['unique-identifier'] instanceof String)) {
        throw new Error("Expected the field `unique-identifier` to be a primitive type in the JSON string but got " + data['unique-identifier']);
      }

      return true;
    }
  }]);

  return AuthMethodUpdateCert;
}();

AuthMethodUpdateCert.RequiredProperties = ["name", "unique-identifier"];
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */

AuthMethodUpdateCert.prototype['access-expires'] = 0;
/**
 * Comma separated list of allowed CORS domains to be validated as part of the authentication flow.
 * @member {String} allowed-cors
 */

AuthMethodUpdateCert.prototype['allowed-cors'] = undefined;
/**
 * Subclaims to include in audit logs, e.g \"--audit-logs-claims email --audit-logs-claims username\"
 * @member {Array.<String>} audit-logs-claims
 */

AuthMethodUpdateCert.prototype['audit-logs-claims'] = undefined;
/**
 * A list of names. At least one must exist in the Common Name. Supports globbing.
 * @member {Array.<String>} bound-common-names
 */

AuthMethodUpdateCert.prototype['bound-common-names'] = undefined;
/**
 * A list of DNS names. At least one must exist in the SANs. Supports globbing.
 * @member {Array.<String>} bound-dns-sans
 */

AuthMethodUpdateCert.prototype['bound-dns-sans'] = undefined;
/**
 * A list of Email Addresses. At least one must exist in the SANs. Supports globbing.
 * @member {Array.<String>} bound-email-sans
 */

AuthMethodUpdateCert.prototype['bound-email-sans'] = undefined;
/**
 * A list of extensions formatted as \"oid:value\". Expects the extension value to be some type of ASN1 encoded string. All values much match. Supports globbing on \"value\".
 * @member {Array.<String>} bound-extensions
 */

AuthMethodUpdateCert.prototype['bound-extensions'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

AuthMethodUpdateCert.prototype['bound-ips'] = undefined;
/**
 * A list of Organizational Units names. At least one must exist in the OU field.
 * @member {Array.<String>} bound-organizational-units
 */

AuthMethodUpdateCert.prototype['bound-organizational-units'] = undefined;
/**
 * A list of URIs. At least one must exist in the SANs. Supports globbing.
 * @member {Array.<String>} bound-uri-sans
 */

AuthMethodUpdateCert.prototype['bound-uri-sans'] = undefined;
/**
 * The certificate data in base64, if no file was provided
 * @member {String} certificate-data
 */

AuthMethodUpdateCert.prototype['certificate-data'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

AuthMethodUpdateCert.prototype['delete_protection'] = undefined;
/**
 * Auth Method description
 * @member {String} description
 */

AuthMethodUpdateCert.prototype['description'] = undefined;
/**
 * How many days before the expiration of the auth method would you like to be notified.
 * @member {Array.<String>} expiration-event-in
 */

AuthMethodUpdateCert.prototype['expiration-event-in'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

AuthMethodUpdateCert.prototype['force-sub-claims'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

AuthMethodUpdateCert.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

AuthMethodUpdateCert.prototype['json'] = false;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 * @default 0
 */

AuthMethodUpdateCert.prototype['jwt-ttl'] = 0;
/**
 * Auth Method name
 * @member {String} name
 */

AuthMethodUpdateCert.prototype['name'] = undefined;
/**
 * Auth Method new name
 * @member {String} new-name
 */

AuthMethodUpdateCert.prototype['new-name'] = undefined;
/**
 * Choose the relevant product type for the auth method [sm, sra, pm, dp, ca]
 * @member {Array.<String>} product-type
 */

AuthMethodUpdateCert.prototype['product-type'] = undefined;
/**
 * A list of revoked cert ids
 * @member {Array.<String>} revoked-cert-ids
 */

AuthMethodUpdateCert.prototype['revoked-cert-ids'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AuthMethodUpdateCert.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AuthMethodUpdateCert.prototype['uid-token'] = undefined;
/**
 * A unique identifier (ID) value should be configured, such as common_name or organizational_unit Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
 * @member {String} unique-identifier
 */

AuthMethodUpdateCert.prototype['unique-identifier'] = undefined;
var _default = AuthMethodUpdateCert;
exports["default"] = _default;
