// akeyless/dist/model/AuthMethodCreateAwsIam.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethodCreateAwsIam model module.
 * @module model/AuthMethodCreateAwsIam
 * @version 5.0.6
 */
var AuthMethodCreateAwsIam = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethodCreateAwsIam</code>.
   * authMethodCreateAwsIam is a command that creates a new Auth Method that will be able to authenticate using AWS IAM credentials.
   * @alias module:model/AuthMethodCreateAwsIam
   * @param boundAwsAccountId {Array.<String>} A list of AWS account-IDs that the access is restricted to
   * @param name {String} Auth Method name
   */
  function AuthMethodCreateAwsIam(boundAwsAccountId, name) {
    _classCallCheck(this, AuthMethodCreateAwsIam);

    AuthMethodCreateAwsIam.initialize(this, boundAwsAccountId, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethodCreateAwsIam, null, [{
    key: "initialize",
    value: function initialize(obj, boundAwsAccountId, name) {
      obj['access-expires'] = 0;
      obj['bound-aws-account-id'] = boundAwsAccountId;
      obj['json'] = false;
      obj['jwt-ttl'] = 0;
      obj['name'] = name;
      obj['sts-url'] = 'https://sts.amazonaws.com';
    }
    /**
     * Constructs a <code>AuthMethodCreateAwsIam</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethodCreateAwsIam} obj Optional instance to populate.
     * @return {module:model/AuthMethodCreateAwsIam} The populated <code>AuthMethodCreateAwsIam</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethodCreateAwsIam();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('audit-logs-claims')) {
          obj['audit-logs-claims'] = _ApiClient["default"].convertToType(data['audit-logs-claims'], ['String']);
        }

        if (data.hasOwnProperty('bound-arn')) {
          obj['bound-arn'] = _ApiClient["default"].convertToType(data['bound-arn'], ['String']);
        }

        if (data.hasOwnProperty('bound-aws-account-id')) {
          obj['bound-aws-account-id'] = _ApiClient["default"].convertToType(data['bound-aws-account-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('bound-resource-id')) {
          obj['bound-resource-id'] = _ApiClient["default"].convertToType(data['bound-resource-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-role-id')) {
          obj['bound-role-id'] = _ApiClient["default"].convertToType(data['bound-role-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-role-name')) {
          obj['bound-role-name'] = _ApiClient["default"].convertToType(data['bound-role-name'], ['String']);
        }

        if (data.hasOwnProperty('bound-user-id')) {
          obj['bound-user-id'] = _ApiClient["default"].convertToType(data['bound-user-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-user-name')) {
          obj['bound-user-name'] = _ApiClient["default"].convertToType(data['bound-user-name'], ['String']);
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('expiration-event-in')) {
          obj['expiration-event-in'] = _ApiClient["default"].convertToType(data['expiration-event-in'], ['String']);
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('product-type')) {
          obj['product-type'] = _ApiClient["default"].convertToType(data['product-type'], ['String']);
        }

        if (data.hasOwnProperty('sts-url')) {
          obj['sts-url'] = _ApiClient["default"].convertToType(data['sts-url'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('unique-identifier')) {
          obj['unique-identifier'] = _ApiClient["default"].convertToType(data['unique-identifier'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AuthMethodCreateAwsIam</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AuthMethodCreateAwsIam</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(AuthMethodCreateAwsIam.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is an array

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (!Array.isArray(data['audit-logs-claims'])) {
        throw new Error("Expected the field `audit-logs-claims` to be an array in the JSON data but got " + data['audit-logs-claims']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-arn'])) {
        throw new Error("Expected the field `bound-arn` to be an array in the JSON data but got " + data['bound-arn']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-aws-account-id'])) {
        throw new Error("Expected the field `bound-aws-account-id` to be an array in the JSON data but got " + data['bound-aws-account-id']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-ips'])) {
        throw new Error("Expected the field `bound-ips` to be an array in the JSON data but got " + data['bound-ips']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-resource-id'])) {
        throw new Error("Expected the field `bound-resource-id` to be an array in the JSON data but got " + data['bound-resource-id']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-role-id'])) {
        throw new Error("Expected the field `bound-role-id` to be an array in the JSON data but got " + data['bound-role-id']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-role-name'])) {
        throw new Error("Expected the field `bound-role-name` to be an array in the JSON data but got " + data['bound-role-name']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-user-id'])) {
        throw new Error("Expected the field `bound-user-id` to be an array in the JSON data but got " + data['bound-user-id']);
      } // ensure the json data is an array


      if (!Array.isArray(data['bound-user-name'])) {
        throw new Error("Expected the field `bound-user-name` to be an array in the JSON data but got " + data['bound-user-name']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is an array


      if (!Array.isArray(data['expiration-event-in'])) {
        throw new Error("Expected the field `expiration-event-in` to be an array in the JSON data but got " + data['expiration-event-in']);
      } // ensure the json data is an array


      if (!Array.isArray(data['gw-bound-ips'])) {
        throw new Error("Expected the field `gw-bound-ips` to be an array in the JSON data but got " + data['gw-bound-ips']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is an array


      if (!Array.isArray(data['product-type'])) {
        throw new Error("Expected the field `product-type` to be an array in the JSON data but got " + data['product-type']);
      } // ensure the json data is a string


      if (data['sts-url'] && !(typeof data['sts-url'] === 'string' || data['sts-url'] instanceof String)) {
        throw new Error("Expected the field `sts-url` to be a primitive type in the JSON string but got " + data['sts-url']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['unique-identifier'] && !(typeof data['unique-identifier'] === 'string' || data['unique-identifier'] instanceof String)) {
        throw new Error("Expected the field `unique-identifier` to be a primitive type in the JSON string but got " + data['unique-identifier']);
      }

      return true;
    }
  }]);

  return AuthMethodCreateAwsIam;
}();

AuthMethodCreateAwsIam.RequiredProperties = ["bound-aws-account-id", "name"];
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */

AuthMethodCreateAwsIam.prototype['access-expires'] = 0;
/**
 * Subclaims to include in audit logs, e.g \"--audit-logs-claims email --audit-logs-claims username\"
 * @member {Array.<String>} audit-logs-claims
 */

AuthMethodCreateAwsIam.prototype['audit-logs-claims'] = undefined;
/**
 * A list of full arns that the access is restricted to
 * @member {Array.<String>} bound-arn
 */

AuthMethodCreateAwsIam.prototype['bound-arn'] = undefined;
/**
 * A list of AWS account-IDs that the access is restricted to
 * @member {Array.<String>} bound-aws-account-id
 */

AuthMethodCreateAwsIam.prototype['bound-aws-account-id'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

AuthMethodCreateAwsIam.prototype['bound-ips'] = undefined;
/**
 * A list of full resource ids that the access is restricted to
 * @member {Array.<String>} bound-resource-id
 */

AuthMethodCreateAwsIam.prototype['bound-resource-id'] = undefined;
/**
 * A list of full role ids that the access is restricted to
 * @member {Array.<String>} bound-role-id
 */

AuthMethodCreateAwsIam.prototype['bound-role-id'] = undefined;
/**
 * A list of full role-name that the access is restricted to
 * @member {Array.<String>} bound-role-name
 */

AuthMethodCreateAwsIam.prototype['bound-role-name'] = undefined;
/**
 * A list of full user ids that the access is restricted to
 * @member {Array.<String>} bound-user-id
 */

AuthMethodCreateAwsIam.prototype['bound-user-id'] = undefined;
/**
 * A list of full user-name that the access is restricted to
 * @member {Array.<String>} bound-user-name
 */

AuthMethodCreateAwsIam.prototype['bound-user-name'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

AuthMethodCreateAwsIam.prototype['delete_protection'] = undefined;
/**
 * Auth Method description
 * @member {String} description
 */

AuthMethodCreateAwsIam.prototype['description'] = undefined;
/**
 * How many days before the expiration of the auth method would you like to be notified.
 * @member {Array.<String>} expiration-event-in
 */

AuthMethodCreateAwsIam.prototype['expiration-event-in'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

AuthMethodCreateAwsIam.prototype['force-sub-claims'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

AuthMethodCreateAwsIam.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

AuthMethodCreateAwsIam.prototype['json'] = false;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 * @default 0
 */

AuthMethodCreateAwsIam.prototype['jwt-ttl'] = 0;
/**
 * Auth Method name
 * @member {String} name
 */

AuthMethodCreateAwsIam.prototype['name'] = undefined;
/**
 * Choose the relevant product type for the auth method [sm, sra, pm, dp, ca]
 * @member {Array.<String>} product-type
 */

AuthMethodCreateAwsIam.prototype['product-type'] = undefined;
/**
 * sts URL
 * @member {String} sts-url
 * @default 'https://sts.amazonaws.com'
 */

AuthMethodCreateAwsIam.prototype['sts-url'] = 'https://sts.amazonaws.com';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AuthMethodCreateAwsIam.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AuthMethodCreateAwsIam.prototype['uid-token'] = undefined;
/**
 * A unique identifier (ID) value which is a \"sub claim\" name that contains details uniquely identifying that resource. This \"sub claim\" is used to distinguish between different identities.
 * @member {String} unique-identifier
 */

AuthMethodCreateAwsIam.prototype['unique-identifier'] = undefined;
var _default = AuthMethodCreateAwsIam;
exports["default"] = _default;
