// akeyless/dist/model/AuthMethodCreateGcp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethodCreateGcp model module.
 * @module model/AuthMethodCreateGcp
 * @version 4.2.2
 */
var AuthMethodCreateGcp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethodCreateGcp</code>.
   * authMethodCreateGcp is a command that creates a new auth method that will be able to authenticate using GCP IAM Service Account credentials or GCE instance credentials.
   * @alias module:model/AuthMethodCreateGcp
   * @param audience {String} The audience to verify in the JWT received by the client
   * @param name {String} Auth Method name
   * @param type {String} Type of the GCP Access Rules
   */
  function AuthMethodCreateGcp(audience, name, type) {
    _classCallCheck(this, AuthMethodCreateGcp);

    AuthMethodCreateGcp.initialize(this, audience, name, type);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethodCreateGcp, null, [{
    key: "initialize",
    value: function initialize(obj, audience, name, type) {
      obj['audience'] = audience;
      obj['name'] = name;
      obj['type'] = type;
    }
    /**
     * Constructs a <code>AuthMethodCreateGcp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethodCreateGcp} obj Optional instance to populate.
     * @return {module:model/AuthMethodCreateGcp} The populated <code>AuthMethodCreateGcp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethodCreateGcp();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('audit-logs-claims')) {
          obj['audit-logs-claims'] = _ApiClient["default"].convertToType(data['audit-logs-claims'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('bound-labels')) {
          obj['bound-labels'] = _ApiClient["default"].convertToType(data['bound-labels'], ['String']);
        }

        if (data.hasOwnProperty('bound-projects')) {
          obj['bound-projects'] = _ApiClient["default"].convertToType(data['bound-projects'], ['String']);
        }

        if (data.hasOwnProperty('bound-regions')) {
          obj['bound-regions'] = _ApiClient["default"].convertToType(data['bound-regions'], ['String']);
        }

        if (data.hasOwnProperty('bound-service-accounts')) {
          obj['bound-service-accounts'] = _ApiClient["default"].convertToType(data['bound-service-accounts'], ['String']);
        }

        if (data.hasOwnProperty('bound-zones')) {
          obj['bound-zones'] = _ApiClient["default"].convertToType(data['bound-zones'], ['String']);
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('product-type')) {
          obj['product-type'] = _ApiClient["default"].convertToType(data['product-type'], ['String']);
        }

        if (data.hasOwnProperty('service-account-creds-data')) {
          obj['service-account-creds-data'] = _ApiClient["default"].convertToType(data['service-account-creds-data'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AuthMethodCreateGcp;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


AuthMethodCreateGcp.prototype['access-expires'] = 0;
/**
 * The audience to verify in the JWT received by the client
 * @member {String} audience
 * @default 'akeyless.io'
 */

AuthMethodCreateGcp.prototype['audience'] = 'akeyless.io';
/**
 * Subclaims to include in audit logs, e.g \"--audit-logs-claims email --audit-logs-claims username\"
 * @member {Array.<String>} audit-logs-claims
 */

AuthMethodCreateGcp.prototype['audit-logs-claims'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

AuthMethodCreateGcp.prototype['bound-ips'] = undefined;
/**
 * A comma-separated list of GCP labels formatted as \"key:value\" strings that must be set on authorized GCE instances. TODO: Because GCP labels are not currently ACL'd ....
 * @member {Array.<String>} bound-labels
 */

AuthMethodCreateGcp.prototype['bound-labels'] = undefined;
/**
 * === Human and Machine authentication section === Array of GCP project IDs. Only entities belonging to any of the provided projects can authenticate.
 * @member {Array.<String>} bound-projects
 */

AuthMethodCreateGcp.prototype['bound-projects'] = undefined;
/**
 * List of regions that a GCE instance must belong to in order to be authenticated. TODO: If bound_instance_groups is provided, it is assumed to be a regional group and the group must belong to this region. If bound_zones are provided, this attribute is ignored.
 * @member {Array.<String>} bound-regions
 */

AuthMethodCreateGcp.prototype['bound-regions'] = undefined;
/**
 * List of service accounts the service account must be part of in order to be authenticated.
 * @member {Array.<String>} bound-service-accounts
 */

AuthMethodCreateGcp.prototype['bound-service-accounts'] = undefined;
/**
 * === Machine authentication section === List of zones that a GCE instance must belong to in order to be authenticated. TODO: If bound_instance_groups is provided, it is assumed to be a zonal group and the group must belong to this zone.
 * @member {Array.<String>} bound-zones
 */

AuthMethodCreateGcp.prototype['bound-zones'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

AuthMethodCreateGcp.prototype['delete_protection'] = undefined;
/**
 * Auth Method description
 * @member {String} description
 */

AuthMethodCreateGcp.prototype['description'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

AuthMethodCreateGcp.prototype['force-sub-claims'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

AuthMethodCreateGcp.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

AuthMethodCreateGcp.prototype['json'] = false;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 * @default 0
 */

AuthMethodCreateGcp.prototype['jwt-ttl'] = 0;
/**
 * Auth Method name
 * @member {String} name
 */

AuthMethodCreateGcp.prototype['name'] = undefined;
/**
 * Choose the relevant product type for the auth method [sm, sra, pm, dp, ca]
 * @member {Array.<String>} product-type
 */

AuthMethodCreateGcp.prototype['product-type'] = undefined;
/**
 * ServiceAccount credentials data instead of giving a file path, base64 encoded
 * @member {String} service-account-creds-data
 */

AuthMethodCreateGcp.prototype['service-account-creds-data'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AuthMethodCreateGcp.prototype['token'] = undefined;
/**
 * Type of the GCP Access Rules
 * @member {String} type
 */

AuthMethodCreateGcp.prototype['type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AuthMethodCreateGcp.prototype['uid-token'] = undefined;
var _default = AuthMethodCreateGcp;
exports["default"] = _default;
