// akeyless/dist/model/AuthMethod.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AuthExpirationEvent = _interopRequireDefault(require("./AuthExpirationEvent"));

var _AuthMethodAccessInfo = _interopRequireDefault(require("./AuthMethodAccessInfo"));

var _AuthMethodAdditionalData = _interopRequireDefault(require("./AuthMethodAdditionalData"));

var _AuthMethodRoleAssociation = _interopRequireDefault(require("./AuthMethodRoleAssociation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethod model module.
 * @module model/AuthMethod
 * @version 5.0.5
 */
var AuthMethod = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethod</code>.
   * @alias module:model/AuthMethod
   */
  function AuthMethod() {
    _classCallCheck(this, AuthMethod);

    AuthMethod.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethod, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AuthMethod</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethod} obj Optional instance to populate.
     * @return {module:model/AuthMethod} The populated <code>AuthMethod</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethod();

        if (data.hasOwnProperty('access_date')) {
          obj['access_date'] = _ApiClient["default"].convertToType(data['access_date'], 'Date');
        }

        if (data.hasOwnProperty('access_date_display')) {
          obj['access_date_display'] = _ApiClient["default"].convertToType(data['access_date_display'], 'String');
        }

        if (data.hasOwnProperty('access_info')) {
          obj['access_info'] = _AuthMethodAccessInfo["default"].constructFromObject(data['access_info']);
        }

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('associated_gw_ids')) {
          obj['associated_gw_ids'] = _ApiClient["default"].convertToType(data['associated_gw_ids'], ['Number']);
        }

        if (data.hasOwnProperty('auth_method_access_id')) {
          obj['auth_method_access_id'] = _ApiClient["default"].convertToType(data['auth_method_access_id'], 'String');
        }

        if (data.hasOwnProperty('auth_method_additional_data')) {
          obj['auth_method_additional_data'] = _AuthMethodAdditionalData["default"].constructFromObject(data['auth_method_additional_data']);
        }

        if (data.hasOwnProperty('auth_method_id')) {
          obj['auth_method_id'] = _ApiClient["default"].convertToType(data['auth_method_id'], 'Number');
        }

        if (data.hasOwnProperty('auth_method_name')) {
          obj['auth_method_name'] = _ApiClient["default"].convertToType(data['auth_method_name'], 'String');
        }

        if (data.hasOwnProperty('auth_method_roles_assoc')) {
          obj['auth_method_roles_assoc'] = _ApiClient["default"].convertToType(data['auth_method_roles_assoc'], [_AuthMethodRoleAssociation["default"]]);
        }

        if (data.hasOwnProperty('client_permissions')) {
          obj['client_permissions'] = _ApiClient["default"].convertToType(data['client_permissions'], ['String']);
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'Boolean');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('expiration_events')) {
          obj['expiration_events'] = _ApiClient["default"].convertToType(data['expiration_events'], [_AuthExpirationEvent["default"]]);
        }

        if (data.hasOwnProperty('is_approved')) {
          obj['is_approved'] = _ApiClient["default"].convertToType(data['is_approved'], 'Boolean');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AuthMethod</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AuthMethod</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_date_display'] && !(typeof data['access_date_display'] === 'string' || data['access_date_display'] instanceof String)) {
        throw new Error("Expected the field `access_date_display` to be a primitive type in the JSON string but got " + data['access_date_display']);
      } // validate the optional field `access_info`


      if (data['access_info']) {
        // data not null
        _AuthMethodAccessInfo["default"].validateJSON(data['access_info']);
      } // ensure the json data is a string


      if (data['account_id'] && !(typeof data['account_id'] === 'string' || data['account_id'] instanceof String)) {
        throw new Error("Expected the field `account_id` to be a primitive type in the JSON string but got " + data['account_id']);
      } // ensure the json data is an array


      if (!Array.isArray(data['associated_gw_ids'])) {
        throw new Error("Expected the field `associated_gw_ids` to be an array in the JSON data but got " + data['associated_gw_ids']);
      } // ensure the json data is a string


      if (data['auth_method_access_id'] && !(typeof data['auth_method_access_id'] === 'string' || data['auth_method_access_id'] instanceof String)) {
        throw new Error("Expected the field `auth_method_access_id` to be a primitive type in the JSON string but got " + data['auth_method_access_id']);
      } // validate the optional field `auth_method_additional_data`


      if (data['auth_method_additional_data']) {
        // data not null
        _AuthMethodAdditionalData["default"].validateJSON(data['auth_method_additional_data']);
      } // ensure the json data is a string


      if (data['auth_method_name'] && !(typeof data['auth_method_name'] === 'string' || data['auth_method_name'] instanceof String)) {
        throw new Error("Expected the field `auth_method_name` to be a primitive type in the JSON string but got " + data['auth_method_name']);
      }

      if (data['auth_method_roles_assoc']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['auth_method_roles_assoc'])) {
          throw new Error("Expected the field `auth_method_roles_assoc` to be an array in the JSON data but got " + data['auth_method_roles_assoc']);
        } // validate the optional field `auth_method_roles_assoc` (array)


        var _iterator = _createForOfIteratorHelper(data['auth_method_roles_assoc']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _AuthMethodRoleAssociation["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is an array


      if (!Array.isArray(data['client_permissions'])) {
        throw new Error("Expected the field `client_permissions` to be an array in the JSON data but got " + data['client_permissions']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      }

      if (data['expiration_events']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['expiration_events'])) {
          throw new Error("Expected the field `expiration_events` to be an array in the JSON data but got " + data['expiration_events']);
        } // validate the optional field `expiration_events` (array)


        var _iterator2 = _createForOfIteratorHelper(data['expiration_events']),
            _step2;

        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;

            _AuthExpirationEvent["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }

        ;
      }

      return true;
    }
  }]);

  return AuthMethod;
}();
/**
 * @member {Date} access_date
 */


AuthMethod.prototype['access_date'] = undefined;
/**
 * @member {String} access_date_display
 */

AuthMethod.prototype['access_date_display'] = undefined;
/**
 * @member {module:model/AuthMethodAccessInfo} access_info
 */

AuthMethod.prototype['access_info'] = undefined;
/**
 * @member {String} account_id
 */

AuthMethod.prototype['account_id'] = undefined;
/**
 * @member {Array.<Number>} associated_gw_ids
 */

AuthMethod.prototype['associated_gw_ids'] = undefined;
/**
 * @member {String} auth_method_access_id
 */

AuthMethod.prototype['auth_method_access_id'] = undefined;
/**
 * @member {module:model/AuthMethodAdditionalData} auth_method_additional_data
 */

AuthMethod.prototype['auth_method_additional_data'] = undefined;
/**
 * @member {Number} auth_method_id
 */

AuthMethod.prototype['auth_method_id'] = undefined;
/**
 * @member {String} auth_method_name
 */

AuthMethod.prototype['auth_method_name'] = undefined;
/**
 * @member {Array.<module:model/AuthMethodRoleAssociation>} auth_method_roles_assoc
 */

AuthMethod.prototype['auth_method_roles_assoc'] = undefined;
/**
 * @member {Array.<String>} client_permissions
 */

AuthMethod.prototype['client_permissions'] = undefined;
/**
 * @member {Date} creation_date
 */

AuthMethod.prototype['creation_date'] = undefined;
/**
 * @member {Boolean} delete_protection
 */

AuthMethod.prototype['delete_protection'] = undefined;
/**
 * @member {String} description
 */

AuthMethod.prototype['description'] = undefined;
/**
 * @member {Array.<module:model/AuthExpirationEvent>} expiration_events
 */

AuthMethod.prototype['expiration_events'] = undefined;
/**
 * @member {Boolean} is_approved
 */

AuthMethod.prototype['is_approved'] = undefined;
/**
 * @member {Date} modification_date
 */

AuthMethod.prototype['modification_date'] = undefined;
var _default = AuthMethod;
exports["default"] = _default;
