// akeyless/dist/model/AuthMethodCreateAzureAD.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethodCreateAzureAD model module.
 * @module model/AuthMethodCreateAzureAD
 * @version 4.2.0
 */
var AuthMethodCreateAzureAD = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethodCreateAzureAD</code>.
   * authMethodCreateAzureAD is a command that creates a new auth method that will be able to authenticate using Azure Active Directory credentials.
   * @alias module:model/AuthMethodCreateAzureAD
   * @param boundTenantId {String} The Azure tenant id that the access is restricted to
   * @param name {String} Auth Method name
   */
  function AuthMethodCreateAzureAD(boundTenantId, name) {
    _classCallCheck(this, AuthMethodCreateAzureAD);

    AuthMethodCreateAzureAD.initialize(this, boundTenantId, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethodCreateAzureAD, null, [{
    key: "initialize",
    value: function initialize(obj, boundTenantId, name) {
      obj['bound-tenant-id'] = boundTenantId;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>AuthMethodCreateAzureAD</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethodCreateAzureAD} obj Optional instance to populate.
     * @return {module:model/AuthMethodCreateAzureAD} The populated <code>AuthMethodCreateAzureAD</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethodCreateAzureAD();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('audit-logs-claims')) {
          obj['audit-logs-claims'] = _ApiClient["default"].convertToType(data['audit-logs-claims'], ['String']);
        }

        if (data.hasOwnProperty('bound-group-id')) {
          obj['bound-group-id'] = _ApiClient["default"].convertToType(data['bound-group-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('bound-providers')) {
          obj['bound-providers'] = _ApiClient["default"].convertToType(data['bound-providers'], ['String']);
        }

        if (data.hasOwnProperty('bound-resource-id')) {
          obj['bound-resource-id'] = _ApiClient["default"].convertToType(data['bound-resource-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-resource-names')) {
          obj['bound-resource-names'] = _ApiClient["default"].convertToType(data['bound-resource-names'], ['String']);
        }

        if (data.hasOwnProperty('bound-resource-types')) {
          obj['bound-resource-types'] = _ApiClient["default"].convertToType(data['bound-resource-types'], ['String']);
        }

        if (data.hasOwnProperty('bound-rg-id')) {
          obj['bound-rg-id'] = _ApiClient["default"].convertToType(data['bound-rg-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-spid')) {
          obj['bound-spid'] = _ApiClient["default"].convertToType(data['bound-spid'], ['String']);
        }

        if (data.hasOwnProperty('bound-sub-id')) {
          obj['bound-sub-id'] = _ApiClient["default"].convertToType(data['bound-sub-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-tenant-id')) {
          obj['bound-tenant-id'] = _ApiClient["default"].convertToType(data['bound-tenant-id'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwks-uri')) {
          obj['jwks-uri'] = _ApiClient["default"].convertToType(data['jwks-uri'], 'String');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('product-type')) {
          obj['product-type'] = _ApiClient["default"].convertToType(data['product-type'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AuthMethodCreateAzureAD;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


AuthMethodCreateAzureAD.prototype['access-expires'] = 0;
/**
 * Deprecated (Deprecated) The audience in the JWT
 * @member {String} audience
 * @default 'https://management.azure.com/'
 */

AuthMethodCreateAzureAD.prototype['audience'] = 'https://management.azure.com/';
/**
 * Subclaims to include in audit logs, e.g \"--audit-logs-claims email --audit-logs-claims username\"
 * @member {Array.<String>} audit-logs-claims
 */

AuthMethodCreateAzureAD.prototype['audit-logs-claims'] = undefined;
/**
 * A list of group ids that the access is restricted to
 * @member {Array.<String>} bound-group-id
 */

AuthMethodCreateAzureAD.prototype['bound-group-id'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

AuthMethodCreateAzureAD.prototype['bound-ips'] = undefined;
/**
 * A list of resource providers that the access is restricted to (e.g, Microsoft.Compute, Microsoft.ManagedIdentity, etc)
 * @member {Array.<String>} bound-providers
 */

AuthMethodCreateAzureAD.prototype['bound-providers'] = undefined;
/**
 * A list of full resource ids that the access is restricted to
 * @member {Array.<String>} bound-resource-id
 */

AuthMethodCreateAzureAD.prototype['bound-resource-id'] = undefined;
/**
 * A list of resource names that the access is restricted to (e.g, a virtual machine name, scale set name, etc).
 * @member {Array.<String>} bound-resource-names
 */

AuthMethodCreateAzureAD.prototype['bound-resource-names'] = undefined;
/**
 * A list of resource types that the access is restricted to (e.g, virtualMachines, userAssignedIdentities, etc)
 * @member {Array.<String>} bound-resource-types
 */

AuthMethodCreateAzureAD.prototype['bound-resource-types'] = undefined;
/**
 * A list of resource groups that the access is restricted to
 * @member {Array.<String>} bound-rg-id
 */

AuthMethodCreateAzureAD.prototype['bound-rg-id'] = undefined;
/**
 * A list of service principal IDs that the access is restricted to
 * @member {Array.<String>} bound-spid
 */

AuthMethodCreateAzureAD.prototype['bound-spid'] = undefined;
/**
 * A list of subscription ids that the access is restricted to
 * @member {Array.<String>} bound-sub-id
 */

AuthMethodCreateAzureAD.prototype['bound-sub-id'] = undefined;
/**
 * The Azure tenant id that the access is restricted to
 * @member {String} bound-tenant-id
 */

AuthMethodCreateAzureAD.prototype['bound-tenant-id'] = undefined;
/**
 * Auth Method description
 * @member {String} description
 */

AuthMethodCreateAzureAD.prototype['description'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

AuthMethodCreateAzureAD.prototype['force-sub-claims'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

AuthMethodCreateAzureAD.prototype['gw-bound-ips'] = undefined;
/**
 * Issuer URL
 * @member {String} issuer
 * @default 'https://sts.windows.net/---bound_tenant_id---'
 */

AuthMethodCreateAzureAD.prototype['issuer'] = 'https://sts.windows.net/---bound_tenant_id---';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

AuthMethodCreateAzureAD.prototype['json'] = false;
/**
 * The URL to the JSON Web Key Set (JWKS) that containing the public keys that should be used to verify any JSON Web Token (JWT) issued by the authorization server.
 * @member {String} jwks-uri
 * @default 'https://login.microsoftonline.com/common/discovery/keys'
 */

AuthMethodCreateAzureAD.prototype['jwks-uri'] = 'https://login.microsoftonline.com/common/discovery/keys';
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 * @default 0
 */

AuthMethodCreateAzureAD.prototype['jwt-ttl'] = 0;
/**
 * Auth Method name
 * @member {String} name
 */

AuthMethodCreateAzureAD.prototype['name'] = undefined;
/**
 * Choose the relevant product type for the auth method [sm, sra, pm, dp, ca]
 * @member {Array.<String>} product-type
 */

AuthMethodCreateAzureAD.prototype['product-type'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AuthMethodCreateAzureAD.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AuthMethodCreateAzureAD.prototype['uid-token'] = undefined;
var _default = AuthMethodCreateAzureAD;
exports["default"] = _default;
