// akeyless/dist/model/AuthOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SystemAccessCredentialsReplyObj = _interopRequireDefault(require("./SystemAccessCredentialsReplyObj"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthOutput model module.
 * @module model/AuthOutput
 * @version 5.0.6
 */
var AuthOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthOutput</code>.
   * @alias module:model/AuthOutput
   */
  function AuthOutput() {
    _classCallCheck(this, AuthOutput);

    AuthOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AuthOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthOutput} obj Optional instance to populate.
     * @return {module:model/AuthOutput} The populated <code>AuthOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthOutput();

        if (data.hasOwnProperty('complete_auth_link')) {
          obj['complete_auth_link'] = _ApiClient["default"].convertToType(data['complete_auth_link'], 'String');
        }

        if (data.hasOwnProperty('creds')) {
          obj['creds'] = _SystemAccessCredentialsReplyObj["default"].constructFromObject(data['creds']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AuthOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AuthOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['complete_auth_link'] && !(typeof data['complete_auth_link'] === 'string' || data['complete_auth_link'] instanceof String)) {
        throw new Error("Expected the field `complete_auth_link` to be a primitive type in the JSON string but got " + data['complete_auth_link']);
      } // validate the optional field `creds`


      if (data['creds']) {
        // data not null
        _SystemAccessCredentialsReplyObj["default"].validateJSON(data['creds']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      }

      return true;
    }
  }]);

  return AuthOutput;
}();
/**
 * @member {String} complete_auth_link
 */


AuthOutput.prototype['complete_auth_link'] = undefined;
/**
 * @member {module:model/SystemAccessCredentialsReplyObj} creds
 */

AuthOutput.prototype['creds'] = undefined;
/**
 * @member {String} token
 */

AuthOutput.prototype['token'] = undefined;
var _default = AuthOutput;
exports["default"] = _default;
