// akeyless/dist/model/AWSGatewayCloudIdentityExternalIdOpt.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AWSGatewayCloudIdentityExternalIdOpt model module.
 * @module model/AWSGatewayCloudIdentityExternalIdOpt
 * @version 5.0.5
 */
var AWSGatewayCloudIdentityExternalIdOpt = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AWSGatewayCloudIdentityExternalIdOpt</code>.
   * @alias module:model/AWSGatewayCloudIdentityExternalIdOpt
   */
  function AWSGatewayCloudIdentityExternalIdOpt() {
    _classCallCheck(this, AWSGatewayCloudIdentityExternalIdOpt);

    AWSGatewayCloudIdentityExternalIdOpt.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AWSGatewayCloudIdentityExternalIdOpt, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AWSGatewayCloudIdentityExternalIdOpt</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AWSGatewayCloudIdentityExternalIdOpt} obj Optional instance to populate.
     * @return {module:model/AWSGatewayCloudIdentityExternalIdOpt} The populated <code>AWSGatewayCloudIdentityExternalIdOpt</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AWSGatewayCloudIdentityExternalIdOpt();

        if (data.hasOwnProperty('generated_external_id')) {
          obj['generated_external_id'] = _ApiClient["default"].convertToType(data['generated_external_id'], 'String');
        }

        if (data.hasOwnProperty('is_enabled')) {
          obj['is_enabled'] = _ApiClient["default"].convertToType(data['is_enabled'], 'Boolean');
        }

        if (data.hasOwnProperty('role_arn')) {
          obj['role_arn'] = _ApiClient["default"].convertToType(data['role_arn'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AWSGatewayCloudIdentityExternalIdOpt</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AWSGatewayCloudIdentityExternalIdOpt</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['generated_external_id'] && !(typeof data['generated_external_id'] === 'string' || data['generated_external_id'] instanceof String)) {
        throw new Error("Expected the field `generated_external_id` to be a primitive type in the JSON string but got " + data['generated_external_id']);
      } // ensure the json data is a string


      if (data['role_arn'] && !(typeof data['role_arn'] === 'string' || data['role_arn'] instanceof String)) {
        throw new Error("Expected the field `role_arn` to be a primitive type in the JSON string but got " + data['role_arn']);
      }

      return true;
    }
  }]);

  return AWSGatewayCloudIdentityExternalIdOpt;
}();
/**
 * @member {String} generated_external_id
 */


AWSGatewayCloudIdentityExternalIdOpt.prototype['generated_external_id'] = undefined;
/**
 * @member {Boolean} is_enabled
 */

AWSGatewayCloudIdentityExternalIdOpt.prototype['is_enabled'] = undefined;
/**
 * @member {String} role_arn
 */

AWSGatewayCloudIdentityExternalIdOpt.prototype['role_arn'] = undefined;
var _default = AWSGatewayCloudIdentityExternalIdOpt;
exports["default"] = _default;
