// akeyless/dist/model/AWSPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AWSPayload model module.
 * @module model/AWSPayload
 * @version 5.0.8
 */
var AWSPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AWSPayload</code>.
   * @alias module:model/AWSPayload
   */
  function AWSPayload() {
    _classCallCheck(this, AWSPayload);

    AWSPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AWSPayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AWSPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AWSPayload} obj Optional instance to populate.
     * @return {module:model/AWSPayload} The populated <code>AWSPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AWSPayload();

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('secret')) {
          obj['secret'] = _ApiClient["default"].convertToType(data['secret'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AWSPayload</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AWSPayload</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['region'] && !(typeof data['region'] === 'string' || data['region'] instanceof String)) {
        throw new Error("Expected the field `region` to be a primitive type in the JSON string but got " + data['region']);
      } // ensure the json data is a string


      if (data['secret'] && !(typeof data['secret'] === 'string' || data['secret'] instanceof String)) {
        throw new Error("Expected the field `secret` to be a primitive type in the JSON string but got " + data['secret']);
      }

      return true;
    }
  }]);

  return AWSPayload;
}();
/**
 * @member {String} key
 */


AWSPayload.prototype['key'] = undefined;
/**
 * @member {String} region
 */

AWSPayload.prototype['region'] = undefined;
/**
 * @member {String} secret
 */

AWSPayload.prototype['secret'] = undefined;
var _default = AWSPayload;
exports["default"] = _default;
