// akeyless/dist/model/AwsS3LogForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AwsS3LogForwardingConfig model module.
 * @module model/AwsS3LogForwardingConfig
 * @version 5.0.3
 */
var AwsS3LogForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AwsS3LogForwardingConfig</code>.
   * @alias module:model/AwsS3LogForwardingConfig
   */
  function AwsS3LogForwardingConfig() {
    _classCallCheck(this, AwsS3LogForwardingConfig);

    AwsS3LogForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AwsS3LogForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AwsS3LogForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AwsS3LogForwardingConfig} obj Optional instance to populate.
     * @return {module:model/AwsS3LogForwardingConfig} The populated <code>AwsS3LogForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AwsS3LogForwardingConfig();

        if (data.hasOwnProperty('aws_access_id')) {
          obj['aws_access_id'] = _ApiClient["default"].convertToType(data['aws_access_id'], 'String');
        }

        if (data.hasOwnProperty('aws_access_key')) {
          obj['aws_access_key'] = _ApiClient["default"].convertToType(data['aws_access_key'], 'String');
        }

        if (data.hasOwnProperty('aws_auth_type')) {
          obj['aws_auth_type'] = _ApiClient["default"].convertToType(data['aws_auth_type'], 'String');
        }

        if (data.hasOwnProperty('aws_region')) {
          obj['aws_region'] = _ApiClient["default"].convertToType(data['aws_region'], 'String');
        }

        if (data.hasOwnProperty('aws_role_arn')) {
          obj['aws_role_arn'] = _ApiClient["default"].convertToType(data['aws_role_arn'], 'String');
        }

        if (data.hasOwnProperty('aws_use_gateway_cloud_identity')) {
          obj['aws_use_gateway_cloud_identity'] = _ApiClient["default"].convertToType(data['aws_use_gateway_cloud_identity'], 'Boolean');
        }

        if (data.hasOwnProperty('bucket_name')) {
          obj['bucket_name'] = _ApiClient["default"].convertToType(data['bucket_name'], 'String');
        }

        if (data.hasOwnProperty('log_folder')) {
          obj['log_folder'] = _ApiClient["default"].convertToType(data['log_folder'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AwsS3LogForwardingConfig</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AwsS3LogForwardingConfig</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['aws_access_id'] && !(typeof data['aws_access_id'] === 'string' || data['aws_access_id'] instanceof String)) {
        throw new Error("Expected the field `aws_access_id` to be a primitive type in the JSON string but got " + data['aws_access_id']);
      } // ensure the json data is a string


      if (data['aws_access_key'] && !(typeof data['aws_access_key'] === 'string' || data['aws_access_key'] instanceof String)) {
        throw new Error("Expected the field `aws_access_key` to be a primitive type in the JSON string but got " + data['aws_access_key']);
      } // ensure the json data is a string


      if (data['aws_auth_type'] && !(typeof data['aws_auth_type'] === 'string' || data['aws_auth_type'] instanceof String)) {
        throw new Error("Expected the field `aws_auth_type` to be a primitive type in the JSON string but got " + data['aws_auth_type']);
      } // ensure the json data is a string


      if (data['aws_region'] && !(typeof data['aws_region'] === 'string' || data['aws_region'] instanceof String)) {
        throw new Error("Expected the field `aws_region` to be a primitive type in the JSON string but got " + data['aws_region']);
      } // ensure the json data is a string


      if (data['aws_role_arn'] && !(typeof data['aws_role_arn'] === 'string' || data['aws_role_arn'] instanceof String)) {
        throw new Error("Expected the field `aws_role_arn` to be a primitive type in the JSON string but got " + data['aws_role_arn']);
      } // ensure the json data is a string


      if (data['bucket_name'] && !(typeof data['bucket_name'] === 'string' || data['bucket_name'] instanceof String)) {
        throw new Error("Expected the field `bucket_name` to be a primitive type in the JSON string but got " + data['bucket_name']);
      } // ensure the json data is a string


      if (data['log_folder'] && !(typeof data['log_folder'] === 'string' || data['log_folder'] instanceof String)) {
        throw new Error("Expected the field `log_folder` to be a primitive type in the JSON string but got " + data['log_folder']);
      }

      return true;
    }
  }]);

  return AwsS3LogForwardingConfig;
}();
/**
 * @member {String} aws_access_id
 */


AwsS3LogForwardingConfig.prototype['aws_access_id'] = undefined;
/**
 * @member {String} aws_access_key
 */

AwsS3LogForwardingConfig.prototype['aws_access_key'] = undefined;
/**
 * @member {String} aws_auth_type
 */

AwsS3LogForwardingConfig.prototype['aws_auth_type'] = undefined;
/**
 * @member {String} aws_region
 */

AwsS3LogForwardingConfig.prototype['aws_region'] = undefined;
/**
 * @member {String} aws_role_arn
 */

AwsS3LogForwardingConfig.prototype['aws_role_arn'] = undefined;
/**
 * deprecated
 * @member {Boolean} aws_use_gateway_cloud_identity
 */

AwsS3LogForwardingConfig.prototype['aws_use_gateway_cloud_identity'] = undefined;
/**
 * @member {String} bucket_name
 */

AwsS3LogForwardingConfig.prototype['bucket_name'] = undefined;
/**
 * @member {String} log_folder
 */

AwsS3LogForwardingConfig.prototype['log_folder'] = undefined;
var _default = AwsS3LogForwardingConfig;
exports["default"] = _default;
