// akeyless/dist/model/AWSTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AWSTargetDetails model module.
 * @module model/AWSTargetDetails
 * @version 5.0.3
 */
var AWSTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AWSTargetDetails</code>.
   * @alias module:model/AWSTargetDetails
   */
  function AWSTargetDetails() {
    _classCallCheck(this, AWSTargetDetails);

    AWSTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AWSTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AWSTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AWSTargetDetails} obj Optional instance to populate.
     * @return {module:model/AWSTargetDetails} The populated <code>AWSTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AWSTargetDetails();

        if (data.hasOwnProperty('aws_access_key_id')) {
          obj['aws_access_key_id'] = _ApiClient["default"].convertToType(data['aws_access_key_id'], 'String');
        }

        if (data.hasOwnProperty('aws_region')) {
          obj['aws_region'] = _ApiClient["default"].convertToType(data['aws_region'], 'String');
        }

        if (data.hasOwnProperty('aws_secret_access_key')) {
          obj['aws_secret_access_key'] = _ApiClient["default"].convertToType(data['aws_secret_access_key'], 'String');
        }

        if (data.hasOwnProperty('aws_session_token')) {
          obj['aws_session_token'] = _ApiClient["default"].convertToType(data['aws_session_token'], 'String');
        }

        if (data.hasOwnProperty('use_gw_cloud_identity')) {
          obj['use_gw_cloud_identity'] = _ApiClient["default"].convertToType(data['use_gw_cloud_identity'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AWSTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AWSTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['aws_access_key_id'] && !(typeof data['aws_access_key_id'] === 'string' || data['aws_access_key_id'] instanceof String)) {
        throw new Error("Expected the field `aws_access_key_id` to be a primitive type in the JSON string but got " + data['aws_access_key_id']);
      } // ensure the json data is a string


      if (data['aws_region'] && !(typeof data['aws_region'] === 'string' || data['aws_region'] instanceof String)) {
        throw new Error("Expected the field `aws_region` to be a primitive type in the JSON string but got " + data['aws_region']);
      } // ensure the json data is a string


      if (data['aws_secret_access_key'] && !(typeof data['aws_secret_access_key'] === 'string' || data['aws_secret_access_key'] instanceof String)) {
        throw new Error("Expected the field `aws_secret_access_key` to be a primitive type in the JSON string but got " + data['aws_secret_access_key']);
      } // ensure the json data is a string


      if (data['aws_session_token'] && !(typeof data['aws_session_token'] === 'string' || data['aws_session_token'] instanceof String)) {
        throw new Error("Expected the field `aws_session_token` to be a primitive type in the JSON string but got " + data['aws_session_token']);
      }

      return true;
    }
  }]);

  return AWSTargetDetails;
}();
/**
 * @member {String} aws_access_key_id
 */


AWSTargetDetails.prototype['aws_access_key_id'] = undefined;
/**
 * @member {String} aws_region
 */

AWSTargetDetails.prototype['aws_region'] = undefined;
/**
 * @member {String} aws_secret_access_key
 */

AWSTargetDetails.prototype['aws_secret_access_key'] = undefined;
/**
 * @member {String} aws_session_token
 */

AWSTargetDetails.prototype['aws_session_token'] = undefined;
/**
 * @member {Boolean} use_gw_cloud_identity
 */

AWSTargetDetails.prototype['use_gw_cloud_identity'] = undefined;
var _default = AWSTargetDetails;
exports["default"] = _default;
