// akeyless/dist/model/AzureLogAnalyticsForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AzureLogAnalyticsForwardingConfig model module.
 * @module model/AzureLogAnalyticsForwardingConfig
 * @version 5.0.1
 */
var AzureLogAnalyticsForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AzureLogAnalyticsForwardingConfig</code>.
   * @alias module:model/AzureLogAnalyticsForwardingConfig
   */
  function AzureLogAnalyticsForwardingConfig() {
    _classCallCheck(this, AzureLogAnalyticsForwardingConfig);

    AzureLogAnalyticsForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AzureLogAnalyticsForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AzureLogAnalyticsForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AzureLogAnalyticsForwardingConfig} obj Optional instance to populate.
     * @return {module:model/AzureLogAnalyticsForwardingConfig} The populated <code>AzureLogAnalyticsForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AzureLogAnalyticsForwardingConfig();

        if (data.hasOwnProperty('azure_workspace_id')) {
          obj['azure_workspace_id'] = _ApiClient["default"].convertToType(data['azure_workspace_id'], 'String');
        }

        if (data.hasOwnProperty('azure_workspace_key')) {
          obj['azure_workspace_key'] = _ApiClient["default"].convertToType(data['azure_workspace_key'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AzureLogAnalyticsForwardingConfig</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AzureLogAnalyticsForwardingConfig</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['azure_workspace_id'] && !(typeof data['azure_workspace_id'] === 'string' || data['azure_workspace_id'] instanceof String)) {
        throw new Error("Expected the field `azure_workspace_id` to be a primitive type in the JSON string but got " + data['azure_workspace_id']);
      } // ensure the json data is a string


      if (data['azure_workspace_key'] && !(typeof data['azure_workspace_key'] === 'string' || data['azure_workspace_key'] instanceof String)) {
        throw new Error("Expected the field `azure_workspace_key` to be a primitive type in the JSON string but got " + data['azure_workspace_key']);
      }

      return true;
    }
  }]);

  return AzureLogAnalyticsForwardingConfig;
}();
/**
 * @member {String} azure_workspace_id
 */


AzureLogAnalyticsForwardingConfig.prototype['azure_workspace_id'] = undefined;
/**
 * @member {String} azure_workspace_key
 */

AzureLogAnalyticsForwardingConfig.prototype['azure_workspace_key'] = undefined;
var _default = AzureLogAnalyticsForwardingConfig;
exports["default"] = _default;
