// akeyless/dist/model/AzureStorage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AzureStorage model module.
 * @module model/AzureStorage
 * @version 5.0.1
 */
var AzureStorage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AzureStorage</code>.
   * @alias module:model/AzureStorage
   */
  function AzureStorage() {
    _classCallCheck(this, AzureStorage);

    AzureStorage.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AzureStorage, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AzureStorage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AzureStorage} obj Optional instance to populate.
     * @return {module:model/AzureStorage} The populated <code>AzureStorage</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AzureStorage();

        if (data.hasOwnProperty('auth_type')) {
          obj['auth_type'] = _ApiClient["default"].convertToType(data['auth_type'], 'String');
        }

        if (data.hasOwnProperty('client_id')) {
          obj['client_id'] = _ApiClient["default"].convertToType(data['client_id'], 'String');
        }

        if (data.hasOwnProperty('client_secret')) {
          obj['client_secret'] = _ApiClient["default"].convertToType(data['client_secret'], 'String');
        }

        if (data.hasOwnProperty('storage_account')) {
          obj['storage_account'] = _ApiClient["default"].convertToType(data['storage_account'], 'String');
        }

        if (data.hasOwnProperty('storage_container_name')) {
          obj['storage_container_name'] = _ApiClient["default"].convertToType(data['storage_container_name'], 'String');
        }

        if (data.hasOwnProperty('tenant_id')) {
          obj['tenant_id'] = _ApiClient["default"].convertToType(data['tenant_id'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AzureStorage</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AzureStorage</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['auth_type'] && !(typeof data['auth_type'] === 'string' || data['auth_type'] instanceof String)) {
        throw new Error("Expected the field `auth_type` to be a primitive type in the JSON string but got " + data['auth_type']);
      } // ensure the json data is a string


      if (data['client_id'] && !(typeof data['client_id'] === 'string' || data['client_id'] instanceof String)) {
        throw new Error("Expected the field `client_id` to be a primitive type in the JSON string but got " + data['client_id']);
      } // ensure the json data is a string


      if (data['client_secret'] && !(typeof data['client_secret'] === 'string' || data['client_secret'] instanceof String)) {
        throw new Error("Expected the field `client_secret` to be a primitive type in the JSON string but got " + data['client_secret']);
      } // ensure the json data is a string


      if (data['storage_account'] && !(typeof data['storage_account'] === 'string' || data['storage_account'] instanceof String)) {
        throw new Error("Expected the field `storage_account` to be a primitive type in the JSON string but got " + data['storage_account']);
      } // ensure the json data is a string


      if (data['storage_container_name'] && !(typeof data['storage_container_name'] === 'string' || data['storage_container_name'] instanceof String)) {
        throw new Error("Expected the field `storage_container_name` to be a primitive type in the JSON string but got " + data['storage_container_name']);
      } // ensure the json data is a string


      if (data['tenant_id'] && !(typeof data['tenant_id'] === 'string' || data['tenant_id'] instanceof String)) {
        throw new Error("Expected the field `tenant_id` to be a primitive type in the JSON string but got " + data['tenant_id']);
      }

      return true;
    }
  }]);

  return AzureStorage;
}();
/**
 * @member {String} auth_type
 */


AzureStorage.prototype['auth_type'] = undefined;
/**
 * @member {String} client_id
 */

AzureStorage.prototype['client_id'] = undefined;
/**
 * @member {String} client_secret
 */

AzureStorage.prototype['client_secret'] = undefined;
/**
 * @member {String} storage_account
 */

AzureStorage.prototype['storage_account'] = undefined;
/**
 * @member {String} storage_container_name
 */

AzureStorage.prototype['storage_container_name'] = undefined;
/**
 * creds
 * @member {String} tenant_id
 */

AzureStorage.prototype['tenant_id'] = undefined;
var _default = AzureStorage;
exports["default"] = _default;
