// akeyless/dist/model/AzureTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AzureTargetDetails model module.
 * @module model/AzureTargetDetails
 * @version 5.0.6
 */
var AzureTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AzureTargetDetails</code>.
   * @alias module:model/AzureTargetDetails
   */
  function AzureTargetDetails() {
    _classCallCheck(this, AzureTargetDetails);

    AzureTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AzureTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AzureTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AzureTargetDetails} obj Optional instance to populate.
     * @return {module:model/AzureTargetDetails} The populated <code>AzureTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AzureTargetDetails();

        if (data.hasOwnProperty('azure_client_id')) {
          obj['azure_client_id'] = _ApiClient["default"].convertToType(data['azure_client_id'], 'String');
        }

        if (data.hasOwnProperty('azure_client_secret')) {
          obj['azure_client_secret'] = _ApiClient["default"].convertToType(data['azure_client_secret'], 'String');
        }

        if (data.hasOwnProperty('azure_resource_group_name')) {
          obj['azure_resource_group_name'] = _ApiClient["default"].convertToType(data['azure_resource_group_name'], 'String');
        }

        if (data.hasOwnProperty('azure_resource_name')) {
          obj['azure_resource_name'] = _ApiClient["default"].convertToType(data['azure_resource_name'], 'String');
        }

        if (data.hasOwnProperty('azure_subscription_id')) {
          obj['azure_subscription_id'] = _ApiClient["default"].convertToType(data['azure_subscription_id'], 'String');
        }

        if (data.hasOwnProperty('azure_tenant_id')) {
          obj['azure_tenant_id'] = _ApiClient["default"].convertToType(data['azure_tenant_id'], 'String');
        }

        if (data.hasOwnProperty('azure_username')) {
          obj['azure_username'] = _ApiClient["default"].convertToType(data['azure_username'], 'String');
        }

        if (data.hasOwnProperty('connection_type')) {
          obj['connection_type'] = _ApiClient["default"].convertToType(data['connection_type'], 'String');
        }

        if (data.hasOwnProperty('expiration_date')) {
          obj['expiration_date'] = _ApiClient["default"].convertToType(data['expiration_date'], 'Date');
        }

        if (data.hasOwnProperty('grace_rotated_secret_key')) {
          obj['grace_rotated_secret_key'] = _ApiClient["default"].convertToType(data['grace_rotated_secret_key'], 'String');
        }

        if (data.hasOwnProperty('use_gw_cloud_identity')) {
          obj['use_gw_cloud_identity'] = _ApiClient["default"].convertToType(data['use_gw_cloud_identity'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>AzureTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AzureTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['azure_client_id'] && !(typeof data['azure_client_id'] === 'string' || data['azure_client_id'] instanceof String)) {
        throw new Error("Expected the field `azure_client_id` to be a primitive type in the JSON string but got " + data['azure_client_id']);
      } // ensure the json data is a string


      if (data['azure_client_secret'] && !(typeof data['azure_client_secret'] === 'string' || data['azure_client_secret'] instanceof String)) {
        throw new Error("Expected the field `azure_client_secret` to be a primitive type in the JSON string but got " + data['azure_client_secret']);
      } // ensure the json data is a string


      if (data['azure_resource_group_name'] && !(typeof data['azure_resource_group_name'] === 'string' || data['azure_resource_group_name'] instanceof String)) {
        throw new Error("Expected the field `azure_resource_group_name` to be a primitive type in the JSON string but got " + data['azure_resource_group_name']);
      } // ensure the json data is a string


      if (data['azure_resource_name'] && !(typeof data['azure_resource_name'] === 'string' || data['azure_resource_name'] instanceof String)) {
        throw new Error("Expected the field `azure_resource_name` to be a primitive type in the JSON string but got " + data['azure_resource_name']);
      } // ensure the json data is a string


      if (data['azure_subscription_id'] && !(typeof data['azure_subscription_id'] === 'string' || data['azure_subscription_id'] instanceof String)) {
        throw new Error("Expected the field `azure_subscription_id` to be a primitive type in the JSON string but got " + data['azure_subscription_id']);
      } // ensure the json data is a string


      if (data['azure_tenant_id'] && !(typeof data['azure_tenant_id'] === 'string' || data['azure_tenant_id'] instanceof String)) {
        throw new Error("Expected the field `azure_tenant_id` to be a primitive type in the JSON string but got " + data['azure_tenant_id']);
      } // ensure the json data is a string


      if (data['azure_username'] && !(typeof data['azure_username'] === 'string' || data['azure_username'] instanceof String)) {
        throw new Error("Expected the field `azure_username` to be a primitive type in the JSON string but got " + data['azure_username']);
      } // ensure the json data is a string


      if (data['connection_type'] && !(typeof data['connection_type'] === 'string' || data['connection_type'] instanceof String)) {
        throw new Error("Expected the field `connection_type` to be a primitive type in the JSON string but got " + data['connection_type']);
      } // ensure the json data is a string


      if (data['grace_rotated_secret_key'] && !(typeof data['grace_rotated_secret_key'] === 'string' || data['grace_rotated_secret_key'] instanceof String)) {
        throw new Error("Expected the field `grace_rotated_secret_key` to be a primitive type in the JSON string but got " + data['grace_rotated_secret_key']);
      }

      return true;
    }
  }]);

  return AzureTargetDetails;
}();
/**
 * @member {String} azure_client_id
 */


AzureTargetDetails.prototype['azure_client_id'] = undefined;
/**
 * @member {String} azure_client_secret
 */

AzureTargetDetails.prototype['azure_client_secret'] = undefined;
/**
 * @member {String} azure_resource_group_name
 */

AzureTargetDetails.prototype['azure_resource_group_name'] = undefined;
/**
 * @member {String} azure_resource_name
 */

AzureTargetDetails.prototype['azure_resource_name'] = undefined;
/**
 * @member {String} azure_subscription_id
 */

AzureTargetDetails.prototype['azure_subscription_id'] = undefined;
/**
 * @member {String} azure_tenant_id
 */

AzureTargetDetails.prototype['azure_tenant_id'] = undefined;
/**
 * @member {String} azure_username
 */

AzureTargetDetails.prototype['azure_username'] = undefined;
/**
 * @member {String} connection_type
 */

AzureTargetDetails.prototype['connection_type'] = undefined;
/**
 * @member {Date} expiration_date
 */

AzureTargetDetails.prototype['expiration_date'] = undefined;
/**
 * @member {String} grace_rotated_secret_key
 */

AzureTargetDetails.prototype['grace_rotated_secret_key'] = undefined;
/**
 * @member {Boolean} use_gw_cloud_identity
 */

AzureTargetDetails.prototype['use_gw_cloud_identity'] = undefined;
var _default = AzureTargetDetails;
exports["default"] = _default;
