// akeyless/dist/model/BastionListEntry.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BastionListEntry model module.
 * @module model/BastionListEntry
 * @version 5.0.6
 */
var BastionListEntry = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BastionListEntry</code>.
   * @alias module:model/BastionListEntry
   */
  function BastionListEntry() {
    _classCallCheck(this, BastionListEntry);

    BastionListEntry.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BastionListEntry, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BastionListEntry</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BastionListEntry} obj Optional instance to populate.
     * @return {module:model/BastionListEntry} The populated <code>BastionListEntry</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BastionListEntry();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('allowed_access_ids')) {
          obj['allowed_access_ids'] = _ApiClient["default"].convertToType(data['allowed_access_ids'], ['String']);
        }

        if (data.hasOwnProperty('allowed_urls')) {
          obj['allowed_urls'] = _ApiClient["default"].convertToType(data['allowed_urls'], ['String']);
        }

        if (data.hasOwnProperty('allowed_urls_per_instance')) {
          obj['allowed_urls_per_instance'] = _ApiClient["default"].convertToType(data['allowed_urls_per_instance'], {
            'String': ['String']
          });
        }

        if (data.hasOwnProperty('bastion_ssh_port')) {
          obj['bastion_ssh_port'] = _ApiClient["default"].convertToType(data['bastion_ssh_port'], 'Number');
        }

        if (data.hasOwnProperty('bastion_urls_per_type')) {
          obj['bastion_urls_per_type'] = _ApiClient["default"].convertToType(data['bastion_urls_per_type'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('cluster_name')) {
          obj['cluster_name'] = _ApiClient["default"].convertToType(data['cluster_name'], 'String');
        }

        if (data.hasOwnProperty('display_name')) {
          obj['display_name'] = _ApiClient["default"].convertToType(data['display_name'], 'String');
        }

        if (data.hasOwnProperty('has_gateway_identity')) {
          obj['has_gateway_identity'] = _ApiClient["default"].convertToType(data['has_gateway_identity'], 'Boolean');
        }

        if (data.hasOwnProperty('last_report')) {
          obj['last_report'] = _ApiClient["default"].convertToType(data['last_report'], 'Date');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>BastionListEntry</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>BastionListEntry</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_id'] && !(typeof data['access_id'] === 'string' || data['access_id'] instanceof String)) {
        throw new Error("Expected the field `access_id` to be a primitive type in the JSON string but got " + data['access_id']);
      } // ensure the json data is an array


      if (!Array.isArray(data['allowed_access_ids'])) {
        throw new Error("Expected the field `allowed_access_ids` to be an array in the JSON data but got " + data['allowed_access_ids']);
      } // ensure the json data is an array


      if (!Array.isArray(data['allowed_urls'])) {
        throw new Error("Expected the field `allowed_urls` to be an array in the JSON data but got " + data['allowed_urls']);
      } // ensure the json data is a string


      if (data['cluster_name'] && !(typeof data['cluster_name'] === 'string' || data['cluster_name'] instanceof String)) {
        throw new Error("Expected the field `cluster_name` to be a primitive type in the JSON string but got " + data['cluster_name']);
      } // ensure the json data is a string


      if (data['display_name'] && !(typeof data['display_name'] === 'string' || data['display_name'] instanceof String)) {
        throw new Error("Expected the field `display_name` to be a primitive type in the JSON string but got " + data['display_name']);
      }

      return true;
    }
  }]);

  return BastionListEntry;
}();
/**
 * @member {String} access_id
 */


BastionListEntry.prototype['access_id'] = undefined;
/**
 * @member {Array.<String>} allowed_access_ids
 */

BastionListEntry.prototype['allowed_access_ids'] = undefined;
/**
 * @member {Array.<String>} allowed_urls
 */

BastionListEntry.prototype['allowed_urls'] = undefined;
/**
 * @member {Object.<String, Array.<String>>} allowed_urls_per_instance
 */

BastionListEntry.prototype['allowed_urls_per_instance'] = undefined;
/**
 * @member {Number} bastion_ssh_port
 */

BastionListEntry.prototype['bastion_ssh_port'] = undefined;
/**
 * @member {Object.<String, String>} bastion_urls_per_type
 */

BastionListEntry.prototype['bastion_urls_per_type'] = undefined;
/**
 * @member {String} cluster_name
 */

BastionListEntry.prototype['cluster_name'] = undefined;
/**
 * @member {String} display_name
 */

BastionListEntry.prototype['display_name'] = undefined;
/**
 * @member {Boolean} has_gateway_identity
 */

BastionListEntry.prototype['has_gateway_identity'] = undefined;
/**
 * @member {Date} last_report
 */

BastionListEntry.prototype['last_report'] = undefined;
var _default = BastionListEntry;
exports["default"] = _default;
