// akeyless/dist/model/BastionConfigReplyObj.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BastionGlobalConf = _interopRequireDefault(require("./BastionGlobalConf"));

var _SshBastionConf = _interopRequireDefault(require("./SshBastionConf"));

var _WebBastionConf = _interopRequireDefault(require("./WebBastionConf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BastionConfigReplyObj model module.
 * @module model/BastionConfigReplyObj
 * @version 5.0.1
 */
var BastionConfigReplyObj = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BastionConfigReplyObj</code>.
   * @alias module:model/BastionConfigReplyObj
   */
  function BastionConfigReplyObj() {
    _classCallCheck(this, BastionConfigReplyObj);

    BastionConfigReplyObj.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BastionConfigReplyObj, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BastionConfigReplyObj</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BastionConfigReplyObj} obj Optional instance to populate.
     * @return {module:model/BastionConfigReplyObj} The populated <code>BastionConfigReplyObj</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BastionConfigReplyObj();

        if (data.hasOwnProperty('api_gateway_url')) {
          obj['api_gateway_url'] = _ApiClient["default"].convertToType(data['api_gateway_url'], 'String');
        }

        if (data.hasOwnProperty('cluster_id')) {
          obj['cluster_id'] = _ApiClient["default"].convertToType(data['cluster_id'], 'String');
        }

        if (data.hasOwnProperty('gator_cluster_id')) {
          obj['gator_cluster_id'] = _ApiClient["default"].convertToType(data['gator_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('global')) {
          obj['global'] = _BastionGlobalConf["default"].constructFromObject(data['global']);
        }

        if (data.hasOwnProperty('ssh_bastion')) {
          obj['ssh_bastion'] = _SshBastionConf["default"].constructFromObject(data['ssh_bastion']);
        }

        if (data.hasOwnProperty('web_bastion')) {
          obj['web_bastion'] = _WebBastionConf["default"].constructFromObject(data['web_bastion']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>BastionConfigReplyObj</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>BastionConfigReplyObj</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['api_gateway_url'] && !(typeof data['api_gateway_url'] === 'string' || data['api_gateway_url'] instanceof String)) {
        throw new Error("Expected the field `api_gateway_url` to be a primitive type in the JSON string but got " + data['api_gateway_url']);
      } // ensure the json data is a string


      if (data['cluster_id'] && !(typeof data['cluster_id'] === 'string' || data['cluster_id'] instanceof String)) {
        throw new Error("Expected the field `cluster_id` to be a primitive type in the JSON string but got " + data['cluster_id']);
      } // validate the optional field `global`


      if (data['global']) {
        // data not null
        _BastionGlobalConf["default"].validateJSON(data['global']);
      } // validate the optional field `ssh_bastion`


      if (data['ssh_bastion']) {
        // data not null
        _SshBastionConf["default"].validateJSON(data['ssh_bastion']);
      } // validate the optional field `web_bastion`


      if (data['web_bastion']) {
        // data not null
        _WebBastionConf["default"].validateJSON(data['web_bastion']);
      }

      return true;
    }
  }]);

  return BastionConfigReplyObj;
}();
/**
 * @member {String} api_gateway_url
 */


BastionConfigReplyObj.prototype['api_gateway_url'] = undefined;
/**
 * @member {String} cluster_id
 */

BastionConfigReplyObj.prototype['cluster_id'] = undefined;
/**
 * @member {Number} gator_cluster_id
 */

BastionConfigReplyObj.prototype['gator_cluster_id'] = undefined;
/**
 * @member {module:model/BastionGlobalConf} global
 */

BastionConfigReplyObj.prototype['global'] = undefined;
/**
 * @member {module:model/SshBastionConf} ssh_bastion
 */

BastionConfigReplyObj.prototype['ssh_bastion'] = undefined;
/**
 * @member {module:model/WebBastionConf} web_bastion
 */

BastionConfigReplyObj.prototype['web_bastion'] = undefined;
var _default = BastionConfigReplyObj;
exports["default"] = _default;
