// akeyless/dist/model/BastionGlobalConf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BastionGlobalConf model module.
 * @module model/BastionGlobalConf
 * @version 5.0.5
 */
var BastionGlobalConf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BastionGlobalConf</code>.
   * @alias module:model/BastionGlobalConf
   */
  function BastionGlobalConf() {
    _classCallCheck(this, BastionGlobalConf);

    BastionGlobalConf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BastionGlobalConf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BastionGlobalConf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BastionGlobalConf} obj Optional instance to populate.
     * @return {module:model/BastionGlobalConf} The populated <code>BastionGlobalConf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BastionGlobalConf();

        if (data.hasOwnProperty('allowed_bastion_urls')) {
          obj['allowed_bastion_urls'] = _ApiClient["default"].convertToType(data['allowed_bastion_urls'], ['String']);
        }

        if (data.hasOwnProperty('legacy_signing_alg')) {
          obj['legacy_signing_alg'] = _ApiClient["default"].convertToType(data['legacy_signing_alg'], 'Boolean');
        }

        if (data.hasOwnProperty('rdp_username_sub_claim')) {
          obj['rdp_username_sub_claim'] = _ApiClient["default"].convertToType(data['rdp_username_sub_claim'], 'String');
        }

        if (data.hasOwnProperty('ssh_username_sub_claim')) {
          obj['ssh_username_sub_claim'] = _ApiClient["default"].convertToType(data['ssh_username_sub_claim'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>BastionGlobalConf</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>BastionGlobalConf</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['allowed_bastion_urls'])) {
        throw new Error("Expected the field `allowed_bastion_urls` to be an array in the JSON data but got " + data['allowed_bastion_urls']);
      } // ensure the json data is a string


      if (data['rdp_username_sub_claim'] && !(typeof data['rdp_username_sub_claim'] === 'string' || data['rdp_username_sub_claim'] instanceof String)) {
        throw new Error("Expected the field `rdp_username_sub_claim` to be a primitive type in the JSON string but got " + data['rdp_username_sub_claim']);
      } // ensure the json data is a string


      if (data['ssh_username_sub_claim'] && !(typeof data['ssh_username_sub_claim'] === 'string' || data['ssh_username_sub_claim'] instanceof String)) {
        throw new Error("Expected the field `ssh_username_sub_claim` to be a primitive type in the JSON string but got " + data['ssh_username_sub_claim']);
      }

      return true;
    }
  }]);

  return BastionGlobalConf;
}();
/**
 * @member {Array.<String>} allowed_bastion_urls
 */


BastionGlobalConf.prototype['allowed_bastion_urls'] = undefined;
/**
 * @member {Boolean} legacy_signing_alg
 */

BastionGlobalConf.prototype['legacy_signing_alg'] = undefined;
/**
 * @member {String} rdp_username_sub_claim
 */

BastionGlobalConf.prototype['rdp_username_sub_claim'] = undefined;
/**
 * @member {String} ssh_username_sub_claim
 */

BastionGlobalConf.prototype['ssh_username_sub_claim'] = undefined;
var _default = BastionGlobalConf;
exports["default"] = _default;
