// akeyless/dist/model/BatchEncryptionResponseLine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BatchEncryptionResponseLine model module.
 * @module model/BatchEncryptionResponseLine
 * @version 5.0.5
 */
var BatchEncryptionResponseLine = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BatchEncryptionResponseLine</code>.
   * @alias module:model/BatchEncryptionResponseLine
   */
  function BatchEncryptionResponseLine() {
    _classCallCheck(this, BatchEncryptionResponseLine);

    BatchEncryptionResponseLine.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BatchEncryptionResponseLine, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BatchEncryptionResponseLine</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchEncryptionResponseLine} obj Optional instance to populate.
     * @return {module:model/BatchEncryptionResponseLine} The populated <code>BatchEncryptionResponseLine</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchEncryptionResponseLine();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }

        if (data.hasOwnProperty('error')) {
          obj['error'] = _ApiClient["default"].convertToType(data['error'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>BatchEncryptionResponseLine</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>BatchEncryptionResponseLine</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['data'] && !(typeof data['data'] === 'string' || data['data'] instanceof String)) {
        throw new Error("Expected the field `data` to be a primitive type in the JSON string but got " + data['data']);
      } // ensure the json data is a string


      if (data['error'] && !(typeof data['error'] === 'string' || data['error'] instanceof String)) {
        throw new Error("Expected the field `error` to be a primitive type in the JSON string but got " + data['error']);
      }

      return true;
    }
  }]);

  return BatchEncryptionResponseLine;
}();
/**
 * @member {String} data
 */


BatchEncryptionResponseLine.prototype['data'] = undefined;
/**
 * @member {String} error
 */

BatchEncryptionResponseLine.prototype['error'] = undefined;
var _default = BatchEncryptionResponseLine;
exports["default"] = _default;
