// akeyless/dist/model/BatchEncryptionRequestLine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BatchEncryptionRequestLine model module.
 * @module model/BatchEncryptionRequestLine
 * @version 5.0.3
 */
var BatchEncryptionRequestLine = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BatchEncryptionRequestLine</code>.
   * @alias module:model/BatchEncryptionRequestLine
   */
  function BatchEncryptionRequestLine() {
    _classCallCheck(this, BatchEncryptionRequestLine);

    BatchEncryptionRequestLine.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BatchEncryptionRequestLine, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BatchEncryptionRequestLine</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchEncryptionRequestLine} obj Optional instance to populate.
     * @return {module:model/BatchEncryptionRequestLine} The populated <code>BatchEncryptionRequestLine</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchEncryptionRequestLine();

        if (data.hasOwnProperty('context')) {
          obj['context'] = _ApiClient["default"].convertToType(data['context'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }

        if (data.hasOwnProperty('item_id')) {
          obj['item_id'] = _ApiClient["default"].convertToType(data['item_id'], 'Number');
        }

        if (data.hasOwnProperty('item_version')) {
          obj['item_version'] = _ApiClient["default"].convertToType(data['item_version'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>BatchEncryptionRequestLine</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>BatchEncryptionRequestLine</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['data'] && !(typeof data['data'] === 'string' || data['data'] instanceof String)) {
        throw new Error("Expected the field `data` to be a primitive type in the JSON string but got " + data['data']);
      }

      return true;
    }
  }]);

  return BatchEncryptionRequestLine;
}();
/**
 * @member {Object.<String, String>} context
 */


BatchEncryptionRequestLine.prototype['context'] = undefined;
/**
 * @member {String} data
 */

BatchEncryptionRequestLine.prototype['data'] = undefined;
/**
 * @member {Number} item_id
 */

BatchEncryptionRequestLine.prototype['item_id'] = undefined;
/**
 * @member {Number} item_version
 */

BatchEncryptionRequestLine.prototype['item_version'] = undefined;
var _default = BatchEncryptionRequestLine;
exports["default"] = _default;
