// akeyless/dist/model/BatchTokenizationRequestLine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BatchTokenizationRequestLine model module.
 * @module model/BatchTokenizationRequestLine
 * @version 4.3.0
 */
var BatchTokenizationRequestLine = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BatchTokenizationRequestLine</code>.
   * @alias module:model/BatchTokenizationRequestLine
   */
  function BatchTokenizationRequestLine() {
    _classCallCheck(this, BatchTokenizationRequestLine);

    BatchTokenizationRequestLine.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BatchTokenizationRequestLine, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BatchTokenizationRequestLine</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchTokenizationRequestLine} obj Optional instance to populate.
     * @return {module:model/BatchTokenizationRequestLine} The populated <code>BatchTokenizationRequestLine</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchTokenizationRequestLine();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }

        if (data.hasOwnProperty('item_id')) {
          obj['item_id'] = _ApiClient["default"].convertToType(data['item_id'], 'Number');
        }

        if (data.hasOwnProperty('tweak')) {
          obj['tweak'] = _ApiClient["default"].convertToType(data['tweak'], 'String');
        }
      }

      return obj;
    }
  }]);

  return BatchTokenizationRequestLine;
}();
/**
 * @member {String} data
 */


BatchTokenizationRequestLine.prototype['data'] = undefined;
/**
 * @member {Number} item_id
 */

BatchTokenizationRequestLine.prototype['item_id'] = undefined;
/**
 * @member {String} tweak
 */

BatchTokenizationRequestLine.prototype['tweak'] = undefined;
var _default = BatchTokenizationRequestLine;
exports["default"] = _default;
