// akeyless/dist/model/CertificateMigration.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificatePayload = _interopRequireDefault(require("./CertificatePayload"));

var _MigrationGeneral = _interopRequireDefault(require("./MigrationGeneral"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateMigration model module.
 * @module model/CertificateMigration
 * @version 5.0.20
 */
var CertificateMigration = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateMigration</code>.
   * @alias module:model/CertificateMigration
   */
  function CertificateMigration() {
    _classCallCheck(this, CertificateMigration);

    CertificateMigration.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateMigration, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateMigration</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateMigration} obj Optional instance to populate.
     * @return {module:model/CertificateMigration} The populated <code>CertificateMigration</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateMigration();

        if (data.hasOwnProperty('general')) {
          obj['general'] = _MigrationGeneral["default"].constructFromObject(data['general']);
        }

        if (data.hasOwnProperty('payload')) {
          obj['payload'] = _CertificatePayload["default"].constructFromObject(data['payload']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CertificateMigration</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CertificateMigration</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `general`
      if (data['general']) {
        // data not null
        _MigrationGeneral["default"].validateJSON(data['general']);
      } // validate the optional field `payload`


      if (data['payload']) {
        // data not null
        _CertificatePayload["default"].validateJSON(data['payload']);
      }

      return true;
    }
  }]);

  return CertificateMigration;
}();
/**
 * @member {module:model/MigrationGeneral} general
 */


CertificateMigration.prototype['general'] = undefined;
/**
 * @member {module:model/CertificatePayload} payload
 */

CertificateMigration.prototype['payload'] = undefined;
var _default = CertificateMigration;
exports["default"] = _default;
