// akeyless/dist/model/CertificateStore.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateStore model module.
 * @module model/CertificateStore
 * @version 5.0.6
 */
var CertificateStore = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateStore</code>.
   * @alias module:model/CertificateStore
   */
  function CertificateStore() {
    _classCallCheck(this, CertificateStore);

    CertificateStore.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateStore, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateStore</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateStore} obj Optional instance to populate.
     * @return {module:model/CertificateStore} The populated <code>CertificateStore</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateStore();

        if (data.hasOwnProperty('certificate_pem')) {
          obj['certificate_pem'] = _ApiClient["default"].convertToType(data['certificate_pem'], 'String');
        }

        if (data.hasOwnProperty('common_name')) {
          obj['common_name'] = _ApiClient["default"].convertToType(data['common_name'], 'String');
        }

        if (data.hasOwnProperty('expiration_date')) {
          obj['expiration_date'] = _ApiClient["default"].convertToType(data['expiration_date'], 'Date');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CertificateStore</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CertificateStore</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['certificate_pem'] && !(typeof data['certificate_pem'] === 'string' || data['certificate_pem'] instanceof String)) {
        throw new Error("Expected the field `certificate_pem` to be a primitive type in the JSON string but got " + data['certificate_pem']);
      } // ensure the json data is a string


      if (data['common_name'] && !(typeof data['common_name'] === 'string' || data['common_name'] instanceof String)) {
        throw new Error("Expected the field `common_name` to be a primitive type in the JSON string but got " + data['common_name']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }

      return true;
    }
  }]);

  return CertificateStore;
}();
/**
 * @member {String} certificate_pem
 */


CertificateStore.prototype['certificate_pem'] = undefined;
/**
 * @member {String} common_name
 */

CertificateStore.prototype['common_name'] = undefined;
/**
 * @member {Date} expiration_date
 */

CertificateStore.prototype['expiration_date'] = undefined;
/**
 * @member {String} name
 */

CertificateStore.prototype['name'] = undefined;
var _default = CertificateStore;
exports["default"] = _default;
