// akeyless/dist/model/CertificateAnalyticAggregation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateAnalyticAggregation model module.
 * @module model/CertificateAnalyticAggregation
 * @version 4.2.1
 */
var CertificateAnalyticAggregation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateAnalyticAggregation</code>.
   * @alias module:model/CertificateAnalyticAggregation
   */
  function CertificateAnalyticAggregation() {
    _classCallCheck(this, CertificateAnalyticAggregation);

    CertificateAnalyticAggregation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateAnalyticAggregation, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateAnalyticAggregation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateAnalyticAggregation} obj Optional instance to populate.
     * @return {module:model/CertificateAnalyticAggregation} The populated <code>CertificateAnalyticAggregation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateAnalyticAggregation();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('ca_counts')) {
          obj['ca_counts'] = _ApiClient["default"].convertToType(data['ca_counts'], {
            'String': 'Number'
          });
        }

        if (data.hasOwnProperty('risk_counts')) {
          obj['risk_counts'] = _ApiClient["default"].convertToType(data['risk_counts'], {
            'String': 'Number'
          });
        }
      }

      return obj;
    }
  }]);

  return CertificateAnalyticAggregation;
}();
/**
 * @member {String} account_id
 */


CertificateAnalyticAggregation.prototype['account_id'] = undefined;
/**
 * @member {Object.<String, Number>} ca_counts
 */

CertificateAnalyticAggregation.prototype['ca_counts'] = undefined;
/**
 * @member {Object.<String, Number>} risk_counts
 */

CertificateAnalyticAggregation.prototype['risk_counts'] = undefined;
var _default = CertificateAnalyticAggregation;
exports["default"] = _default;
