// akeyless/dist/model/CertificatePayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateExpirationEvent = _interopRequireDefault(require("./CertificateExpirationEvent"));

var _CertificateScanTarget = _interopRequireDefault(require("./CertificateScanTarget"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificatePayload model module.
 * @module model/CertificatePayload
 * @version 5.0.20
 */
var CertificatePayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificatePayload</code>.
   * @alias module:model/CertificatePayload
   */
  function CertificatePayload() {
    _classCallCheck(this, CertificatePayload);

    CertificatePayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificatePayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificatePayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificatePayload} obj Optional instance to populate.
     * @return {module:model/CertificatePayload} The populated <code>CertificatePayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificatePayload();

        if (data.hasOwnProperty('expiration_events')) {
          obj['expiration_events'] = _ApiClient["default"].convertToType(data['expiration_events'], [_CertificateExpirationEvent["default"]]);
        }

        if (data.hasOwnProperty('folder')) {
          obj['folder'] = _ApiClient["default"].convertToType(data['folder'], 'String');
        }

        if (data.hasOwnProperty('max_dial_timeout')) {
          obj['max_dial_timeout'] = _ApiClient["default"].convertToType(data['max_dial_timeout'], 'Number');
        }

        if (data.hasOwnProperty('max_scan_duration')) {
          obj['max_scan_duration'] = _ApiClient["default"].convertToType(data['max_scan_duration'], 'Number');
        }

        if (data.hasOwnProperty('max_workers')) {
          obj['max_workers'] = _ApiClient["default"].convertToType(data['max_workers'], 'Number');
        }

        if (data.hasOwnProperty('port_ranges')) {
          obj['port_ranges'] = _ApiClient["default"].convertToType(data['port_ranges'], 'String');
        }

        if (data.hasOwnProperty('targets')) {
          obj['targets'] = _ApiClient["default"].convertToType(data['targets'], [_CertificateScanTarget["default"]]);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CertificatePayload</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CertificatePayload</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['expiration_events']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['expiration_events'])) {
          throw new Error("Expected the field `expiration_events` to be an array in the JSON data but got " + data['expiration_events']);
        } // validate the optional field `expiration_events` (array)


        var _iterator = _createForOfIteratorHelper(data['expiration_events']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _CertificateExpirationEvent["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is a string


      if (data['folder'] && !(typeof data['folder'] === 'string' || data['folder'] instanceof String)) {
        throw new Error("Expected the field `folder` to be a primitive type in the JSON string but got " + data['folder']);
      } // ensure the json data is a string


      if (data['port_ranges'] && !(typeof data['port_ranges'] === 'string' || data['port_ranges'] instanceof String)) {
        throw new Error("Expected the field `port_ranges` to be a primitive type in the JSON string but got " + data['port_ranges']);
      }

      if (data['targets']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['targets'])) {
          throw new Error("Expected the field `targets` to be an array in the JSON data but got " + data['targets']);
        } // validate the optional field `targets` (array)


        var _iterator2 = _createForOfIteratorHelper(data['targets']),
            _step2;

        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;

            _CertificateScanTarget["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }

        ;
      }

      return true;
    }
  }]);

  return CertificatePayload;
}();
/**
 * @member {Array.<module:model/CertificateExpirationEvent>} expiration_events
 */


CertificatePayload.prototype['expiration_events'] = undefined;
/**
 * @member {String} folder
 */

CertificatePayload.prototype['folder'] = undefined;
/**
 * @member {Number} max_dial_timeout
 */

CertificatePayload.prototype['max_dial_timeout'] = undefined;
/**
 * @member {Number} max_scan_duration
 */

CertificatePayload.prototype['max_scan_duration'] = undefined;
/**
 * @member {Number} max_workers
 */

CertificatePayload.prototype['max_workers'] = undefined;
/**
 * @member {String} port_ranges
 */

CertificatePayload.prototype['port_ranges'] = undefined;
/**
 * @member {Array.<module:model/CertificateScanTarget>} targets
 */

CertificatePayload.prototype['targets'] = undefined;
var _default = CertificatePayload;
exports["default"] = _default;
