// akeyless/dist/model/CertificateExpirationEventsSettings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateExpirationEvent = _interopRequireDefault(require("./CertificateExpirationEvent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateExpirationEventsSettings model module.
 * @module model/CertificateExpirationEventsSettings
 * @version 5.0.14
 */
var CertificateExpirationEventsSettings = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateExpirationEventsSettings</code>.
   * @alias module:model/CertificateExpirationEventsSettings
   */
  function CertificateExpirationEventsSettings() {
    _classCallCheck(this, CertificateExpirationEventsSettings);

    CertificateExpirationEventsSettings.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateExpirationEventsSettings, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateExpirationEventsSettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateExpirationEventsSettings} obj Optional instance to populate.
     * @return {module:model/CertificateExpirationEventsSettings} The populated <code>CertificateExpirationEventsSettings</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateExpirationEventsSettings();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'Boolean');
        }

        if (data.hasOwnProperty('expirations_list')) {
          obj['expirations_list'] = _ApiClient["default"].convertToType(data['expirations_list'], [_CertificateExpirationEvent["default"]]);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CertificateExpirationEventsSettings</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CertificateExpirationEventsSettings</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['expirations_list']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['expirations_list'])) {
          throw new Error("Expected the field `expirations_list` to be an array in the JSON data but got " + data['expirations_list']);
        } // validate the optional field `expirations_list` (array)


        var _iterator = _createForOfIteratorHelper(data['expirations_list']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _CertificateExpirationEvent["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      }

      return true;
    }
  }]);

  return CertificateExpirationEventsSettings;
}();
/**
 * @member {Boolean} enable
 */


CertificateExpirationEventsSettings.prototype['enable'] = undefined;
/**
 * ExpirationEventsList is the default expiration events for the account
 * @member {Array.<module:model/CertificateExpirationEvent>} expirations_list
 */

CertificateExpirationEventsSettings.prototype['expirations_list'] = undefined;
var _default = CertificateExpirationEventsSettings;
exports["default"] = _default;
