// akeyless/dist/model/CertificateDiscovery.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateDiscovery model module.
 * @module model/CertificateDiscovery
 * @version 5.0.20
 */
var CertificateDiscovery = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateDiscovery</code>.
   * CertificateDiscovery is a command that discovery certificates
   * @alias module:model/CertificateDiscovery
   * @param hosts {String} A comma separated list of IPs, CIDR ranges, or DNS names to discovery
   * @param targetLocation {String} The folder where the results will be saved
   */
  function CertificateDiscovery(hosts, targetLocation) {
    _classCallCheck(this, CertificateDiscovery);

    CertificateDiscovery.initialize(this, hosts, targetLocation);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateDiscovery, null, [{
    key: "initialize",
    value: function initialize(obj, hosts, targetLocation) {
      obj['debug'] = false;
      obj['hosts'] = hosts;
      obj['json'] = false;
      obj['port-ranges'] = '443';
      obj['target-location'] = targetLocation;
    }
    /**
     * Constructs a <code>CertificateDiscovery</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateDiscovery} obj Optional instance to populate.
     * @return {module:model/CertificateDiscovery} The populated <code>CertificateDiscovery</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateDiscovery();

        if (data.hasOwnProperty('debug')) {
          obj['debug'] = _ApiClient["default"].convertToType(data['debug'], 'Boolean');
        }

        if (data.hasOwnProperty('expiration-event-in')) {
          obj['expiration-event-in'] = _ApiClient["default"].convertToType(data['expiration-event-in'], ['String']);
        }

        if (data.hasOwnProperty('hosts')) {
          obj['hosts'] = _ApiClient["default"].convertToType(data['hosts'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('port-ranges')) {
          obj['port-ranges'] = _ApiClient["default"].convertToType(data['port-ranges'], 'String');
        }

        if (data.hasOwnProperty('protection-key')) {
          obj['protection-key'] = _ApiClient["default"].convertToType(data['protection-key'], 'String');
        }

        if (data.hasOwnProperty('target-location')) {
          obj['target-location'] = _ApiClient["default"].convertToType(data['target-location'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CertificateDiscovery</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CertificateDiscovery</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(CertificateDiscovery.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is an array

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (!Array.isArray(data['expiration-event-in'])) {
        throw new Error("Expected the field `expiration-event-in` to be an array in the JSON data but got " + data['expiration-event-in']);
      } // ensure the json data is a string


      if (data['hosts'] && !(typeof data['hosts'] === 'string' || data['hosts'] instanceof String)) {
        throw new Error("Expected the field `hosts` to be a primitive type in the JSON string but got " + data['hosts']);
      } // ensure the json data is a string


      if (data['port-ranges'] && !(typeof data['port-ranges'] === 'string' || data['port-ranges'] instanceof String)) {
        throw new Error("Expected the field `port-ranges` to be a primitive type in the JSON string but got " + data['port-ranges']);
      } // ensure the json data is a string


      if (data['protection-key'] && !(typeof data['protection-key'] === 'string' || data['protection-key'] instanceof String)) {
        throw new Error("Expected the field `protection-key` to be a primitive type in the JSON string but got " + data['protection-key']);
      } // ensure the json data is a string


      if (data['target-location'] && !(typeof data['target-location'] === 'string' || data['target-location'] instanceof String)) {
        throw new Error("Expected the field `target-location` to be a primitive type in the JSON string but got " + data['target-location']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return CertificateDiscovery;
}();

CertificateDiscovery.RequiredProperties = ["hosts", "target-location"];
/**
 * Debug mode
 * @member {Boolean} debug
 * @default false
 */

CertificateDiscovery.prototype['debug'] = false;
/**
 * How many days before the expiration of the certificate would you like to be notified.
 * @member {Array.<String>} expiration-event-in
 */

CertificateDiscovery.prototype['expiration-event-in'] = undefined;
/**
 * A comma separated list of IPs, CIDR ranges, or DNS names to discovery
 * @member {String} hosts
 */

CertificateDiscovery.prototype['hosts'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CertificateDiscovery.prototype['json'] = false;
/**
 * A comma separated list of port ranges Examples: \"80,443\" or \"80,443,8080-8090\" or \"443\"
 * @member {String} port-ranges
 * @default '443'
 */

CertificateDiscovery.prototype['port-ranges'] = '443';
/**
 * The name of the key that protects the certificate value
 * @member {String} protection-key
 */

CertificateDiscovery.prototype['protection-key'] = undefined;
/**
 * The folder where the results will be saved
 * @member {String} target-location
 */

CertificateDiscovery.prototype['target-location'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CertificateDiscovery.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CertificateDiscovery.prototype['uid-token'] = undefined;
var _default = CertificateDiscovery;
exports["default"] = _default;
