// akeyless/dist/model/CertificateTemplateInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateTemplateInfo model module.
 * @module model/CertificateTemplateInfo
 * @version 5.0.5
 */
var CertificateTemplateInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateTemplateInfo</code>.
   * @alias module:model/CertificateTemplateInfo
   */
  function CertificateTemplateInfo() {
    _classCallCheck(this, CertificateTemplateInfo);

    CertificateTemplateInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateTemplateInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateTemplateInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateTemplateInfo} obj Optional instance to populate.
     * @return {module:model/CertificateTemplateInfo} The populated <code>CertificateTemplateInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateTemplateInfo();

        if (data.hasOwnProperty('common_name')) {
          obj['common_name'] = _ApiClient["default"].convertToType(data['common_name'], 'String');
        }

        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }

        if (data.hasOwnProperty('csr_cnf_base_64')) {
          obj['csr_cnf_base_64'] = _ApiClient["default"].convertToType(data['csr_cnf_base_64'], 'String');
        }

        if (data.hasOwnProperty('digest_algo')) {
          obj['digest_algo'] = _ApiClient["default"].convertToType(data['digest_algo'], 'String');
        }

        if (data.hasOwnProperty('locality')) {
          obj['locality'] = _ApiClient["default"].convertToType(data['locality'], 'String');
        }

        if (data.hasOwnProperty('organization')) {
          obj['organization'] = _ApiClient["default"].convertToType(data['organization'], 'String');
        }

        if (data.hasOwnProperty('province')) {
          obj['province'] = _ApiClient["default"].convertToType(data['province'], 'String');
        }

        if (data.hasOwnProperty('self_signed_enabled')) {
          obj['self_signed_enabled'] = _ApiClient["default"].convertToType(data['self_signed_enabled'], 'Boolean');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CertificateTemplateInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CertificateTemplateInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['common_name'] && !(typeof data['common_name'] === 'string' || data['common_name'] instanceof String)) {
        throw new Error("Expected the field `common_name` to be a primitive type in the JSON string but got " + data['common_name']);
      } // ensure the json data is a string


      if (data['country'] && !(typeof data['country'] === 'string' || data['country'] instanceof String)) {
        throw new Error("Expected the field `country` to be a primitive type in the JSON string but got " + data['country']);
      } // ensure the json data is a string


      if (data['csr_cnf_base_64'] && !(typeof data['csr_cnf_base_64'] === 'string' || data['csr_cnf_base_64'] instanceof String)) {
        throw new Error("Expected the field `csr_cnf_base_64` to be a primitive type in the JSON string but got " + data['csr_cnf_base_64']);
      } // ensure the json data is a string


      if (data['digest_algo'] && !(typeof data['digest_algo'] === 'string' || data['digest_algo'] instanceof String)) {
        throw new Error("Expected the field `digest_algo` to be a primitive type in the JSON string but got " + data['digest_algo']);
      } // ensure the json data is a string


      if (data['locality'] && !(typeof data['locality'] === 'string' || data['locality'] instanceof String)) {
        throw new Error("Expected the field `locality` to be a primitive type in the JSON string but got " + data['locality']);
      } // ensure the json data is a string


      if (data['organization'] && !(typeof data['organization'] === 'string' || data['organization'] instanceof String)) {
        throw new Error("Expected the field `organization` to be a primitive type in the JSON string but got " + data['organization']);
      } // ensure the json data is a string


      if (data['province'] && !(typeof data['province'] === 'string' || data['province'] instanceof String)) {
        throw new Error("Expected the field `province` to be a primitive type in the JSON string but got " + data['province']);
      }

      return true;
    }
  }]);

  return CertificateTemplateInfo;
}();
/**
 * @member {String} common_name
 */


CertificateTemplateInfo.prototype['common_name'] = undefined;
/**
 * @member {String} country
 */

CertificateTemplateInfo.prototype['country'] = undefined;
/**
 * @member {String} csr_cnf_base_64
 */

CertificateTemplateInfo.prototype['csr_cnf_base_64'] = undefined;
/**
 * @member {String} digest_algo
 */

CertificateTemplateInfo.prototype['digest_algo'] = undefined;
/**
 * @member {String} locality
 */

CertificateTemplateInfo.prototype['locality'] = undefined;
/**
 * @member {String} organization
 */

CertificateTemplateInfo.prototype['organization'] = undefined;
/**
 * @member {String} province
 */

CertificateTemplateInfo.prototype['province'] = undefined;
/**
 * @member {Boolean} self_signed_enabled
 */

CertificateTemplateInfo.prototype['self_signed_enabled'] = undefined;
/**
 * @member {Number} ttl
 */

CertificateTemplateInfo.prototype['ttl'] = undefined;
var _default = CertificateTemplateInfo;
exports["default"] = _default;
