// akeyless/dist/model/CertificateScanTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateScanTarget model module.
 * @module model/CertificateScanTarget
 * @version 5.0.20
 */
var CertificateScanTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateScanTarget</code>.
   * @alias module:model/CertificateScanTarget
   */
  function CertificateScanTarget() {
    _classCallCheck(this, CertificateScanTarget);

    CertificateScanTarget.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateScanTarget, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateScanTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateScanTarget} obj Optional instance to populate.
     * @return {module:model/CertificateScanTarget} The populated <code>CertificateScanTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateScanTarget();

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('portRanges')) {
          obj['portRanges'] = _ApiClient["default"].convertToType(data['portRanges'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CertificateScanTarget</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CertificateScanTarget</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['host'] && !(typeof data['host'] === 'string' || data['host'] instanceof String)) {
        throw new Error("Expected the field `host` to be a primitive type in the JSON string but got " + data['host']);
      } // ensure the json data is a string


      if (data['portRanges'] && !(typeof data['portRanges'] === 'string' || data['portRanges'] instanceof String)) {
        throw new Error("Expected the field `portRanges` to be a primitive type in the JSON string but got " + data['portRanges']);
      }

      return true;
    }
  }]);

  return CertificateScanTarget;
}();
/**
 * @member {String} host
 */


CertificateScanTarget.prototype['host'] = undefined;
/**
 * @member {String} portRanges
 */

CertificateScanTarget.prototype['portRanges'] = undefined;
var _default = CertificateScanTarget;
exports["default"] = _default;
