// akeyless/dist/model/CfInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CfInfo model module.
 * @module model/CfInfo
 * @version 5.0.14
 */
var CfInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CfInfo</code>.
   * @alias module:model/CfInfo
   */
  function CfInfo() {
    _classCallCheck(this, CfInfo);

    CfInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CfInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CfInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CfInfo} obj Optional instance to populate.
     * @return {module:model/CfInfo} The populated <code>CfInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CfInfo();

        if (data.hasOwnProperty('cf_name')) {
          obj['cf_name'] = _ApiClient["default"].convertToType(data['cf_name'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('hash')) {
          obj['hash'] = _ApiClient["default"].convertToType(data['hash'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CfInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CfInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['cf_name'] && !(typeof data['cf_name'] === 'string' || data['cf_name'] instanceof String)) {
        throw new Error("Expected the field `cf_name` to be a primitive type in the JSON string but got " + data['cf_name']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['hash'] && !(typeof data['hash'] === 'string' || data['hash'] instanceof String)) {
        throw new Error("Expected the field `hash` to be a primitive type in the JSON string but got " + data['hash']);
      } // ensure the json data is a string


      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }

      return true;
    }
  }]);

  return CfInfo;
}();
/**
 * @member {String} cf_name
 */


CfInfo.prototype['cf_name'] = undefined;
/**
 * @member {String} description
 */

CfInfo.prototype['description'] = undefined;
/**
 * @member {String} hash
 */

CfInfo.prototype['hash'] = undefined;
/**
 * @member {String} id
 */

CfInfo.prototype['id'] = undefined;
var _default = CfInfo;
exports["default"] = _default;
