// akeyless/dist/model/ClientsUsageReport.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ClientUsageInfo = _interopRequireDefault(require("./ClientUsageInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClientsUsageReport model module.
 * @module model/ClientsUsageReport
 * @version 5.0.5
 */
var ClientsUsageReport = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClientsUsageReport</code>.
   * @alias module:model/ClientsUsageReport
   */
  function ClientsUsageReport() {
    _classCallCheck(this, ClientsUsageReport);

    ClientsUsageReport.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClientsUsageReport, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClientsUsageReport</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ClientsUsageReport} obj Optional instance to populate.
     * @return {module:model/ClientsUsageReport} The populated <code>ClientsUsageReport</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClientsUsageReport();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('clients')) {
          obj['clients'] = _ApiClient["default"].convertToType(data['clients'], [_ClientUsageInfo["default"]]);
        }

        if (data.hasOwnProperty('product')) {
          obj['product'] = _ApiClient["default"].convertToType(data['product'], 'String');
        }

        if (data.hasOwnProperty('time')) {
          obj['time'] = _ApiClient["default"].convertToType(data['time'], 'Date');
        }

        if (data.hasOwnProperty('total_clients')) {
          obj['total_clients'] = _ApiClient["default"].convertToType(data['total_clients'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ClientsUsageReport</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ClientsUsageReport</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['account_id'] && !(typeof data['account_id'] === 'string' || data['account_id'] instanceof String)) {
        throw new Error("Expected the field `account_id` to be a primitive type in the JSON string but got " + data['account_id']);
      }

      if (data['clients']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['clients'])) {
          throw new Error("Expected the field `clients` to be an array in the JSON data but got " + data['clients']);
        } // validate the optional field `clients` (array)


        var _iterator = _createForOfIteratorHelper(data['clients']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _ClientUsageInfo["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is a string


      if (data['product'] && !(typeof data['product'] === 'string' || data['product'] instanceof String)) {
        throw new Error("Expected the field `product` to be a primitive type in the JSON string but got " + data['product']);
      }

      return true;
    }
  }]);

  return ClientsUsageReport;
}();
/**
 * @member {String} account_id
 */


ClientsUsageReport.prototype['account_id'] = undefined;
/**
 * @member {Array.<module:model/ClientUsageInfo>} clients
 */

ClientsUsageReport.prototype['clients'] = undefined;
/**
 * @member {String} product
 */

ClientsUsageReport.prototype['product'] = undefined;
/**
 * @member {Date} time
 */

ClientsUsageReport.prototype['time'] = undefined;
/**
 * @member {Number} total_clients
 */

ClientsUsageReport.prototype['total_clients'] = undefined;
var _default = ClientsUsageReport;
exports["default"] = _default;
