// akeyless/dist/model/ClientUsageInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClientUsageInfo model module.
 * @module model/ClientUsageInfo
 * @version 5.0.0
 */
var ClientUsageInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClientUsageInfo</code>.
   * @alias module:model/ClientUsageInfo
   */
  function ClientUsageInfo() {
    _classCallCheck(this, ClientUsageInfo);

    ClientUsageInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClientUsageInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClientUsageInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ClientUsageInfo} obj Optional instance to populate.
     * @return {module:model/ClientUsageInfo} The populated <code>ClientUsageInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClientUsageInfo();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('access_type')) {
          obj['access_type'] = _ApiClient["default"].convertToType(data['access_type'], 'String');
        }

        if (data.hasOwnProperty('auth_method_name')) {
          obj['auth_method_name'] = _ApiClient["default"].convertToType(data['auth_method_name'], 'String');
        }

        if (data.hasOwnProperty('client_unique_id')) {
          obj['client_unique_id'] = _ApiClient["default"].convertToType(data['client_unique_id'], 'String');
        }

        if (data.hasOwnProperty('exceeded_clients')) {
          obj['exceeded_clients'] = _ApiClient["default"].convertToType(data['exceeded_clients'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ClientUsageInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ClientUsageInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_id'] && !(typeof data['access_id'] === 'string' || data['access_id'] instanceof String)) {
        throw new Error("Expected the field `access_id` to be a primitive type in the JSON string but got " + data['access_id']);
      } // ensure the json data is a string


      if (data['access_type'] && !(typeof data['access_type'] === 'string' || data['access_type'] instanceof String)) {
        throw new Error("Expected the field `access_type` to be a primitive type in the JSON string but got " + data['access_type']);
      } // ensure the json data is a string


      if (data['auth_method_name'] && !(typeof data['auth_method_name'] === 'string' || data['auth_method_name'] instanceof String)) {
        throw new Error("Expected the field `auth_method_name` to be a primitive type in the JSON string but got " + data['auth_method_name']);
      } // ensure the json data is a string


      if (data['client_unique_id'] && !(typeof data['client_unique_id'] === 'string' || data['client_unique_id'] instanceof String)) {
        throw new Error("Expected the field `client_unique_id` to be a primitive type in the JSON string but got " + data['client_unique_id']);
      }

      return true;
    }
  }]);

  return ClientUsageInfo;
}();
/**
 * @member {String} access_id
 */


ClientUsageInfo.prototype['access_id'] = undefined;
/**
 * @member {String} access_type
 */

ClientUsageInfo.prototype['access_type'] = undefined;
/**
 * @member {String} auth_method_name
 */

ClientUsageInfo.prototype['auth_method_name'] = undefined;
/**
 * @member {String} client_unique_id
 */

ClientUsageInfo.prototype['client_unique_id'] = undefined;
/**
 * @member {Number} exceeded_clients
 */

ClientUsageInfo.prototype['exceeded_clients'] = undefined;
var _default = ClientUsageInfo;
exports["default"] = _default;
