// akeyless/dist/model/ClientsUsageReport.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ClientUsageInfo = _interopRequireDefault(require("./ClientUsageInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClientsUsageReport model module.
 * @module model/ClientsUsageReport
 * @version 4.1.0
 */
var ClientsUsageReport = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClientsUsageReport</code>.
   * @alias module:model/ClientsUsageReport
   */
  function ClientsUsageReport() {
    _classCallCheck(this, ClientsUsageReport);

    ClientsUsageReport.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClientsUsageReport, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClientsUsageReport</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ClientsUsageReport} obj Optional instance to populate.
     * @return {module:model/ClientsUsageReport} The populated <code>ClientsUsageReport</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClientsUsageReport();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('clients')) {
          obj['clients'] = _ApiClient["default"].convertToType(data['clients'], [_ClientUsageInfo["default"]]);
        }

        if (data.hasOwnProperty('product')) {
          obj['product'] = _ApiClient["default"].convertToType(data['product'], 'String');
        }

        if (data.hasOwnProperty('time')) {
          obj['time'] = _ApiClient["default"].convertToType(data['time'], 'Date');
        }

        if (data.hasOwnProperty('total_clients')) {
          obj['total_clients'] = _ApiClient["default"].convertToType(data['total_clients'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ClientsUsageReport;
}();
/**
 * @member {String} account_id
 */


ClientsUsageReport.prototype['account_id'] = undefined;
/**
 * @member {Array.<module:model/ClientUsageInfo>} clients
 */

ClientsUsageReport.prototype['clients'] = undefined;
/**
 * @member {String} product
 */

ClientsUsageReport.prototype['product'] = undefined;
/**
 * @member {Date} time
 */

ClientsUsageReport.prototype['time'] = undefined;
/**
 * @member {Number} total_clients
 */

ClientsUsageReport.prototype['total_clients'] = undefined;
var _default = ClientsUsageReport;
exports["default"] = _default;
