// akeyless/dist/model/ConfigHash.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConfigHash model module.
 * @module model/ConfigHash
 * @version 5.0.5
 */
var ConfigHash = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConfigHash</code>.
   * @alias module:model/ConfigHash
   */
  function ConfigHash() {
    _classCallCheck(this, ConfigHash);

    ConfigHash.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConfigHash, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConfigHash</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ConfigHash} obj Optional instance to populate.
     * @return {module:model/ConfigHash} The populated <code>ConfigHash</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConfigHash();

        if (data.hasOwnProperty('admins')) {
          obj['admins'] = _ApiClient["default"].convertToType(data['admins'], 'String');
        }

        if (data.hasOwnProperty('cache')) {
          obj['cache'] = _ApiClient["default"].convertToType(data['cache'], 'String');
        }

        if (data.hasOwnProperty('customer_fragements')) {
          obj['customer_fragements'] = _ApiClient["default"].convertToType(data['customer_fragements'], 'String');
        }

        if (data.hasOwnProperty('general')) {
          obj['general'] = _ApiClient["default"].convertToType(data['general'], 'String');
        }

        if (data.hasOwnProperty('k8s_auths')) {
          obj['k8s_auths'] = _ApiClient["default"].convertToType(data['k8s_auths'], 'String');
        }

        if (data.hasOwnProperty('kmip')) {
          obj['kmip'] = _ApiClient["default"].convertToType(data['kmip'], 'String');
        }

        if (data.hasOwnProperty('ldap')) {
          obj['ldap'] = _ApiClient["default"].convertToType(data['ldap'], 'String');
        }

        if (data.hasOwnProperty('leadership')) {
          obj['leadership'] = _ApiClient["default"].convertToType(data['leadership'], 'String');
        }

        if (data.hasOwnProperty('log_forwarding')) {
          obj['log_forwarding'] = _ApiClient["default"].convertToType(data['log_forwarding'], 'String');
        }

        if (data.hasOwnProperty('m_queue')) {
          obj['m_queue'] = _ApiClient["default"].convertToType(data['m_queue'], 'String');
        }

        if (data.hasOwnProperty('migration_status')) {
          obj['migration_status'] = _ApiClient["default"].convertToType(data['migration_status'], 'String');
        }

        if (data.hasOwnProperty('migrations')) {
          obj['migrations'] = _ApiClient["default"].convertToType(data['migrations'], 'String');
        }

        if (data.hasOwnProperty('producers')) {
          obj['producers'] = _ApiClient["default"].convertToType(data['producers'], Object);
        }

        if (data.hasOwnProperty('producers_status')) {
          obj['producers_status'] = _ApiClient["default"].convertToType(data['producers_status'], 'String');
        }

        if (data.hasOwnProperty('rotators')) {
          obj['rotators'] = _ApiClient["default"].convertToType(data['rotators'], Object);
        }

        if (data.hasOwnProperty('saml')) {
          obj['saml'] = _ApiClient["default"].convertToType(data['saml'], 'String');
        }

        if (data.hasOwnProperty('universal_identity')) {
          obj['universal_identity'] = _ApiClient["default"].convertToType(data['universal_identity'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ConfigHash</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ConfigHash</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['admins'] && !(typeof data['admins'] === 'string' || data['admins'] instanceof String)) {
        throw new Error("Expected the field `admins` to be a primitive type in the JSON string but got " + data['admins']);
      } // ensure the json data is a string


      if (data['cache'] && !(typeof data['cache'] === 'string' || data['cache'] instanceof String)) {
        throw new Error("Expected the field `cache` to be a primitive type in the JSON string but got " + data['cache']);
      } // ensure the json data is a string


      if (data['customer_fragements'] && !(typeof data['customer_fragements'] === 'string' || data['customer_fragements'] instanceof String)) {
        throw new Error("Expected the field `customer_fragements` to be a primitive type in the JSON string but got " + data['customer_fragements']);
      } // ensure the json data is a string


      if (data['general'] && !(typeof data['general'] === 'string' || data['general'] instanceof String)) {
        throw new Error("Expected the field `general` to be a primitive type in the JSON string but got " + data['general']);
      } // ensure the json data is a string


      if (data['k8s_auths'] && !(typeof data['k8s_auths'] === 'string' || data['k8s_auths'] instanceof String)) {
        throw new Error("Expected the field `k8s_auths` to be a primitive type in the JSON string but got " + data['k8s_auths']);
      } // ensure the json data is a string


      if (data['kmip'] && !(typeof data['kmip'] === 'string' || data['kmip'] instanceof String)) {
        throw new Error("Expected the field `kmip` to be a primitive type in the JSON string but got " + data['kmip']);
      } // ensure the json data is a string


      if (data['ldap'] && !(typeof data['ldap'] === 'string' || data['ldap'] instanceof String)) {
        throw new Error("Expected the field `ldap` to be a primitive type in the JSON string but got " + data['ldap']);
      } // ensure the json data is a string


      if (data['leadership'] && !(typeof data['leadership'] === 'string' || data['leadership'] instanceof String)) {
        throw new Error("Expected the field `leadership` to be a primitive type in the JSON string but got " + data['leadership']);
      } // ensure the json data is a string


      if (data['log_forwarding'] && !(typeof data['log_forwarding'] === 'string' || data['log_forwarding'] instanceof String)) {
        throw new Error("Expected the field `log_forwarding` to be a primitive type in the JSON string but got " + data['log_forwarding']);
      } // ensure the json data is a string


      if (data['m_queue'] && !(typeof data['m_queue'] === 'string' || data['m_queue'] instanceof String)) {
        throw new Error("Expected the field `m_queue` to be a primitive type in the JSON string but got " + data['m_queue']);
      } // ensure the json data is a string


      if (data['migration_status'] && !(typeof data['migration_status'] === 'string' || data['migration_status'] instanceof String)) {
        throw new Error("Expected the field `migration_status` to be a primitive type in the JSON string but got " + data['migration_status']);
      } // ensure the json data is a string


      if (data['migrations'] && !(typeof data['migrations'] === 'string' || data['migrations'] instanceof String)) {
        throw new Error("Expected the field `migrations` to be a primitive type in the JSON string but got " + data['migrations']);
      } // ensure the json data is a string


      if (data['producers_status'] && !(typeof data['producers_status'] === 'string' || data['producers_status'] instanceof String)) {
        throw new Error("Expected the field `producers_status` to be a primitive type in the JSON string but got " + data['producers_status']);
      } // ensure the json data is a string


      if (data['saml'] && !(typeof data['saml'] === 'string' || data['saml'] instanceof String)) {
        throw new Error("Expected the field `saml` to be a primitive type in the JSON string but got " + data['saml']);
      } // ensure the json data is a string


      if (data['universal_identity'] && !(typeof data['universal_identity'] === 'string' || data['universal_identity'] instanceof String)) {
        throw new Error("Expected the field `universal_identity` to be a primitive type in the JSON string but got " + data['universal_identity']);
      }

      return true;
    }
  }]);

  return ConfigHash;
}();
/**
 * @member {String} admins
 */


ConfigHash.prototype['admins'] = undefined;
/**
 * @member {String} cache
 */

ConfigHash.prototype['cache'] = undefined;
/**
 * @member {String} customer_fragements
 */

ConfigHash.prototype['customer_fragements'] = undefined;
/**
 * @member {String} general
 */

ConfigHash.prototype['general'] = undefined;
/**
 * @member {String} k8s_auths
 */

ConfigHash.prototype['k8s_auths'] = undefined;
/**
 * @member {String} kmip
 */

ConfigHash.prototype['kmip'] = undefined;
/**
 * @member {String} ldap
 */

ConfigHash.prototype['ldap'] = undefined;
/**
 * @member {String} leadership
 */

ConfigHash.prototype['leadership'] = undefined;
/**
 * @member {String} log_forwarding
 */

ConfigHash.prototype['log_forwarding'] = undefined;
/**
 * @member {String} m_queue
 */

ConfigHash.prototype['m_queue'] = undefined;
/**
 * @member {String} migration_status
 */

ConfigHash.prototype['migration_status'] = undefined;
/**
 * @member {String} migrations
 */

ConfigHash.prototype['migrations'] = undefined;
/**
 * @member {Object} producers
 */

ConfigHash.prototype['producers'] = undefined;
/**
 * @member {String} producers_status
 */

ConfigHash.prototype['producers_status'] = undefined;
/**
 * @member {Object} rotators
 */

ConfigHash.prototype['rotators'] = undefined;
/**
 * @member {String} saml
 */

ConfigHash.prototype['saml'] = undefined;
/**
 * @member {String} universal_identity
 */

ConfigHash.prototype['universal_identity'] = undefined;
var _default = ConfigHash;
exports["default"] = _default;
