// akeyless/dist/model/CreateOidcApp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateOidcApp model module.
 * @module model/CreateOidcApp
 * @version 5.0.14
 */
var CreateOidcApp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateOidcApp</code>.
   * @alias module:model/CreateOidcApp
   * @param name {String} OIDC application name
   */
  function CreateOidcApp(name) {
    _classCallCheck(this, CreateOidcApp);

    CreateOidcApp.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateOidcApp, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['accessibility'] = 'regular';
      obj['json'] = false;
      obj['name'] = name;
      obj['scopes'] = 'openid';
    }
    /**
     * Constructs a <code>CreateOidcApp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateOidcApp} obj Optional instance to populate.
     * @return {module:model/CreateOidcApp} The populated <code>CreateOidcApp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateOidcApp();

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'String');
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('item-custom-fields')) {
          obj['item-custom-fields'] = _ApiClient["default"].convertToType(data['item-custom-fields'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('permission-assignment')) {
          obj['permission-assignment'] = _ApiClient["default"].convertToType(data['permission-assignment'], 'String');
        }

        if (data.hasOwnProperty('public')) {
          obj['public'] = _ApiClient["default"].convertToType(data['public'], 'Boolean');
        }

        if (data.hasOwnProperty('redirect-uris')) {
          obj['redirect-uris'] = _ApiClient["default"].convertToType(data['redirect-uris'], 'String');
        }

        if (data.hasOwnProperty('scopes')) {
          obj['scopes'] = _ApiClient["default"].convertToType(data['scopes'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CreateOidcApp</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CreateOidcApp</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(CreateOidcApp.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['accessibility'] && !(typeof data['accessibility'] === 'string' || data['accessibility'] instanceof String)) {
        throw new Error("Expected the field `accessibility` to be a primitive type in the JSON string but got " + data['accessibility']);
      } // ensure the json data is a string


      if (data['audience'] && !(typeof data['audience'] === 'string' || data['audience'] instanceof String)) {
        throw new Error("Expected the field `audience` to be a primitive type in the JSON string but got " + data['audience']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['metadata'] && !(typeof data['metadata'] === 'string' || data['metadata'] instanceof String)) {
        throw new Error("Expected the field `metadata` to be a primitive type in the JSON string but got " + data['metadata']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['permission-assignment'] && !(typeof data['permission-assignment'] === 'string' || data['permission-assignment'] instanceof String)) {
        throw new Error("Expected the field `permission-assignment` to be a primitive type in the JSON string but got " + data['permission-assignment']);
      } // ensure the json data is a string


      if (data['redirect-uris'] && !(typeof data['redirect-uris'] === 'string' || data['redirect-uris'] instanceof String)) {
        throw new Error("Expected the field `redirect-uris` to be a primitive type in the JSON string but got " + data['redirect-uris']);
      } // ensure the json data is a string


      if (data['scopes'] && !(typeof data['scopes'] === 'string' || data['scopes'] instanceof String)) {
        throw new Error("Expected the field `scopes` to be a primitive type in the JSON string but got " + data['scopes']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return CreateOidcApp;
}();

CreateOidcApp.RequiredProperties = ["name"];
/**
 * for personal password manager
 * @member {String} accessibility
 * @default 'regular'
 */

CreateOidcApp.prototype['accessibility'] = 'regular';
/**
 * A comma separated list of allowed audiences
 * @member {String} audience
 */

CreateOidcApp.prototype['audience'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

CreateOidcApp.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

CreateOidcApp.prototype['description'] = undefined;
/**
 * Additional custom fields to associate with the item
 * @member {Object.<String, String>} item-custom-fields
 */

CreateOidcApp.prototype['item-custom-fields'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateOidcApp.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the OIDC application (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateOidcApp.prototype['key'] = undefined;
/**
 * Deprecated - use description
 * @member {String} metadata
 */

CreateOidcApp.prototype['metadata'] = undefined;
/**
 * OIDC application name
 * @member {String} name
 */

CreateOidcApp.prototype['name'] = undefined;
/**
 * A json string defining the access permission assignment for this app
 * @member {String} permission-assignment
 */

CreateOidcApp.prototype['permission-assignment'] = undefined;
/**
 * Set to true if the app is public (cannot keep secrets)
 * @member {Boolean} public
 */

CreateOidcApp.prototype['public'] = undefined;
/**
 * A comma separated list of allowed redirect uris
 * @member {String} redirect-uris
 */

CreateOidcApp.prototype['redirect-uris'] = undefined;
/**
 * A comma separated list of allowed scopes
 * @member {String} scopes
 * @default 'openid'
 */

CreateOidcApp.prototype['scopes'] = 'openid';
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

CreateOidcApp.prototype['tags'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateOidcApp.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateOidcApp.prototype['uid-token'] = undefined;
var _default = CreateOidcApp;
exports["default"] = _default;
