// akeyless/dist/model/CreateRole.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateRole model module.
 * @module model/CreateRole
 * @version 5.0.3
 */
var CreateRole = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateRole</code>.
   * @alias module:model/CreateRole
   * @param name {String} Role name
   */
  function CreateRole(name) {
    _classCallCheck(this, CreateRole);

    CreateRole.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateRole, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['json'] = false;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateRole</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateRole} obj Optional instance to populate.
     * @return {module:model/CreateRole} The populated <code>CreateRole</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateRole();

        if (data.hasOwnProperty('analytics-access')) {
          obj['analytics-access'] = _ApiClient["default"].convertToType(data['analytics-access'], 'String');
        }

        if (data.hasOwnProperty('audit-access')) {
          obj['audit-access'] = _ApiClient["default"].convertToType(data['audit-access'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('event-center-access')) {
          obj['event-center-access'] = _ApiClient["default"].convertToType(data['event-center-access'], 'String');
        }

        if (data.hasOwnProperty('event-forwarders-access')) {
          obj['event-forwarders-access'] = _ApiClient["default"].convertToType(data['event-forwarders-access'], 'String');
        }

        if (data.hasOwnProperty('gw-analytics-access')) {
          obj['gw-analytics-access'] = _ApiClient["default"].convertToType(data['gw-analytics-access'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('sra-reports-access')) {
          obj['sra-reports-access'] = _ApiClient["default"].convertToType(data['sra-reports-access'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('usage-reports-access')) {
          obj['usage-reports-access'] = _ApiClient["default"].convertToType(data['usage-reports-access'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CreateRole</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CreateRole</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(CreateRole.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['analytics-access'] && !(typeof data['analytics-access'] === 'string' || data['analytics-access'] instanceof String)) {
        throw new Error("Expected the field `analytics-access` to be a primitive type in the JSON string but got " + data['analytics-access']);
      } // ensure the json data is a string


      if (data['audit-access'] && !(typeof data['audit-access'] === 'string' || data['audit-access'] instanceof String)) {
        throw new Error("Expected the field `audit-access` to be a primitive type in the JSON string but got " + data['audit-access']);
      } // ensure the json data is a string


      if (data['comment'] && !(typeof data['comment'] === 'string' || data['comment'] instanceof String)) {
        throw new Error("Expected the field `comment` to be a primitive type in the JSON string but got " + data['comment']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['event-center-access'] && !(typeof data['event-center-access'] === 'string' || data['event-center-access'] instanceof String)) {
        throw new Error("Expected the field `event-center-access` to be a primitive type in the JSON string but got " + data['event-center-access']);
      } // ensure the json data is a string


      if (data['event-forwarders-access'] && !(typeof data['event-forwarders-access'] === 'string' || data['event-forwarders-access'] instanceof String)) {
        throw new Error("Expected the field `event-forwarders-access` to be a primitive type in the JSON string but got " + data['event-forwarders-access']);
      } // ensure the json data is a string


      if (data['gw-analytics-access'] && !(typeof data['gw-analytics-access'] === 'string' || data['gw-analytics-access'] instanceof String)) {
        throw new Error("Expected the field `gw-analytics-access` to be a primitive type in the JSON string but got " + data['gw-analytics-access']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['sra-reports-access'] && !(typeof data['sra-reports-access'] === 'string' || data['sra-reports-access'] instanceof String)) {
        throw new Error("Expected the field `sra-reports-access` to be a primitive type in the JSON string but got " + data['sra-reports-access']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['usage-reports-access'] && !(typeof data['usage-reports-access'] === 'string' || data['usage-reports-access'] instanceof String)) {
        throw new Error("Expected the field `usage-reports-access` to be a primitive type in the JSON string but got " + data['usage-reports-access']);
      }

      return true;
    }
  }]);

  return CreateRole;
}();

CreateRole.RequiredProperties = ["name"];
/**
 * Allow this role to view analytics. Currently only 'none', 'own', 'all' values are supported, allowing associated auth methods to view reports produced by the same auth methods.
 * @member {String} analytics-access
 */

CreateRole.prototype['analytics-access'] = undefined;
/**
 * Allow this role to view audit logs. Currently only 'none', 'own' and 'all' values are supported, allowing associated auth methods to view audit logs produced by the same auth methods.
 * @member {String} audit-access
 */

CreateRole.prototype['audit-access'] = undefined;
/**
 * Deprecated - use description
 * @member {String} comment
 */

CreateRole.prototype['comment'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

CreateRole.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

CreateRole.prototype['description'] = undefined;
/**
 * Allow this role to view Event Center. Currently only 'none', 'own' and 'all' values are supported
 * @member {String} event-center-access
 */

CreateRole.prototype['event-center-access'] = undefined;
/**
 * Allow this role to manage Event Forwarders. Currently only 'none' and 'all' values are supported.
 * @member {String} event-forwarders-access
 */

CreateRole.prototype['event-forwarders-access'] = undefined;
/**
 * Allow this role to view gw analytics. Currently only 'none', 'own', 'all' values are supported, allowing associated auth methods to view reports produced by the same auth methods.
 * @member {String} gw-analytics-access
 */

CreateRole.prototype['gw-analytics-access'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateRole.prototype['json'] = false;
/**
 * Role name
 * @member {String} name
 */

CreateRole.prototype['name'] = undefined;
/**
 * Allow this role to view SRA Clusters. Currently only 'none', 'own', 'all' values are supported.
 * @member {String} sra-reports-access
 */

CreateRole.prototype['sra-reports-access'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateRole.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateRole.prototype['uid-token'] = undefined;
/**
 * Allow this role to view Usage Report. Currently only 'none' and 'all' values are supported.
 * @member {String} usage-reports-access
 */

CreateRole.prototype['usage-reports-access'] = undefined;
var _default = CreateRole;
exports["default"] = _default;
