// akeyless/dist/model/CreateUserEvent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateUserEvent model module.
 * @module model/CreateUserEvent
 * @version 5.0.14
 */
var CreateUserEvent = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateUserEvent</code>.
   * @alias module:model/CreateUserEvent
   * @param eventType {String} 
   * @param itemName {String} EventItemName Event item name
   * @param itemType {String} EventItemType Event item type
   */
  function CreateUserEvent(eventType, itemName, itemType) {
    _classCallCheck(this, CreateUserEvent);

    CreateUserEvent.initialize(this, eventType, itemName, itemType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateUserEvent, null, [{
    key: "initialize",
    value: function initialize(obj, eventType, itemName, itemType) {
      obj['event-type'] = eventType;
      obj['item-name'] = itemName;
      obj['item-type'] = itemType;
      obj['json'] = false;
    }
    /**
     * Constructs a <code>CreateUserEvent</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateUserEvent} obj Optional instance to populate.
     * @return {module:model/CreateUserEvent} The populated <code>CreateUserEvent</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateUserEvent();

        if (data.hasOwnProperty('capabilities')) {
          obj['capabilities'] = _ApiClient["default"].convertToType(data['capabilities'], ['String']);
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('event-source')) {
          obj['event-source'] = _ApiClient["default"].convertToType(data['event-source'], 'String');
        }

        if (data.hasOwnProperty('event-type')) {
          obj['event-type'] = _ApiClient["default"].convertToType(data['event-type'], 'String');
        }

        if (data.hasOwnProperty('item-name')) {
          obj['item-name'] = _ApiClient["default"].convertToType(data['item-name'], 'String');
        }

        if (data.hasOwnProperty('item-type')) {
          obj['item-type'] = _ApiClient["default"].convertToType(data['item-type'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('request-access-ttl')) {
          obj['request-access-ttl'] = _ApiClient["default"].convertToType(data['request-access-ttl'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CreateUserEvent</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CreateUserEvent</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(CreateUserEvent.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is an array

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (!Array.isArray(data['capabilities'])) {
        throw new Error("Expected the field `capabilities` to be an array in the JSON data but got " + data['capabilities']);
      } // ensure the json data is a string


      if (data['comment'] && !(typeof data['comment'] === 'string' || data['comment'] instanceof String)) {
        throw new Error("Expected the field `comment` to be a primitive type in the JSON string but got " + data['comment']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['event-source'] && !(typeof data['event-source'] === 'string' || data['event-source'] instanceof String)) {
        throw new Error("Expected the field `event-source` to be a primitive type in the JSON string but got " + data['event-source']);
      } // ensure the json data is a string


      if (data['event-type'] && !(typeof data['event-type'] === 'string' || data['event-type'] instanceof String)) {
        throw new Error("Expected the field `event-type` to be a primitive type in the JSON string but got " + data['event-type']);
      } // ensure the json data is a string


      if (data['item-name'] && !(typeof data['item-name'] === 'string' || data['item-name'] instanceof String)) {
        throw new Error("Expected the field `item-name` to be a primitive type in the JSON string but got " + data['item-name']);
      } // ensure the json data is a string


      if (data['item-type'] && !(typeof data['item-type'] === 'string' || data['item-type'] instanceof String)) {
        throw new Error("Expected the field `item-type` to be a primitive type in the JSON string but got " + data['item-type']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return CreateUserEvent;
}();

CreateUserEvent.RequiredProperties = ["event-type", "item-name", "item-type"];
/**
 * List of the required capabilities options: [read, update, delete,sra_transparently_connect]. Relevant only for request-access event types
 * @member {Array.<String>} capabilities
 */

CreateUserEvent.prototype['capabilities'] = undefined;
/**
 * Deprecated - use description
 * @member {String} comment
 */

CreateUserEvent.prototype['comment'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

CreateUserEvent.prototype['description'] = undefined;
/**
 * @member {String} event-source
 */

CreateUserEvent.prototype['event-source'] = undefined;
/**
 * @member {String} event-type
 */

CreateUserEvent.prototype['event-type'] = undefined;
/**
 * EventItemName Event item name
 * @member {String} item-name
 */

CreateUserEvent.prototype['item-name'] = undefined;
/**
 * EventItemType Event item type
 * @member {String} item-type
 */

CreateUserEvent.prototype['item-type'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateUserEvent.prototype['json'] = false;
/**
 * For how long to grant the requested access, in minutes
 * @member {Number} request-access-ttl
 */

CreateUserEvent.prototype['request-access-ttl'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateUserEvent.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateUserEvent.prototype['uid-token'] = undefined;
var _default = CreateUserEvent;
exports["default"] = _default;
