// akeyless/dist/model/CreateUSC.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateUSC model module.
 * @module model/CreateUSC
 * @version 5.0.8
 */
var CreateUSC = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateUSC</code>.
   * CreateUSC is a command that creates a Universal Secrets Connector
   * @alias module:model/CreateUSC
   * @param name {String} Universal Secrets Connector name
   * @param targetToAssociate {String} Target Universal Secrets Connector to connect
   */
  function CreateUSC(name, targetToAssociate) {
    _classCallCheck(this, CreateUSC);

    CreateUSC.initialize(this, name, targetToAssociate);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateUSC, null, [{
    key: "initialize",
    value: function initialize(obj, name, targetToAssociate) {
      obj['json'] = false;
      obj['name'] = name;
      obj['target-to-associate'] = targetToAssociate;
      obj['use-prefix-as-filter'] = 'false';
    }
    /**
     * Constructs a <code>CreateUSC</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateUSC} obj Optional instance to populate.
     * @return {module:model/CreateUSC} The populated <code>CreateUSC</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateUSC();

        if (data.hasOwnProperty('azure-kv-name')) {
          obj['azure-kv-name'] = _ApiClient["default"].convertToType(data['azure-kv-name'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('k8s-namespace')) {
          obj['k8s-namespace'] = _ApiClient["default"].convertToType(data['k8s-namespace'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-to-associate')) {
          obj['target-to-associate'] = _ApiClient["default"].convertToType(data['target-to-associate'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('usc-prefix')) {
          obj['usc-prefix'] = _ApiClient["default"].convertToType(data['usc-prefix'], 'String');
        }

        if (data.hasOwnProperty('use-prefix-as-filter')) {
          obj['use-prefix-as-filter'] = _ApiClient["default"].convertToType(data['use-prefix-as-filter'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CreateUSC</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CreateUSC</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(CreateUSC.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['azure-kv-name'] && !(typeof data['azure-kv-name'] === 'string' || data['azure-kv-name'] instanceof String)) {
        throw new Error("Expected the field `azure-kv-name` to be a primitive type in the JSON string but got " + data['azure-kv-name']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['k8s-namespace'] && !(typeof data['k8s-namespace'] === 'string' || data['k8s-namespace'] instanceof String)) {
        throw new Error("Expected the field `k8s-namespace` to be a primitive type in the JSON string but got " + data['k8s-namespace']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-to-associate'] && !(typeof data['target-to-associate'] === 'string' || data['target-to-associate'] instanceof String)) {
        throw new Error("Expected the field `target-to-associate` to be a primitive type in the JSON string but got " + data['target-to-associate']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['usc-prefix'] && !(typeof data['usc-prefix'] === 'string' || data['usc-prefix'] instanceof String)) {
        throw new Error("Expected the field `usc-prefix` to be a primitive type in the JSON string but got " + data['usc-prefix']);
      } // ensure the json data is a string


      if (data['use-prefix-as-filter'] && !(typeof data['use-prefix-as-filter'] === 'string' || data['use-prefix-as-filter'] instanceof String)) {
        throw new Error("Expected the field `use-prefix-as-filter` to be a primitive type in the JSON string but got " + data['use-prefix-as-filter']);
      }

      return true;
    }
  }]);

  return CreateUSC;
}();

CreateUSC.RequiredProperties = ["name", "target-to-associate"];
/**
 * Azure Key Vault name (Relevant only for Azure targets)
 * @member {String} azure-kv-name
 */

CreateUSC.prototype['azure-kv-name'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

CreateUSC.prototype['delete_protection'] = undefined;
/**
 * Description of the Universal Secrets Connector
 * @member {String} description
 */

CreateUSC.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateUSC.prototype['json'] = false;
/**
 * K8s namespace (Relevant to Kubernetes targets)
 * @member {String} k8s-namespace
 */

CreateUSC.prototype['k8s-namespace'] = undefined;
/**
 * Universal Secrets Connector name
 * @member {String} name
 */

CreateUSC.prototype['name'] = undefined;
/**
 * List of the tags attached to this Universal Secrets Connector
 * @member {Array.<String>} tags
 */

CreateUSC.prototype['tags'] = undefined;
/**
 * Target Universal Secrets Connector to connect
 * @member {String} target-to-associate
 */

CreateUSC.prototype['target-to-associate'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateUSC.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateUSC.prototype['uid-token'] = undefined;
/**
 * Prefix for all secrets created in AWS Secrets Manager
 * @member {String} usc-prefix
 */

CreateUSC.prototype['usc-prefix'] = undefined;
/**
 * Whether to filter the USC secret list using the specified usc-prefix [true/false]
 * @member {String} use-prefix-as-filter
 * @default 'false'
 */

CreateUSC.prototype['use-prefix-as-filter'] = 'false';
var _default = CreateUSC;
exports["default"] = _default;
