// akeyless/dist/model/CreateUSCOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateUSCOutput model module.
 * @module model/CreateUSCOutput
 * @version 5.0.14
 */
var CreateUSCOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateUSCOutput</code>.
   * @alias module:model/CreateUSCOutput
   */
  function CreateUSCOutput() {
    _classCallCheck(this, CreateUSCOutput);

    CreateUSCOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateUSCOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CreateUSCOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateUSCOutput} obj Optional instance to populate.
     * @return {module:model/CreateUSCOutput} The populated <code>CreateUSCOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateUSCOutput();

        if (data.hasOwnProperty('universal_secrets_connector_id')) {
          obj['universal_secrets_connector_id'] = _ApiClient["default"].convertToType(data['universal_secrets_connector_id'], 'Number');
        }

        if (data.hasOwnProperty('universal_secrets_connector_name')) {
          obj['universal_secrets_connector_name'] = _ApiClient["default"].convertToType(data['universal_secrets_connector_name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>CreateUSCOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CreateUSCOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['universal_secrets_connector_name'] && !(typeof data['universal_secrets_connector_name'] === 'string' || data['universal_secrets_connector_name'] instanceof String)) {
        throw new Error("Expected the field `universal_secrets_connector_name` to be a primitive type in the JSON string but got " + data['universal_secrets_connector_name']);
      }

      return true;
    }
  }]);

  return CreateUSCOutput;
}();
/**
 * @member {Number} universal_secrets_connector_id
 */


CreateUSCOutput.prototype['universal_secrets_connector_id'] = undefined;
/**
 * @member {String} universal_secrets_connector_name
 */

CreateUSCOutput.prototype['universal_secrets_connector_name'] = undefined;
var _default = CreateUSCOutput;
exports["default"] = _default;
