// akeyless/dist/model/CreateAuthMethodOCI.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAuthMethodOCI model module.
 * @module model/CreateAuthMethodOCI
 * @version 4.2.1
 */
var CreateAuthMethodOCI = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAuthMethodOCI</code>.
   * createAuthMethodOCI is a command that Creates a new Oracle Auth Method that will be used in the account using OCI principle and groups. [Deprecated: Use auth-method-create-oci command]
   * @alias module:model/CreateAuthMethodOCI
   * @param groupOcid {Array.<String>} A list of required groups ocids
   * @param name {String} Auth Method name
   * @param tenantOcid {String} The Oracle Cloud tenant ID
   */
  function CreateAuthMethodOCI(groupOcid, name, tenantOcid) {
    _classCallCheck(this, CreateAuthMethodOCI);

    CreateAuthMethodOCI.initialize(this, groupOcid, name, tenantOcid);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAuthMethodOCI, null, [{
    key: "initialize",
    value: function initialize(obj, groupOcid, name, tenantOcid) {
      obj['group-ocid'] = groupOcid;
      obj['name'] = name;
      obj['tenant-ocid'] = tenantOcid;
    }
    /**
     * Constructs a <code>CreateAuthMethodOCI</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAuthMethodOCI} obj Optional instance to populate.
     * @return {module:model/CreateAuthMethodOCI} The populated <code>CreateAuthMethodOCI</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAuthMethodOCI();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('audit-logs-claims')) {
          obj['audit-logs-claims'] = _ApiClient["default"].convertToType(data['audit-logs-claims'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('group-ocid')) {
          obj['group-ocid'] = _ApiClient["default"].convertToType(data['group-ocid'], ['String']);
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('product-type')) {
          obj['product-type'] = _ApiClient["default"].convertToType(data['product-type'], ['String']);
        }

        if (data.hasOwnProperty('tenant-ocid')) {
          obj['tenant-ocid'] = _ApiClient["default"].convertToType(data['tenant-ocid'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAuthMethodOCI;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


CreateAuthMethodOCI.prototype['access-expires'] = 0;
/**
 * Subclaims to include in audit logs, e.g \"--audit-logs-claims email --audit-logs-claims username\"
 * @member {Array.<String>} audit-logs-claims
 */

CreateAuthMethodOCI.prototype['audit-logs-claims'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

CreateAuthMethodOCI.prototype['bound-ips'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

CreateAuthMethodOCI.prototype['delete_protection'] = undefined;
/**
 * Auth Method description
 * @member {String} description
 */

CreateAuthMethodOCI.prototype['description'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

CreateAuthMethodOCI.prototype['force-sub-claims'] = undefined;
/**
 * A list of required groups ocids
 * @member {Array.<String>} group-ocid
 */

CreateAuthMethodOCI.prototype['group-ocid'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

CreateAuthMethodOCI.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreateAuthMethodOCI.prototype['json'] = false;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 * @default 0
 */

CreateAuthMethodOCI.prototype['jwt-ttl'] = 0;
/**
 * Auth Method name
 * @member {String} name
 */

CreateAuthMethodOCI.prototype['name'] = undefined;
/**
 * Choose the relevant product type for the auth method [sm, sra, pm, dp, ca]
 * @member {Array.<String>} product-type
 */

CreateAuthMethodOCI.prototype['product-type'] = undefined;
/**
 * The Oracle Cloud tenant ID
 * @member {String} tenant-ocid
 */

CreateAuthMethodOCI.prototype['tenant-ocid'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAuthMethodOCI.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAuthMethodOCI.prototype['uid-token'] = undefined;
var _default = CreateAuthMethodOCI;
exports["default"] = _default;
