// akeyless/dist/model/CreatePasskey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreatePasskey model module.
 * @module model/CreatePasskey
 * @version 4.3.0
 */
var CreatePasskey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreatePasskey</code>.
   * CreatePasskey is a command that creates a new passkey
   * @alias module:model/CreatePasskey
   * @param alg {String} Passkey type; options: [EC256, EC384, EC512]
   * @param name {String} ClassicKey name
   */
  function CreatePasskey(alg, name) {
    _classCallCheck(this, CreatePasskey);

    CreatePasskey.initialize(this, alg, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreatePasskey, null, [{
    key: "initialize",
    value: function initialize(obj, alg, name) {
      obj['alg'] = alg;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreatePasskey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreatePasskey} obj Optional instance to populate.
     * @return {module:model/CreatePasskey} The populated <code>CreatePasskey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreatePasskey();

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'String');
        }

        if (data.hasOwnProperty('alg')) {
          obj['alg'] = _ApiClient["default"].convertToType(data['alg'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('origin-url')) {
          obj['origin-url'] = _ApiClient["default"].convertToType(data['origin-url'], ['String']);
        }

        if (data.hasOwnProperty('protection-key-name')) {
          obj['protection-key-name'] = _ApiClient["default"].convertToType(data['protection-key-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreatePasskey;
}();
/**
 * for personal password manager
 * @member {String} accessibility
 * @default 'regular'
 */


CreatePasskey.prototype['accessibility'] = 'regular';
/**
 * Passkey type; options: [EC256, EC384, EC512]
 * @member {String} alg
 * @default 'EC256'
 */

CreatePasskey.prototype['alg'] = 'EC256';
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

CreatePasskey.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

CreatePasskey.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

CreatePasskey.prototype['json'] = false;
/**
 * ClassicKey name
 * @member {String} name
 */

CreatePasskey.prototype['name'] = undefined;
/**
 * Originating websites for this passkey
 * @member {Array.<String>} origin-url
 */

CreatePasskey.prototype['origin-url'] = undefined;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection-key-name
 */

CreatePasskey.prototype['protection-key-name'] = undefined;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

CreatePasskey.prototype['tags'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreatePasskey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreatePasskey.prototype['uid-token'] = undefined;
/**
 * For Password Management use
 * @member {String} username
 */

CreatePasskey.prototype['username'] = undefined;
var _default = CreatePasskey;
exports["default"] = _default;
