// akeyless/dist/model/CustomerFragmentConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CustomerFragmentConfig model module.
 * @module model/CustomerFragmentConfig
 * @version 3.6.2
 */
var CustomerFragmentConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CustomerFragmentConfig</code>.
   * @alias module:model/CustomerFragmentConfig
   */
  function CustomerFragmentConfig() {
    _classCallCheck(this, CustomerFragmentConfig);

    CustomerFragmentConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CustomerFragmentConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CustomerFragmentConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CustomerFragmentConfig} obj Optional instance to populate.
     * @return {module:model/CustomerFragmentConfig} The populated <code>CustomerFragmentConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CustomerFragmentConfig();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('fragment_type')) {
          obj['fragment_type'] = _ApiClient["default"].convertToType(data['fragment_type'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('key_label')) {
          obj['key_label'] = _ApiClient["default"].convertToType(data['key_label'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CustomerFragmentConfig;
}();
/**
 * @member {String} description
 */


CustomerFragmentConfig.prototype['description'] = undefined;
/**
 * @member {String} fragment_type
 */

CustomerFragmentConfig.prototype['fragment_type'] = undefined;
/**
 * @member {String} id
 */

CustomerFragmentConfig.prototype['id'] = undefined;
/**
 * @member {String} key_label
 */

CustomerFragmentConfig.prototype['key_label'] = undefined;
/**
 * @member {String} name
 */

CustomerFragmentConfig.prototype['name'] = undefined;
/**
 * @member {String} value
 */

CustomerFragmentConfig.prototype['value'] = undefined;
var _default = CustomerFragmentConfig;
exports["default"] = _default;
