// akeyless/dist/model/DatadogForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DatadogForwardingConfig model module.
 * @module model/DatadogForwardingConfig
 * @version 5.0.3
 */
var DatadogForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DatadogForwardingConfig</code>.
   * @alias module:model/DatadogForwardingConfig
   */
  function DatadogForwardingConfig() {
    _classCallCheck(this, DatadogForwardingConfig);

    DatadogForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DatadogForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DatadogForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DatadogForwardingConfig} obj Optional instance to populate.
     * @return {module:model/DatadogForwardingConfig} The populated <code>DatadogForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DatadogForwardingConfig();

        if (data.hasOwnProperty('datadog_api_key')) {
          obj['datadog_api_key'] = _ApiClient["default"].convertToType(data['datadog_api_key'], 'String');
        }

        if (data.hasOwnProperty('datadog_host')) {
          obj['datadog_host'] = _ApiClient["default"].convertToType(data['datadog_host'], 'String');
        }

        if (data.hasOwnProperty('datadog_log_service')) {
          obj['datadog_log_service'] = _ApiClient["default"].convertToType(data['datadog_log_service'], 'String');
        }

        if (data.hasOwnProperty('datadog_log_source')) {
          obj['datadog_log_source'] = _ApiClient["default"].convertToType(data['datadog_log_source'], 'String');
        }

        if (data.hasOwnProperty('datadog_log_tags')) {
          obj['datadog_log_tags'] = _ApiClient["default"].convertToType(data['datadog_log_tags'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>DatadogForwardingConfig</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DatadogForwardingConfig</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['datadog_api_key'] && !(typeof data['datadog_api_key'] === 'string' || data['datadog_api_key'] instanceof String)) {
        throw new Error("Expected the field `datadog_api_key` to be a primitive type in the JSON string but got " + data['datadog_api_key']);
      } // ensure the json data is a string


      if (data['datadog_host'] && !(typeof data['datadog_host'] === 'string' || data['datadog_host'] instanceof String)) {
        throw new Error("Expected the field `datadog_host` to be a primitive type in the JSON string but got " + data['datadog_host']);
      } // ensure the json data is a string


      if (data['datadog_log_service'] && !(typeof data['datadog_log_service'] === 'string' || data['datadog_log_service'] instanceof String)) {
        throw new Error("Expected the field `datadog_log_service` to be a primitive type in the JSON string but got " + data['datadog_log_service']);
      } // ensure the json data is a string


      if (data['datadog_log_source'] && !(typeof data['datadog_log_source'] === 'string' || data['datadog_log_source'] instanceof String)) {
        throw new Error("Expected the field `datadog_log_source` to be a primitive type in the JSON string but got " + data['datadog_log_source']);
      } // ensure the json data is a string


      if (data['datadog_log_tags'] && !(typeof data['datadog_log_tags'] === 'string' || data['datadog_log_tags'] instanceof String)) {
        throw new Error("Expected the field `datadog_log_tags` to be a primitive type in the JSON string but got " + data['datadog_log_tags']);
      }

      return true;
    }
  }]);

  return DatadogForwardingConfig;
}();
/**
 * @member {String} datadog_api_key
 */


DatadogForwardingConfig.prototype['datadog_api_key'] = undefined;
/**
 * @member {String} datadog_host
 */

DatadogForwardingConfig.prototype['datadog_host'] = undefined;
/**
 * @member {String} datadog_log_service
 */

DatadogForwardingConfig.prototype['datadog_log_service'] = undefined;
/**
 * @member {String} datadog_log_source
 */

DatadogForwardingConfig.prototype['datadog_log_source'] = undefined;
/**
 * @member {String} datadog_log_tags
 */

DatadogForwardingConfig.prototype['datadog_log_tags'] = undefined;
var _default = DatadogForwardingConfig;
exports["default"] = _default;
