// akeyless/dist/model/DecryptFileOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DecryptFileOutput model module.
 * @module model/DecryptFileOutput
 * @version 5.0.14
 */
var DecryptFileOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DecryptFileOutput</code>.
   * @alias module:model/DecryptFileOutput
   */
  function DecryptFileOutput() {
    _classCallCheck(this, DecryptFileOutput);

    DecryptFileOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DecryptFileOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DecryptFileOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DecryptFileOutput} obj Optional instance to populate.
     * @return {module:model/DecryptFileOutput} The populated <code>DecryptFileOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DecryptFileOutput();

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }

        if (data.hasOwnProperty('plain_text')) {
          obj['plain_text'] = _ApiClient["default"].convertToType(data['plain_text'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>DecryptFileOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DecryptFileOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['path'] && !(typeof data['path'] === 'string' || data['path'] instanceof String)) {
        throw new Error("Expected the field `path` to be a primitive type in the JSON string but got " + data['path']);
      } // ensure the json data is a string


      if (data['plain_text'] && !(typeof data['plain_text'] === 'string' || data['plain_text'] instanceof String)) {
        throw new Error("Expected the field `plain_text` to be a primitive type in the JSON string but got " + data['plain_text']);
      }

      return true;
    }
  }]);

  return DecryptFileOutput;
}();
/**
 * @member {String} path
 */


DecryptFileOutput.prototype['path'] = undefined;
/**
 * @member {String} plain_text
 */

DecryptFileOutput.prototype['plain_text'] = undefined;
var _default = DecryptFileOutput;
exports["default"] = _default;
