// akeyless/dist/model/DefaultConfigPart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DefaultConfigPart model module.
 * @module model/DefaultConfigPart
 * @version 5.0.14
 */
var DefaultConfigPart = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DefaultConfigPart</code>.
   * @alias module:model/DefaultConfigPart
   */
  function DefaultConfigPart() {
    _classCallCheck(this, DefaultConfigPart);

    DefaultConfigPart.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DefaultConfigPart, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DefaultConfigPart</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DefaultConfigPart} obj Optional instance to populate.
     * @return {module:model/DefaultConfigPart} The populated <code>DefaultConfigPart</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DefaultConfigPart();

        if (data.hasOwnProperty('certificate_access_id')) {
          obj['certificate_access_id'] = _ApiClient["default"].convertToType(data['certificate_access_id'], 'String');
        }

        if (data.hasOwnProperty('default_protection_key_id')) {
          obj['default_protection_key_id'] = _ApiClient["default"].convertToType(data['default_protection_key_id'], 'Number');
        }

        if (data.hasOwnProperty('default_secret_location')) {
          obj['default_secret_location'] = _ApiClient["default"].convertToType(data['default_secret_location'], 'String');
        }

        if (data.hasOwnProperty('oidc_access_id')) {
          obj['oidc_access_id'] = _ApiClient["default"].convertToType(data['oidc_access_id'], 'String');
        }

        if (data.hasOwnProperty('saml_access_id')) {
          obj['saml_access_id'] = _ApiClient["default"].convertToType(data['saml_access_id'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>DefaultConfigPart</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DefaultConfigPart</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['certificate_access_id'] && !(typeof data['certificate_access_id'] === 'string' || data['certificate_access_id'] instanceof String)) {
        throw new Error("Expected the field `certificate_access_id` to be a primitive type in the JSON string but got " + data['certificate_access_id']);
      } // ensure the json data is a string


      if (data['default_secret_location'] && !(typeof data['default_secret_location'] === 'string' || data['default_secret_location'] instanceof String)) {
        throw new Error("Expected the field `default_secret_location` to be a primitive type in the JSON string but got " + data['default_secret_location']);
      } // ensure the json data is a string


      if (data['oidc_access_id'] && !(typeof data['oidc_access_id'] === 'string' || data['oidc_access_id'] instanceof String)) {
        throw new Error("Expected the field `oidc_access_id` to be a primitive type in the JSON string but got " + data['oidc_access_id']);
      } // ensure the json data is a string


      if (data['saml_access_id'] && !(typeof data['saml_access_id'] === 'string' || data['saml_access_id'] instanceof String)) {
        throw new Error("Expected the field `saml_access_id` to be a primitive type in the JSON string but got " + data['saml_access_id']);
      }

      return true;
    }
  }]);

  return DefaultConfigPart;
}();
/**
 * @member {String} certificate_access_id
 */


DefaultConfigPart.prototype['certificate_access_id'] = undefined;
/**
 * @member {Number} default_protection_key_id
 */

DefaultConfigPart.prototype['default_protection_key_id'] = undefined;
/**
 * @member {String} default_secret_location
 */

DefaultConfigPart.prototype['default_secret_location'] = undefined;
/**
 * @member {String} oidc_access_id
 */

DefaultConfigPart.prototype['oidc_access_id'] = undefined;
/**
 * @member {String} saml_access_id
 */

DefaultConfigPart.prototype['saml_access_id'] = undefined;
var _default = DefaultConfigPart;
exports["default"] = _default;
