// akeyless/dist/model/DynamicSecretUpdateChef.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretUpdateChef model module.
 * @module model/DynamicSecretUpdateChef
 * @version 4.2.2
 */
var DynamicSecretUpdateChef = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretUpdateChef</code>.
   * dynamicSecretUpdateChef is a command that updates chef dynamic secret
   * @alias module:model/DynamicSecretUpdateChef
   * @param name {String} Dynamic secret name
   */
  function DynamicSecretUpdateChef(name) {
    _classCallCheck(this, DynamicSecretUpdateChef);

    DynamicSecretUpdateChef.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretUpdateChef, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>DynamicSecretUpdateChef</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretUpdateChef} obj Optional instance to populate.
     * @return {module:model/DynamicSecretUpdateChef} The populated <code>DynamicSecretUpdateChef</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretUpdateChef();

        if (data.hasOwnProperty('chef-orgs')) {
          obj['chef-orgs'] = _ApiClient["default"].convertToType(data['chef-orgs'], 'String');
        }

        if (data.hasOwnProperty('chef-server-key')) {
          obj['chef-server-key'] = _ApiClient["default"].convertToType(data['chef-server-key'], 'String');
        }

        if (data.hasOwnProperty('chef-server-url')) {
          obj['chef-server-url'] = _ApiClient["default"].convertToType(data['chef-server-url'], 'String');
        }

        if (data.hasOwnProperty('chef-server-username')) {
          obj['chef-server-username'] = _ApiClient["default"].convertToType(data['chef-server-username'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password-length')) {
          obj['password-length'] = _ApiClient["default"].convertToType(data['password-length'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('skip-ssl')) {
          obj['skip-ssl'] = _ApiClient["default"].convertToType(data['skip-ssl'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DynamicSecretUpdateChef;
}();
/**
 * Organizations
 * @member {String} chef-orgs
 */


DynamicSecretUpdateChef.prototype['chef-orgs'] = undefined;
/**
 * Server key
 * @member {String} chef-server-key
 */

DynamicSecretUpdateChef.prototype['chef-server-key'] = undefined;
/**
 * Server URL
 * @member {String} chef-server-url
 */

DynamicSecretUpdateChef.prototype['chef-server-url'] = undefined;
/**
 * Server username
 * @member {String} chef-server-username
 */

DynamicSecretUpdateChef.prototype['chef-server-username'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

DynamicSecretUpdateChef.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

DynamicSecretUpdateChef.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

DynamicSecretUpdateChef.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

DynamicSecretUpdateChef.prototype['name'] = undefined;
/**
 * Dynamic secret new name
 * @member {String} new-name
 */

DynamicSecretUpdateChef.prototype['new-name'] = undefined;
/**
 * The length of the password to be generated
 * @member {String} password-length
 */

DynamicSecretUpdateChef.prototype['password-length'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

DynamicSecretUpdateChef.prototype['producer-encryption-key-name'] = undefined;
/**
 * Skip SSL
 * @member {Boolean} skip-ssl
 * @default true
 */

DynamicSecretUpdateChef.prototype['skip-ssl'] = true;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

DynamicSecretUpdateChef.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

DynamicSecretUpdateChef.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DynamicSecretUpdateChef.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DynamicSecretUpdateChef.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

DynamicSecretUpdateChef.prototype['user-ttl'] = '60m';
var _default = DynamicSecretUpdateChef;
exports["default"] = _default;
