// akeyless/dist/model/DynamicSecretCreateLdap.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretCreateLdap model module.
 * @module model/DynamicSecretCreateLdap
 * @version 4.2.2
 */
var DynamicSecretCreateLdap = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretCreateLdap</code>.
   * dynamicSecretCreateLdap is a command that creates ldap dynamic secret
   * @alias module:model/DynamicSecretCreateLdap
   * @param name {String} Dynamic secret name
   */
  function DynamicSecretCreateLdap(name) {
    _classCallCheck(this, DynamicSecretCreateLdap);

    DynamicSecretCreateLdap.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretCreateLdap, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>DynamicSecretCreateLdap</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretCreateLdap} obj Optional instance to populate.
     * @return {module:model/DynamicSecretCreateLdap} The populated <code>DynamicSecretCreateLdap</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretCreateLdap();

        if (data.hasOwnProperty('ProviderType')) {
          obj['ProviderType'] = _ApiClient["default"].convertToType(data['ProviderType'], 'String');
        }

        if (data.hasOwnProperty('bind-dn')) {
          obj['bind-dn'] = _ApiClient["default"].convertToType(data['bind-dn'], 'String');
        }

        if (data.hasOwnProperty('bind-dn-password')) {
          obj['bind-dn-password'] = _ApiClient["default"].convertToType(data['bind-dn-password'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('external-username')) {
          obj['external-username'] = _ApiClient["default"].convertToType(data['external-username'], 'String');
        }

        if (data.hasOwnProperty('group-dn')) {
          obj['group-dn'] = _ApiClient["default"].convertToType(data['group-dn'], 'String');
        }

        if (data.hasOwnProperty('host-provider')) {
          obj['host-provider'] = _ApiClient["default"].convertToType(data['host-provider'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('ldap-ca-cert')) {
          obj['ldap-ca-cert'] = _ApiClient["default"].convertToType(data['ldap-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('ldap-url')) {
          obj['ldap-url'] = _ApiClient["default"].convertToType(data['ldap-url'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password-length')) {
          obj['password-length'] = _ApiClient["default"].convertToType(data['password-length'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('secure-access-rd-gateway-server')) {
          obj['secure-access-rd-gateway-server'] = _ApiClient["default"].convertToType(data['secure-access-rd-gateway-server'], 'String');
        }

        if (data.hasOwnProperty('secure-access-rdp-domain')) {
          obj['secure-access-rdp-domain'] = _ApiClient["default"].convertToType(data['secure-access-rdp-domain'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('token-expiration')) {
          obj['token-expiration'] = _ApiClient["default"].convertToType(data['token-expiration'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-attribute')) {
          obj['user-attribute'] = _ApiClient["default"].convertToType(data['user-attribute'], 'String');
        }

        if (data.hasOwnProperty('user-dn')) {
          obj['user-dn'] = _ApiClient["default"].convertToType(data['user-dn'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DynamicSecretCreateLdap;
}();
/**
 * @member {String} ProviderType
 */


DynamicSecretCreateLdap.prototype['ProviderType'] = undefined;
/**
 * Bind DN
 * @member {String} bind-dn
 */

DynamicSecretCreateLdap.prototype['bind-dn'] = undefined;
/**
 * Bind DN Password
 * @member {String} bind-dn-password
 */

DynamicSecretCreateLdap.prototype['bind-dn-password'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

DynamicSecretCreateLdap.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

DynamicSecretCreateLdap.prototype['description'] = undefined;
/**
 * Externally provided username [true/false]
 * @member {String} external-username
 * @default 'false'
 */

DynamicSecretCreateLdap.prototype['external-username'] = 'false';
/**
 * Group DN which the temporary user should be added
 * @member {String} group-dn
 */

DynamicSecretCreateLdap.prototype['group-dn'] = undefined;
/**
 * Host provider type [explicit/target], Default Host provider is explicit, Relevant only for Secure Remote Access of ssh cert issuer, ldap rotated secret and ldap dynamic secret
 * @member {String} host-provider
 */

DynamicSecretCreateLdap.prototype['host-provider'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

DynamicSecretCreateLdap.prototype['json'] = false;
/**
 * CA Certificate File Content
 * @member {String} ldap-ca-cert
 */

DynamicSecretCreateLdap.prototype['ldap-ca-cert'] = undefined;
/**
 * LDAP Server URL
 * @member {String} ldap-url
 */

DynamicSecretCreateLdap.prototype['ldap-url'] = undefined;
/**
 * Dynamic secret name
 * @member {String} name
 */

DynamicSecretCreateLdap.prototype['name'] = undefined;
/**
 * The length of the password to be generated
 * @member {String} password-length
 */

DynamicSecretCreateLdap.prototype['password-length'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

DynamicSecretCreateLdap.prototype['producer-encryption-key-name'] = undefined;
/**
 * Enable/Disable secure remote access [true/false]
 * @member {String} secure-access-enable
 */

DynamicSecretCreateLdap.prototype['secure-access-enable'] = undefined;
/**
 * Target servers for connections (In case of Linked Target association, host(s) will inherit Linked Target hosts - Relevant only for Dynamic Secrets/producers)
 * @member {Array.<String>} secure-access-host
 */

DynamicSecretCreateLdap.prototype['secure-access-host'] = undefined;
/**
 * RD Gateway server
 * @member {String} secure-access-rd-gateway-server
 */

DynamicSecretCreateLdap.prototype['secure-access-rd-gateway-server'] = undefined;
/**
 * Required when the Dynamic Secret is used for a domain user
 * @member {String} secure-access-rdp-domain
 */

DynamicSecretCreateLdap.prototype['secure-access-rdp-domain'] = undefined;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

DynamicSecretCreateLdap.prototype['tags'] = undefined;
/**
 * A list of linked targets to be associated, Relevant only for Secure Remote Access for ssh cert issuer, ldap rotated secret and ldap dynamic secret, To specify multiple targets use argument multiple times
 * @member {Array.<String>} target
 */

DynamicSecretCreateLdap.prototype['target'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

DynamicSecretCreateLdap.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DynamicSecretCreateLdap.prototype['token'] = undefined;
/**
 * Token expiration
 * @member {String} token-expiration
 */

DynamicSecretCreateLdap.prototype['token-expiration'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DynamicSecretCreateLdap.prototype['uid-token'] = undefined;
/**
 * User Attribute
 * @member {String} user-attribute
 */

DynamicSecretCreateLdap.prototype['user-attribute'] = undefined;
/**
 * User DN
 * @member {String} user-dn
 */

DynamicSecretCreateLdap.prototype['user-dn'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

DynamicSecretCreateLdap.prototype['user-ttl'] = '60m';
var _default = DynamicSecretCreateLdap;
exports["default"] = _default;
