// akeyless/dist/model/DynamicSecretProducerInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretProducerInfo model module.
 * @module model/DynamicSecretProducerInfo
 * @version 5.0.0
 */
var DynamicSecretProducerInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretProducerInfo</code>.
   * DynamicSecretProducerInfo The dynamic secret producer info This parameter relevant and required only in case of create update dynamic secret.
   * @alias module:model/DynamicSecretProducerInfo
   */
  function DynamicSecretProducerInfo() {
    _classCallCheck(this, DynamicSecretProducerInfo);

    DynamicSecretProducerInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretProducerInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DynamicSecretProducerInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretProducerInfo} obj Optional instance to populate.
     * @return {module:model/DynamicSecretProducerInfo} The populated <code>DynamicSecretProducerInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretProducerInfo();

        if (data.hasOwnProperty('failure_message')) {
          obj['failure_message'] = _ApiClient["default"].convertToType(data['failure_message'], 'String');
        }

        if (data.hasOwnProperty('gw_cluster_id')) {
          obj['gw_cluster_id'] = _ApiClient["default"].convertToType(data['gw_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('k8s_allowed_namespaces')) {
          obj['k8s_allowed_namespaces'] = _ApiClient["default"].convertToType(data['k8s_allowed_namespaces'], 'String');
        }

        if (data.hasOwnProperty('k8s_dynamic_mode')) {
          obj['k8s_dynamic_mode'] = _ApiClient["default"].convertToType(data['k8s_dynamic_mode'], 'Boolean');
        }

        if (data.hasOwnProperty('producer_last_keep_alive')) {
          obj['producer_last_keep_alive'] = _ApiClient["default"].convertToType(data['producer_last_keep_alive'], 'String');
        }

        if (data.hasOwnProperty('producer_metadata')) {
          obj['producer_metadata'] = _ApiClient["default"].convertToType(data['producer_metadata'], 'String');
        }

        if (data.hasOwnProperty('producer_status')) {
          obj['producer_status'] = _ApiClient["default"].convertToType(data['producer_status'], 'String');
        }

        if (data.hasOwnProperty('producer_type')) {
          obj['producer_type'] = _ApiClient["default"].convertToType(data['producer_type'], 'String');
        }

        if (data.hasOwnProperty('user_ttl')) {
          obj['user_ttl'] = _ApiClient["default"].convertToType(data['user_ttl'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>DynamicSecretProducerInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DynamicSecretProducerInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['failure_message'] && !(typeof data['failure_message'] === 'string' || data['failure_message'] instanceof String)) {
        throw new Error("Expected the field `failure_message` to be a primitive type in the JSON string but got " + data['failure_message']);
      } // ensure the json data is a string


      if (data['k8s_allowed_namespaces'] && !(typeof data['k8s_allowed_namespaces'] === 'string' || data['k8s_allowed_namespaces'] instanceof String)) {
        throw new Error("Expected the field `k8s_allowed_namespaces` to be a primitive type in the JSON string but got " + data['k8s_allowed_namespaces']);
      } // ensure the json data is a string


      if (data['producer_last_keep_alive'] && !(typeof data['producer_last_keep_alive'] === 'string' || data['producer_last_keep_alive'] instanceof String)) {
        throw new Error("Expected the field `producer_last_keep_alive` to be a primitive type in the JSON string but got " + data['producer_last_keep_alive']);
      } // ensure the json data is a string


      if (data['producer_metadata'] && !(typeof data['producer_metadata'] === 'string' || data['producer_metadata'] instanceof String)) {
        throw new Error("Expected the field `producer_metadata` to be a primitive type in the JSON string but got " + data['producer_metadata']);
      } // ensure the json data is a string


      if (data['producer_status'] && !(typeof data['producer_status'] === 'string' || data['producer_status'] instanceof String)) {
        throw new Error("Expected the field `producer_status` to be a primitive type in the JSON string but got " + data['producer_status']);
      } // ensure the json data is a string


      if (data['producer_type'] && !(typeof data['producer_type'] === 'string' || data['producer_type'] instanceof String)) {
        throw new Error("Expected the field `producer_type` to be a primitive type in the JSON string but got " + data['producer_type']);
      } // ensure the json data is a string


      if (data['user_ttl'] && !(typeof data['user_ttl'] === 'string' || data['user_ttl'] instanceof String)) {
        throw new Error("Expected the field `user_ttl` to be a primitive type in the JSON string but got " + data['user_ttl']);
      }

      return true;
    }
  }]);

  return DynamicSecretProducerInfo;
}();
/**
 * @member {String} failure_message
 */


DynamicSecretProducerInfo.prototype['failure_message'] = undefined;
/**
 * @member {Number} gw_cluster_id
 */

DynamicSecretProducerInfo.prototype['gw_cluster_id'] = undefined;
/**
 * Relevant only for generic k8s producer
 * @member {String} k8s_allowed_namespaces
 */

DynamicSecretProducerInfo.prototype['k8s_allowed_namespaces'] = undefined;
/**
 * Relevant only for generic k8s producer
 * @member {Boolean} k8s_dynamic_mode
 */

DynamicSecretProducerInfo.prototype['k8s_dynamic_mode'] = undefined;
/**
 * @member {String} producer_last_keep_alive
 */

DynamicSecretProducerInfo.prototype['producer_last_keep_alive'] = undefined;
/**
 * @member {String} producer_metadata
 */

DynamicSecretProducerInfo.prototype['producer_metadata'] = undefined;
/**
 * RotationStatus defines types of rotation Status
 * @member {String} producer_status
 */

DynamicSecretProducerInfo.prototype['producer_status'] = undefined;
/**
 * @member {String} producer_type
 */

DynamicSecretProducerInfo.prototype['producer_type'] = undefined;
/**
 * @member {String} user_ttl
 */

DynamicSecretProducerInfo.prototype['user_ttl'] = undefined;
var _default = DynamicSecretProducerInfo;
exports["default"] = _default;
