// akeyless/dist/model/DynamicSecretTmpCredsDelete.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretTmpCredsDelete model module.
 * @module model/DynamicSecretTmpCredsDelete
 * @version 4.1.0
 */
var DynamicSecretTmpCredsDelete = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretTmpCredsDelete</code>.
   * dynamicSecretTmpCredsDelete is a command that deletes dynamic secret temp creds
   * @alias module:model/DynamicSecretTmpCredsDelete
   * @param name {String} Dynamic secret name
   * @param tmpCredsId {String} Tmp Creds ID
   */
  function DynamicSecretTmpCredsDelete(name, tmpCredsId) {
    _classCallCheck(this, DynamicSecretTmpCredsDelete);

    DynamicSecretTmpCredsDelete.initialize(this, name, tmpCredsId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretTmpCredsDelete, null, [{
    key: "initialize",
    value: function initialize(obj, name, tmpCredsId) {
      obj['name'] = name;
      obj['tmp-creds-id'] = tmpCredsId;
    }
    /**
     * Constructs a <code>DynamicSecretTmpCredsDelete</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretTmpCredsDelete} obj Optional instance to populate.
     * @return {module:model/DynamicSecretTmpCredsDelete} The populated <code>DynamicSecretTmpCredsDelete</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretTmpCredsDelete();

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('revoke-all')) {
          obj['revoke-all'] = _ApiClient["default"].convertToType(data['revoke-all'], 'Boolean');
        }

        if (data.hasOwnProperty('soft-delete')) {
          obj['soft-delete'] = _ApiClient["default"].convertToType(data['soft-delete'], 'Boolean');
        }

        if (data.hasOwnProperty('tmp-creds-id')) {
          obj['tmp-creds-id'] = _ApiClient["default"].convertToType(data['tmp-creds-id'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DynamicSecretTmpCredsDelete;
}();
/**
 * Host
 * @member {String} host
 */


DynamicSecretTmpCredsDelete.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

DynamicSecretTmpCredsDelete.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

DynamicSecretTmpCredsDelete.prototype['name'] = undefined;
/**
 * Revoke All Temp Creds
 * @member {Boolean} revoke-all
 */

DynamicSecretTmpCredsDelete.prototype['revoke-all'] = undefined;
/**
 * Soft Delete
 * @member {Boolean} soft-delete
 */

DynamicSecretTmpCredsDelete.prototype['soft-delete'] = undefined;
/**
 * Tmp Creds ID
 * @member {String} tmp-creds-id
 * @default 'demo_default_tmp_creds_id_for_sdk_bc'
 */

DynamicSecretTmpCredsDelete.prototype['tmp-creds-id'] = 'demo_default_tmp_creds_id_for_sdk_bc';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DynamicSecretTmpCredsDelete.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DynamicSecretTmpCredsDelete.prototype['uid-token'] = undefined;
var _default = DynamicSecretTmpCredsDelete;
exports["default"] = _default;
