// akeyless/dist/model/DynamicSecretUpdateVenafi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretUpdateVenafi model module.
 * @module model/DynamicSecretUpdateVenafi
 * @version 5.0.3
 */
var DynamicSecretUpdateVenafi = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretUpdateVenafi</code>.
   * dynamicSecretUpdateVenafi is a command that updates a Venafi dynamic secret to dynamically update certificates generated by Venafi or have Akeyless generated certificates using PKI be monitored by Venafi
   * @alias module:model/DynamicSecretUpdateVenafi
   * @param name {String} Dynamic secret name
   */
  function DynamicSecretUpdateVenafi(name) {
    _classCallCheck(this, DynamicSecretUpdateVenafi);

    DynamicSecretUpdateVenafi.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretUpdateVenafi, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['admin-rotation-interval-days'] = 0;
      obj['enable-admin-rotation'] = false;
      obj['json'] = false;
      obj['name'] = name;
      obj['user-ttl'] = '2160h';
      obj['venafi-client-id'] = 'akeyless';
    }
    /**
     * Constructs a <code>DynamicSecretUpdateVenafi</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretUpdateVenafi} obj Optional instance to populate.
     * @return {module:model/DynamicSecretUpdateVenafi} The populated <code>DynamicSecretUpdateVenafi</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretUpdateVenafi();

        if (data.hasOwnProperty('admin-rotation-interval-days')) {
          obj['admin-rotation-interval-days'] = _ApiClient["default"].convertToType(data['admin-rotation-interval-days'], 'Number');
        }

        if (data.hasOwnProperty('allow-subdomains')) {
          obj['allow-subdomains'] = _ApiClient["default"].convertToType(data['allow-subdomains'], 'Boolean');
        }

        if (data.hasOwnProperty('allowed-domains')) {
          obj['allowed-domains'] = _ApiClient["default"].convertToType(data['allowed-domains'], ['String']);
        }

        if (data.hasOwnProperty('auto-generated-folder')) {
          obj['auto-generated-folder'] = _ApiClient["default"].convertToType(data['auto-generated-folder'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('enable-admin-rotation')) {
          obj['enable-admin-rotation'] = _ApiClient["default"].convertToType(data['enable-admin-rotation'], 'Boolean');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('root-first-in-chain')) {
          obj['root-first-in-chain'] = _ApiClient["default"].convertToType(data['root-first-in-chain'], 'Boolean');
        }

        if (data.hasOwnProperty('sign-using-akeyless-pki')) {
          obj['sign-using-akeyless-pki'] = _ApiClient["default"].convertToType(data['sign-using-akeyless-pki'], 'Boolean');
        }

        if (data.hasOwnProperty('signer-key-name')) {
          obj['signer-key-name'] = _ApiClient["default"].convertToType(data['signer-key-name'], 'String');
        }

        if (data.hasOwnProperty('store-private-key')) {
          obj['store-private-key'] = _ApiClient["default"].convertToType(data['store-private-key'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('venafi-access-token')) {
          obj['venafi-access-token'] = _ApiClient["default"].convertToType(data['venafi-access-token'], 'String');
        }

        if (data.hasOwnProperty('venafi-api-key')) {
          obj['venafi-api-key'] = _ApiClient["default"].convertToType(data['venafi-api-key'], 'String');
        }

        if (data.hasOwnProperty('venafi-baseurl')) {
          obj['venafi-baseurl'] = _ApiClient["default"].convertToType(data['venafi-baseurl'], 'String');
        }

        if (data.hasOwnProperty('venafi-client-id')) {
          obj['venafi-client-id'] = _ApiClient["default"].convertToType(data['venafi-client-id'], 'String');
        }

        if (data.hasOwnProperty('venafi-refresh-token')) {
          obj['venafi-refresh-token'] = _ApiClient["default"].convertToType(data['venafi-refresh-token'], 'String');
        }

        if (data.hasOwnProperty('venafi-use-tpp')) {
          obj['venafi-use-tpp'] = _ApiClient["default"].convertToType(data['venafi-use-tpp'], 'Boolean');
        }

        if (data.hasOwnProperty('venafi-zone')) {
          obj['venafi-zone'] = _ApiClient["default"].convertToType(data['venafi-zone'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>DynamicSecretUpdateVenafi</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DynamicSecretUpdateVenafi</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(DynamicSecretUpdateVenafi.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is an array

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (!Array.isArray(data['allowed-domains'])) {
        throw new Error("Expected the field `allowed-domains` to be an array in the JSON data but got " + data['allowed-domains']);
      } // ensure the json data is a string


      if (data['auto-generated-folder'] && !(typeof data['auto-generated-folder'] === 'string' || data['auto-generated-folder'] instanceof String)) {
        throw new Error("Expected the field `auto-generated-folder` to be a primitive type in the JSON string but got " + data['auto-generated-folder']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['producer-encryption-key-name'] && !(typeof data['producer-encryption-key-name'] === 'string' || data['producer-encryption-key-name'] instanceof String)) {
        throw new Error("Expected the field `producer-encryption-key-name` to be a primitive type in the JSON string but got " + data['producer-encryption-key-name']);
      } // ensure the json data is a string


      if (data['signer-key-name'] && !(typeof data['signer-key-name'] === 'string' || data['signer-key-name'] instanceof String)) {
        throw new Error("Expected the field `signer-key-name` to be a primitive type in the JSON string but got " + data['signer-key-name']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-ttl'] && !(typeof data['user-ttl'] === 'string' || data['user-ttl'] instanceof String)) {
        throw new Error("Expected the field `user-ttl` to be a primitive type in the JSON string but got " + data['user-ttl']);
      } // ensure the json data is a string


      if (data['venafi-access-token'] && !(typeof data['venafi-access-token'] === 'string' || data['venafi-access-token'] instanceof String)) {
        throw new Error("Expected the field `venafi-access-token` to be a primitive type in the JSON string but got " + data['venafi-access-token']);
      } // ensure the json data is a string


      if (data['venafi-api-key'] && !(typeof data['venafi-api-key'] === 'string' || data['venafi-api-key'] instanceof String)) {
        throw new Error("Expected the field `venafi-api-key` to be a primitive type in the JSON string but got " + data['venafi-api-key']);
      } // ensure the json data is a string


      if (data['venafi-baseurl'] && !(typeof data['venafi-baseurl'] === 'string' || data['venafi-baseurl'] instanceof String)) {
        throw new Error("Expected the field `venafi-baseurl` to be a primitive type in the JSON string but got " + data['venafi-baseurl']);
      } // ensure the json data is a string


      if (data['venafi-client-id'] && !(typeof data['venafi-client-id'] === 'string' || data['venafi-client-id'] instanceof String)) {
        throw new Error("Expected the field `venafi-client-id` to be a primitive type in the JSON string but got " + data['venafi-client-id']);
      } // ensure the json data is a string


      if (data['venafi-refresh-token'] && !(typeof data['venafi-refresh-token'] === 'string' || data['venafi-refresh-token'] instanceof String)) {
        throw new Error("Expected the field `venafi-refresh-token` to be a primitive type in the JSON string but got " + data['venafi-refresh-token']);
      } // ensure the json data is a string


      if (data['venafi-zone'] && !(typeof data['venafi-zone'] === 'string' || data['venafi-zone'] instanceof String)) {
        throw new Error("Expected the field `venafi-zone` to be a primitive type in the JSON string but got " + data['venafi-zone']);
      }

      return true;
    }
  }]);

  return DynamicSecretUpdateVenafi;
}();

DynamicSecretUpdateVenafi.RequiredProperties = ["name"];
/**
 * Admin credentials rotation interval (days)
 * @member {Number} admin-rotation-interval-days
 * @default 0
 */

DynamicSecretUpdateVenafi.prototype['admin-rotation-interval-days'] = 0;
/**
 * Allow subdomains
 * @member {Boolean} allow-subdomains
 */

DynamicSecretUpdateVenafi.prototype['allow-subdomains'] = undefined;
/**
 * Allowed domains
 * @member {Array.<String>} allowed-domains
 */

DynamicSecretUpdateVenafi.prototype['allowed-domains'] = undefined;
/**
 * Auto generated folder
 * @member {String} auto-generated-folder
 */

DynamicSecretUpdateVenafi.prototype['auto-generated-folder'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

DynamicSecretUpdateVenafi.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

DynamicSecretUpdateVenafi.prototype['description'] = undefined;
/**
 * Automatic admin credentials rotation
 * @member {Boolean} enable-admin-rotation
 * @default false
 */

DynamicSecretUpdateVenafi.prototype['enable-admin-rotation'] = false;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

DynamicSecretUpdateVenafi.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

DynamicSecretUpdateVenafi.prototype['name'] = undefined;
/**
 * Dynamic secret name
 * @member {String} new-name
 */

DynamicSecretUpdateVenafi.prototype['new-name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

DynamicSecretUpdateVenafi.prototype['producer-encryption-key-name'] = undefined;
/**
 * Root first in chain
 * @member {Boolean} root-first-in-chain
 */

DynamicSecretUpdateVenafi.prototype['root-first-in-chain'] = undefined;
/**
 * Use Akeyless PKI issuer or Venafi issuer
 * @member {Boolean} sign-using-akeyless-pki
 */

DynamicSecretUpdateVenafi.prototype['sign-using-akeyless-pki'] = undefined;
/**
 * Signer key name
 * @member {String} signer-key-name
 */

DynamicSecretUpdateVenafi.prototype['signer-key-name'] = undefined;
/**
 * Store private key
 * @member {Boolean} store-private-key
 */

DynamicSecretUpdateVenafi.prototype['store-private-key'] = undefined;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

DynamicSecretUpdateVenafi.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

DynamicSecretUpdateVenafi.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DynamicSecretUpdateVenafi.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DynamicSecretUpdateVenafi.prototype['uid-token'] = undefined;
/**
 * User TTL in time.Duration format (2160h / 129600m / etc...). When using sign-using-akeyless-pki certificates created will have this validity period, otherwise the user-ttl is taken from the Validity Period field of the Zone's' Issuing Template. When using cert-manager it is advised to have a TTL of above 60 days (1440h). For more information - https://cert-manager.io/docs/usage/certificate/
 * @member {String} user-ttl
 * @default '2160h'
 */

DynamicSecretUpdateVenafi.prototype['user-ttl'] = '2160h';
/**
 * Venafi Access Token to use to access the TPP environment (Relevant when using TPP)
 * @member {String} venafi-access-token
 */

DynamicSecretUpdateVenafi.prototype['venafi-access-token'] = undefined;
/**
 * Venafi API key
 * @member {String} venafi-api-key
 */

DynamicSecretUpdateVenafi.prototype['venafi-api-key'] = undefined;
/**
 * Venafi Baseurl
 * @member {String} venafi-baseurl
 */

DynamicSecretUpdateVenafi.prototype['venafi-baseurl'] = undefined;
/**
 * Venafi Client ID that was used when the access token was generated
 * @member {String} venafi-client-id
 * @default 'akeyless'
 */

DynamicSecretUpdateVenafi.prototype['venafi-client-id'] = 'akeyless';
/**
 * Venafi Refresh Token to use when the Access Token is expired (Relevant when using TPP)
 * @member {String} venafi-refresh-token
 */

DynamicSecretUpdateVenafi.prototype['venafi-refresh-token'] = undefined;
/**
 * Venafi using TPP
 * @member {Boolean} venafi-use-tpp
 */

DynamicSecretUpdateVenafi.prototype['venafi-use-tpp'] = undefined;
/**
 * Venafi Zone
 * @member {String} venafi-zone
 */

DynamicSecretUpdateVenafi.prototype['venafi-zone'] = undefined;
var _default = DynamicSecretUpdateVenafi;
exports["default"] = _default;
