// akeyless/dist/model/DynamicSecretCreateCustom.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretCreateCustom model module.
 * @module model/DynamicSecretCreateCustom
 * @version 5.0.3
 */
var DynamicSecretCreateCustom = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretCreateCustom</code>.
   * @alias module:model/DynamicSecretCreateCustom
   * @param createSyncUrl {String} URL of an endpoint that implements /sync/create method, for example https://webhook.example.com/sync/create
   * @param name {String} Dynamic secret name
   * @param revokeSyncUrl {String} URL of an endpoint that implements /sync/revoke method, for example https://webhook.example.com/sync/revoke
   */
  function DynamicSecretCreateCustom(createSyncUrl, name, revokeSyncUrl) {
    _classCallCheck(this, DynamicSecretCreateCustom);

    DynamicSecretCreateCustom.initialize(this, createSyncUrl, name, revokeSyncUrl);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretCreateCustom, null, [{
    key: "initialize",
    value: function initialize(obj, createSyncUrl, name, revokeSyncUrl) {
      obj['create-sync-url'] = createSyncUrl;
      obj['enable-admin-rotation'] = false;
      obj['json'] = false;
      obj['name'] = name;
      obj['revoke-sync-url'] = revokeSyncUrl;
      obj['timeout-sec'] = 60;
      obj['user-ttl'] = '60m';
    }
    /**
     * Constructs a <code>DynamicSecretCreateCustom</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretCreateCustom} obj Optional instance to populate.
     * @return {module:model/DynamicSecretCreateCustom} The populated <code>DynamicSecretCreateCustom</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretCreateCustom();

        if (data.hasOwnProperty('admin-rotation-interval-days')) {
          obj['admin-rotation-interval-days'] = _ApiClient["default"].convertToType(data['admin-rotation-interval-days'], 'Number');
        }

        if (data.hasOwnProperty('create-sync-url')) {
          obj['create-sync-url'] = _ApiClient["default"].convertToType(data['create-sync-url'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('enable-admin-rotation')) {
          obj['enable-admin-rotation'] = _ApiClient["default"].convertToType(data['enable-admin-rotation'], 'Boolean');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('payload')) {
          obj['payload'] = _ApiClient["default"].convertToType(data['payload'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('revoke-sync-url')) {
          obj['revoke-sync-url'] = _ApiClient["default"].convertToType(data['revoke-sync-url'], 'String');
        }

        if (data.hasOwnProperty('rotate-sync-url')) {
          obj['rotate-sync-url'] = _ApiClient["default"].convertToType(data['rotate-sync-url'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('timeout-sec')) {
          obj['timeout-sec'] = _ApiClient["default"].convertToType(data['timeout-sec'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>DynamicSecretCreateCustom</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DynamicSecretCreateCustom</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(DynamicSecretCreateCustom.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['create-sync-url'] && !(typeof data['create-sync-url'] === 'string' || data['create-sync-url'] instanceof String)) {
        throw new Error("Expected the field `create-sync-url` to be a primitive type in the JSON string but got " + data['create-sync-url']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['payload'] && !(typeof data['payload'] === 'string' || data['payload'] instanceof String)) {
        throw new Error("Expected the field `payload` to be a primitive type in the JSON string but got " + data['payload']);
      } // ensure the json data is a string


      if (data['producer-encryption-key-name'] && !(typeof data['producer-encryption-key-name'] === 'string' || data['producer-encryption-key-name'] instanceof String)) {
        throw new Error("Expected the field `producer-encryption-key-name` to be a primitive type in the JSON string but got " + data['producer-encryption-key-name']);
      } // ensure the json data is a string


      if (data['revoke-sync-url'] && !(typeof data['revoke-sync-url'] === 'string' || data['revoke-sync-url'] instanceof String)) {
        throw new Error("Expected the field `revoke-sync-url` to be a primitive type in the JSON string but got " + data['revoke-sync-url']);
      } // ensure the json data is a string


      if (data['rotate-sync-url'] && !(typeof data['rotate-sync-url'] === 'string' || data['rotate-sync-url'] instanceof String)) {
        throw new Error("Expected the field `rotate-sync-url` to be a primitive type in the JSON string but got " + data['rotate-sync-url']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-ttl'] && !(typeof data['user-ttl'] === 'string' || data['user-ttl'] instanceof String)) {
        throw new Error("Expected the field `user-ttl` to be a primitive type in the JSON string but got " + data['user-ttl']);
      }

      return true;
    }
  }]);

  return DynamicSecretCreateCustom;
}();

DynamicSecretCreateCustom.RequiredProperties = ["create-sync-url", "name", "revoke-sync-url"];
/**
 * Define rotation interval in days
 * @member {Number} admin-rotation-interval-days
 */

DynamicSecretCreateCustom.prototype['admin-rotation-interval-days'] = undefined;
/**
 * URL of an endpoint that implements /sync/create method, for example https://webhook.example.com/sync/create
 * @member {String} create-sync-url
 */

DynamicSecretCreateCustom.prototype['create-sync-url'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

DynamicSecretCreateCustom.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

DynamicSecretCreateCustom.prototype['description'] = undefined;
/**
 * Should admin credentials be rotated
 * @member {Boolean} enable-admin-rotation
 * @default false
 */

DynamicSecretCreateCustom.prototype['enable-admin-rotation'] = false;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

DynamicSecretCreateCustom.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

DynamicSecretCreateCustom.prototype['name'] = undefined;
/**
 * Secret payload to be sent with each create/revoke webhook request
 * @member {String} payload
 */

DynamicSecretCreateCustom.prototype['payload'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

DynamicSecretCreateCustom.prototype['producer-encryption-key-name'] = undefined;
/**
 * URL of an endpoint that implements /sync/revoke method, for example https://webhook.example.com/sync/revoke
 * @member {String} revoke-sync-url
 */

DynamicSecretCreateCustom.prototype['revoke-sync-url'] = undefined;
/**
 * URL of an endpoint that implements /sync/rotate method, for example https://webhook.example.com/sync/rotate
 * @member {String} rotate-sync-url
 */

DynamicSecretCreateCustom.prototype['rotate-sync-url'] = undefined;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

DynamicSecretCreateCustom.prototype['tags'] = undefined;
/**
 * Maximum allowed time in seconds for the webhook to return the results
 * @member {Number} timeout-sec
 * @default 60
 */

DynamicSecretCreateCustom.prototype['timeout-sec'] = 60;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DynamicSecretCreateCustom.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DynamicSecretCreateCustom.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

DynamicSecretCreateCustom.prototype['user-ttl'] = '60m';
var _default = DynamicSecretCreateCustom;
exports["default"] = _default;
