// akeyless/dist/model/DynamicSecretTmpCredsUpdate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretTmpCredsUpdate model module.
 * @module model/DynamicSecretTmpCredsUpdate
 * @version 4.3.0
 */
var DynamicSecretTmpCredsUpdate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretTmpCredsUpdate</code>.
   * dynamicSecretTmpCredsUpdate is a command that updates dynamic secret temp creds
   * @alias module:model/DynamicSecretTmpCredsUpdate
   * @param host {String} Host
   * @param name {String} Dynamic secret name
   * @param newTtlMin {Number} New TTL in Minutes
   * @param tmpCredsId {String} Tmp Creds ID
   */
  function DynamicSecretTmpCredsUpdate(host, name, newTtlMin, tmpCredsId) {
    _classCallCheck(this, DynamicSecretTmpCredsUpdate);

    DynamicSecretTmpCredsUpdate.initialize(this, host, name, newTtlMin, tmpCredsId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretTmpCredsUpdate, null, [{
    key: "initialize",
    value: function initialize(obj, host, name, newTtlMin, tmpCredsId) {
      obj['host'] = host;
      obj['name'] = name;
      obj['new-ttl-min'] = newTtlMin;
      obj['tmp-creds-id'] = tmpCredsId;
    }
    /**
     * Constructs a <code>DynamicSecretTmpCredsUpdate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretTmpCredsUpdate} obj Optional instance to populate.
     * @return {module:model/DynamicSecretTmpCredsUpdate} The populated <code>DynamicSecretTmpCredsUpdate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretTmpCredsUpdate();

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-ttl-min')) {
          obj['new-ttl-min'] = _ApiClient["default"].convertToType(data['new-ttl-min'], 'Number');
        }

        if (data.hasOwnProperty('tmp-creds-id')) {
          obj['tmp-creds-id'] = _ApiClient["default"].convertToType(data['tmp-creds-id'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DynamicSecretTmpCredsUpdate;
}();
/**
 * Host
 * @member {String} host
 */


DynamicSecretTmpCredsUpdate.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

DynamicSecretTmpCredsUpdate.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

DynamicSecretTmpCredsUpdate.prototype['name'] = undefined;
/**
 * New TTL in Minutes
 * @member {Number} new-ttl-min
 */

DynamicSecretTmpCredsUpdate.prototype['new-ttl-min'] = undefined;
/**
 * Tmp Creds ID
 * @member {String} tmp-creds-id
 */

DynamicSecretTmpCredsUpdate.prototype['tmp-creds-id'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DynamicSecretTmpCredsUpdate.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DynamicSecretTmpCredsUpdate.prototype['uid-token'] = undefined;
var _default = DynamicSecretTmpCredsUpdate;
exports["default"] = _default;
