// akeyless/dist/model/EncryptFile.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EncryptFile model module.
 * @module model/EncryptFile
 * @version 5.0.14
 */
var EncryptFile = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EncryptFile</code>.
   * @alias module:model/EncryptFile
   * @param _in {String} Path to the file to be encrypted. If not provided, the content will be taken from stdin
   * @param keyName {String} The name of the key to use in the encryption process
   */
  function EncryptFile(_in, keyName) {
    _classCallCheck(this, EncryptFile);

    EncryptFile.initialize(this, _in, keyName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EncryptFile, null, [{
    key: "initialize",
    value: function initialize(obj, _in, keyName) {
      obj['in'] = _in;
      obj['json'] = false;
      obj['key-name'] = keyName;
    }
    /**
     * Constructs a <code>EncryptFile</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EncryptFile} obj Optional instance to populate.
     * @return {module:model/EncryptFile} The populated <code>EncryptFile</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EncryptFile();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('in')) {
          obj['in'] = _ApiClient["default"].convertToType(data['in'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('out')) {
          obj['out'] = _ApiClient["default"].convertToType(data['out'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>EncryptFile</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>EncryptFile</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(EncryptFile.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['display-id'] && !(typeof data['display-id'] === 'string' || data['display-id'] instanceof String)) {
        throw new Error("Expected the field `display-id` to be a primitive type in the JSON string but got " + data['display-id']);
      } // ensure the json data is a string


      if (data['in'] && !(typeof data['in'] === 'string' || data['in'] instanceof String)) {
        throw new Error("Expected the field `in` to be a primitive type in the JSON string but got " + data['in']);
      } // ensure the json data is a string


      if (data['key-name'] && !(typeof data['key-name'] === 'string' || data['key-name'] instanceof String)) {
        throw new Error("Expected the field `key-name` to be a primitive type in the JSON string but got " + data['key-name']);
      } // ensure the json data is a string


      if (data['out'] && !(typeof data['out'] === 'string' || data['out'] instanceof String)) {
        throw new Error("Expected the field `out` to be a primitive type in the JSON string but got " + data['out']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return EncryptFile;
}();

EncryptFile.RequiredProperties = ["in", "key-name"];
/**
 * The display id of the key to use in the encryption process
 * @member {String} display-id
 */

EncryptFile.prototype['display-id'] = undefined;
/**
 * Path to the file to be encrypted. If not provided, the content will be taken from stdin
 * @member {String} in
 */

EncryptFile.prototype['in'] = undefined;
/**
 * The item id of the key to use in the encryption process
 * @member {Number} item-id
 */

EncryptFile.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

EncryptFile.prototype['json'] = false;
/**
 * The name of the key to use in the encryption process
 * @member {String} key-name
 */

EncryptFile.prototype['key-name'] = undefined;
/**
 * Path to the output file. If not provided, the output will be sent to stdout
 * @member {String} out
 */

EncryptFile.prototype['out'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

EncryptFile.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

EncryptFile.prototype['uid-token'] = undefined;
var _default = EncryptFile;
exports["default"] = _default;
