// akeyless/dist/model/EventForwarderUpdateWebhook.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EventForwarderUpdateWebhook model module.
 * @module model/EventForwarderUpdateWebhook
 * @version 4.2.2
 */
var EventForwarderUpdateWebhook = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EventForwarderUpdateWebhook</code>.
   * eventForwarderUpdateWebhook is a command that updates webhook event forwarder
   * @alias module:model/EventForwarderUpdateWebhook
   * @param gatewaysEventSourceLocations {Array.<String>} Event sources
   * @param name {String} EventForwarder name
   */
  function EventForwarderUpdateWebhook(gatewaysEventSourceLocations, name) {
    _classCallCheck(this, EventForwarderUpdateWebhook);

    EventForwarderUpdateWebhook.initialize(this, gatewaysEventSourceLocations, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EventForwarderUpdateWebhook, null, [{
    key: "initialize",
    value: function initialize(obj, gatewaysEventSourceLocations, name) {
      obj['gateways-event-source-locations'] = gatewaysEventSourceLocations;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>EventForwarderUpdateWebhook</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EventForwarderUpdateWebhook} obj Optional instance to populate.
     * @return {module:model/EventForwarderUpdateWebhook} The populated <code>EventForwarderUpdateWebhook</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EventForwarderUpdateWebhook();

        if (data.hasOwnProperty('auth-methods-event-source-locations')) {
          obj['auth-methods-event-source-locations'] = _ApiClient["default"].convertToType(data['auth-methods-event-source-locations'], ['String']);
        }

        if (data.hasOwnProperty('auth-token')) {
          obj['auth-token'] = _ApiClient["default"].convertToType(data['auth-token'], 'String');
        }

        if (data.hasOwnProperty('auth-type')) {
          obj['auth-type'] = _ApiClient["default"].convertToType(data['auth-type'], 'String');
        }

        if (data.hasOwnProperty('client-cert-data')) {
          obj['client-cert-data'] = _ApiClient["default"].convertToType(data['client-cert-data'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('event-types')) {
          obj['event-types'] = _ApiClient["default"].convertToType(data['event-types'], ['String']);
        }

        if (data.hasOwnProperty('gateways-event-source-locations')) {
          obj['gateways-event-source-locations'] = _ApiClient["default"].convertToType(data['gateways-event-source-locations'], ['String']);
        }

        if (data.hasOwnProperty('items-event-source-locations')) {
          obj['items-event-source-locations'] = _ApiClient["default"].convertToType(data['items-event-source-locations'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('private-key-data')) {
          obj['private-key-data'] = _ApiClient["default"].convertToType(data['private-key-data'], 'String');
        }

        if (data.hasOwnProperty('server-certificates')) {
          obj['server-certificates'] = _ApiClient["default"].convertToType(data['server-certificates'], 'String');
        }

        if (data.hasOwnProperty('targets-event-source-locations')) {
          obj['targets-event-source-locations'] = _ApiClient["default"].convertToType(data['targets-event-source-locations'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return EventForwarderUpdateWebhook;
}();
/**
 * Auth Method Event sources
 * @member {Array.<String>} auth-methods-event-source-locations
 */


EventForwarderUpdateWebhook.prototype['auth-methods-event-source-locations'] = undefined;
/**
 * Base64 encoded Token string for authentication type Token
 * @member {String} auth-token
 */

EventForwarderUpdateWebhook.prototype['auth-token'] = undefined;
/**
 * The Webhook authentication type [user-pass, token, certificate]
 * @member {String} auth-type
 * @default 'user-pass'
 */

EventForwarderUpdateWebhook.prototype['auth-type'] = 'user-pass';
/**
 * Base64 encoded PEM certificate, relevant for certificate auth-type
 * @member {String} client-cert-data
 */

EventForwarderUpdateWebhook.prototype['client-cert-data'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

EventForwarderUpdateWebhook.prototype['description'] = undefined;
/**
 * Enable/Disable Event Forwarder [true/false]
 * @member {String} enable
 * @default 'true'
 */

EventForwarderUpdateWebhook.prototype['enable'] = 'true';
/**
 * List of event types to notify about [request-access, certificate-pending-expiration, certificate-expired, certificate-provisioning-success, certificate-provisioning-failure, auth-method-pending-expiration, auth-method-expired, rotated-secret-success, rotated-secret-failure, dynamic-secret-failure, multi-auth-failure, uid-rotation-failure, apply-justification, email-auth-method-approved, usage, rotation-usage, gateway-inactive, static-secret-updated]
 * @member {Array.<String>} event-types
 */

EventForwarderUpdateWebhook.prototype['event-types'] = undefined;
/**
 * Event sources
 * @member {Array.<String>} gateways-event-source-locations
 */

EventForwarderUpdateWebhook.prototype['gateways-event-source-locations'] = undefined;
/**
 * Items Event sources
 * @member {Array.<String>} items-event-source-locations
 */

EventForwarderUpdateWebhook.prototype['items-event-source-locations'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

EventForwarderUpdateWebhook.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

EventForwarderUpdateWebhook.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the EventForwarder secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

EventForwarderUpdateWebhook.prototype['key'] = undefined;
/**
 * EventForwarder name
 * @member {String} name
 */

EventForwarderUpdateWebhook.prototype['name'] = undefined;
/**
 * New EventForwarder name
 * @member {String} new-name
 */

EventForwarderUpdateWebhook.prototype['new-name'] = undefined;
/**
 * Password for authentication relevant for user-pass auth-type
 * @member {String} password
 */

EventForwarderUpdateWebhook.prototype['password'] = undefined;
/**
 * Base64 encoded PEM RSA Private Key, relevant for certificate auth-type
 * @member {String} private-key-data
 */

EventForwarderUpdateWebhook.prototype['private-key-data'] = undefined;
/**
 * Base64 encoded PEM certificate of the Webhook
 * @member {String} server-certificates
 */

EventForwarderUpdateWebhook.prototype['server-certificates'] = undefined;
/**
 * Targets Event sources
 * @member {Array.<String>} targets-event-source-locations
 */

EventForwarderUpdateWebhook.prototype['targets-event-source-locations'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

EventForwarderUpdateWebhook.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

EventForwarderUpdateWebhook.prototype['uid-token'] = undefined;
/**
 * Webhook URL
 * @member {String} url
 */

EventForwarderUpdateWebhook.prototype['url'] = undefined;
/**
 * Username for authentication relevant for user-pass auth-type
 * @member {String} username
 */

EventForwarderUpdateWebhook.prototype['username'] = undefined;
var _default = EventForwarderUpdateWebhook;
exports["default"] = _default;
