// akeyless/dist/model/ExportClassicKeyOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ExportClassicKeyOutput model module.
 * @module model/ExportClassicKeyOutput
 * @version 5.0.8
 */
var ExportClassicKeyOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ExportClassicKeyOutput</code>.
   * @alias module:model/ExportClassicKeyOutput
   */
  function ExportClassicKeyOutput() {
    _classCallCheck(this, ExportClassicKeyOutput);

    ExportClassicKeyOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ExportClassicKeyOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ExportClassicKeyOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ExportClassicKeyOutput} obj Optional instance to populate.
     * @return {module:model/ExportClassicKeyOutput} The populated <code>ExportClassicKeyOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ExportClassicKeyOutput();

        if (data.hasOwnProperty('certificatePem')) {
          obj['certificatePem'] = _ApiClient["default"].convertToType(data['certificatePem'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('ssh')) {
          obj['ssh'] = _ApiClient["default"].convertToType(data['ssh'], 'String');
        }

        if (data.hasOwnProperty('wrapping-iv')) {
          obj['wrapping-iv'] = _ApiClient["default"].convertToType(data['wrapping-iv'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ExportClassicKeyOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ExportClassicKeyOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['certificatePem'] && !(typeof data['certificatePem'] === 'string' || data['certificatePem'] instanceof String)) {
        throw new Error("Expected the field `certificatePem` to be a primitive type in the JSON string but got " + data['certificatePem']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['ssh'] && !(typeof data['ssh'] === 'string' || data['ssh'] instanceof String)) {
        throw new Error("Expected the field `ssh` to be a primitive type in the JSON string but got " + data['ssh']);
      } // ensure the json data is a string


      if (data['wrapping-iv'] && !(typeof data['wrapping-iv'] === 'string' || data['wrapping-iv'] instanceof String)) {
        throw new Error("Expected the field `wrapping-iv` to be a primitive type in the JSON string but got " + data['wrapping-iv']);
      }

      return true;
    }
  }]);

  return ExportClassicKeyOutput;
}();
/**
 * @member {String} certificatePem
 */


ExportClassicKeyOutput.prototype['certificatePem'] = undefined;
/**
 * @member {String} key
 */

ExportClassicKeyOutput.prototype['key'] = undefined;
/**
 * @member {String} ssh
 */

ExportClassicKeyOutput.prototype['ssh'] = undefined;
/**
 * @member {String} wrapping-iv
 */

ExportClassicKeyOutput.prototype['wrapping-iv'] = undefined;
var _default = ExportClassicKeyOutput;
exports["default"] = _default;
