// akeyless/dist/model/ExternalKMSKeyId.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ExternalKMSKeyId model module.
 * @module model/ExternalKMSKeyId
 * @version 5.0.5
 */
var ExternalKMSKeyId = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ExternalKMSKeyId</code>.
   * @alias module:model/ExternalKMSKeyId
   */
  function ExternalKMSKeyId() {
    _classCallCheck(this, ExternalKMSKeyId);

    ExternalKMSKeyId.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ExternalKMSKeyId, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ExternalKMSKeyId</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ExternalKMSKeyId} obj Optional instance to populate.
     * @return {module:model/ExternalKMSKeyId} The populated <code>ExternalKMSKeyId</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ExternalKMSKeyId();

        if (data.hasOwnProperty('key_id')) {
          obj['key_id'] = _ApiClient["default"].convertToType(data['key_id'], 'String');
        }

        if (data.hasOwnProperty('key_reference')) {
          obj['key_reference'] = _ApiClient["default"].convertToType(data['key_reference'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ExternalKMSKeyId</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ExternalKMSKeyId</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['key_id'] && !(typeof data['key_id'] === 'string' || data['key_id'] instanceof String)) {
        throw new Error("Expected the field `key_id` to be a primitive type in the JSON string but got " + data['key_id']);
      } // ensure the json data is a string


      if (data['key_reference'] && !(typeof data['key_reference'] === 'string' || data['key_reference'] instanceof String)) {
        throw new Error("Expected the field `key_reference` to be a primitive type in the JSON string but got " + data['key_reference']);
      }

      return true;
    }
  }]);

  return ExternalKMSKeyId;
}();
/**
 * @member {String} key_id
 */


ExternalKMSKeyId.prototype['key_id'] = undefined;
/**
 * @member {String} key_reference
 */

ExternalKMSKeyId.prototype['key_reference'] = undefined;
var _default = ExternalKMSKeyId;
exports["default"] = _default;
