// akeyless/dist/model/GatewayUpdateProducerGcp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerGcp model module.
 * @module model/GatewayUpdateProducerGcp
 * @version 5.0.20
 */
var GatewayUpdateProducerGcp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerGcp</code>.
   * gatewayUpdateProducerGcp is a command that updates a GCP producer [Deprecated: Use dynamic-secret-update-gcp command]
   * @alias module:model/GatewayUpdateProducerGcp
   * @param name {String} Dynamic secret name
   */
  function GatewayUpdateProducerGcp(name) {
    _classCallCheck(this, GatewayUpdateProducerGcp);

    GatewayUpdateProducerGcp.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerGcp, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['fixed-user-claim-keyname'] = 'ext_email';
      obj['json'] = false;
      obj['name'] = name;
      obj['secure-access-web-browsing'] = false;
      obj['secure-access-web-proxy'] = false;
      obj['service-account-type'] = 'fixed';
      obj['user-ttl'] = '60m';
    }
    /**
     * Constructs a <code>GatewayUpdateProducerGcp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerGcp} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerGcp} The populated <code>GatewayUpdateProducerGcp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerGcp();

        if (data.hasOwnProperty('access-type')) {
          obj['access-type'] = _ApiClient["default"].convertToType(data['access-type'], 'String');
        }

        if (data.hasOwnProperty('custom-username-template')) {
          obj['custom-username-template'] = _ApiClient["default"].convertToType(data['custom-username-template'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('fixed-user-claim-keyname')) {
          obj['fixed-user-claim-keyname'] = _ApiClient["default"].convertToType(data['fixed-user-claim-keyname'], 'String');
        }

        if (data.hasOwnProperty('gcp-cred-type')) {
          obj['gcp-cred-type'] = _ApiClient["default"].convertToType(data['gcp-cred-type'], 'String');
        }

        if (data.hasOwnProperty('gcp-key')) {
          obj['gcp-key'] = _ApiClient["default"].convertToType(data['gcp-key'], 'String');
        }

        if (data.hasOwnProperty('gcp-key-algo')) {
          obj['gcp-key-algo'] = _ApiClient["default"].convertToType(data['gcp-key-algo'], 'String');
        }

        if (data.hasOwnProperty('gcp-project-id')) {
          obj['gcp-project-id'] = _ApiClient["default"].convertToType(data['gcp-project-id'], 'String');
        }

        if (data.hasOwnProperty('gcp-sa-email')) {
          obj['gcp-sa-email'] = _ApiClient["default"].convertToType(data['gcp-sa-email'], 'String');
        }

        if (data.hasOwnProperty('gcp-token-scopes')) {
          obj['gcp-token-scopes'] = _ApiClient["default"].convertToType(data['gcp-token-scopes'], 'String');
        }

        if (data.hasOwnProperty('item-custom-fields')) {
          obj['item-custom-fields'] = _ApiClient["default"].convertToType(data['item-custom-fields'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('role-binding')) {
          obj['role-binding'] = _ApiClient["default"].convertToType(data['role-binding'], 'String');
        }

        if (data.hasOwnProperty('role-names')) {
          obj['role-names'] = _ApiClient["default"].convertToType(data['role-names'], 'String');
        }

        if (data.hasOwnProperty('secure-access-delay')) {
          obj['secure-access-delay'] = _ApiClient["default"].convertToType(data['secure-access-delay'], 'Number');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-url')) {
          obj['secure-access-url'] = _ApiClient["default"].convertToType(data['secure-access-url'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web-browsing')) {
          obj['secure-access-web-browsing'] = _ApiClient["default"].convertToType(data['secure-access-web-browsing'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-proxy')) {
          obj['secure-access-web-proxy'] = _ApiClient["default"].convertToType(data['secure-access-web-proxy'], 'Boolean');
        }

        if (data.hasOwnProperty('service-account-type')) {
          obj['service-account-type'] = _ApiClient["default"].convertToType(data['service-account-type'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayUpdateProducerGcp</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayUpdateProducerGcp</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GatewayUpdateProducerGcp.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['access-type'] && !(typeof data['access-type'] === 'string' || data['access-type'] instanceof String)) {
        throw new Error("Expected the field `access-type` to be a primitive type in the JSON string but got " + data['access-type']);
      } // ensure the json data is a string


      if (data['custom-username-template'] && !(typeof data['custom-username-template'] === 'string' || data['custom-username-template'] instanceof String)) {
        throw new Error("Expected the field `custom-username-template` to be a primitive type in the JSON string but got " + data['custom-username-template']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['fixed-user-claim-keyname'] && !(typeof data['fixed-user-claim-keyname'] === 'string' || data['fixed-user-claim-keyname'] instanceof String)) {
        throw new Error("Expected the field `fixed-user-claim-keyname` to be a primitive type in the JSON string but got " + data['fixed-user-claim-keyname']);
      } // ensure the json data is a string


      if (data['gcp-cred-type'] && !(typeof data['gcp-cred-type'] === 'string' || data['gcp-cred-type'] instanceof String)) {
        throw new Error("Expected the field `gcp-cred-type` to be a primitive type in the JSON string but got " + data['gcp-cred-type']);
      } // ensure the json data is a string


      if (data['gcp-key'] && !(typeof data['gcp-key'] === 'string' || data['gcp-key'] instanceof String)) {
        throw new Error("Expected the field `gcp-key` to be a primitive type in the JSON string but got " + data['gcp-key']);
      } // ensure the json data is a string


      if (data['gcp-key-algo'] && !(typeof data['gcp-key-algo'] === 'string' || data['gcp-key-algo'] instanceof String)) {
        throw new Error("Expected the field `gcp-key-algo` to be a primitive type in the JSON string but got " + data['gcp-key-algo']);
      } // ensure the json data is a string


      if (data['gcp-project-id'] && !(typeof data['gcp-project-id'] === 'string' || data['gcp-project-id'] instanceof String)) {
        throw new Error("Expected the field `gcp-project-id` to be a primitive type in the JSON string but got " + data['gcp-project-id']);
      } // ensure the json data is a string


      if (data['gcp-sa-email'] && !(typeof data['gcp-sa-email'] === 'string' || data['gcp-sa-email'] instanceof String)) {
        throw new Error("Expected the field `gcp-sa-email` to be a primitive type in the JSON string but got " + data['gcp-sa-email']);
      } // ensure the json data is a string


      if (data['gcp-token-scopes'] && !(typeof data['gcp-token-scopes'] === 'string' || data['gcp-token-scopes'] instanceof String)) {
        throw new Error("Expected the field `gcp-token-scopes` to be a primitive type in the JSON string but got " + data['gcp-token-scopes']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['producer-encryption-key-name'] && !(typeof data['producer-encryption-key-name'] === 'string' || data['producer-encryption-key-name'] instanceof String)) {
        throw new Error("Expected the field `producer-encryption-key-name` to be a primitive type in the JSON string but got " + data['producer-encryption-key-name']);
      } // ensure the json data is a string


      if (data['role-binding'] && !(typeof data['role-binding'] === 'string' || data['role-binding'] instanceof String)) {
        throw new Error("Expected the field `role-binding` to be a primitive type in the JSON string but got " + data['role-binding']);
      } // ensure the json data is a string


      if (data['role-names'] && !(typeof data['role-names'] === 'string' || data['role-names'] instanceof String)) {
        throw new Error("Expected the field `role-names` to be a primitive type in the JSON string but got " + data['role-names']);
      } // ensure the json data is a string


      if (data['secure-access-enable'] && !(typeof data['secure-access-enable'] === 'string' || data['secure-access-enable'] instanceof String)) {
        throw new Error("Expected the field `secure-access-enable` to be a primitive type in the JSON string but got " + data['secure-access-enable']);
      } // ensure the json data is a string


      if (data['secure-access-url'] && !(typeof data['secure-access-url'] === 'string' || data['secure-access-url'] instanceof String)) {
        throw new Error("Expected the field `secure-access-url` to be a primitive type in the JSON string but got " + data['secure-access-url']);
      } // ensure the json data is a string


      if (data['service-account-type'] && !(typeof data['service-account-type'] === 'string' || data['service-account-type'] instanceof String)) {
        throw new Error("Expected the field `service-account-type` to be a primitive type in the JSON string but got " + data['service-account-type']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-ttl'] && !(typeof data['user-ttl'] === 'string' || data['user-ttl'] instanceof String)) {
        throw new Error("Expected the field `user-ttl` to be a primitive type in the JSON string but got " + data['user-ttl']);
      }

      return true;
    }
  }]);

  return GatewayUpdateProducerGcp;
}();

GatewayUpdateProducerGcp.RequiredProperties = ["name"];
/**
 * @member {String} access-type
 */

GatewayUpdateProducerGcp.prototype['access-type'] = undefined;
/**
 * Customize how temporary usernames are generated using go template
 * @member {String} custom-username-template
 */

GatewayUpdateProducerGcp.prototype['custom-username-template'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

GatewayUpdateProducerGcp.prototype['delete_protection'] = undefined;
/**
 * For externally provided users, denotes the key-name of IdP claim to extract the username from (Relevant only when --access-type=external)
 * @member {String} fixed-user-claim-keyname
 * @default 'ext_email'
 */

GatewayUpdateProducerGcp.prototype['fixed-user-claim-keyname'] = 'ext_email';
/**
 * @member {String} gcp-cred-type
 */

GatewayUpdateProducerGcp.prototype['gcp-cred-type'] = undefined;
/**
 * Base64-encoded service account private key text
 * @member {String} gcp-key
 */

GatewayUpdateProducerGcp.prototype['gcp-key'] = undefined;
/**
 * Service account key algorithm, e.g. KEY_ALG_RSA_1024 (Relevant only when --access-type=sa and --gcp-cred-type=key)
 * @member {String} gcp-key-algo
 */

GatewayUpdateProducerGcp.prototype['gcp-key-algo'] = undefined;
/**
 * GCP Project ID override for dynamic secret operations
 * @member {String} gcp-project-id
 */

GatewayUpdateProducerGcp.prototype['gcp-project-id'] = undefined;
/**
 * The email of the fixed service account to generate keys or tokens for (Relevant only when --access-type=sa and --service-account-type=fixed)
 * @member {String} gcp-sa-email
 */

GatewayUpdateProducerGcp.prototype['gcp-sa-email'] = undefined;
/**
 * Access token scopes list, e.g. scope1,scope2 (Relevant only when --access-type=sa; required when --gcp-cred-type=token)
 * @member {String} gcp-token-scopes
 */

GatewayUpdateProducerGcp.prototype['gcp-token-scopes'] = undefined;
/**
 * Additional custom fields to associate with the item
 * @member {Object.<String, String>} item-custom-fields
 */

GatewayUpdateProducerGcp.prototype['item-custom-fields'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateProducerGcp.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

GatewayUpdateProducerGcp.prototype['name'] = undefined;
/**
 * Dynamic secret name
 * @member {String} new-name
 */

GatewayUpdateProducerGcp.prototype['new-name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerGcp.prototype['producer-encryption-key-name'] = undefined;
/**
 * Role binding definitions in JSON format (Relevant only when --access-type=sa and --service-account-type=dynamic)
 * @member {String} role-binding
 */

GatewayUpdateProducerGcp.prototype['role-binding'] = undefined;
/**
 * Comma-separated list of GCP roles to assign to the user (Relevant only when --access-type=external)
 * @member {String} role-names
 */

GatewayUpdateProducerGcp.prototype['role-names'] = undefined;
/**
 * The delay duration, in seconds, to wait after generating just-in-time credentials. Accepted range: 0-120 seconds
 * @member {Number} secure-access-delay
 */

GatewayUpdateProducerGcp.prototype['secure-access-delay'] = undefined;
/**
 * Enable/Disable secure remote access [true/false]
 * @member {String} secure-access-enable
 */

GatewayUpdateProducerGcp.prototype['secure-access-enable'] = undefined;
/**
 * Destination URL to inject secrets
 * @member {String} secure-access-url
 */

GatewayUpdateProducerGcp.prototype['secure-access-url'] = undefined;
/**
 * Secure browser via Akeyless's Secure Remote Access (SRA)
 * @member {Boolean} secure-access-web-browsing
 * @default false
 */

GatewayUpdateProducerGcp.prototype['secure-access-web-browsing'] = false;
/**
 * Web-Proxy via Akeyless's Secure Remote Access (SRA)
 * @member {Boolean} secure-access-web-proxy
 * @default false
 */

GatewayUpdateProducerGcp.prototype['secure-access-web-proxy'] = false;
/**
 * The type of the GCP service account. Options [fixed, dynamic] (Relevant only when --access-type=sa)
 * @member {String} service-account-type
 * @default 'fixed'
 */

GatewayUpdateProducerGcp.prototype['service-account-type'] = 'fixed';
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerGcp.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerGcp.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerGcp.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerGcp.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerGcp.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerGcp;
exports["default"] = _default;
