// akeyless/dist/model/GatewayUpdateProducerEks.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerEks model module.
 * @module model/GatewayUpdateProducerEks
 * @version 5.0.5
 */
var GatewayUpdateProducerEks = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerEks</code>.
   * gatewayUpdateProducerEks is a command that updates eks producer [Deprecated: Use dynamic-secret-update-eks command]
   * @alias module:model/GatewayUpdateProducerEks
   * @param name {String} Dynamic secret name
   */
  function GatewayUpdateProducerEks(name) {
    _classCallCheck(this, GatewayUpdateProducerEks);

    GatewayUpdateProducerEks.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerEks, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['eks-region'] = 'us-east-2';
      obj['json'] = false;
      obj['name'] = name;
      obj['secure-access-web'] = false;
      obj['user-ttl'] = '15m';
    }
    /**
     * Constructs a <code>GatewayUpdateProducerEks</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerEks} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerEks} The populated <code>GatewayUpdateProducerEks</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerEks();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('eks-access-key-id')) {
          obj['eks-access-key-id'] = _ApiClient["default"].convertToType(data['eks-access-key-id'], 'String');
        }

        if (data.hasOwnProperty('eks-assume-role')) {
          obj['eks-assume-role'] = _ApiClient["default"].convertToType(data['eks-assume-role'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-ca-cert')) {
          obj['eks-cluster-ca-cert'] = _ApiClient["default"].convertToType(data['eks-cluster-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-endpoint')) {
          obj['eks-cluster-endpoint'] = _ApiClient["default"].convertToType(data['eks-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-name')) {
          obj['eks-cluster-name'] = _ApiClient["default"].convertToType(data['eks-cluster-name'], 'String');
        }

        if (data.hasOwnProperty('eks-region')) {
          obj['eks-region'] = _ApiClient["default"].convertToType(data['eks-region'], 'String');
        }

        if (data.hasOwnProperty('eks-secret-access-key')) {
          obj['eks-secret-access-key'] = _ApiClient["default"].convertToType(data['eks-secret-access-key'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-allow-port-forwading')) {
          obj['secure-access-allow-port-forwading'] = _ApiClient["default"].convertToType(data['secure-access-allow-port-forwading'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-certificate-issuer')) {
          obj['secure-access-certificate-issuer'] = _ApiClient["default"].convertToType(data['secure-access-certificate-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-cluster-endpoint')) {
          obj['secure-access-cluster-endpoint'] = _ApiClient["default"].convertToType(data['secure-access-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('secure-access-delay')) {
          obj['secure-access-delay'] = _ApiClient["default"].convertToType(data['secure-access-delay'], 'Number');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayUpdateProducerEks</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayUpdateProducerEks</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GatewayUpdateProducerEks.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['eks-access-key-id'] && !(typeof data['eks-access-key-id'] === 'string' || data['eks-access-key-id'] instanceof String)) {
        throw new Error("Expected the field `eks-access-key-id` to be a primitive type in the JSON string but got " + data['eks-access-key-id']);
      } // ensure the json data is a string


      if (data['eks-assume-role'] && !(typeof data['eks-assume-role'] === 'string' || data['eks-assume-role'] instanceof String)) {
        throw new Error("Expected the field `eks-assume-role` to be a primitive type in the JSON string but got " + data['eks-assume-role']);
      } // ensure the json data is a string


      if (data['eks-cluster-ca-cert'] && !(typeof data['eks-cluster-ca-cert'] === 'string' || data['eks-cluster-ca-cert'] instanceof String)) {
        throw new Error("Expected the field `eks-cluster-ca-cert` to be a primitive type in the JSON string but got " + data['eks-cluster-ca-cert']);
      } // ensure the json data is a string


      if (data['eks-cluster-endpoint'] && !(typeof data['eks-cluster-endpoint'] === 'string' || data['eks-cluster-endpoint'] instanceof String)) {
        throw new Error("Expected the field `eks-cluster-endpoint` to be a primitive type in the JSON string but got " + data['eks-cluster-endpoint']);
      } // ensure the json data is a string


      if (data['eks-cluster-name'] && !(typeof data['eks-cluster-name'] === 'string' || data['eks-cluster-name'] instanceof String)) {
        throw new Error("Expected the field `eks-cluster-name` to be a primitive type in the JSON string but got " + data['eks-cluster-name']);
      } // ensure the json data is a string


      if (data['eks-region'] && !(typeof data['eks-region'] === 'string' || data['eks-region'] instanceof String)) {
        throw new Error("Expected the field `eks-region` to be a primitive type in the JSON string but got " + data['eks-region']);
      } // ensure the json data is a string


      if (data['eks-secret-access-key'] && !(typeof data['eks-secret-access-key'] === 'string' || data['eks-secret-access-key'] instanceof String)) {
        throw new Error("Expected the field `eks-secret-access-key` to be a primitive type in the JSON string but got " + data['eks-secret-access-key']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['producer-encryption-key-name'] && !(typeof data['producer-encryption-key-name'] === 'string' || data['producer-encryption-key-name'] instanceof String)) {
        throw new Error("Expected the field `producer-encryption-key-name` to be a primitive type in the JSON string but got " + data['producer-encryption-key-name']);
      } // ensure the json data is a string


      if (data['secure-access-bastion-issuer'] && !(typeof data['secure-access-bastion-issuer'] === 'string' || data['secure-access-bastion-issuer'] instanceof String)) {
        throw new Error("Expected the field `secure-access-bastion-issuer` to be a primitive type in the JSON string but got " + data['secure-access-bastion-issuer']);
      } // ensure the json data is a string


      if (data['secure-access-certificate-issuer'] && !(typeof data['secure-access-certificate-issuer'] === 'string' || data['secure-access-certificate-issuer'] instanceof String)) {
        throw new Error("Expected the field `secure-access-certificate-issuer` to be a primitive type in the JSON string but got " + data['secure-access-certificate-issuer']);
      } // ensure the json data is a string


      if (data['secure-access-cluster-endpoint'] && !(typeof data['secure-access-cluster-endpoint'] === 'string' || data['secure-access-cluster-endpoint'] instanceof String)) {
        throw new Error("Expected the field `secure-access-cluster-endpoint` to be a primitive type in the JSON string but got " + data['secure-access-cluster-endpoint']);
      } // ensure the json data is a string


      if (data['secure-access-enable'] && !(typeof data['secure-access-enable'] === 'string' || data['secure-access-enable'] instanceof String)) {
        throw new Error("Expected the field `secure-access-enable` to be a primitive type in the JSON string but got " + data['secure-access-enable']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-ttl'] && !(typeof data['user-ttl'] === 'string' || data['user-ttl'] instanceof String)) {
        throw new Error("Expected the field `user-ttl` to be a primitive type in the JSON string but got " + data['user-ttl']);
      }

      return true;
    }
  }]);

  return GatewayUpdateProducerEks;
}();

GatewayUpdateProducerEks.RequiredProperties = ["name"];
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

GatewayUpdateProducerEks.prototype['delete_protection'] = undefined;
/**
 * Access Key ID
 * @member {String} eks-access-key-id
 */

GatewayUpdateProducerEks.prototype['eks-access-key-id'] = undefined;
/**
 * IAM assume role
 * @member {String} eks-assume-role
 */

GatewayUpdateProducerEks.prototype['eks-assume-role'] = undefined;
/**
 * EKS cluster CA certificate
 * @member {String} eks-cluster-ca-cert
 */

GatewayUpdateProducerEks.prototype['eks-cluster-ca-cert'] = undefined;
/**
 * EKS cluster URL endpoint
 * @member {String} eks-cluster-endpoint
 */

GatewayUpdateProducerEks.prototype['eks-cluster-endpoint'] = undefined;
/**
 * EKS cluster name
 * @member {String} eks-cluster-name
 */

GatewayUpdateProducerEks.prototype['eks-cluster-name'] = undefined;
/**
 * Region
 * @member {String} eks-region
 * @default 'us-east-2'
 */

GatewayUpdateProducerEks.prototype['eks-region'] = 'us-east-2';
/**
 * Secret Access Key
 * @member {String} eks-secret-access-key
 */

GatewayUpdateProducerEks.prototype['eks-secret-access-key'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateProducerEks.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

GatewayUpdateProducerEks.prototype['name'] = undefined;
/**
 * Dynamic secret name
 * @member {String} new-name
 */

GatewayUpdateProducerEks.prototype['new-name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerEks.prototype['producer-encryption-key-name'] = undefined;
/**
 * Enable Port forwarding while using CLI access
 * @member {Boolean} secure-access-allow-port-forwading
 */

GatewayUpdateProducerEks.prototype['secure-access-allow-port-forwading'] = undefined;
/**
 * Deprecated. use secure-access-certificate-issuer
 * @member {String} secure-access-bastion-issuer
 */

GatewayUpdateProducerEks.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * Path to the SSH Certificate Issuer for your Akeyless Secure Access
 * @member {String} secure-access-certificate-issuer
 */

GatewayUpdateProducerEks.prototype['secure-access-certificate-issuer'] = undefined;
/**
 * The K8s cluster endpoint URL
 * @member {String} secure-access-cluster-endpoint
 */

GatewayUpdateProducerEks.prototype['secure-access-cluster-endpoint'] = undefined;
/**
 * The delay duration, in seconds, to wait after generating just-in-time credentials. Accepted range: 0-120 seconds
 * @member {Number} secure-access-delay
 */

GatewayUpdateProducerEks.prototype['secure-access-delay'] = undefined;
/**
 * Enable/Disable secure remote access [true/false]
 * @member {String} secure-access-enable
 */

GatewayUpdateProducerEks.prototype['secure-access-enable'] = undefined;
/**
 * Enable Web Secure Remote Access
 * @member {Boolean} secure-access-web
 * @default false
 */

GatewayUpdateProducerEks.prototype['secure-access-web'] = false;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerEks.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerEks.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerEks.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerEks.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '15m'
 */

GatewayUpdateProducerEks.prototype['user-ttl'] = '15m';
var _default = GatewayUpdateProducerEks;
exports["default"] = _default;
