// akeyless/dist/model/GatewayUpdateRemoteAccessDesktopApp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateRemoteAccessDesktopApp model module.
 * @module model/GatewayUpdateRemoteAccessDesktopApp
 * @version 5.0.5
 */
var GatewayUpdateRemoteAccessDesktopApp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateRemoteAccessDesktopApp</code>.
   * gatewayUpdateRemoteAccessDesktopApp is a command that update remote access desktop app config
   * @alias module:model/GatewayUpdateRemoteAccessDesktopApp
   */
  function GatewayUpdateRemoteAccessDesktopApp() {
    _classCallCheck(this, GatewayUpdateRemoteAccessDesktopApp);

    GatewayUpdateRemoteAccessDesktopApp.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateRemoteAccessDesktopApp, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
    }
    /**
     * Constructs a <code>GatewayUpdateRemoteAccessDesktopApp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateRemoteAccessDesktopApp} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateRemoteAccessDesktopApp} The populated <code>GatewayUpdateRemoteAccessDesktopApp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateRemoteAccessDesktopApp();

        if (data.hasOwnProperty('desktop-app-secure-web-access-url')) {
          obj['desktop-app-secure-web-access-url'] = _ApiClient["default"].convertToType(data['desktop-app-secure-web-access-url'], 'String');
        }

        if (data.hasOwnProperty('desktop-app-secure-web-proxy')) {
          obj['desktop-app-secure-web-proxy'] = _ApiClient["default"].convertToType(data['desktop-app-secure-web-proxy'], 'String');
        }

        if (data.hasOwnProperty('desktop-app-ssh-cert-issuer')) {
          obj['desktop-app-ssh-cert-issuer'] = _ApiClient["default"].convertToType(data['desktop-app-ssh-cert-issuer'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayUpdateRemoteAccessDesktopApp</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayUpdateRemoteAccessDesktopApp</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['desktop-app-secure-web-access-url'] && !(typeof data['desktop-app-secure-web-access-url'] === 'string' || data['desktop-app-secure-web-access-url'] instanceof String)) {
        throw new Error("Expected the field `desktop-app-secure-web-access-url` to be a primitive type in the JSON string but got " + data['desktop-app-secure-web-access-url']);
      } // ensure the json data is a string


      if (data['desktop-app-secure-web-proxy'] && !(typeof data['desktop-app-secure-web-proxy'] === 'string' || data['desktop-app-secure-web-proxy'] instanceof String)) {
        throw new Error("Expected the field `desktop-app-secure-web-proxy` to be a primitive type in the JSON string but got " + data['desktop-app-secure-web-proxy']);
      } // ensure the json data is a string


      if (data['desktop-app-ssh-cert-issuer'] && !(typeof data['desktop-app-ssh-cert-issuer'] === 'string' || data['desktop-app-ssh-cert-issuer'] instanceof String)) {
        throw new Error("Expected the field `desktop-app-ssh-cert-issuer` to be a primitive type in the JSON string but got " + data['desktop-app-ssh-cert-issuer']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return GatewayUpdateRemoteAccessDesktopApp;
}();
/**
 * @member {String} desktop-app-secure-web-access-url
 */


GatewayUpdateRemoteAccessDesktopApp.prototype['desktop-app-secure-web-access-url'] = undefined;
/**
 * @member {String} desktop-app-secure-web-proxy
 */

GatewayUpdateRemoteAccessDesktopApp.prototype['desktop-app-secure-web-proxy'] = undefined;
/**
 * @member {String} desktop-app-ssh-cert-issuer
 */

GatewayUpdateRemoteAccessDesktopApp.prototype['desktop-app-ssh-cert-issuer'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateRemoteAccessDesktopApp.prototype['json'] = false;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateRemoteAccessDesktopApp.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateRemoteAccessDesktopApp.prototype['uid-token'] = undefined;
var _default = GatewayUpdateRemoteAccessDesktopApp;
exports["default"] = _default;
