// akeyless/dist/model/GatewayUpdateLogForwardingLogstash.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateLogForwardingLogstash model module.
 * @module model/GatewayUpdateLogForwardingLogstash
 * @version 4.1.0
 */
var GatewayUpdateLogForwardingLogstash = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateLogForwardingLogstash</code>.
   * gatewayUpdateLogForwardingLogstash is a command that updates log forwarding config (logstash target)
   * @alias module:model/GatewayUpdateLogForwardingLogstash
   */
  function GatewayUpdateLogForwardingLogstash() {
    _classCallCheck(this, GatewayUpdateLogForwardingLogstash);

    GatewayUpdateLogForwardingLogstash.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateLogForwardingLogstash, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayUpdateLogForwardingLogstash</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateLogForwardingLogstash} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateLogForwardingLogstash} The populated <code>GatewayUpdateLogForwardingLogstash</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateLogForwardingLogstash();

        if (data.hasOwnProperty('dns')) {
          obj['dns'] = _ApiClient["default"].convertToType(data['dns'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('enable-tls')) {
          obj['enable-tls'] = _ApiClient["default"].convertToType(data['enable-tls'], 'Boolean');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('protocol')) {
          obj['protocol'] = _ApiClient["default"].convertToType(data['protocol'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('tls-certificate')) {
          obj['tls-certificate'] = _ApiClient["default"].convertToType(data['tls-certificate'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateLogForwardingLogstash;
}();
/**
 * Logstash dns
 * @member {String} dns
 */


GatewayUpdateLogForwardingLogstash.prototype['dns'] = undefined;
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */

GatewayUpdateLogForwardingLogstash.prototype['enable'] = 'true';
/**
 * Enable tls
 * @member {Boolean} enable-tls
 */

GatewayUpdateLogForwardingLogstash.prototype['enable-tls'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateLogForwardingLogstash.prototype['json'] = false;
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GatewayUpdateLogForwardingLogstash.prototype['output-format'] = 'text';
/**
 * Logstash protocol [tcp/udp]
 * @member {String} protocol
 */

GatewayUpdateLogForwardingLogstash.prototype['protocol'] = undefined;
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GatewayUpdateLogForwardingLogstash.prototype['pull-interval'] = '10';
/**
 * Logstash tls certificate
 * @member {String} tls-certificate
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingLogstash.prototype['tls-certificate'] = 'use-existing';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateLogForwardingLogstash.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateLogForwardingLogstash.prototype['uid-token'] = undefined;
var _default = GatewayUpdateLogForwardingLogstash;
exports["default"] = _default;
