// akeyless/dist/model/GatewayUpdateLogForwardingElasticsearch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateLogForwardingElasticsearch model module.
 * @module model/GatewayUpdateLogForwardingElasticsearch
 * @version 5.0.8
 */
var GatewayUpdateLogForwardingElasticsearch = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateLogForwardingElasticsearch</code>.
   * gatewayUpdateLogForwardingElasticsearch is a command that updates log forwarding config (elasticsearch target)
   * @alias module:model/GatewayUpdateLogForwardingElasticsearch
   */
  function GatewayUpdateLogForwardingElasticsearch() {
    _classCallCheck(this, GatewayUpdateLogForwardingElasticsearch);

    GatewayUpdateLogForwardingElasticsearch.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateLogForwardingElasticsearch, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['enable'] = 'true';
      obj['json'] = false;
      obj['output-format'] = 'text';
      obj['pull-interval'] = '10';
      obj['tls-certificate'] = 'use-existing';
    }
    /**
     * Constructs a <code>GatewayUpdateLogForwardingElasticsearch</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateLogForwardingElasticsearch} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateLogForwardingElasticsearch} The populated <code>GatewayUpdateLogForwardingElasticsearch</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateLogForwardingElasticsearch();

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('auth-type')) {
          obj['auth-type'] = _ApiClient["default"].convertToType(data['auth-type'], 'String');
        }

        if (data.hasOwnProperty('cloud-id')) {
          obj['cloud-id'] = _ApiClient["default"].convertToType(data['cloud-id'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('enable-tls')) {
          obj['enable-tls'] = _ApiClient["default"].convertToType(data['enable-tls'], 'Boolean');
        }

        if (data.hasOwnProperty('index')) {
          obj['index'] = _ApiClient["default"].convertToType(data['index'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('nodes')) {
          obj['nodes'] = _ApiClient["default"].convertToType(data['nodes'], 'String');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('server-type')) {
          obj['server-type'] = _ApiClient["default"].convertToType(data['server-type'], 'String');
        }

        if (data.hasOwnProperty('tls-certificate')) {
          obj['tls-certificate'] = _ApiClient["default"].convertToType(data['tls-certificate'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-name')) {
          obj['user-name'] = _ApiClient["default"].convertToType(data['user-name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayUpdateLogForwardingElasticsearch</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayUpdateLogForwardingElasticsearch</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['api-key'] && !(typeof data['api-key'] === 'string' || data['api-key'] instanceof String)) {
        throw new Error("Expected the field `api-key` to be a primitive type in the JSON string but got " + data['api-key']);
      } // ensure the json data is a string


      if (data['auth-type'] && !(typeof data['auth-type'] === 'string' || data['auth-type'] instanceof String)) {
        throw new Error("Expected the field `auth-type` to be a primitive type in the JSON string but got " + data['auth-type']);
      } // ensure the json data is a string


      if (data['cloud-id'] && !(typeof data['cloud-id'] === 'string' || data['cloud-id'] instanceof String)) {
        throw new Error("Expected the field `cloud-id` to be a primitive type in the JSON string but got " + data['cloud-id']);
      } // ensure the json data is a string


      if (data['enable'] && !(typeof data['enable'] === 'string' || data['enable'] instanceof String)) {
        throw new Error("Expected the field `enable` to be a primitive type in the JSON string but got " + data['enable']);
      } // ensure the json data is a string


      if (data['index'] && !(typeof data['index'] === 'string' || data['index'] instanceof String)) {
        throw new Error("Expected the field `index` to be a primitive type in the JSON string but got " + data['index']);
      } // ensure the json data is a string


      if (data['nodes'] && !(typeof data['nodes'] === 'string' || data['nodes'] instanceof String)) {
        throw new Error("Expected the field `nodes` to be a primitive type in the JSON string but got " + data['nodes']);
      } // ensure the json data is a string


      if (data['output-format'] && !(typeof data['output-format'] === 'string' || data['output-format'] instanceof String)) {
        throw new Error("Expected the field `output-format` to be a primitive type in the JSON string but got " + data['output-format']);
      } // ensure the json data is a string


      if (data['password'] && !(typeof data['password'] === 'string' || data['password'] instanceof String)) {
        throw new Error("Expected the field `password` to be a primitive type in the JSON string but got " + data['password']);
      } // ensure the json data is a string


      if (data['pull-interval'] && !(typeof data['pull-interval'] === 'string' || data['pull-interval'] instanceof String)) {
        throw new Error("Expected the field `pull-interval` to be a primitive type in the JSON string but got " + data['pull-interval']);
      } // ensure the json data is a string


      if (data['server-type'] && !(typeof data['server-type'] === 'string' || data['server-type'] instanceof String)) {
        throw new Error("Expected the field `server-type` to be a primitive type in the JSON string but got " + data['server-type']);
      } // ensure the json data is a string


      if (data['tls-certificate'] && !(typeof data['tls-certificate'] === 'string' || data['tls-certificate'] instanceof String)) {
        throw new Error("Expected the field `tls-certificate` to be a primitive type in the JSON string but got " + data['tls-certificate']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-name'] && !(typeof data['user-name'] === 'string' || data['user-name'] instanceof String)) {
        throw new Error("Expected the field `user-name` to be a primitive type in the JSON string but got " + data['user-name']);
      }

      return true;
    }
  }]);

  return GatewayUpdateLogForwardingElasticsearch;
}();
/**
 * Elasticsearch api key relevant only for api_key auth-type
 * @member {String} api-key
 */


GatewayUpdateLogForwardingElasticsearch.prototype['api-key'] = undefined;
/**
 * Elasticsearch auth type [api_key/password]
 * @member {String} auth-type
 */

GatewayUpdateLogForwardingElasticsearch.prototype['auth-type'] = undefined;
/**
 * Elasticsearch cloud id relevant only for cloud server-type
 * @member {String} cloud-id
 */

GatewayUpdateLogForwardingElasticsearch.prototype['cloud-id'] = undefined;
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */

GatewayUpdateLogForwardingElasticsearch.prototype['enable'] = 'true';
/**
 * Enable tls
 * @member {Boolean} enable-tls
 */

GatewayUpdateLogForwardingElasticsearch.prototype['enable-tls'] = undefined;
/**
 * Elasticsearch index
 * @member {String} index
 */

GatewayUpdateLogForwardingElasticsearch.prototype['index'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateLogForwardingElasticsearch.prototype['json'] = false;
/**
 * Elasticsearch nodes relevant only for nodes server-type
 * @member {String} nodes
 */

GatewayUpdateLogForwardingElasticsearch.prototype['nodes'] = undefined;
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GatewayUpdateLogForwardingElasticsearch.prototype['output-format'] = 'text';
/**
 * Elasticsearch password relevant only for password auth-type
 * @member {String} password
 */

GatewayUpdateLogForwardingElasticsearch.prototype['password'] = undefined;
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GatewayUpdateLogForwardingElasticsearch.prototype['pull-interval'] = '10';
/**
 * Elasticsearch server type [cloud/nodes]
 * @member {String} server-type
 */

GatewayUpdateLogForwardingElasticsearch.prototype['server-type'] = undefined;
/**
 * Elasticsearch tls certificate
 * @member {String} tls-certificate
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingElasticsearch.prototype['tls-certificate'] = 'use-existing';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateLogForwardingElasticsearch.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateLogForwardingElasticsearch.prototype['uid-token'] = undefined;
/**
 * Elasticsearch user name relevant only for password auth-type
 * @member {String} user-name
 */

GatewayUpdateLogForwardingElasticsearch.prototype['user-name'] = undefined;
var _default = GatewayUpdateLogForwardingElasticsearch;
exports["default"] = _default;
