// akeyless/dist/model/GatewayMessageQueueInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayMessageQueueInfo model module.
 * @module model/GatewayMessageQueueInfo
 * @version 5.0.14
 */
var GatewayMessageQueueInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayMessageQueueInfo</code>.
   * @alias module:model/GatewayMessageQueueInfo
   */
  function GatewayMessageQueueInfo() {
    _classCallCheck(this, GatewayMessageQueueInfo);

    GatewayMessageQueueInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayMessageQueueInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayMessageQueueInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayMessageQueueInfo} obj Optional instance to populate.
     * @return {module:model/GatewayMessageQueueInfo} The populated <code>GatewayMessageQueueInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayMessageQueueInfo();

        if (data.hasOwnProperty('broadcast_queue_name_a')) {
          obj['broadcast_queue_name_a'] = _ApiClient["default"].convertToType(data['broadcast_queue_name_a'], 'String');
        }

        if (data.hasOwnProperty('mq_type')) {
          obj['mq_type'] = _ApiClient["default"].convertToType(data['mq_type'], 'String');
        }

        if (data.hasOwnProperty('queue_name')) {
          obj['queue_name'] = _ApiClient["default"].convertToType(data['queue_name'], 'String');
        }

        if (data.hasOwnProperty('queue_url')) {
          obj['queue_url'] = _ApiClient["default"].convertToType(data['queue_url'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayMessageQueueInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayMessageQueueInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['broadcast_queue_name_a'] && !(typeof data['broadcast_queue_name_a'] === 'string' || data['broadcast_queue_name_a'] instanceof String)) {
        throw new Error("Expected the field `broadcast_queue_name_a` to be a primitive type in the JSON string but got " + data['broadcast_queue_name_a']);
      } // ensure the json data is a string


      if (data['mq_type'] && !(typeof data['mq_type'] === 'string' || data['mq_type'] instanceof String)) {
        throw new Error("Expected the field `mq_type` to be a primitive type in the JSON string but got " + data['mq_type']);
      } // ensure the json data is a string


      if (data['queue_name'] && !(typeof data['queue_name'] === 'string' || data['queue_name'] instanceof String)) {
        throw new Error("Expected the field `queue_name` to be a primitive type in the JSON string but got " + data['queue_name']);
      } // ensure the json data is a string


      if (data['queue_url'] && !(typeof data['queue_url'] === 'string' || data['queue_url'] instanceof String)) {
        throw new Error("Expected the field `queue_url` to be a primitive type in the JSON string but got " + data['queue_url']);
      }

      return true;
    }
  }]);

  return GatewayMessageQueueInfo;
}();
/**
 * @member {String} broadcast_queue_name_a
 */


GatewayMessageQueueInfo.prototype['broadcast_queue_name_a'] = undefined;
/**
 * @member {String} mq_type
 */

GatewayMessageQueueInfo.prototype['mq_type'] = undefined;
/**
 * @member {String} queue_name
 */

GatewayMessageQueueInfo.prototype['queue_name'] = undefined;
/**
 * @member {String} queue_url
 */

GatewayMessageQueueInfo.prototype['queue_url'] = undefined;
var _default = GatewayMessageQueueInfo;
exports["default"] = _default;
