// akeyless/dist/model/GatewayUpdateLogForwardingSumologic.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateLogForwardingSumologic model module.
 * @module model/GatewayUpdateLogForwardingSumologic
 * @version 4.2.2
 */
var GatewayUpdateLogForwardingSumologic = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateLogForwardingSumologic</code>.
   * gatewayUpdateLogForwardingSumologic is a command that updates log forwarding config (sumologic target)
   * @alias module:model/GatewayUpdateLogForwardingSumologic
   */
  function GatewayUpdateLogForwardingSumologic() {
    _classCallCheck(this, GatewayUpdateLogForwardingSumologic);

    GatewayUpdateLogForwardingSumologic.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateLogForwardingSumologic, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayUpdateLogForwardingSumologic</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateLogForwardingSumologic} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateLogForwardingSumologic} The populated <code>GatewayUpdateLogForwardingSumologic</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateLogForwardingSumologic();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('endpoint')) {
          obj['endpoint'] = _ApiClient["default"].convertToType(data['endpoint'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('sumologic-tags')) {
          obj['sumologic-tags'] = _ApiClient["default"].convertToType(data['sumologic-tags'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateLogForwardingSumologic;
}();
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */


GatewayUpdateLogForwardingSumologic.prototype['enable'] = 'true';
/**
 * Sumologic endpoint URL
 * @member {String} endpoint
 */

GatewayUpdateLogForwardingSumologic.prototype['endpoint'] = undefined;
/**
 * Sumologic host
 * @member {String} host
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingSumologic.prototype['host'] = 'use-existing';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateLogForwardingSumologic.prototype['json'] = false;
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GatewayUpdateLogForwardingSumologic.prototype['output-format'] = 'text';
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GatewayUpdateLogForwardingSumologic.prototype['pull-interval'] = '10';
/**
 * A comma-separated list of Sumologic tags
 * @member {String} sumologic-tags
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingSumologic.prototype['sumologic-tags'] = 'use-existing';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateLogForwardingSumologic.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateLogForwardingSumologic.prototype['uid-token'] = undefined;
var _default = GatewayUpdateLogForwardingSumologic;
exports["default"] = _default;
