// akeyless/dist/model/GatewayGetDefaultsOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayGetDefaultsOutput model module.
 * @module model/GatewayGetDefaultsOutput
 * @version 4.3.0
 */
var GatewayGetDefaultsOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayGetDefaultsOutput</code>.
   * @alias module:model/GatewayGetDefaultsOutput
   */
  function GatewayGetDefaultsOutput() {
    _classCallCheck(this, GatewayGetDefaultsOutput);

    GatewayGetDefaultsOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayGetDefaultsOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayGetDefaultsOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayGetDefaultsOutput} obj Optional instance to populate.
     * @return {module:model/GatewayGetDefaultsOutput} The populated <code>GatewayGetDefaultsOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayGetDefaultsOutput();

        if (data.hasOwnProperty('certificate_access_id')) {
          obj['certificate_access_id'] = _ApiClient["default"].convertToType(data['certificate_access_id'], 'String');
        }

        if (data.hasOwnProperty('default_protection_key_id')) {
          obj['default_protection_key_id'] = _ApiClient["default"].convertToType(data['default_protection_key_id'], 'String');
        }

        if (data.hasOwnProperty('notify_on_status_change')) {
          obj['notify_on_status_change'] = _ApiClient["default"].convertToType(data['notify_on_status_change'], 'Boolean');
        }

        if (data.hasOwnProperty('oidc_access_id')) {
          obj['oidc_access_id'] = _ApiClient["default"].convertToType(data['oidc_access_id'], 'String');
        }

        if (data.hasOwnProperty('saml_access_id')) {
          obj['saml_access_id'] = _ApiClient["default"].convertToType(data['saml_access_id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayGetDefaultsOutput;
}();
/**
 * @member {String} certificate_access_id
 */


GatewayGetDefaultsOutput.prototype['certificate_access_id'] = undefined;
/**
 * @member {String} default_protection_key_id
 */

GatewayGetDefaultsOutput.prototype['default_protection_key_id'] = undefined;
/**
 * @member {Boolean} notify_on_status_change
 */

GatewayGetDefaultsOutput.prototype['notify_on_status_change'] = undefined;
/**
 * @member {String} oidc_access_id
 */

GatewayGetDefaultsOutput.prototype['oidc_access_id'] = undefined;
/**
 * @member {String} saml_access_id
 */

GatewayGetDefaultsOutput.prototype['saml_access_id'] = undefined;
var _default = GatewayGetDefaultsOutput;
exports["default"] = _default;
