// akeyless/dist/model/GatewayUpdateLogForwardingSplunk.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateLogForwardingSplunk model module.
 * @module model/GatewayUpdateLogForwardingSplunk
 * @version 4.2.2
 */
var GatewayUpdateLogForwardingSplunk = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateLogForwardingSplunk</code>.
   * gatewayUpdateLogForwardingSplunk is a command that updates log forwarding config (splunk target)
   * @alias module:model/GatewayUpdateLogForwardingSplunk
   */
  function GatewayUpdateLogForwardingSplunk() {
    _classCallCheck(this, GatewayUpdateLogForwardingSplunk);

    GatewayUpdateLogForwardingSplunk.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateLogForwardingSplunk, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayUpdateLogForwardingSplunk</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateLogForwardingSplunk} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateLogForwardingSplunk} The populated <code>GatewayUpdateLogForwardingSplunk</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateLogForwardingSplunk();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('enable-tls')) {
          obj['enable-tls'] = _ApiClient["default"].convertToType(data['enable-tls'], 'Boolean');
        }

        if (data.hasOwnProperty('index')) {
          obj['index'] = _ApiClient["default"].convertToType(data['index'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('source-type')) {
          obj['source-type'] = _ApiClient["default"].convertToType(data['source-type'], 'String');
        }

        if (data.hasOwnProperty('splunk-token')) {
          obj['splunk-token'] = _ApiClient["default"].convertToType(data['splunk-token'], 'String');
        }

        if (data.hasOwnProperty('splunk-url')) {
          obj['splunk-url'] = _ApiClient["default"].convertToType(data['splunk-url'], 'String');
        }

        if (data.hasOwnProperty('tls-certificate')) {
          obj['tls-certificate'] = _ApiClient["default"].convertToType(data['tls-certificate'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateLogForwardingSplunk;
}();
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */


GatewayUpdateLogForwardingSplunk.prototype['enable'] = 'true';
/**
 * Enable tls
 * @member {Boolean} enable-tls
 */

GatewayUpdateLogForwardingSplunk.prototype['enable-tls'] = undefined;
/**
 * Splunk index
 * @member {String} index
 */

GatewayUpdateLogForwardingSplunk.prototype['index'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateLogForwardingSplunk.prototype['json'] = false;
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GatewayUpdateLogForwardingSplunk.prototype['output-format'] = 'text';
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GatewayUpdateLogForwardingSplunk.prototype['pull-interval'] = '10';
/**
 * Splunk source
 * @member {String} source
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingSplunk.prototype['source'] = 'use-existing';
/**
 * Splunk source type
 * @member {String} source-type
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingSplunk.prototype['source-type'] = 'use-existing';
/**
 * Splunk token
 * @member {String} splunk-token
 */

GatewayUpdateLogForwardingSplunk.prototype['splunk-token'] = undefined;
/**
 * Splunk server URL
 * @member {String} splunk-url
 */

GatewayUpdateLogForwardingSplunk.prototype['splunk-url'] = undefined;
/**
 * Splunk tls certificate
 * @member {String} tls-certificate
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingSplunk.prototype['tls-certificate'] = 'use-existing';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateLogForwardingSplunk.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateLogForwardingSplunk.prototype['uid-token'] = undefined;
var _default = GatewayUpdateLogForwardingSplunk;
exports["default"] = _default;
