// akeyless/dist/model/GatewayCreateProducerGcp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerGcp model module.
 * @module model/GatewayCreateProducerGcp
 * @version 5.0.6
 */
var GatewayCreateProducerGcp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerGcp</code>.
   * gatewayCreateProducerGcp is a command that creates a GCP producer [Deprecated: Use dynamic-secret-create-gcp command]
   * @alias module:model/GatewayCreateProducerGcp
   * @param name {String} Dynamic secret name
   * @param serviceAccountType {String} The type of the gcp dynamic secret. Options[fixed, dynamic]
   */
  function GatewayCreateProducerGcp(name, serviceAccountType) {
    _classCallCheck(this, GatewayCreateProducerGcp);

    GatewayCreateProducerGcp.initialize(this, name, serviceAccountType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerGcp, null, [{
    key: "initialize",
    value: function initialize(obj, name, serviceAccountType) {
      obj['json'] = false;
      obj['name'] = name;
      obj['service-account-type'] = serviceAccountType || 'fixed';
      obj['user-ttl'] = '60m';
    }
    /**
     * Constructs a <code>GatewayCreateProducerGcp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerGcp} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerGcp} The populated <code>GatewayCreateProducerGcp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerGcp();

        if (data.hasOwnProperty('custom-username-template')) {
          obj['custom-username-template'] = _ApiClient["default"].convertToType(data['custom-username-template'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('gcp-cred-type')) {
          obj['gcp-cred-type'] = _ApiClient["default"].convertToType(data['gcp-cred-type'], 'String');
        }

        if (data.hasOwnProperty('gcp-key')) {
          obj['gcp-key'] = _ApiClient["default"].convertToType(data['gcp-key'], 'String');
        }

        if (data.hasOwnProperty('gcp-key-algo')) {
          obj['gcp-key-algo'] = _ApiClient["default"].convertToType(data['gcp-key-algo'], 'String');
        }

        if (data.hasOwnProperty('gcp-sa-email')) {
          obj['gcp-sa-email'] = _ApiClient["default"].convertToType(data['gcp-sa-email'], 'String');
        }

        if (data.hasOwnProperty('gcp-token-scopes')) {
          obj['gcp-token-scopes'] = _ApiClient["default"].convertToType(data['gcp-token-scopes'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('role-binding')) {
          obj['role-binding'] = _ApiClient["default"].convertToType(data['role-binding'], 'String');
        }

        if (data.hasOwnProperty('service-account-type')) {
          obj['service-account-type'] = _ApiClient["default"].convertToType(data['service-account-type'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayCreateProducerGcp</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayCreateProducerGcp</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GatewayCreateProducerGcp.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['custom-username-template'] && !(typeof data['custom-username-template'] === 'string' || data['custom-username-template'] instanceof String)) {
        throw new Error("Expected the field `custom-username-template` to be a primitive type in the JSON string but got " + data['custom-username-template']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['gcp-cred-type'] && !(typeof data['gcp-cred-type'] === 'string' || data['gcp-cred-type'] instanceof String)) {
        throw new Error("Expected the field `gcp-cred-type` to be a primitive type in the JSON string but got " + data['gcp-cred-type']);
      } // ensure the json data is a string


      if (data['gcp-key'] && !(typeof data['gcp-key'] === 'string' || data['gcp-key'] instanceof String)) {
        throw new Error("Expected the field `gcp-key` to be a primitive type in the JSON string but got " + data['gcp-key']);
      } // ensure the json data is a string


      if (data['gcp-key-algo'] && !(typeof data['gcp-key-algo'] === 'string' || data['gcp-key-algo'] instanceof String)) {
        throw new Error("Expected the field `gcp-key-algo` to be a primitive type in the JSON string but got " + data['gcp-key-algo']);
      } // ensure the json data is a string


      if (data['gcp-sa-email'] && !(typeof data['gcp-sa-email'] === 'string' || data['gcp-sa-email'] instanceof String)) {
        throw new Error("Expected the field `gcp-sa-email` to be a primitive type in the JSON string but got " + data['gcp-sa-email']);
      } // ensure the json data is a string


      if (data['gcp-token-scopes'] && !(typeof data['gcp-token-scopes'] === 'string' || data['gcp-token-scopes'] instanceof String)) {
        throw new Error("Expected the field `gcp-token-scopes` to be a primitive type in the JSON string but got " + data['gcp-token-scopes']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['producer-encryption-key-name'] && !(typeof data['producer-encryption-key-name'] === 'string' || data['producer-encryption-key-name'] instanceof String)) {
        throw new Error("Expected the field `producer-encryption-key-name` to be a primitive type in the JSON string but got " + data['producer-encryption-key-name']);
      } // ensure the json data is a string


      if (data['role-binding'] && !(typeof data['role-binding'] === 'string' || data['role-binding'] instanceof String)) {
        throw new Error("Expected the field `role-binding` to be a primitive type in the JSON string but got " + data['role-binding']);
      } // ensure the json data is a string


      if (data['service-account-type'] && !(typeof data['service-account-type'] === 'string' || data['service-account-type'] instanceof String)) {
        throw new Error("Expected the field `service-account-type` to be a primitive type in the JSON string but got " + data['service-account-type']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-ttl'] && !(typeof data['user-ttl'] === 'string' || data['user-ttl'] instanceof String)) {
        throw new Error("Expected the field `user-ttl` to be a primitive type in the JSON string but got " + data['user-ttl']);
      }

      return true;
    }
  }]);

  return GatewayCreateProducerGcp;
}();

GatewayCreateProducerGcp.RequiredProperties = ["name", "service-account-type"];
/**
 * Customize how temporary usernames are generated using go template
 * @member {String} custom-username-template
 */

GatewayCreateProducerGcp.prototype['custom-username-template'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

GatewayCreateProducerGcp.prototype['delete_protection'] = undefined;
/**
 * @member {String} gcp-cred-type
 */

GatewayCreateProducerGcp.prototype['gcp-cred-type'] = undefined;
/**
 * Base64-encoded service account private key text
 * @member {String} gcp-key
 */

GatewayCreateProducerGcp.prototype['gcp-key'] = undefined;
/**
 * Service account key algorithm, e.g. KEY_ALG_RSA_1024
 * @member {String} gcp-key-algo
 */

GatewayCreateProducerGcp.prototype['gcp-key-algo'] = undefined;
/**
 * The email of the fixed service acocunt to generate keys or tokens for. (revelant for service-account-type=fixed)
 * @member {String} gcp-sa-email
 */

GatewayCreateProducerGcp.prototype['gcp-sa-email'] = undefined;
/**
 * Access token scopes list, e.g. scope1,scope2
 * @member {String} gcp-token-scopes
 */

GatewayCreateProducerGcp.prototype['gcp-token-scopes'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayCreateProducerGcp.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

GatewayCreateProducerGcp.prototype['name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerGcp.prototype['producer-encryption-key-name'] = undefined;
/**
 * Role binding definitions in json format
 * @member {String} role-binding
 */

GatewayCreateProducerGcp.prototype['role-binding'] = undefined;
/**
 * The type of the gcp dynamic secret. Options[fixed, dynamic]
 * @member {String} service-account-type
 * @default 'fixed'
 */

GatewayCreateProducerGcp.prototype['service-account-type'] = 'fixed';
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

GatewayCreateProducerGcp.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerGcp.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerGcp.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerGcp.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerGcp.prototype['user-ttl'] = '60m';
var _default = GatewayCreateProducerGcp;
exports["default"] = _default;
