// akeyless/dist/model/GatewayUpdateProducerGithub.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerGithub model module.
 * @module model/GatewayUpdateProducerGithub
 * @version 5.0.6
 */
var GatewayUpdateProducerGithub = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerGithub</code>.
   * gatewayUpdateProducerGithub is a command that updates github producer [Deprecated: Use dynamic-secret-update-github command]
   * @alias module:model/GatewayUpdateProducerGithub
   * @param name {String} Dynamic secret name
   */
  function GatewayUpdateProducerGithub(name) {
    _classCallCheck(this, GatewayUpdateProducerGithub);

    GatewayUpdateProducerGithub.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerGithub, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['github-base-url'] = 'https://api.github.com/';
      obj['json'] = false;
      obj['name'] = name;
      obj['token-ttl'] = '60m';
    }
    /**
     * Constructs a <code>GatewayUpdateProducerGithub</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerGithub} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerGithub} The populated <code>GatewayUpdateProducerGithub</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerGithub();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('github-app-id')) {
          obj['github-app-id'] = _ApiClient["default"].convertToType(data['github-app-id'], 'Number');
        }

        if (data.hasOwnProperty('github-app-private-key')) {
          obj['github-app-private-key'] = _ApiClient["default"].convertToType(data['github-app-private-key'], 'String');
        }

        if (data.hasOwnProperty('github-base-url')) {
          obj['github-base-url'] = _ApiClient["default"].convertToType(data['github-base-url'], 'String');
        }

        if (data.hasOwnProperty('installation-id')) {
          obj['installation-id'] = _ApiClient["default"].convertToType(data['installation-id'], 'Number');
        }

        if (data.hasOwnProperty('installation-organization')) {
          obj['installation-organization'] = _ApiClient["default"].convertToType(data['installation-organization'], 'String');
        }

        if (data.hasOwnProperty('installation-repository')) {
          obj['installation-repository'] = _ApiClient["default"].convertToType(data['installation-repository'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('token-permissions')) {
          obj['token-permissions'] = _ApiClient["default"].convertToType(data['token-permissions'], ['String']);
        }

        if (data.hasOwnProperty('token-repositories')) {
          obj['token-repositories'] = _ApiClient["default"].convertToType(data['token-repositories'], ['String']);
        }

        if (data.hasOwnProperty('token-ttl')) {
          obj['token-ttl'] = _ApiClient["default"].convertToType(data['token-ttl'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayUpdateProducerGithub</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayUpdateProducerGithub</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GatewayUpdateProducerGithub.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['github-app-private-key'] && !(typeof data['github-app-private-key'] === 'string' || data['github-app-private-key'] instanceof String)) {
        throw new Error("Expected the field `github-app-private-key` to be a primitive type in the JSON string but got " + data['github-app-private-key']);
      } // ensure the json data is a string


      if (data['github-base-url'] && !(typeof data['github-base-url'] === 'string' || data['github-base-url'] instanceof String)) {
        throw new Error("Expected the field `github-base-url` to be a primitive type in the JSON string but got " + data['github-base-url']);
      } // ensure the json data is a string


      if (data['installation-organization'] && !(typeof data['installation-organization'] === 'string' || data['installation-organization'] instanceof String)) {
        throw new Error("Expected the field `installation-organization` to be a primitive type in the JSON string but got " + data['installation-organization']);
      } // ensure the json data is a string


      if (data['installation-repository'] && !(typeof data['installation-repository'] === 'string' || data['installation-repository'] instanceof String)) {
        throw new Error("Expected the field `installation-repository` to be a primitive type in the JSON string but got " + data['installation-repository']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is an array


      if (!Array.isArray(data['token-permissions'])) {
        throw new Error("Expected the field `token-permissions` to be an array in the JSON data but got " + data['token-permissions']);
      } // ensure the json data is an array


      if (!Array.isArray(data['token-repositories'])) {
        throw new Error("Expected the field `token-repositories` to be an array in the JSON data but got " + data['token-repositories']);
      } // ensure the json data is a string


      if (data['token-ttl'] && !(typeof data['token-ttl'] === 'string' || data['token-ttl'] instanceof String)) {
        throw new Error("Expected the field `token-ttl` to be a primitive type in the JSON string but got " + data['token-ttl']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return GatewayUpdateProducerGithub;
}();

GatewayUpdateProducerGithub.RequiredProperties = ["name"];
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

GatewayUpdateProducerGithub.prototype['delete_protection'] = undefined;
/**
 * Github app id
 * @member {Number} github-app-id
 */

GatewayUpdateProducerGithub.prototype['github-app-id'] = undefined;
/**
 * App private key
 * @member {String} github-app-private-key
 */

GatewayUpdateProducerGithub.prototype['github-app-private-key'] = undefined;
/**
 * Base URL
 * @member {String} github-base-url
 * @default 'https://api.github.com/'
 */

GatewayUpdateProducerGithub.prototype['github-base-url'] = 'https://api.github.com/';
/**
 * GitHub application installation id
 * @member {Number} installation-id
 */

GatewayUpdateProducerGithub.prototype['installation-id'] = undefined;
/**
 * Optional, mutually exclusive with installation id, GitHub organization name
 * @member {String} installation-organization
 */

GatewayUpdateProducerGithub.prototype['installation-organization'] = undefined;
/**
 * Optional, mutually exclusive with installation id, GitHub repository '<owner>/<repo-name>'
 * @member {String} installation-repository
 */

GatewayUpdateProducerGithub.prototype['installation-repository'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateProducerGithub.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

GatewayUpdateProducerGithub.prototype['name'] = undefined;
/**
 * Dynamic secret name
 * @member {String} new-name
 */

GatewayUpdateProducerGithub.prototype['new-name'] = undefined;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerGithub.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerGithub.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerGithub.prototype['token'] = undefined;
/**
 * Optional - installation token's allowed permissions
 * @member {Array.<String>} token-permissions
 */

GatewayUpdateProducerGithub.prototype['token-permissions'] = undefined;
/**
 * Optional - installation token's allowed repositories
 * @member {Array.<String>} token-repositories
 */

GatewayUpdateProducerGithub.prototype['token-repositories'] = undefined;
/**
 * Token TTL
 * @member {String} token-ttl
 * @default '60m'
 */

GatewayUpdateProducerGithub.prototype['token-ttl'] = '60m';
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerGithub.prototype['uid-token'] = undefined;
var _default = GatewayUpdateProducerGithub;
exports["default"] = _default;
