// akeyless/dist/model/GatewayUpdateProducerCassandra.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerCassandra model module.
 * @module model/GatewayUpdateProducerCassandra
 * @version 5.0.6
 */
var GatewayUpdateProducerCassandra = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerCassandra</code>.
   * gatewayUpdateProducerCassandra is a command that updates a Cassandra producer [Deprecated: Use dynamic-secret-update-cassandra command]
   * @alias module:model/GatewayUpdateProducerCassandra
   * @param name {String} Dynamic secret name
   */
  function GatewayUpdateProducerCassandra(name) {
    _classCallCheck(this, GatewayUpdateProducerCassandra);

    GatewayUpdateProducerCassandra.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerCassandra, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['cassandra-port'] = '9042';
      obj['json'] = false;
      obj['name'] = name;
      obj['ssl'] = false;
      obj['user-ttl'] = '60m';
    }
    /**
     * Constructs a <code>GatewayUpdateProducerCassandra</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerCassandra} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerCassandra} The populated <code>GatewayUpdateProducerCassandra</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerCassandra();

        if (data.hasOwnProperty('cassandra-creation-statements')) {
          obj['cassandra-creation-statements'] = _ApiClient["default"].convertToType(data['cassandra-creation-statements'], 'String');
        }

        if (data.hasOwnProperty('cassandra-hosts')) {
          obj['cassandra-hosts'] = _ApiClient["default"].convertToType(data['cassandra-hosts'], 'String');
        }

        if (data.hasOwnProperty('cassandra-password')) {
          obj['cassandra-password'] = _ApiClient["default"].convertToType(data['cassandra-password'], 'String');
        }

        if (data.hasOwnProperty('cassandra-port')) {
          obj['cassandra-port'] = _ApiClient["default"].convertToType(data['cassandra-port'], 'String');
        }

        if (data.hasOwnProperty('cassandra-username')) {
          obj['cassandra-username'] = _ApiClient["default"].convertToType(data['cassandra-username'], 'String');
        }

        if (data.hasOwnProperty('custom-username-template')) {
          obj['custom-username-template'] = _ApiClient["default"].convertToType(data['custom-username-template'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password-length')) {
          obj['password-length'] = _ApiClient["default"].convertToType(data['password-length'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('ssl')) {
          obj['ssl'] = _ApiClient["default"].convertToType(data['ssl'], 'Boolean');
        }

        if (data.hasOwnProperty('ssl-certificate')) {
          obj['ssl-certificate'] = _ApiClient["default"].convertToType(data['ssl-certificate'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayUpdateProducerCassandra</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayUpdateProducerCassandra</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GatewayUpdateProducerCassandra.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['cassandra-creation-statements'] && !(typeof data['cassandra-creation-statements'] === 'string' || data['cassandra-creation-statements'] instanceof String)) {
        throw new Error("Expected the field `cassandra-creation-statements` to be a primitive type in the JSON string but got " + data['cassandra-creation-statements']);
      } // ensure the json data is a string


      if (data['cassandra-hosts'] && !(typeof data['cassandra-hosts'] === 'string' || data['cassandra-hosts'] instanceof String)) {
        throw new Error("Expected the field `cassandra-hosts` to be a primitive type in the JSON string but got " + data['cassandra-hosts']);
      } // ensure the json data is a string


      if (data['cassandra-password'] && !(typeof data['cassandra-password'] === 'string' || data['cassandra-password'] instanceof String)) {
        throw new Error("Expected the field `cassandra-password` to be a primitive type in the JSON string but got " + data['cassandra-password']);
      } // ensure the json data is a string


      if (data['cassandra-port'] && !(typeof data['cassandra-port'] === 'string' || data['cassandra-port'] instanceof String)) {
        throw new Error("Expected the field `cassandra-port` to be a primitive type in the JSON string but got " + data['cassandra-port']);
      } // ensure the json data is a string


      if (data['cassandra-username'] && !(typeof data['cassandra-username'] === 'string' || data['cassandra-username'] instanceof String)) {
        throw new Error("Expected the field `cassandra-username` to be a primitive type in the JSON string but got " + data['cassandra-username']);
      } // ensure the json data is a string


      if (data['custom-username-template'] && !(typeof data['custom-username-template'] === 'string' || data['custom-username-template'] instanceof String)) {
        throw new Error("Expected the field `custom-username-template` to be a primitive type in the JSON string but got " + data['custom-username-template']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['password-length'] && !(typeof data['password-length'] === 'string' || data['password-length'] instanceof String)) {
        throw new Error("Expected the field `password-length` to be a primitive type in the JSON string but got " + data['password-length']);
      } // ensure the json data is a string


      if (data['producer-encryption-key-name'] && !(typeof data['producer-encryption-key-name'] === 'string' || data['producer-encryption-key-name'] instanceof String)) {
        throw new Error("Expected the field `producer-encryption-key-name` to be a primitive type in the JSON string but got " + data['producer-encryption-key-name']);
      } // ensure the json data is a string


      if (data['ssl-certificate'] && !(typeof data['ssl-certificate'] === 'string' || data['ssl-certificate'] instanceof String)) {
        throw new Error("Expected the field `ssl-certificate` to be a primitive type in the JSON string but got " + data['ssl-certificate']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-ttl'] && !(typeof data['user-ttl'] === 'string' || data['user-ttl'] instanceof String)) {
        throw new Error("Expected the field `user-ttl` to be a primitive type in the JSON string but got " + data['user-ttl']);
      }

      return true;
    }
  }]);

  return GatewayUpdateProducerCassandra;
}();

GatewayUpdateProducerCassandra.RequiredProperties = ["name"];
/**
 * Cassandra creation statements
 * @member {String} cassandra-creation-statements
 */

GatewayUpdateProducerCassandra.prototype['cassandra-creation-statements'] = undefined;
/**
 * Cassandra hosts IP or addresses, comma separated
 * @member {String} cassandra-hosts
 */

GatewayUpdateProducerCassandra.prototype['cassandra-hosts'] = undefined;
/**
 * Cassandra superuser password
 * @member {String} cassandra-password
 */

GatewayUpdateProducerCassandra.prototype['cassandra-password'] = undefined;
/**
 * Cassandra port
 * @member {String} cassandra-port
 * @default '9042'
 */

GatewayUpdateProducerCassandra.prototype['cassandra-port'] = '9042';
/**
 * Cassandra superuser username
 * @member {String} cassandra-username
 */

GatewayUpdateProducerCassandra.prototype['cassandra-username'] = undefined;
/**
 * Customize how temporary usernames are generated using go template
 * @member {String} custom-username-template
 */

GatewayUpdateProducerCassandra.prototype['custom-username-template'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

GatewayUpdateProducerCassandra.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateProducerCassandra.prototype['json'] = false;
/**
 * Dynamic secret name
 * @member {String} name
 */

GatewayUpdateProducerCassandra.prototype['name'] = undefined;
/**
 * Dynamic secret name
 * @member {String} new-name
 */

GatewayUpdateProducerCassandra.prototype['new-name'] = undefined;
/**
 * The length of the password to be generated
 * @member {String} password-length
 */

GatewayUpdateProducerCassandra.prototype['password-length'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerCassandra.prototype['producer-encryption-key-name'] = undefined;
/**
 * Enable/Disable SSL [true/false]
 * @member {Boolean} ssl
 * @default false
 */

GatewayUpdateProducerCassandra.prototype['ssl'] = false;
/**
 * SSL CA certificate in base64 encoding generated from a trusted Certificate Authority (CA)
 * @member {String} ssl-certificate
 */

GatewayUpdateProducerCassandra.prototype['ssl-certificate'] = undefined;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerCassandra.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerCassandra.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerCassandra.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerCassandra.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerCassandra.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerCassandra;
exports["default"] = _default;
