// akeyless/dist/model/GatewayNameInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayNameInfo model module.
 * @module model/GatewayNameInfo
 * @version 5.0.3
 */
var GatewayNameInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayNameInfo</code>.
   * @alias module:model/GatewayNameInfo
   */
  function GatewayNameInfo() {
    _classCallCheck(this, GatewayNameInfo);

    GatewayNameInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayNameInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayNameInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayNameInfo} obj Optional instance to populate.
     * @return {module:model/GatewayNameInfo} The populated <code>GatewayNameInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayNameInfo();

        if (data.hasOwnProperty('cluster_display_name')) {
          obj['cluster_display_name'] = _ApiClient["default"].convertToType(data['cluster_display_name'], 'String');
        }

        if (data.hasOwnProperty('cluster_name')) {
          obj['cluster_name'] = _ApiClient["default"].convertToType(data['cluster_name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayNameInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayNameInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['cluster_display_name'] && !(typeof data['cluster_display_name'] === 'string' || data['cluster_display_name'] instanceof String)) {
        throw new Error("Expected the field `cluster_display_name` to be a primitive type in the JSON string but got " + data['cluster_display_name']);
      } // ensure the json data is a string


      if (data['cluster_name'] && !(typeof data['cluster_name'] === 'string' || data['cluster_name'] instanceof String)) {
        throw new Error("Expected the field `cluster_name` to be a primitive type in the JSON string but got " + data['cluster_name']);
      }

      return true;
    }
  }]);

  return GatewayNameInfo;
}();
/**
 * @member {String} cluster_display_name
 */


GatewayNameInfo.prototype['cluster_display_name'] = undefined;
/**
 * @member {String} cluster_name
 */

GatewayNameInfo.prototype['cluster_name'] = undefined;
var _default = GatewayNameInfo;
exports["default"] = _default;
