// akeyless/dist/model/GatewayUpdateLogForwardingAwsS3.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateLogForwardingAwsS3 model module.
 * @module model/GatewayUpdateLogForwardingAwsS3
 * @version 4.2.2
 */
var GatewayUpdateLogForwardingAwsS3 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateLogForwardingAwsS3</code>.
   * gatewayUpdateLogForwardingAwsS3 is a command that updates log forwarding config (aws-s3 target)
   * @alias module:model/GatewayUpdateLogForwardingAwsS3
   */
  function GatewayUpdateLogForwardingAwsS3() {
    _classCallCheck(this, GatewayUpdateLogForwardingAwsS3);

    GatewayUpdateLogForwardingAwsS3.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateLogForwardingAwsS3, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayUpdateLogForwardingAwsS3</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateLogForwardingAwsS3} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateLogForwardingAwsS3} The populated <code>GatewayUpdateLogForwardingAwsS3</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateLogForwardingAwsS3();

        if (data.hasOwnProperty('access-id')) {
          obj['access-id'] = _ApiClient["default"].convertToType(data['access-id'], 'String');
        }

        if (data.hasOwnProperty('access-key')) {
          obj['access-key'] = _ApiClient["default"].convertToType(data['access-key'], 'String');
        }

        if (data.hasOwnProperty('auth-type')) {
          obj['auth-type'] = _ApiClient["default"].convertToType(data['auth-type'], 'String');
        }

        if (data.hasOwnProperty('bucket-name')) {
          obj['bucket-name'] = _ApiClient["default"].convertToType(data['bucket-name'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('log-folder')) {
          obj['log-folder'] = _ApiClient["default"].convertToType(data['log-folder'], 'String');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('role-arn')) {
          obj['role-arn'] = _ApiClient["default"].convertToType(data['role-arn'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateLogForwardingAwsS3;
}();
/**
 * AWS access id relevant for access_key auth-type
 * @member {String} access-id
 */


GatewayUpdateLogForwardingAwsS3.prototype['access-id'] = undefined;
/**
 * AWS access key relevant for access_key auth-type
 * @member {String} access-key
 */

GatewayUpdateLogForwardingAwsS3.prototype['access-key'] = undefined;
/**
 * AWS auth type [access_key/cloud_id/assume_role]
 * @member {String} auth-type
 */

GatewayUpdateLogForwardingAwsS3.prototype['auth-type'] = undefined;
/**
 * AWS S3 bucket name
 * @member {String} bucket-name
 */

GatewayUpdateLogForwardingAwsS3.prototype['bucket-name'] = undefined;
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */

GatewayUpdateLogForwardingAwsS3.prototype['enable'] = 'true';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateLogForwardingAwsS3.prototype['json'] = false;
/**
 * AWS S3 destination folder for logs
 * @member {String} log-folder
 * @default 'use-existing'
 */

GatewayUpdateLogForwardingAwsS3.prototype['log-folder'] = 'use-existing';
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GatewayUpdateLogForwardingAwsS3.prototype['output-format'] = 'text';
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GatewayUpdateLogForwardingAwsS3.prototype['pull-interval'] = '10';
/**
 * AWS region
 * @member {String} region
 */

GatewayUpdateLogForwardingAwsS3.prototype['region'] = undefined;
/**
 * AWS role arn relevant for assume_role auth-type
 * @member {String} role-arn
 */

GatewayUpdateLogForwardingAwsS3.prototype['role-arn'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateLogForwardingAwsS3.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateLogForwardingAwsS3.prototype['uid-token'] = undefined;
var _default = GatewayUpdateLogForwardingAwsS3;
exports["default"] = _default;
