// akeyless/dist/model/GatewayPartialUpdateK8SAuthConfigOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ConfigChange = _interopRequireDefault(require("./ConfigChange"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayPartialUpdateK8SAuthConfigOutput model module.
 * @module model/GatewayPartialUpdateK8SAuthConfigOutput
 * @version 4.2.1
 */
var GatewayPartialUpdateK8SAuthConfigOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayPartialUpdateK8SAuthConfigOutput</code>.
   * @alias module:model/GatewayPartialUpdateK8SAuthConfigOutput
   */
  function GatewayPartialUpdateK8SAuthConfigOutput() {
    _classCallCheck(this, GatewayPartialUpdateK8SAuthConfigOutput);

    GatewayPartialUpdateK8SAuthConfigOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayPartialUpdateK8SAuthConfigOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayPartialUpdateK8SAuthConfigOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayPartialUpdateK8SAuthConfigOutput} obj Optional instance to populate.
     * @return {module:model/GatewayPartialUpdateK8SAuthConfigOutput} The populated <code>GatewayPartialUpdateK8SAuthConfigOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayPartialUpdateK8SAuthConfigOutput();

        if (data.hasOwnProperty('cluster_id')) {
          obj['cluster_id'] = _ApiClient["default"].convertToType(data['cluster_id'], 'String');
        }

        if (data.hasOwnProperty('parts_change')) {
          obj['parts_change'] = _ConfigChange["default"].constructFromObject(data['parts_change']);
        }

        if (data.hasOwnProperty('total_hash')) {
          obj['total_hash'] = _ApiClient["default"].convertToType(data['total_hash'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayPartialUpdateK8SAuthConfigOutput;
}();
/**
 * @member {String} cluster_id
 */


GatewayPartialUpdateK8SAuthConfigOutput.prototype['cluster_id'] = undefined;
/**
 * @member {module:model/ConfigChange} parts_change
 */

GatewayPartialUpdateK8SAuthConfigOutput.prototype['parts_change'] = undefined;
/**
 * @member {String} total_hash
 */

GatewayPartialUpdateK8SAuthConfigOutput.prototype['total_hash'] = undefined;
var _default = GatewayPartialUpdateK8SAuthConfigOutput;
exports["default"] = _default;
