// akeyless/dist/model/GatewayUpdateLogForwardingLogzIo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateLogForwardingLogzIo model module.
 * @module model/GatewayUpdateLogForwardingLogzIo
 * @version 4.3.0
 */
var GatewayUpdateLogForwardingLogzIo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateLogForwardingLogzIo</code>.
   * gatewayUpdateLogForwardingLogzIo is a command that updates log forwarding config (logz-io target)
   * @alias module:model/GatewayUpdateLogForwardingLogzIo
   */
  function GatewayUpdateLogForwardingLogzIo() {
    _classCallCheck(this, GatewayUpdateLogForwardingLogzIo);

    GatewayUpdateLogForwardingLogzIo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateLogForwardingLogzIo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayUpdateLogForwardingLogzIo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateLogForwardingLogzIo} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateLogForwardingLogzIo} The populated <code>GatewayUpdateLogForwardingLogzIo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateLogForwardingLogzIo();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('logz-io-token')) {
          obj['logz-io-token'] = _ApiClient["default"].convertToType(data['logz-io-token'], 'String');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('protocol')) {
          obj['protocol'] = _ApiClient["default"].convertToType(data['protocol'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateLogForwardingLogzIo;
}();
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */


GatewayUpdateLogForwardingLogzIo.prototype['enable'] = 'true';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateLogForwardingLogzIo.prototype['json'] = false;
/**
 * Logz-io token
 * @member {String} logz-io-token
 */

GatewayUpdateLogForwardingLogzIo.prototype['logz-io-token'] = undefined;
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GatewayUpdateLogForwardingLogzIo.prototype['output-format'] = 'text';
/**
 * LogzIo protocol [tcp/https]
 * @member {String} protocol
 */

GatewayUpdateLogForwardingLogzIo.prototype['protocol'] = undefined;
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GatewayUpdateLogForwardingLogzIo.prototype['pull-interval'] = '10';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateLogForwardingLogzIo.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateLogForwardingLogzIo.prototype['uid-token'] = undefined;
var _default = GatewayUpdateLogForwardingLogzIo;
exports["default"] = _default;
