// akeyless/dist/model/GatewayUpdateProducerMSSQL.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerMSSQL model module.
 * @module model/GatewayUpdateProducerMSSQL
 * @version 5.0.14
 */
var GatewayUpdateProducerMSSQL = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerMSSQL</code>.
   * gatewayUpdateProducerMSSQL is a command that updates mssql producer [Deprecated: Use dynamic-secret-update-mssql command]
   * @alias module:model/GatewayUpdateProducerMSSQL
   * @param name {String} Dynamic secret name
   */
  function GatewayUpdateProducerMSSQL(name) {
    _classCallCheck(this, GatewayUpdateProducerMSSQL);

    GatewayUpdateProducerMSSQL.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerMSSQL, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['json'] = false;
      obj['mssql-host'] = '127.0.0.1';
      obj['mssql-port'] = '1433';
      obj['name'] = name;
      obj['secure-access-web'] = false;
      obj['user-ttl'] = '60m';
    }
    /**
     * Constructs a <code>GatewayUpdateProducerMSSQL</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerMSSQL} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerMSSQL} The populated <code>GatewayUpdateProducerMSSQL</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerMSSQL();

        if (data.hasOwnProperty('custom-username-template')) {
          obj['custom-username-template'] = _ApiClient["default"].convertToType(data['custom-username-template'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('item-custom-fields')) {
          obj['item-custom-fields'] = _ApiClient["default"].convertToType(data['item-custom-fields'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('mssql-create-statements')) {
          obj['mssql-create-statements'] = _ApiClient["default"].convertToType(data['mssql-create-statements'], 'String');
        }

        if (data.hasOwnProperty('mssql-dbname')) {
          obj['mssql-dbname'] = _ApiClient["default"].convertToType(data['mssql-dbname'], 'String');
        }

        if (data.hasOwnProperty('mssql-host')) {
          obj['mssql-host'] = _ApiClient["default"].convertToType(data['mssql-host'], 'String');
        }

        if (data.hasOwnProperty('mssql-password')) {
          obj['mssql-password'] = _ApiClient["default"].convertToType(data['mssql-password'], 'String');
        }

        if (data.hasOwnProperty('mssql-port')) {
          obj['mssql-port'] = _ApiClient["default"].convertToType(data['mssql-port'], 'String');
        }

        if (data.hasOwnProperty('mssql-revocation-statements')) {
          obj['mssql-revocation-statements'] = _ApiClient["default"].convertToType(data['mssql-revocation-statements'], 'String');
        }

        if (data.hasOwnProperty('mssql-username')) {
          obj['mssql-username'] = _ApiClient["default"].convertToType(data['mssql-username'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password-length')) {
          obj['password-length'] = _ApiClient["default"].convertToType(data['password-length'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-certificate-issuer')) {
          obj['secure-access-certificate-issuer'] = _ApiClient["default"].convertToType(data['secure-access-certificate-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-db-name')) {
          obj['secure-access-db-name'] = _ApiClient["default"].convertToType(data['secure-access-db-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-db-schema')) {
          obj['secure-access-db-schema'] = _ApiClient["default"].convertToType(data['secure-access-db-schema'], 'String');
        }

        if (data.hasOwnProperty('secure-access-delay')) {
          obj['secure-access-delay'] = _ApiClient["default"].convertToType(data['secure-access-delay'], 'Number');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GatewayUpdateProducerMSSQL</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GatewayUpdateProducerMSSQL</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GatewayUpdateProducerMSSQL.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['custom-username-template'] && !(typeof data['custom-username-template'] === 'string' || data['custom-username-template'] instanceof String)) {
        throw new Error("Expected the field `custom-username-template` to be a primitive type in the JSON string but got " + data['custom-username-template']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['mssql-create-statements'] && !(typeof data['mssql-create-statements'] === 'string' || data['mssql-create-statements'] instanceof String)) {
        throw new Error("Expected the field `mssql-create-statements` to be a primitive type in the JSON string but got " + data['mssql-create-statements']);
      } // ensure the json data is a string


      if (data['mssql-dbname'] && !(typeof data['mssql-dbname'] === 'string' || data['mssql-dbname'] instanceof String)) {
        throw new Error("Expected the field `mssql-dbname` to be a primitive type in the JSON string but got " + data['mssql-dbname']);
      } // ensure the json data is a string


      if (data['mssql-host'] && !(typeof data['mssql-host'] === 'string' || data['mssql-host'] instanceof String)) {
        throw new Error("Expected the field `mssql-host` to be a primitive type in the JSON string but got " + data['mssql-host']);
      } // ensure the json data is a string


      if (data['mssql-password'] && !(typeof data['mssql-password'] === 'string' || data['mssql-password'] instanceof String)) {
        throw new Error("Expected the field `mssql-password` to be a primitive type in the JSON string but got " + data['mssql-password']);
      } // ensure the json data is a string


      if (data['mssql-port'] && !(typeof data['mssql-port'] === 'string' || data['mssql-port'] instanceof String)) {
        throw new Error("Expected the field `mssql-port` to be a primitive type in the JSON string but got " + data['mssql-port']);
      } // ensure the json data is a string


      if (data['mssql-revocation-statements'] && !(typeof data['mssql-revocation-statements'] === 'string' || data['mssql-revocation-statements'] instanceof String)) {
        throw new Error("Expected the field `mssql-revocation-statements` to be a primitive type in the JSON string but got " + data['mssql-revocation-statements']);
      } // ensure the json data is a string


      if (data['mssql-username'] && !(typeof data['mssql-username'] === 'string' || data['mssql-username'] instanceof String)) {
        throw new Error("Expected the field `mssql-username` to be a primitive type in the JSON string but got " + data['mssql-username']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['password-length'] && !(typeof data['password-length'] === 'string' || data['password-length'] instanceof String)) {
        throw new Error("Expected the field `password-length` to be a primitive type in the JSON string but got " + data['password-length']);
      } // ensure the json data is a string


      if (data['producer-encryption-key-name'] && !(typeof data['producer-encryption-key-name'] === 'string' || data['producer-encryption-key-name'] instanceof String)) {
        throw new Error("Expected the field `producer-encryption-key-name` to be a primitive type in the JSON string but got " + data['producer-encryption-key-name']);
      } // ensure the json data is a string


      if (data['secure-access-bastion-issuer'] && !(typeof data['secure-access-bastion-issuer'] === 'string' || data['secure-access-bastion-issuer'] instanceof String)) {
        throw new Error("Expected the field `secure-access-bastion-issuer` to be a primitive type in the JSON string but got " + data['secure-access-bastion-issuer']);
      } // ensure the json data is a string


      if (data['secure-access-certificate-issuer'] && !(typeof data['secure-access-certificate-issuer'] === 'string' || data['secure-access-certificate-issuer'] instanceof String)) {
        throw new Error("Expected the field `secure-access-certificate-issuer` to be a primitive type in the JSON string but got " + data['secure-access-certificate-issuer']);
      } // ensure the json data is a string


      if (data['secure-access-db-name'] && !(typeof data['secure-access-db-name'] === 'string' || data['secure-access-db-name'] instanceof String)) {
        throw new Error("Expected the field `secure-access-db-name` to be a primitive type in the JSON string but got " + data['secure-access-db-name']);
      } // ensure the json data is a string


      if (data['secure-access-db-schema'] && !(typeof data['secure-access-db-schema'] === 'string' || data['secure-access-db-schema'] instanceof String)) {
        throw new Error("Expected the field `secure-access-db-schema` to be a primitive type in the JSON string but got " + data['secure-access-db-schema']);
      } // ensure the json data is a string


      if (data['secure-access-enable'] && !(typeof data['secure-access-enable'] === 'string' || data['secure-access-enable'] instanceof String)) {
        throw new Error("Expected the field `secure-access-enable` to be a primitive type in the JSON string but got " + data['secure-access-enable']);
      } // ensure the json data is an array


      if (!Array.isArray(data['secure-access-host'])) {
        throw new Error("Expected the field `secure-access-host` to be an array in the JSON data but got " + data['secure-access-host']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-ttl'] && !(typeof data['user-ttl'] === 'string' || data['user-ttl'] instanceof String)) {
        throw new Error("Expected the field `user-ttl` to be a primitive type in the JSON string but got " + data['user-ttl']);
      }

      return true;
    }
  }]);

  return GatewayUpdateProducerMSSQL;
}();

GatewayUpdateProducerMSSQL.RequiredProperties = ["name"];
/**
 * Customize how temporary usernames are generated using go template
 * @member {String} custom-username-template
 */

GatewayUpdateProducerMSSQL.prototype['custom-username-template'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

GatewayUpdateProducerMSSQL.prototype['delete_protection'] = undefined;
/**
 * Additional custom fields to associate with the item
 * @member {Object.<String, String>} item-custom-fields
 */

GatewayUpdateProducerMSSQL.prototype['item-custom-fields'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GatewayUpdateProducerMSSQL.prototype['json'] = false;
/**
 * MSSQL Creation statements
 * @member {String} mssql-create-statements
 */

GatewayUpdateProducerMSSQL.prototype['mssql-create-statements'] = undefined;
/**
 * MSSQL Name
 * @member {String} mssql-dbname
 */

GatewayUpdateProducerMSSQL.prototype['mssql-dbname'] = undefined;
/**
 * MSSQL Host
 * @member {String} mssql-host
 * @default '127.0.0.1'
 */

GatewayUpdateProducerMSSQL.prototype['mssql-host'] = '127.0.0.1';
/**
 * MSSQL Password
 * @member {String} mssql-password
 */

GatewayUpdateProducerMSSQL.prototype['mssql-password'] = undefined;
/**
 * MSSQL Port
 * @member {String} mssql-port
 * @default '1433'
 */

GatewayUpdateProducerMSSQL.prototype['mssql-port'] = '1433';
/**
 * MSSQL Revocation statements
 * @member {String} mssql-revocation-statements
 */

GatewayUpdateProducerMSSQL.prototype['mssql-revocation-statements'] = undefined;
/**
 * MSSQL Username
 * @member {String} mssql-username
 */

GatewayUpdateProducerMSSQL.prototype['mssql-username'] = undefined;
/**
 * Dynamic secret name
 * @member {String} name
 */

GatewayUpdateProducerMSSQL.prototype['name'] = undefined;
/**
 * Dynamic secret name
 * @member {String} new-name
 */

GatewayUpdateProducerMSSQL.prototype['new-name'] = undefined;
/**
 * The length of the password to be generated
 * @member {String} password-length
 */

GatewayUpdateProducerMSSQL.prototype['password-length'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerMSSQL.prototype['producer-encryption-key-name'] = undefined;
/**
 * Deprecated. use secure-access-certificate-issuer
 * @member {String} secure-access-bastion-issuer
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * Path to the SSH Certificate Issuer for your Akeyless Secure Access
 * @member {String} secure-access-certificate-issuer
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-certificate-issuer'] = undefined;
/**
 * The DB name (relevant only for DB Dynamic-Secret)
 * @member {String} secure-access-db-name
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-db-name'] = undefined;
/**
 * The DB schema
 * @member {String} secure-access-db-schema
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-db-schema'] = undefined;
/**
 * The delay duration, in seconds, to wait after generating just-in-time credentials. Accepted range: 0-120 seconds
 * @member {Number} secure-access-delay
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-delay'] = undefined;
/**
 * Enable/Disable secure remote access [true/false]
 * @member {String} secure-access-enable
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-enable'] = undefined;
/**
 * Target DB servers for connections (In case of Linked Target association, host(s) will inherit Linked Target hosts)
 * @member {Array.<String>} secure-access-host
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-host'] = undefined;
/**
 * Enable Web Secure Remote Access
 * @member {Boolean} secure-access-web
 * @default false
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-web'] = false;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerMSSQL.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerMSSQL.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerMSSQL.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerMSSQL.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerMSSQL.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerMSSQL;
exports["default"] = _default;
