// akeyless/dist/model/GenCustomerFragment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenCustomerFragment model module.
 * @module model/GenCustomerFragment
 * @version 5.0.5
 */
var GenCustomerFragment = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenCustomerFragment</code>.
   * @alias module:model/GenCustomerFragment
   */
  function GenCustomerFragment() {
    _classCallCheck(this, GenCustomerFragment);

    GenCustomerFragment.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenCustomerFragment, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
      obj['type'] = 'standard';
    }
    /**
     * Constructs a <code>GenCustomerFragment</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenCustomerFragment} obj Optional instance to populate.
     * @return {module:model/GenCustomerFragment} The populated <code>GenCustomerFragment</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenCustomerFragment();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('hsm-key-label')) {
          obj['hsm-key-label'] = _ApiClient["default"].convertToType(data['hsm-key-label'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenCustomerFragment</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenCustomerFragment</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['hsm-key-label'] && !(typeof data['hsm-key-label'] === 'string' || data['hsm-key-label'] instanceof String)) {
        throw new Error("Expected the field `hsm-key-label` to be a primitive type in the JSON string but got " + data['hsm-key-label']);
      } // ensure the json data is a string


      if (data['metadata'] && !(typeof data['metadata'] === 'string' || data['metadata'] instanceof String)) {
        throw new Error("Expected the field `metadata` to be a primitive type in the JSON string but got " + data['metadata']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['type'] && !(typeof data['type'] === 'string' || data['type'] instanceof String)) {
        throw new Error("Expected the field `type` to be a primitive type in the JSON string but got " + data['type']);
      }

      return true;
    }
  }]);

  return GenCustomerFragment;
}();
/**
 * Description of the object
 * @member {String} description
 */


GenCustomerFragment.prototype['description'] = undefined;
/**
 * The label of the hsm key to use for customer fragment operations (relevant for hsm_wrapped/hsm_protected customer fragments)
 * @member {String} hsm-key-label
 */

GenCustomerFragment.prototype['hsm-key-label'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GenCustomerFragment.prototype['json'] = false;
/**
 * Deprecated - use description
 * @member {String} metadata
 */

GenCustomerFragment.prototype['metadata'] = undefined;
/**
 * Customer fragment name
 * @member {String} name
 */

GenCustomerFragment.prototype['name'] = undefined;
/**
 * Customer fragment type [standard/hsm_wrapped/hsm_secured]
 * @member {String} type
 * @default 'standard'
 */

GenCustomerFragment.prototype['type'] = 'standard';
var _default = GenCustomerFragment;
exports["default"] = _default;
