// akeyless/dist/model/GenerateCA.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenerateCA model module.
 * @module model/GenerateCA
 * @version 5.0.1
 */
var GenerateCA = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenerateCA</code>.
   * GenerateCA is a command that creates a new PKI CA and Intermediate issuers
   * @alias module:model/GenerateCA
   * @param allowedDomains {String} A list of the allowed domains that clients can request to be included in the certificate (in a comma-delimited list)
   * @param pkiChainName {String} PKI chain name
   * @param ttl {String} The maximum requested Time To Live for issued certificate by default in seconds, supported formats are s,m,h,d
   */
  function GenerateCA(allowedDomains, pkiChainName, ttl) {
    _classCallCheck(this, GenerateCA);

    GenerateCA.initialize(this, allowedDomains, pkiChainName, ttl);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenerateCA, null, [{
    key: "initialize",
    value: function initialize(obj, allowedDomains, pkiChainName, ttl) {
      obj['allowed-domains'] = allowedDomains;
      obj['extended-key-usage'] = 'serverauth,clientauth';
      obj['json'] = false;
      obj['max-path-len'] = 1;
      obj['pki-chain-name'] = pkiChainName;
      obj['split-level'] = 3;
      obj['ttl'] = ttl;
    }
    /**
     * Constructs a <code>GenerateCA</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenerateCA} obj Optional instance to populate.
     * @return {module:model/GenerateCA} The populated <code>GenerateCA</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenerateCA();

        if (data.hasOwnProperty('alg')) {
          obj['alg'] = _ApiClient["default"].convertToType(data['alg'], 'String');
        }

        if (data.hasOwnProperty('allowed-domains')) {
          obj['allowed-domains'] = _ApiClient["default"].convertToType(data['allowed-domains'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('extended-key-usage')) {
          obj['extended-key-usage'] = _ApiClient["default"].convertToType(data['extended-key-usage'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-type')) {
          obj['key-type'] = _ApiClient["default"].convertToType(data['key-type'], 'String');
        }

        if (data.hasOwnProperty('max-path-len')) {
          obj['max-path-len'] = _ApiClient["default"].convertToType(data['max-path-len'], 'Number');
        }

        if (data.hasOwnProperty('pki-chain-name')) {
          obj['pki-chain-name'] = _ApiClient["default"].convertToType(data['pki-chain-name'], 'String');
        }

        if (data.hasOwnProperty('protection-key-name')) {
          obj['protection-key-name'] = _ApiClient["default"].convertToType(data['protection-key-name'], 'String');
        }

        if (data.hasOwnProperty('split-level')) {
          obj['split-level'] = _ApiClient["default"].convertToType(data['split-level'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenerateCA</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenerateCA</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GenerateCA.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['alg'] && !(typeof data['alg'] === 'string' || data['alg'] instanceof String)) {
        throw new Error("Expected the field `alg` to be a primitive type in the JSON string but got " + data['alg']);
      } // ensure the json data is a string


      if (data['allowed-domains'] && !(typeof data['allowed-domains'] === 'string' || data['allowed-domains'] instanceof String)) {
        throw new Error("Expected the field `allowed-domains` to be a primitive type in the JSON string but got " + data['allowed-domains']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['extended-key-usage'] && !(typeof data['extended-key-usage'] === 'string' || data['extended-key-usage'] instanceof String)) {
        throw new Error("Expected the field `extended-key-usage` to be a primitive type in the JSON string but got " + data['extended-key-usage']);
      } // ensure the json data is a string


      if (data['key-type'] && !(typeof data['key-type'] === 'string' || data['key-type'] instanceof String)) {
        throw new Error("Expected the field `key-type` to be a primitive type in the JSON string but got " + data['key-type']);
      } // ensure the json data is a string


      if (data['pki-chain-name'] && !(typeof data['pki-chain-name'] === 'string' || data['pki-chain-name'] instanceof String)) {
        throw new Error("Expected the field `pki-chain-name` to be a primitive type in the JSON string but got " + data['pki-chain-name']);
      } // ensure the json data is a string


      if (data['protection-key-name'] && !(typeof data['protection-key-name'] === 'string' || data['protection-key-name'] instanceof String)) {
        throw new Error("Expected the field `protection-key-name` to be a primitive type in the JSON string but got " + data['protection-key-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['ttl'] && !(typeof data['ttl'] === 'string' || data['ttl'] instanceof String)) {
        throw new Error("Expected the field `ttl` to be a primitive type in the JSON string but got " + data['ttl']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return GenerateCA;
}();

GenerateCA.RequiredProperties = ["allowed-domains", "pki-chain-name", "ttl"];
/**
 * @member {String} alg
 */

GenerateCA.prototype['alg'] = undefined;
/**
 * A list of the allowed domains that clients can request to be included in the certificate (in a comma-delimited list)
 * @member {String} allowed-domains
 */

GenerateCA.prototype['allowed-domains'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

GenerateCA.prototype['delete_protection'] = undefined;
/**
 * A comma sepereted list of extended key usage for the intermediate (serverauth / clientauth / codesigning)
 * @member {String} extended-key-usage
 * @default 'serverauth,clientauth'
 */

GenerateCA.prototype['extended-key-usage'] = 'serverauth,clientauth';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GenerateCA.prototype['json'] = false;
/**
 * @member {String} key-type
 */

GenerateCA.prototype['key-type'] = undefined;
/**
 * The maximum number of intermediate certificates that can appear in a certification path
 * @member {Number} max-path-len
 * @default 1
 */

GenerateCA.prototype['max-path-len'] = 1;
/**
 * PKI chain name
 * @member {String} pki-chain-name
 */

GenerateCA.prototype['pki-chain-name'] = undefined;
/**
 * The name of a key that used to encrypt the secrets values (if empty, the account default protectionKey key will be used)
 * @member {String} protection-key-name
 */

GenerateCA.prototype['protection-key-name'] = undefined;
/**
 * The number of fragments that the item will be split into
 * @member {Number} split-level
 * @default 3
 */

GenerateCA.prototype['split-level'] = 3;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GenerateCA.prototype['token'] = undefined;
/**
 * The maximum requested Time To Live for issued certificate by default in seconds, supported formats are s,m,h,d
 * @member {String} ttl
 */

GenerateCA.prototype['ttl'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GenerateCA.prototype['uid-token'] = undefined;
var _default = GenerateCA;
exports["default"] = _default;
