// akeyless/dist/model/GenerateCsr.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenerateCsr model module.
 * @module model/GenerateCsr
 * @version 5.0.5
 */
var GenerateCsr = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenerateCsr</code>.
   * @alias module:model/GenerateCsr
   * @param commonName {String} The common name to be included in the CSR certificate
   * @param keyType {String} The type of the key to generate (classic-key/dfc)
   * @param name {String} The key name
   */
  function GenerateCsr(commonName, keyType, name) {
    _classCallCheck(this, GenerateCsr);

    GenerateCsr.initialize(this, commonName, keyType, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenerateCsr, null, [{
    key: "initialize",
    value: function initialize(obj, commonName, keyType, name) {
      obj['common-name'] = commonName;
      obj['export-private-key'] = false;
      obj['json'] = false;
      obj['key-type'] = keyType || 'classic-key';
      obj['name'] = name;
      obj['split-level'] = 3;
    }
    /**
     * Constructs a <code>GenerateCsr</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenerateCsr} obj Optional instance to populate.
     * @return {module:model/GenerateCsr} The populated <code>GenerateCsr</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenerateCsr();

        if (data.hasOwnProperty('alg')) {
          obj['alg'] = _ApiClient["default"].convertToType(data['alg'], 'String');
        }

        if (data.hasOwnProperty('alt-names')) {
          obj['alt-names'] = _ApiClient["default"].convertToType(data['alt-names'], 'String');
        }

        if (data.hasOwnProperty('certificate-type')) {
          obj['certificate-type'] = _ApiClient["default"].convertToType(data['certificate-type'], 'String');
        }

        if (data.hasOwnProperty('city')) {
          obj['city'] = _ApiClient["default"].convertToType(data['city'], 'String');
        }

        if (data.hasOwnProperty('common-name')) {
          obj['common-name'] = _ApiClient["default"].convertToType(data['common-name'], 'String');
        }

        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }

        if (data.hasOwnProperty('critical')) {
          obj['critical'] = _ApiClient["default"].convertToType(data['critical'], 'Boolean');
        }

        if (data.hasOwnProperty('dep')) {
          obj['dep'] = _ApiClient["default"].convertToType(data['dep'], 'String');
        }

        if (data.hasOwnProperty('email-addresses')) {
          obj['email-addresses'] = _ApiClient["default"].convertToType(data['email-addresses'], 'String');
        }

        if (data.hasOwnProperty('export-private-key')) {
          obj['export-private-key'] = _ApiClient["default"].convertToType(data['export-private-key'], 'Boolean');
        }

        if (data.hasOwnProperty('generate-key')) {
          obj['generate-key'] = _ApiClient["default"].convertToType(data['generate-key'], 'Boolean');
        }

        if (data.hasOwnProperty('ip-addresses')) {
          obj['ip-addresses'] = _ApiClient["default"].convertToType(data['ip-addresses'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-type')) {
          obj['key-type'] = _ApiClient["default"].convertToType(data['key-type'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('org')) {
          obj['org'] = _ApiClient["default"].convertToType(data['org'], 'String');
        }

        if (data.hasOwnProperty('split-level')) {
          obj['split-level'] = _ApiClient["default"].convertToType(data['split-level'], 'Number');
        }

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('uri-sans')) {
          obj['uri-sans'] = _ApiClient["default"].convertToType(data['uri-sans'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenerateCsr</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenerateCsr</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GenerateCsr.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['alg'] && !(typeof data['alg'] === 'string' || data['alg'] instanceof String)) {
        throw new Error("Expected the field `alg` to be a primitive type in the JSON string but got " + data['alg']);
      } // ensure the json data is a string


      if (data['alt-names'] && !(typeof data['alt-names'] === 'string' || data['alt-names'] instanceof String)) {
        throw new Error("Expected the field `alt-names` to be a primitive type in the JSON string but got " + data['alt-names']);
      } // ensure the json data is a string


      if (data['certificate-type'] && !(typeof data['certificate-type'] === 'string' || data['certificate-type'] instanceof String)) {
        throw new Error("Expected the field `certificate-type` to be a primitive type in the JSON string but got " + data['certificate-type']);
      } // ensure the json data is a string


      if (data['city'] && !(typeof data['city'] === 'string' || data['city'] instanceof String)) {
        throw new Error("Expected the field `city` to be a primitive type in the JSON string but got " + data['city']);
      } // ensure the json data is a string


      if (data['common-name'] && !(typeof data['common-name'] === 'string' || data['common-name'] instanceof String)) {
        throw new Error("Expected the field `common-name` to be a primitive type in the JSON string but got " + data['common-name']);
      } // ensure the json data is a string


      if (data['country'] && !(typeof data['country'] === 'string' || data['country'] instanceof String)) {
        throw new Error("Expected the field `country` to be a primitive type in the JSON string but got " + data['country']);
      } // ensure the json data is a string


      if (data['dep'] && !(typeof data['dep'] === 'string' || data['dep'] instanceof String)) {
        throw new Error("Expected the field `dep` to be a primitive type in the JSON string but got " + data['dep']);
      } // ensure the json data is a string


      if (data['email-addresses'] && !(typeof data['email-addresses'] === 'string' || data['email-addresses'] instanceof String)) {
        throw new Error("Expected the field `email-addresses` to be a primitive type in the JSON string but got " + data['email-addresses']);
      } // ensure the json data is a string


      if (data['ip-addresses'] && !(typeof data['ip-addresses'] === 'string' || data['ip-addresses'] instanceof String)) {
        throw new Error("Expected the field `ip-addresses` to be a primitive type in the JSON string but got " + data['ip-addresses']);
      } // ensure the json data is a string


      if (data['key-type'] && !(typeof data['key-type'] === 'string' || data['key-type'] instanceof String)) {
        throw new Error("Expected the field `key-type` to be a primitive type in the JSON string but got " + data['key-type']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['org'] && !(typeof data['org'] === 'string' || data['org'] instanceof String)) {
        throw new Error("Expected the field `org` to be a primitive type in the JSON string but got " + data['org']);
      } // ensure the json data is a string


      if (data['state'] && !(typeof data['state'] === 'string' || data['state'] instanceof String)) {
        throw new Error("Expected the field `state` to be a primitive type in the JSON string but got " + data['state']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['uri-sans'] && !(typeof data['uri-sans'] === 'string' || data['uri-sans'] instanceof String)) {
        throw new Error("Expected the field `uri-sans` to be a primitive type in the JSON string but got " + data['uri-sans']);
      }

      return true;
    }
  }]);

  return GenerateCsr;
}();

GenerateCsr.RequiredProperties = ["common-name", "key-type", "name"];
/**
 * @member {String} alg
 */

GenerateCsr.prototype['alg'] = undefined;
/**
 * A comma-separated list of dns alternative names
 * @member {String} alt-names
 */

GenerateCsr.prototype['alt-names'] = undefined;
/**
 * The certificate type to be included in the CSR certificate (ssl-client/ssl-server/certificate-signing)
 * @member {String} certificate-type
 */

GenerateCsr.prototype['certificate-type'] = undefined;
/**
 * The city to be included in the CSR certificate
 * @member {String} city
 */

GenerateCsr.prototype['city'] = undefined;
/**
 * The common name to be included in the CSR certificate
 * @member {String} common-name
 */

GenerateCsr.prototype['common-name'] = undefined;
/**
 * The country to be included in the CSR certificate
 * @member {String} country
 */

GenerateCsr.prototype['country'] = undefined;
/**
 * Add critical to the key usage extension (will be false if not added)
 * @member {Boolean} critical
 */

GenerateCsr.prototype['critical'] = undefined;
/**
 * The department to be included in the CSR certificate
 * @member {String} dep
 */

GenerateCsr.prototype['dep'] = undefined;
/**
 * A comma-separated list of email addresses alternative names
 * @member {String} email-addresses
 */

GenerateCsr.prototype['email-addresses'] = undefined;
/**
 * The flag to indicate if the private key should be exported
 * @member {Boolean} export-private-key
 * @default false
 */

GenerateCsr.prototype['export-private-key'] = false;
/**
 * Generate a new classic key for the csr
 * @member {Boolean} generate-key
 */

GenerateCsr.prototype['generate-key'] = undefined;
/**
 * A comma-separated list of ip addresses alternative names
 * @member {String} ip-addresses
 */

GenerateCsr.prototype['ip-addresses'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GenerateCsr.prototype['json'] = false;
/**
 * The type of the key to generate (classic-key/dfc)
 * @member {String} key-type
 * @default 'classic-key'
 */

GenerateCsr.prototype['key-type'] = 'classic-key';
/**
 * The key name
 * @member {String} name
 */

GenerateCsr.prototype['name'] = undefined;
/**
 * The organization to be included in the CSR certificate
 * @member {String} org
 */

GenerateCsr.prototype['org'] = undefined;
/**
 * The number of fragments that the item will be split into (not includes customer fragment)
 * @member {Number} split-level
 * @default 3
 */

GenerateCsr.prototype['split-level'] = 3;
/**
 * The state to be included in the CSR certificate
 * @member {String} state
 */

GenerateCsr.prototype['state'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GenerateCsr.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GenerateCsr.prototype['uid-token'] = undefined;
/**
 * A comma-separated list of uri alternative names
 * @member {String} uri-sans
 */

GenerateCsr.prototype['uri-sans'] = undefined;
var _default = GenerateCsr;
exports["default"] = _default;
