// akeyless/dist/model/GenerateCAOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenerateCAOutput model module.
 * @module model/GenerateCAOutput
 * @version 5.0.14
 */
var GenerateCAOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenerateCAOutput</code>.
   * @alias module:model/GenerateCAOutput
   */
  function GenerateCAOutput() {
    _classCallCheck(this, GenerateCAOutput);

    GenerateCAOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenerateCAOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GenerateCAOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenerateCAOutput} obj Optional instance to populate.
     * @return {module:model/GenerateCAOutput} The populated <code>GenerateCAOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenerateCAOutput();

        if (data.hasOwnProperty('intermediate_certificate_name')) {
          obj['intermediate_certificate_name'] = _ApiClient["default"].convertToType(data['intermediate_certificate_name'], 'String');
        }

        if (data.hasOwnProperty('intermediate_issuer_name')) {
          obj['intermediate_issuer_name'] = _ApiClient["default"].convertToType(data['intermediate_issuer_name'], 'String');
        }

        if (data.hasOwnProperty('intermediate_key_name')) {
          obj['intermediate_key_name'] = _ApiClient["default"].convertToType(data['intermediate_key_name'], 'String');
        }

        if (data.hasOwnProperty('root_certificate_name')) {
          obj['root_certificate_name'] = _ApiClient["default"].convertToType(data['root_certificate_name'], 'String');
        }

        if (data.hasOwnProperty('root_issuer_name')) {
          obj['root_issuer_name'] = _ApiClient["default"].convertToType(data['root_issuer_name'], 'String');
        }

        if (data.hasOwnProperty('root_key_name')) {
          obj['root_key_name'] = _ApiClient["default"].convertToType(data['root_key_name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenerateCAOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenerateCAOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['intermediate_certificate_name'] && !(typeof data['intermediate_certificate_name'] === 'string' || data['intermediate_certificate_name'] instanceof String)) {
        throw new Error("Expected the field `intermediate_certificate_name` to be a primitive type in the JSON string but got " + data['intermediate_certificate_name']);
      } // ensure the json data is a string


      if (data['intermediate_issuer_name'] && !(typeof data['intermediate_issuer_name'] === 'string' || data['intermediate_issuer_name'] instanceof String)) {
        throw new Error("Expected the field `intermediate_issuer_name` to be a primitive type in the JSON string but got " + data['intermediate_issuer_name']);
      } // ensure the json data is a string


      if (data['intermediate_key_name'] && !(typeof data['intermediate_key_name'] === 'string' || data['intermediate_key_name'] instanceof String)) {
        throw new Error("Expected the field `intermediate_key_name` to be a primitive type in the JSON string but got " + data['intermediate_key_name']);
      } // ensure the json data is a string


      if (data['root_certificate_name'] && !(typeof data['root_certificate_name'] === 'string' || data['root_certificate_name'] instanceof String)) {
        throw new Error("Expected the field `root_certificate_name` to be a primitive type in the JSON string but got " + data['root_certificate_name']);
      } // ensure the json data is a string


      if (data['root_issuer_name'] && !(typeof data['root_issuer_name'] === 'string' || data['root_issuer_name'] instanceof String)) {
        throw new Error("Expected the field `root_issuer_name` to be a primitive type in the JSON string but got " + data['root_issuer_name']);
      } // ensure the json data is a string


      if (data['root_key_name'] && !(typeof data['root_key_name'] === 'string' || data['root_key_name'] instanceof String)) {
        throw new Error("Expected the field `root_key_name` to be a primitive type in the JSON string but got " + data['root_key_name']);
      }

      return true;
    }
  }]);

  return GenerateCAOutput;
}();
/**
 * @member {String} intermediate_certificate_name
 */


GenerateCAOutput.prototype['intermediate_certificate_name'] = undefined;
/**
 * @member {String} intermediate_issuer_name
 */

GenerateCAOutput.prototype['intermediate_issuer_name'] = undefined;
/**
 * @member {String} intermediate_key_name
 */

GenerateCAOutput.prototype['intermediate_key_name'] = undefined;
/**
 * @member {String} root_certificate_name
 */

GenerateCAOutput.prototype['root_certificate_name'] = undefined;
/**
 * @member {String} root_issuer_name
 */

GenerateCAOutput.prototype['root_issuer_name'] = undefined;
/**
 * @member {String} root_key_name
 */

GenerateCAOutput.prototype['root_key_name'] = undefined;
var _default = GenerateCAOutput;
exports["default"] = _default;
