// akeyless/dist/model/GenerateAcmeEabOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GenerateAcmeEabOutput model module.
 * @module model/GenerateAcmeEabOutput
 * @version 5.0.3
 */
var GenerateAcmeEabOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GenerateAcmeEabOutput</code>.
   * @alias module:model/GenerateAcmeEabOutput
   */
  function GenerateAcmeEabOutput() {
    _classCallCheck(this, GenerateAcmeEabOutput);

    GenerateAcmeEabOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GenerateAcmeEabOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GenerateAcmeEabOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GenerateAcmeEabOutput} obj Optional instance to populate.
     * @return {module:model/GenerateAcmeEabOutput} The populated <code>GenerateAcmeEabOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GenerateAcmeEabOutput();

        if (data.hasOwnProperty('expires_at')) {
          obj['expires_at'] = _ApiClient["default"].convertToType(data['expires_at'], 'Date');
        }

        if (data.hasOwnProperty('kid')) {
          obj['kid'] = _ApiClient["default"].convertToType(data['kid'], 'String');
        }

        if (data.hasOwnProperty('mac_key')) {
          obj['mac_key'] = _ApiClient["default"].convertToType(data['mac_key'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GenerateAcmeEabOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GenerateAcmeEabOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['kid'] && !(typeof data['kid'] === 'string' || data['kid'] instanceof String)) {
        throw new Error("Expected the field `kid` to be a primitive type in the JSON string but got " + data['kid']);
      } // ensure the json data is a string


      if (data['mac_key'] && !(typeof data['mac_key'] === 'string' || data['mac_key'] instanceof String)) {
        throw new Error("Expected the field `mac_key` to be a primitive type in the JSON string but got " + data['mac_key']);
      }

      return true;
    }
  }]);

  return GenerateAcmeEabOutput;
}();
/**
 * @member {Date} expires_at
 */


GenerateAcmeEabOutput.prototype['expires_at'] = undefined;
/**
 * @member {String} kid
 */

GenerateAcmeEabOutput.prototype['kid'] = undefined;
/**
 * @member {String} mac_key
 */

GenerateAcmeEabOutput.prototype['mac_key'] = undefined;
var _default = GenerateAcmeEabOutput;
exports["default"] = _default;
