// akeyless/dist/model/GetAccountSettingsCommandOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccountGeneralSettings = _interopRequireDefault(require("./AccountGeneralSettings"));

var _AccountObjectVersionSettingsOutput = _interopRequireDefault(require("./AccountObjectVersionSettingsOutput"));

var _CustomerFullAddress = _interopRequireDefault(require("./CustomerFullAddress"));

var _SmInfo = _interopRequireDefault(require("./SmInfo"));

var _SraInfo = _interopRequireDefault(require("./SraInfo"));

var _SystemAccessCredsSettings = _interopRequireDefault(require("./SystemAccessCredsSettings"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetAccountSettingsCommandOutput model module.
 * @module model/GetAccountSettingsCommandOutput
 * @version 5.0.0
 */
var GetAccountSettingsCommandOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetAccountSettingsCommandOutput</code>.
   * @alias module:model/GetAccountSettingsCommandOutput
   */
  function GetAccountSettingsCommandOutput() {
    _classCallCheck(this, GetAccountSettingsCommandOutput);

    GetAccountSettingsCommandOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetAccountSettingsCommandOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetAccountSettingsCommandOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetAccountSettingsCommandOutput} obj Optional instance to populate.
     * @return {module:model/GetAccountSettingsCommandOutput} The populated <code>GetAccountSettingsCommandOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetAccountSettingsCommandOutput();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('address')) {
          obj['address'] = _CustomerFullAddress["default"].constructFromObject(data['address']);
        }

        if (data.hasOwnProperty('company_name')) {
          obj['company_name'] = _ApiClient["default"].convertToType(data['company_name'], 'String');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('general_settings')) {
          obj['general_settings'] = _AccountGeneralSettings["default"].constructFromObject(data['general_settings']);
        }

        if (data.hasOwnProperty('object_version_settings')) {
          obj['object_version_settings'] = _AccountObjectVersionSettingsOutput["default"].constructFromObject(data['object_version_settings']);
        }

        if (data.hasOwnProperty('phone')) {
          obj['phone'] = _ApiClient["default"].convertToType(data['phone'], 'String');
        }

        if (data.hasOwnProperty('secret_management')) {
          obj['secret_management'] = _SmInfo["default"].constructFromObject(data['secret_management']);
        }

        if (data.hasOwnProperty('secure_remote_access')) {
          obj['secure_remote_access'] = _SraInfo["default"].constructFromObject(data['secure_remote_access']);
        }

        if (data.hasOwnProperty('system_access_creds_settings')) {
          obj['system_access_creds_settings'] = _SystemAccessCredsSettings["default"].constructFromObject(data['system_access_creds_settings']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetAccountSettingsCommandOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetAccountSettingsCommandOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['account_id'] && !(typeof data['account_id'] === 'string' || data['account_id'] instanceof String)) {
        throw new Error("Expected the field `account_id` to be a primitive type in the JSON string but got " + data['account_id']);
      } // validate the optional field `address`


      if (data['address']) {
        // data not null
        _CustomerFullAddress["default"].validateJSON(data['address']);
      } // ensure the json data is a string


      if (data['company_name'] && !(typeof data['company_name'] === 'string' || data['company_name'] instanceof String)) {
        throw new Error("Expected the field `company_name` to be a primitive type in the JSON string but got " + data['company_name']);
      } // ensure the json data is a string


      if (data['email'] && !(typeof data['email'] === 'string' || data['email'] instanceof String)) {
        throw new Error("Expected the field `email` to be a primitive type in the JSON string but got " + data['email']);
      } // validate the optional field `general_settings`


      if (data['general_settings']) {
        // data not null
        _AccountGeneralSettings["default"].validateJSON(data['general_settings']);
      } // validate the optional field `object_version_settings`


      if (data['object_version_settings']) {
        // data not null
        _AccountObjectVersionSettingsOutput["default"].validateJSON(data['object_version_settings']);
      } // ensure the json data is a string


      if (data['phone'] && !(typeof data['phone'] === 'string' || data['phone'] instanceof String)) {
        throw new Error("Expected the field `phone` to be a primitive type in the JSON string but got " + data['phone']);
      } // validate the optional field `secret_management`


      if (data['secret_management']) {
        // data not null
        _SmInfo["default"].validateJSON(data['secret_management']);
      } // validate the optional field `secure_remote_access`


      if (data['secure_remote_access']) {
        // data not null
        _SraInfo["default"].validateJSON(data['secure_remote_access']);
      } // validate the optional field `system_access_creds_settings`


      if (data['system_access_creds_settings']) {
        // data not null
        _SystemAccessCredsSettings["default"].validateJSON(data['system_access_creds_settings']);
      }

      return true;
    }
  }]);

  return GetAccountSettingsCommandOutput;
}();
/**
 * @member {String} account_id
 */


GetAccountSettingsCommandOutput.prototype['account_id'] = undefined;
/**
 * @member {module:model/CustomerFullAddress} address
 */

GetAccountSettingsCommandOutput.prototype['address'] = undefined;
/**
 * @member {String} company_name
 */

GetAccountSettingsCommandOutput.prototype['company_name'] = undefined;
/**
 * @member {String} email
 */

GetAccountSettingsCommandOutput.prototype['email'] = undefined;
/**
 * @member {module:model/AccountGeneralSettings} general_settings
 */

GetAccountSettingsCommandOutput.prototype['general_settings'] = undefined;
/**
 * @member {module:model/AccountObjectVersionSettingsOutput} object_version_settings
 */

GetAccountSettingsCommandOutput.prototype['object_version_settings'] = undefined;
/**
 * @member {String} phone
 */

GetAccountSettingsCommandOutput.prototype['phone'] = undefined;
/**
 * @member {module:model/SmInfo} secret_management
 */

GetAccountSettingsCommandOutput.prototype['secret_management'] = undefined;
/**
 * @member {module:model/SraInfo} secure_remote_access
 */

GetAccountSettingsCommandOutput.prototype['secure_remote_access'] = undefined;
/**
 * @member {module:model/SystemAccessCredsSettings} system_access_creds_settings
 */

GetAccountSettingsCommandOutput.prototype['system_access_creds_settings'] = undefined;
var _default = GetAccountSettingsCommandOutput;
exports["default"] = _default;
