// akeyless/dist/model/GetFolderOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetFolderOutput model module.
 * @module model/GetFolderOutput
 * @version 5.0.14
 */
var GetFolderOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetFolderOutput</code>.
   * GetFolderOutput is the result of the getFolder operation
   * @alias module:model/GetFolderOutput
   */
  function GetFolderOutput() {
    _classCallCheck(this, GetFolderOutput);

    GetFolderOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetFolderOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetFolderOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetFolderOutput} obj Optional instance to populate.
     * @return {module:model/GetFolderOutput} The populated <code>GetFolderOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetFolderOutput();

        if (data.hasOwnProperty('access_date')) {
          obj['access_date'] = _ApiClient["default"].convertToType(data['access_date'], 'Date');
        }

        if (data.hasOwnProperty('access_date_display')) {
          obj['access_date_display'] = _ApiClient["default"].convertToType(data['access_date_display'], 'String');
        }

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'Number');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'Boolean');
        }

        if (data.hasOwnProperty('folder_id')) {
          obj['folder_id'] = _ApiClient["default"].convertToType(data['folder_id'], 'Number');
        }

        if (data.hasOwnProperty('folder_name')) {
          obj['folder_name'] = _ApiClient["default"].convertToType(data['folder_name'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetFolderOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetFolderOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_date_display'] && !(typeof data['access_date_display'] === 'string' || data['access_date_display'] instanceof String)) {
        throw new Error("Expected the field `access_date_display` to be a primitive type in the JSON string but got " + data['access_date_display']);
      } // ensure the json data is a string


      if (data['folder_name'] && !(typeof data['folder_name'] === 'string' || data['folder_name'] instanceof String)) {
        throw new Error("Expected the field `folder_name` to be a primitive type in the JSON string but got " + data['folder_name']);
      } // ensure the json data is a string


      if (data['metadata'] && !(typeof data['metadata'] === 'string' || data['metadata'] instanceof String)) {
        throw new Error("Expected the field `metadata` to be a primitive type in the JSON string but got " + data['metadata']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      }

      return true;
    }
  }]);

  return GetFolderOutput;
}();
/**
 * @member {Date} access_date
 */


GetFolderOutput.prototype['access_date'] = undefined;
/**
 * @member {String} access_date_display
 */

GetFolderOutput.prototype['access_date_display'] = undefined;
/**
 * @member {Number} accessibility
 */

GetFolderOutput.prototype['accessibility'] = undefined;
/**
 * @member {Date} creation_date
 */

GetFolderOutput.prototype['creation_date'] = undefined;
/**
 * @member {Boolean} delete_protection
 */

GetFolderOutput.prototype['delete_protection'] = undefined;
/**
 * @member {Number} folder_id
 */

GetFolderOutput.prototype['folder_id'] = undefined;
/**
 * @member {String} folder_name
 */

GetFolderOutput.prototype['folder_name'] = undefined;
/**
 * @member {String} metadata
 */

GetFolderOutput.prototype['metadata'] = undefined;
/**
 * @member {Date} modification_date
 */

GetFolderOutput.prototype['modification_date'] = undefined;
/**
 * @member {Array.<String>} tags
 */

GetFolderOutput.prototype['tags'] = undefined;
var _default = GetFolderOutput;
exports["default"] = _default;
