// akeyless/dist/model/GetGroupOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccessPermissionAssignment = _interopRequireDefault(require("./AccessPermissionAssignment"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetGroupOutput model module.
 * @module model/GetGroupOutput
 * @version 5.0.8
 */
var GetGroupOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetGroupOutput</code>.
   * @alias module:model/GetGroupOutput
   */
  function GetGroupOutput() {
    _classCallCheck(this, GetGroupOutput);

    GetGroupOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetGroupOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetGroupOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetGroupOutput} obj Optional instance to populate.
     * @return {module:model/GetGroupOutput} The populated <code>GetGroupOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetGroupOutput();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('group_alias')) {
          obj['group_alias'] = _ApiClient["default"].convertToType(data['group_alias'], 'String');
        }

        if (data.hasOwnProperty('group_id')) {
          obj['group_id'] = _ApiClient["default"].convertToType(data['group_id'], 'String');
        }

        if (data.hasOwnProperty('group_name')) {
          obj['group_name'] = _ApiClient["default"].convertToType(data['group_name'], 'String');
        }

        if (data.hasOwnProperty('is_subclaims_with_operator')) {
          obj['is_subclaims_with_operator'] = _ApiClient["default"].convertToType(data['is_subclaims_with_operator'], 'Boolean');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('user_assignments')) {
          obj['user_assignments'] = _ApiClient["default"].convertToType(data['user_assignments'], [_AccessPermissionAssignment["default"]]);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetGroupOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetGroupOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['account_id'] && !(typeof data['account_id'] === 'string' || data['account_id'] instanceof String)) {
        throw new Error("Expected the field `account_id` to be a primitive type in the JSON string but got " + data['account_id']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['group_alias'] && !(typeof data['group_alias'] === 'string' || data['group_alias'] instanceof String)) {
        throw new Error("Expected the field `group_alias` to be a primitive type in the JSON string but got " + data['group_alias']);
      } // ensure the json data is a string


      if (data['group_id'] && !(typeof data['group_id'] === 'string' || data['group_id'] instanceof String)) {
        throw new Error("Expected the field `group_id` to be a primitive type in the JSON string but got " + data['group_id']);
      } // ensure the json data is a string


      if (data['group_name'] && !(typeof data['group_name'] === 'string' || data['group_name'] instanceof String)) {
        throw new Error("Expected the field `group_name` to be a primitive type in the JSON string but got " + data['group_name']);
      }

      if (data['user_assignments']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['user_assignments'])) {
          throw new Error("Expected the field `user_assignments` to be an array in the JSON data but got " + data['user_assignments']);
        } // validate the optional field `user_assignments` (array)


        var _iterator = _createForOfIteratorHelper(data['user_assignments']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _AccessPermissionAssignment["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      }

      return true;
    }
  }]);

  return GetGroupOutput;
}();
/**
 * @member {String} account_id
 */


GetGroupOutput.prototype['account_id'] = undefined;
/**
 * @member {Date} creation_date
 */

GetGroupOutput.prototype['creation_date'] = undefined;
/**
 * @member {String} description
 */

GetGroupOutput.prototype['description'] = undefined;
/**
 * @member {String} group_alias
 */

GetGroupOutput.prototype['group_alias'] = undefined;
/**
 * @member {String} group_id
 */

GetGroupOutput.prototype['group_id'] = undefined;
/**
 * @member {String} group_name
 */

GetGroupOutput.prototype['group_name'] = undefined;
/**
 * @member {Boolean} is_subclaims_with_operator
 */

GetGroupOutput.prototype['is_subclaims_with_operator'] = undefined;
/**
 * @member {Date} modification_date
 */

GetGroupOutput.prototype['modification_date'] = undefined;
/**
 * @member {Array.<module:model/AccessPermissionAssignment>} user_assignments
 */

GetGroupOutput.prototype['user_assignments'] = undefined;
var _default = GetGroupOutput;
exports["default"] = _default;
