// akeyless/dist/model/GetKubeExecCredsOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ClientData = _interopRequireDefault(require("./ClientData"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetKubeExecCredsOutput model module.
 * @module model/GetKubeExecCredsOutput
 * @version 5.0.3
 */
var GetKubeExecCredsOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetKubeExecCredsOutput</code>.
   * @alias module:model/GetKubeExecCredsOutput
   */
  function GetKubeExecCredsOutput() {
    _classCallCheck(this, GetKubeExecCredsOutput);

    GetKubeExecCredsOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetKubeExecCredsOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetKubeExecCredsOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetKubeExecCredsOutput} obj Optional instance to populate.
     * @return {module:model/GetKubeExecCredsOutput} The populated <code>GetKubeExecCredsOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetKubeExecCredsOutput();

        if (data.hasOwnProperty('apiVersion')) {
          obj['apiVersion'] = _ApiClient["default"].convertToType(data['apiVersion'], 'String');
        }

        if (data.hasOwnProperty('kind')) {
          obj['kind'] = _ApiClient["default"].convertToType(data['kind'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ClientData["default"].constructFromObject(data['status']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetKubeExecCredsOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetKubeExecCredsOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['apiVersion'] && !(typeof data['apiVersion'] === 'string' || data['apiVersion'] instanceof String)) {
        throw new Error("Expected the field `apiVersion` to be a primitive type in the JSON string but got " + data['apiVersion']);
      } // ensure the json data is a string


      if (data['kind'] && !(typeof data['kind'] === 'string' || data['kind'] instanceof String)) {
        throw new Error("Expected the field `kind` to be a primitive type in the JSON string but got " + data['kind']);
      } // validate the optional field `status`


      if (data['status']) {
        // data not null
        _ClientData["default"].validateJSON(data['status']);
      }

      return true;
    }
  }]);

  return GetKubeExecCredsOutput;
}();
/**
 * @member {String} apiVersion
 */


GetKubeExecCredsOutput.prototype['apiVersion'] = undefined;
/**
 * @member {String} kind
 */

GetKubeExecCredsOutput.prototype['kind'] = undefined;
/**
 * @member {module:model/ClientData} status
 */

GetKubeExecCredsOutput.prototype['status'] = undefined;
var _default = GetKubeExecCredsOutput;
exports["default"] = _default;
