// akeyless/dist/model/GetRSAPublicOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetRSAPublicOutput model module.
 * @module model/GetRSAPublicOutput
 * @version 5.0.1
 */
var GetRSAPublicOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRSAPublicOutput</code>.
   * @alias module:model/GetRSAPublicOutput
   */
  function GetRSAPublicOutput() {
    _classCallCheck(this, GetRSAPublicOutput);

    GetRSAPublicOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetRSAPublicOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetRSAPublicOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRSAPublicOutput} obj Optional instance to populate.
     * @return {module:model/GetRSAPublicOutput} The populated <code>GetRSAPublicOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRSAPublicOutput();

        if (data.hasOwnProperty('pem')) {
          obj['pem'] = _ApiClient["default"].convertToType(data['pem'], 'String');
        }

        if (data.hasOwnProperty('raw')) {
          obj['raw'] = _ApiClient["default"].convertToType(data['raw'], 'String');
        }

        if (data.hasOwnProperty('ssh')) {
          obj['ssh'] = _ApiClient["default"].convertToType(data['ssh'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetRSAPublicOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetRSAPublicOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['pem'] && !(typeof data['pem'] === 'string' || data['pem'] instanceof String)) {
        throw new Error("Expected the field `pem` to be a primitive type in the JSON string but got " + data['pem']);
      } // ensure the json data is a string


      if (data['raw'] && !(typeof data['raw'] === 'string' || data['raw'] instanceof String)) {
        throw new Error("Expected the field `raw` to be a primitive type in the JSON string but got " + data['raw']);
      } // ensure the json data is a string


      if (data['ssh'] && !(typeof data['ssh'] === 'string' || data['ssh'] instanceof String)) {
        throw new Error("Expected the field `ssh` to be a primitive type in the JSON string but got " + data['ssh']);
      }

      return true;
    }
  }]);

  return GetRSAPublicOutput;
}();
/**
 * @member {String} pem
 */


GetRSAPublicOutput.prototype['pem'] = undefined;
/**
 * @member {String} raw
 */

GetRSAPublicOutput.prototype['raw'] = undefined;
/**
 * @member {String} ssh
 */

GetRSAPublicOutput.prototype['ssh'] = undefined;
var _default = GetRSAPublicOutput;
exports["default"] = _default;
