// akeyless/dist/model/GetSSHCertificateOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetSSHCertificateOutput model module.
 * @module model/GetSSHCertificateOutput
 * @version 5.0.6
 */
var GetSSHCertificateOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetSSHCertificateOutput</code>.
   * @alias module:model/GetSSHCertificateOutput
   */
  function GetSSHCertificateOutput() {
    _classCallCheck(this, GetSSHCertificateOutput);

    GetSSHCertificateOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetSSHCertificateOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetSSHCertificateOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetSSHCertificateOutput} obj Optional instance to populate.
     * @return {module:model/GetSSHCertificateOutput} The populated <code>GetSSHCertificateOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetSSHCertificateOutput();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetSSHCertificateOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetSSHCertificateOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['data'] && !(typeof data['data'] === 'string' || data['data'] instanceof String)) {
        throw new Error("Expected the field `data` to be a primitive type in the JSON string but got " + data['data']);
      } // ensure the json data is a string


      if (data['path'] && !(typeof data['path'] === 'string' || data['path'] instanceof String)) {
        throw new Error("Expected the field `path` to be a primitive type in the JSON string but got " + data['path']);
      }

      return true;
    }
  }]);

  return GetSSHCertificateOutput;
}();
/**
 * @member {String} data
 */


GetSSHCertificateOutput.prototype['data'] = undefined;
/**
 * @member {String} path
 */

GetSSHCertificateOutput.prototype['path'] = undefined;
var _default = GetSSHCertificateOutput;
exports["default"] = _default;
