// akeyless/dist/model/GetSSHCertificate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetSSHCertificate model module.
 * @module model/GetSSHCertificate
 * @version 5.0.3
 */
var GetSSHCertificate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetSSHCertificate</code>.
   * @alias module:model/GetSSHCertificate
   * @param certIssuerName {String} The name of the SSH certificate issuer
   * @param certUsername {String} The username to sign in the SSH certificate
   */
  function GetSSHCertificate(certIssuerName, certUsername) {
    _classCallCheck(this, GetSSHCertificate);

    GetSSHCertificate.initialize(this, certIssuerName, certUsername);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetSSHCertificate, null, [{
    key: "initialize",
    value: function initialize(obj, certIssuerName, certUsername) {
      obj['cert-issuer-name'] = certIssuerName;
      obj['cert-username'] = certUsername || '-';
      obj['json'] = false;
      obj['legacy-signing-alg-name'] = false;
    }
    /**
     * Constructs a <code>GetSSHCertificate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetSSHCertificate} obj Optional instance to populate.
     * @return {module:model/GetSSHCertificate} The populated <code>GetSSHCertificate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetSSHCertificate();

        if (data.hasOwnProperty('cert-issuer-name')) {
          obj['cert-issuer-name'] = _ApiClient["default"].convertToType(data['cert-issuer-name'], 'String');
        }

        if (data.hasOwnProperty('cert-username')) {
          obj['cert-username'] = _ApiClient["default"].convertToType(data['cert-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('legacy-signing-alg-name')) {
          obj['legacy-signing-alg-name'] = _ApiClient["default"].convertToType(data['legacy-signing-alg-name'], 'Boolean');
        }

        if (data.hasOwnProperty('public-key-data')) {
          obj['public-key-data'] = _ApiClient["default"].convertToType(data['public-key-data'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GetSSHCertificate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GetSSHCertificate</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(GetSSHCertificate.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['cert-issuer-name'] && !(typeof data['cert-issuer-name'] === 'string' || data['cert-issuer-name'] instanceof String)) {
        throw new Error("Expected the field `cert-issuer-name` to be a primitive type in the JSON string but got " + data['cert-issuer-name']);
      } // ensure the json data is a string


      if (data['cert-username'] && !(typeof data['cert-username'] === 'string' || data['cert-username'] instanceof String)) {
        throw new Error("Expected the field `cert-username` to be a primitive type in the JSON string but got " + data['cert-username']);
      } // ensure the json data is a string


      if (data['public-key-data'] && !(typeof data['public-key-data'] === 'string' || data['public-key-data'] instanceof String)) {
        throw new Error("Expected the field `public-key-data` to be a primitive type in the JSON string but got " + data['public-key-data']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return GetSSHCertificate;
}();

GetSSHCertificate.RequiredProperties = ["cert-issuer-name", "cert-username"];
/**
 * The name of the SSH certificate issuer
 * @member {String} cert-issuer-name
 */

GetSSHCertificate.prototype['cert-issuer-name'] = undefined;
/**
 * The username to sign in the SSH certificate
 * @member {String} cert-username
 * @default '-'
 */

GetSSHCertificate.prototype['cert-username'] = '-';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GetSSHCertificate.prototype['json'] = false;
/**
 * Set this option to output legacy ('ssh-rsa-cert-v01@openssh.com') signing algorithm name in the certificate.
 * @member {Boolean} legacy-signing-alg-name
 * @default false
 */

GetSSHCertificate.prototype['legacy-signing-alg-name'] = false;
/**
 * SSH public key file contents. If this option is used, the certificate will be printed to stdout
 * @member {String} public-key-data
 */

GetSSHCertificate.prototype['public-key-data'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GetSSHCertificate.prototype['token'] = undefined;
/**
 * Updated certificate lifetime in seconds (must be less than the Certificate Issuer default TTL)
 * @member {Number} ttl
 */

GetSSHCertificate.prototype['ttl'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GetSSHCertificate.prototype['uid-token'] = undefined;
var _default = GetSSHCertificate;
exports["default"] = _default;
