// akeyless/dist/model/GitlabTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GitlabTargetDetails model module.
 * @module model/GitlabTargetDetails
 * @version 4.3.0
 */
var GitlabTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GitlabTargetDetails</code>.
   * GitlabTargetDetails
   * @alias module:model/GitlabTargetDetails
   */
  function GitlabTargetDetails() {
    _classCallCheck(this, GitlabTargetDetails);

    GitlabTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GitlabTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GitlabTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GitlabTargetDetails} obj Optional instance to populate.
     * @return {module:model/GitlabTargetDetails} The populated <code>GitlabTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GitlabTargetDetails();

        if (data.hasOwnProperty('gitlab_access_token')) {
          obj['gitlab_access_token'] = _ApiClient["default"].convertToType(data['gitlab_access_token'], 'String');
        }

        if (data.hasOwnProperty('gitlab_certificate')) {
          obj['gitlab_certificate'] = _ApiClient["default"].convertToType(data['gitlab_certificate'], 'String');
        }

        if (data.hasOwnProperty('gitlab_url')) {
          obj['gitlab_url'] = _ApiClient["default"].convertToType(data['gitlab_url'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GitlabTargetDetails;
}();
/**
 * @member {String} gitlab_access_token
 */


GitlabTargetDetails.prototype['gitlab_access_token'] = undefined;
/**
 * @member {String} gitlab_certificate
 */

GitlabTargetDetails.prototype['gitlab_certificate'] = undefined;
/**
 * @member {String} gitlab_url
 */

GitlabTargetDetails.prototype['gitlab_url'] = undefined;
var _default = GitlabTargetDetails;
exports["default"] = _default;
