// akeyless/dist/model/GKETargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GKETargetDetails model module.
 * @module model/GKETargetDetails
 * @version 5.0.8
 */
var GKETargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GKETargetDetails</code>.
   * GKETargetDetails defines details related to connecting to a GKE (Google Kubernetes Engine) target
   * @alias module:model/GKETargetDetails
   */
  function GKETargetDetails() {
    _classCallCheck(this, GKETargetDetails);

    GKETargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GKETargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GKETargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GKETargetDetails} obj Optional instance to populate.
     * @return {module:model/GKETargetDetails} The populated <code>GKETargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GKETargetDetails();

        if (data.hasOwnProperty('gke_cluster_ca_certificate')) {
          obj['gke_cluster_ca_certificate'] = _ApiClient["default"].convertToType(data['gke_cluster_ca_certificate'], 'String');
        }

        if (data.hasOwnProperty('gke_cluster_endpoint')) {
          obj['gke_cluster_endpoint'] = _ApiClient["default"].convertToType(data['gke_cluster_endpoint'], 'String');
        }

        if (data.hasOwnProperty('gke_cluster_name')) {
          obj['gke_cluster_name'] = _ApiClient["default"].convertToType(data['gke_cluster_name'], 'String');
        }

        if (data.hasOwnProperty('gke_service_account_key')) {
          obj['gke_service_account_key'] = _ApiClient["default"].convertToType(data['gke_service_account_key'], 'String');
        }

        if (data.hasOwnProperty('gke_service_account_name')) {
          obj['gke_service_account_name'] = _ApiClient["default"].convertToType(data['gke_service_account_name'], 'String');
        }

        if (data.hasOwnProperty('use_gw_cloud_identity')) {
          obj['use_gw_cloud_identity'] = _ApiClient["default"].convertToType(data['use_gw_cloud_identity'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GKETargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GKETargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['gke_cluster_ca_certificate'] && !(typeof data['gke_cluster_ca_certificate'] === 'string' || data['gke_cluster_ca_certificate'] instanceof String)) {
        throw new Error("Expected the field `gke_cluster_ca_certificate` to be a primitive type in the JSON string but got " + data['gke_cluster_ca_certificate']);
      } // ensure the json data is a string


      if (data['gke_cluster_endpoint'] && !(typeof data['gke_cluster_endpoint'] === 'string' || data['gke_cluster_endpoint'] instanceof String)) {
        throw new Error("Expected the field `gke_cluster_endpoint` to be a primitive type in the JSON string but got " + data['gke_cluster_endpoint']);
      } // ensure the json data is a string


      if (data['gke_cluster_name'] && !(typeof data['gke_cluster_name'] === 'string' || data['gke_cluster_name'] instanceof String)) {
        throw new Error("Expected the field `gke_cluster_name` to be a primitive type in the JSON string but got " + data['gke_cluster_name']);
      } // ensure the json data is a string


      if (data['gke_service_account_key'] && !(typeof data['gke_service_account_key'] === 'string' || data['gke_service_account_key'] instanceof String)) {
        throw new Error("Expected the field `gke_service_account_key` to be a primitive type in the JSON string but got " + data['gke_service_account_key']);
      } // ensure the json data is a string


      if (data['gke_service_account_name'] && !(typeof data['gke_service_account_name'] === 'string' || data['gke_service_account_name'] instanceof String)) {
        throw new Error("Expected the field `gke_service_account_name` to be a primitive type in the JSON string but got " + data['gke_service_account_name']);
      }

      return true;
    }
  }]);

  return GKETargetDetails;
}();
/**
 * @member {String} gke_cluster_ca_certificate
 */


GKETargetDetails.prototype['gke_cluster_ca_certificate'] = undefined;
/**
 * @member {String} gke_cluster_endpoint
 */

GKETargetDetails.prototype['gke_cluster_endpoint'] = undefined;
/**
 * @member {String} gke_cluster_name
 */

GKETargetDetails.prototype['gke_cluster_name'] = undefined;
/**
 * @member {String} gke_service_account_key
 */

GKETargetDetails.prototype['gke_service_account_key'] = undefined;
/**
 * @member {String} gke_service_account_name
 */

GKETargetDetails.prototype['gke_service_account_name'] = undefined;
/**
 * @member {Boolean} use_gw_cloud_identity
 */

GKETargetDetails.prototype['use_gw_cloud_identity'] = undefined;
var _default = GKETargetDetails;
exports["default"] = _default;
