// akeyless/dist/model/GwUpdateRemoteAccessSessionLogsDatadog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GwUpdateRemoteAccessSessionLogsDatadog model module.
 * @module model/GwUpdateRemoteAccessSessionLogsDatadog
 * @version 5.0.6
 */
var GwUpdateRemoteAccessSessionLogsDatadog = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GwUpdateRemoteAccessSessionLogsDatadog</code>.
   * gwUpdateRemoteAccessSessionLogsDatadog is a command that updates session log forwarding config (datadog target)
   * @alias module:model/GwUpdateRemoteAccessSessionLogsDatadog
   */
  function GwUpdateRemoteAccessSessionLogsDatadog() {
    _classCallCheck(this, GwUpdateRemoteAccessSessionLogsDatadog);

    GwUpdateRemoteAccessSessionLogsDatadog.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GwUpdateRemoteAccessSessionLogsDatadog, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['enable'] = 'true';
      obj['json'] = false;
      obj['log-service'] = 'use-existing';
      obj['log-source'] = 'use-existing';
      obj['log-tags'] = 'use-existing';
      obj['output-format'] = 'text';
      obj['pull-interval'] = '10';
    }
    /**
     * Constructs a <code>GwUpdateRemoteAccessSessionLogsDatadog</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GwUpdateRemoteAccessSessionLogsDatadog} obj Optional instance to populate.
     * @return {module:model/GwUpdateRemoteAccessSessionLogsDatadog} The populated <code>GwUpdateRemoteAccessSessionLogsDatadog</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GwUpdateRemoteAccessSessionLogsDatadog();

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('log-service')) {
          obj['log-service'] = _ApiClient["default"].convertToType(data['log-service'], 'String');
        }

        if (data.hasOwnProperty('log-source')) {
          obj['log-source'] = _ApiClient["default"].convertToType(data['log-source'], 'String');
        }

        if (data.hasOwnProperty('log-tags')) {
          obj['log-tags'] = _ApiClient["default"].convertToType(data['log-tags'], 'String');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GwUpdateRemoteAccessSessionLogsDatadog</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GwUpdateRemoteAccessSessionLogsDatadog</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['api-key'] && !(typeof data['api-key'] === 'string' || data['api-key'] instanceof String)) {
        throw new Error("Expected the field `api-key` to be a primitive type in the JSON string but got " + data['api-key']);
      } // ensure the json data is a string


      if (data['enable'] && !(typeof data['enable'] === 'string' || data['enable'] instanceof String)) {
        throw new Error("Expected the field `enable` to be a primitive type in the JSON string but got " + data['enable']);
      } // ensure the json data is a string


      if (data['host'] && !(typeof data['host'] === 'string' || data['host'] instanceof String)) {
        throw new Error("Expected the field `host` to be a primitive type in the JSON string but got " + data['host']);
      } // ensure the json data is a string


      if (data['log-service'] && !(typeof data['log-service'] === 'string' || data['log-service'] instanceof String)) {
        throw new Error("Expected the field `log-service` to be a primitive type in the JSON string but got " + data['log-service']);
      } // ensure the json data is a string


      if (data['log-source'] && !(typeof data['log-source'] === 'string' || data['log-source'] instanceof String)) {
        throw new Error("Expected the field `log-source` to be a primitive type in the JSON string but got " + data['log-source']);
      } // ensure the json data is a string


      if (data['log-tags'] && !(typeof data['log-tags'] === 'string' || data['log-tags'] instanceof String)) {
        throw new Error("Expected the field `log-tags` to be a primitive type in the JSON string but got " + data['log-tags']);
      } // ensure the json data is a string


      if (data['output-format'] && !(typeof data['output-format'] === 'string' || data['output-format'] instanceof String)) {
        throw new Error("Expected the field `output-format` to be a primitive type in the JSON string but got " + data['output-format']);
      } // ensure the json data is a string


      if (data['pull-interval'] && !(typeof data['pull-interval'] === 'string' || data['pull-interval'] instanceof String)) {
        throw new Error("Expected the field `pull-interval` to be a primitive type in the JSON string but got " + data['pull-interval']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return GwUpdateRemoteAccessSessionLogsDatadog;
}();
/**
 * Datadog api key
 * @member {String} api-key
 */


GwUpdateRemoteAccessSessionLogsDatadog.prototype['api-key'] = undefined;
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['enable'] = 'true';
/**
 * Datadog host
 * @member {String} host
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['json'] = false;
/**
 * Datadog log service
 * @member {String} log-service
 * @default 'use-existing'
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['log-service'] = 'use-existing';
/**
 * Datadog log source
 * @member {String} log-source
 * @default 'use-existing'
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['log-source'] = 'use-existing';
/**
 * A comma-separated list of Datadog log tags formatted as \"key:value\" strings
 * @member {String} log-tags
 * @default 'use-existing'
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['log-tags'] = 'use-existing';
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['output-format'] = 'text';
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['pull-interval'] = '10';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GwUpdateRemoteAccessSessionLogsDatadog.prototype['uid-token'] = undefined;
var _default = GwUpdateRemoteAccessSessionLogsDatadog;
exports["default"] = _default;
