// akeyless/dist/model/GwUpdateRemoteAccessSessionLogsAzureAnalytics.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GwUpdateRemoteAccessSessionLogsAzureAnalytics model module.
 * @module model/GwUpdateRemoteAccessSessionLogsAzureAnalytics
 * @version 4.3.0
 */
var GwUpdateRemoteAccessSessionLogsAzureAnalytics = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GwUpdateRemoteAccessSessionLogsAzureAnalytics</code>.
   * gwUpdateRemoteAccessSessionLogsAzureAnalytics is a command that updates session log forwarding config (azure-analytics target)
   * @alias module:model/GwUpdateRemoteAccessSessionLogsAzureAnalytics
   */
  function GwUpdateRemoteAccessSessionLogsAzureAnalytics() {
    _classCallCheck(this, GwUpdateRemoteAccessSessionLogsAzureAnalytics);

    GwUpdateRemoteAccessSessionLogsAzureAnalytics.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GwUpdateRemoteAccessSessionLogsAzureAnalytics, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GwUpdateRemoteAccessSessionLogsAzureAnalytics</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GwUpdateRemoteAccessSessionLogsAzureAnalytics} obj Optional instance to populate.
     * @return {module:model/GwUpdateRemoteAccessSessionLogsAzureAnalytics} The populated <code>GwUpdateRemoteAccessSessionLogsAzureAnalytics</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GwUpdateRemoteAccessSessionLogsAzureAnalytics();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('workspace-id')) {
          obj['workspace-id'] = _ApiClient["default"].convertToType(data['workspace-id'], 'String');
        }

        if (data.hasOwnProperty('workspace-key')) {
          obj['workspace-key'] = _ApiClient["default"].convertToType(data['workspace-key'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GwUpdateRemoteAccessSessionLogsAzureAnalytics;
}();
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */


GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['enable'] = 'true';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['json'] = false;
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['output-format'] = 'text';
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['pull-interval'] = '10';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['uid-token'] = undefined;
/**
 * Azure workspace id
 * @member {String} workspace-id
 */

GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['workspace-id'] = undefined;
/**
 * Azure workspace key
 * @member {String} workspace-key
 */

GwUpdateRemoteAccessSessionLogsAzureAnalytics.prototype['workspace-key'] = undefined;
var _default = GwUpdateRemoteAccessSessionLogsAzureAnalytics;
exports["default"] = _default;
