// akeyless/dist/model/GwUpdateRemoteAccessSessionLogsLogzIo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GwUpdateRemoteAccessSessionLogsLogzIo model module.
 * @module model/GwUpdateRemoteAccessSessionLogsLogzIo
 * @version 5.0.3
 */
var GwUpdateRemoteAccessSessionLogsLogzIo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GwUpdateRemoteAccessSessionLogsLogzIo</code>.
   * gwUpdateRemoteAccessSessionLogsLogzIo is a command that updates session log forwarding config (logz-io target)
   * @alias module:model/GwUpdateRemoteAccessSessionLogsLogzIo
   */
  function GwUpdateRemoteAccessSessionLogsLogzIo() {
    _classCallCheck(this, GwUpdateRemoteAccessSessionLogsLogzIo);

    GwUpdateRemoteAccessSessionLogsLogzIo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GwUpdateRemoteAccessSessionLogsLogzIo, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['enable'] = 'true';
      obj['json'] = false;
      obj['output-format'] = 'text';
      obj['pull-interval'] = '10';
    }
    /**
     * Constructs a <code>GwUpdateRemoteAccessSessionLogsLogzIo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GwUpdateRemoteAccessSessionLogsLogzIo} obj Optional instance to populate.
     * @return {module:model/GwUpdateRemoteAccessSessionLogsLogzIo} The populated <code>GwUpdateRemoteAccessSessionLogsLogzIo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GwUpdateRemoteAccessSessionLogsLogzIo();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('logz-io-token')) {
          obj['logz-io-token'] = _ApiClient["default"].convertToType(data['logz-io-token'], 'String');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('protocol')) {
          obj['protocol'] = _ApiClient["default"].convertToType(data['protocol'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GwUpdateRemoteAccessSessionLogsLogzIo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GwUpdateRemoteAccessSessionLogsLogzIo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['enable'] && !(typeof data['enable'] === 'string' || data['enable'] instanceof String)) {
        throw new Error("Expected the field `enable` to be a primitive type in the JSON string but got " + data['enable']);
      } // ensure the json data is a string


      if (data['logz-io-token'] && !(typeof data['logz-io-token'] === 'string' || data['logz-io-token'] instanceof String)) {
        throw new Error("Expected the field `logz-io-token` to be a primitive type in the JSON string but got " + data['logz-io-token']);
      } // ensure the json data is a string


      if (data['output-format'] && !(typeof data['output-format'] === 'string' || data['output-format'] instanceof String)) {
        throw new Error("Expected the field `output-format` to be a primitive type in the JSON string but got " + data['output-format']);
      } // ensure the json data is a string


      if (data['protocol'] && !(typeof data['protocol'] === 'string' || data['protocol'] instanceof String)) {
        throw new Error("Expected the field `protocol` to be a primitive type in the JSON string but got " + data['protocol']);
      } // ensure the json data is a string


      if (data['pull-interval'] && !(typeof data['pull-interval'] === 'string' || data['pull-interval'] instanceof String)) {
        throw new Error("Expected the field `pull-interval` to be a primitive type in the JSON string but got " + data['pull-interval']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return GwUpdateRemoteAccessSessionLogsLogzIo;
}();
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */


GwUpdateRemoteAccessSessionLogsLogzIo.prototype['enable'] = 'true';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GwUpdateRemoteAccessSessionLogsLogzIo.prototype['json'] = false;
/**
 * Logz-io token
 * @member {String} logz-io-token
 */

GwUpdateRemoteAccessSessionLogsLogzIo.prototype['logz-io-token'] = undefined;
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GwUpdateRemoteAccessSessionLogsLogzIo.prototype['output-format'] = 'text';
/**
 * LogzIo protocol [tcp/https]
 * @member {String} protocol
 */

GwUpdateRemoteAccessSessionLogsLogzIo.prototype['protocol'] = undefined;
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GwUpdateRemoteAccessSessionLogsLogzIo.prototype['pull-interval'] = '10';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GwUpdateRemoteAccessSessionLogsLogzIo.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GwUpdateRemoteAccessSessionLogsLogzIo.prototype['uid-token'] = undefined;
var _default = GwUpdateRemoteAccessSessionLogsLogzIo;
exports["default"] = _default;
