// akeyless/dist/model/GwUpdateRemoteAccessSessionLogsAwsS3.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GwUpdateRemoteAccessSessionLogsAwsS3 model module.
 * @module model/GwUpdateRemoteAccessSessionLogsAwsS3
 * @version 5.0.14
 */
var GwUpdateRemoteAccessSessionLogsAwsS3 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GwUpdateRemoteAccessSessionLogsAwsS3</code>.
   * gwUpdateRemoteAccessSessionLogsAwsS3 is a command that updates session log forwarding config (aws-s3 target)
   * @alias module:model/GwUpdateRemoteAccessSessionLogsAwsS3
   */
  function GwUpdateRemoteAccessSessionLogsAwsS3() {
    _classCallCheck(this, GwUpdateRemoteAccessSessionLogsAwsS3);

    GwUpdateRemoteAccessSessionLogsAwsS3.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GwUpdateRemoteAccessSessionLogsAwsS3, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['enable'] = 'true';
      obj['json'] = false;
      obj['log-folder'] = 'use-existing';
      obj['output-format'] = 'text';
      obj['pull-interval'] = '10';
    }
    /**
     * Constructs a <code>GwUpdateRemoteAccessSessionLogsAwsS3</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GwUpdateRemoteAccessSessionLogsAwsS3} obj Optional instance to populate.
     * @return {module:model/GwUpdateRemoteAccessSessionLogsAwsS3} The populated <code>GwUpdateRemoteAccessSessionLogsAwsS3</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GwUpdateRemoteAccessSessionLogsAwsS3();

        if (data.hasOwnProperty('access-id')) {
          obj['access-id'] = _ApiClient["default"].convertToType(data['access-id'], 'String');
        }

        if (data.hasOwnProperty('access-key')) {
          obj['access-key'] = _ApiClient["default"].convertToType(data['access-key'], 'String');
        }

        if (data.hasOwnProperty('auth-type')) {
          obj['auth-type'] = _ApiClient["default"].convertToType(data['auth-type'], 'String');
        }

        if (data.hasOwnProperty('bucket-name')) {
          obj['bucket-name'] = _ApiClient["default"].convertToType(data['bucket-name'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('log-folder')) {
          obj['log-folder'] = _ApiClient["default"].convertToType(data['log-folder'], 'String');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('role-arn')) {
          obj['role-arn'] = _ApiClient["default"].convertToType(data['role-arn'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>GwUpdateRemoteAccessSessionLogsAwsS3</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>GwUpdateRemoteAccessSessionLogsAwsS3</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access-id'] && !(typeof data['access-id'] === 'string' || data['access-id'] instanceof String)) {
        throw new Error("Expected the field `access-id` to be a primitive type in the JSON string but got " + data['access-id']);
      } // ensure the json data is a string


      if (data['access-key'] && !(typeof data['access-key'] === 'string' || data['access-key'] instanceof String)) {
        throw new Error("Expected the field `access-key` to be a primitive type in the JSON string but got " + data['access-key']);
      } // ensure the json data is a string


      if (data['auth-type'] && !(typeof data['auth-type'] === 'string' || data['auth-type'] instanceof String)) {
        throw new Error("Expected the field `auth-type` to be a primitive type in the JSON string but got " + data['auth-type']);
      } // ensure the json data is a string


      if (data['bucket-name'] && !(typeof data['bucket-name'] === 'string' || data['bucket-name'] instanceof String)) {
        throw new Error("Expected the field `bucket-name` to be a primitive type in the JSON string but got " + data['bucket-name']);
      } // ensure the json data is a string


      if (data['enable'] && !(typeof data['enable'] === 'string' || data['enable'] instanceof String)) {
        throw new Error("Expected the field `enable` to be a primitive type in the JSON string but got " + data['enable']);
      } // ensure the json data is a string


      if (data['log-folder'] && !(typeof data['log-folder'] === 'string' || data['log-folder'] instanceof String)) {
        throw new Error("Expected the field `log-folder` to be a primitive type in the JSON string but got " + data['log-folder']);
      } // ensure the json data is a string


      if (data['output-format'] && !(typeof data['output-format'] === 'string' || data['output-format'] instanceof String)) {
        throw new Error("Expected the field `output-format` to be a primitive type in the JSON string but got " + data['output-format']);
      } // ensure the json data is a string


      if (data['pull-interval'] && !(typeof data['pull-interval'] === 'string' || data['pull-interval'] instanceof String)) {
        throw new Error("Expected the field `pull-interval` to be a primitive type in the JSON string but got " + data['pull-interval']);
      } // ensure the json data is a string


      if (data['region'] && !(typeof data['region'] === 'string' || data['region'] instanceof String)) {
        throw new Error("Expected the field `region` to be a primitive type in the JSON string but got " + data['region']);
      } // ensure the json data is a string


      if (data['role-arn'] && !(typeof data['role-arn'] === 'string' || data['role-arn'] instanceof String)) {
        throw new Error("Expected the field `role-arn` to be a primitive type in the JSON string but got " + data['role-arn']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return GwUpdateRemoteAccessSessionLogsAwsS3;
}();
/**
 * AWS access id relevant for access_key auth-type
 * @member {String} access-id
 */


GwUpdateRemoteAccessSessionLogsAwsS3.prototype['access-id'] = undefined;
/**
 * AWS access key relevant for access_key auth-type
 * @member {String} access-key
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['access-key'] = undefined;
/**
 * AWS auth type [access_key/cloud_id/assume_role]
 * @member {String} auth-type
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['auth-type'] = undefined;
/**
 * AWS S3 bucket name
 * @member {String} bucket-name
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['bucket-name'] = undefined;
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['enable'] = 'true';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['json'] = false;
/**
 * AWS S3 destination folder for logs
 * @member {String} log-folder
 * @default 'use-existing'
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['log-folder'] = 'use-existing';
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['output-format'] = 'text';
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['pull-interval'] = '10';
/**
 * AWS region
 * @member {String} region
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['region'] = undefined;
/**
 * AWS role arn relevant for assume_role auth-type
 * @member {String} role-arn
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['role-arn'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GwUpdateRemoteAccessSessionLogsAwsS3.prototype['uid-token'] = undefined;
var _default = GwUpdateRemoteAccessSessionLogsAwsS3;
exports["default"] = _default;
