// akeyless/dist/model/GwUpdateRemoteAccessSessionLogsSyslog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GwUpdateRemoteAccessSessionLogsSyslog model module.
 * @module model/GwUpdateRemoteAccessSessionLogsSyslog
 * @version 4.3.0
 */
var GwUpdateRemoteAccessSessionLogsSyslog = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GwUpdateRemoteAccessSessionLogsSyslog</code>.
   * gwUpdateRemoteAccessSessionLogsSyslog is a command that updates session log forwarding config (syslog target)
   * @alias module:model/GwUpdateRemoteAccessSessionLogsSyslog
   */
  function GwUpdateRemoteAccessSessionLogsSyslog() {
    _classCallCheck(this, GwUpdateRemoteAccessSessionLogsSyslog);

    GwUpdateRemoteAccessSessionLogsSyslog.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GwUpdateRemoteAccessSessionLogsSyslog, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GwUpdateRemoteAccessSessionLogsSyslog</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GwUpdateRemoteAccessSessionLogsSyslog} obj Optional instance to populate.
     * @return {module:model/GwUpdateRemoteAccessSessionLogsSyslog} The populated <code>GwUpdateRemoteAccessSessionLogsSyslog</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GwUpdateRemoteAccessSessionLogsSyslog();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('enable-tls')) {
          obj['enable-tls'] = _ApiClient["default"].convertToType(data['enable-tls'], 'Boolean');
        }

        if (data.hasOwnProperty('formatter')) {
          obj['formatter'] = _ApiClient["default"].convertToType(data['formatter'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('network')) {
          obj['network'] = _ApiClient["default"].convertToType(data['network'], 'String');
        }

        if (data.hasOwnProperty('output-format')) {
          obj['output-format'] = _ApiClient["default"].convertToType(data['output-format'], 'String');
        }

        if (data.hasOwnProperty('pull-interval')) {
          obj['pull-interval'] = _ApiClient["default"].convertToType(data['pull-interval'], 'String');
        }

        if (data.hasOwnProperty('target-tag')) {
          obj['target-tag'] = _ApiClient["default"].convertToType(data['target-tag'], 'String');
        }

        if (data.hasOwnProperty('tls-certificate')) {
          obj['tls-certificate'] = _ApiClient["default"].convertToType(data['tls-certificate'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GwUpdateRemoteAccessSessionLogsSyslog;
}();
/**
 * Enable Log Forwarding [true/false]
 * @member {String} enable
 * @default 'true'
 */


GwUpdateRemoteAccessSessionLogsSyslog.prototype['enable'] = 'true';
/**
 * Enable tls relevant only for network type TCP
 * @member {Boolean} enable-tls
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['enable-tls'] = undefined;
/**
 * Syslog formatter [text/cef]
 * @member {String} formatter
 * @default 'text'
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['formatter'] = 'text';
/**
 * Syslog host
 * @member {String} host
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['json'] = false;
/**
 * Syslog network [tcp/udp]
 * @member {String} network
 * @default 'tcp'
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['network'] = 'tcp';
/**
 * Logs format [text/json]
 * @member {String} output-format
 * @default 'text'
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['output-format'] = 'text';
/**
 * Pull interval in seconds
 * @member {String} pull-interval
 * @default '10'
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['pull-interval'] = '10';
/**
 * Syslog target tag
 * @member {String} target-tag
 * @default 'use-existing'
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['target-tag'] = 'use-existing';
/**
 * Syslog tls certificate
 * @member {String} tls-certificate
 * @default 'use-existing'
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['tls-certificate'] = 'use-existing';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GwUpdateRemoteAccessSessionLogsSyslog.prototype['uid-token'] = undefined;
var _default = GwUpdateRemoteAccessSessionLogsSyslog;
exports["default"] = _default;
