// akeyless/dist/model/HashiVaultTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The HashiVaultTargetDetails model module.
 * @module model/HashiVaultTargetDetails
 * @version 5.0.1
 */
var HashiVaultTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>HashiVaultTargetDetails</code>.
   * HashiVaultTargetDetails
   * @alias module:model/HashiVaultTargetDetails
   */
  function HashiVaultTargetDetails() {
    _classCallCheck(this, HashiVaultTargetDetails);

    HashiVaultTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(HashiVaultTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>HashiVaultTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/HashiVaultTargetDetails} obj Optional instance to populate.
     * @return {module:model/HashiVaultTargetDetails} The populated <code>HashiVaultTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new HashiVaultTargetDetails();

        if (data.hasOwnProperty('vault_namespaces')) {
          obj['vault_namespaces'] = _ApiClient["default"].convertToType(data['vault_namespaces'], 'String');
        }

        if (data.hasOwnProperty('vault_token')) {
          obj['vault_token'] = _ApiClient["default"].convertToType(data['vault_token'], 'String');
        }

        if (data.hasOwnProperty('vault_url')) {
          obj['vault_url'] = _ApiClient["default"].convertToType(data['vault_url'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>HashiVaultTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>HashiVaultTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['vault_namespaces'] && !(typeof data['vault_namespaces'] === 'string' || data['vault_namespaces'] instanceof String)) {
        throw new Error("Expected the field `vault_namespaces` to be a primitive type in the JSON string but got " + data['vault_namespaces']);
      } // ensure the json data is a string


      if (data['vault_token'] && !(typeof data['vault_token'] === 'string' || data['vault_token'] instanceof String)) {
        throw new Error("Expected the field `vault_token` to be a primitive type in the JSON string but got " + data['vault_token']);
      } // ensure the json data is a string


      if (data['vault_url'] && !(typeof data['vault_url'] === 'string' || data['vault_url'] instanceof String)) {
        throw new Error("Expected the field `vault_url` to be a primitive type in the JSON string but got " + data['vault_url']);
      }

      return true;
    }
  }]);

  return HashiVaultTargetDetails;
}();
/**
 * @member {String} vault_namespaces
 */


HashiVaultTargetDetails.prototype['vault_namespaces'] = undefined;
/**
 * @member {String} vault_token
 */

HashiVaultTargetDetails.prototype['vault_token'] = undefined;
/**
 * @member {String} vault_url
 */

HashiVaultTargetDetails.prototype['vault_url'] = undefined;
var _default = HashiVaultTargetDetails;
exports["default"] = _default;
