// akeyless/dist/model/ImporterInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ImporterInfo model module.
 * @module model/ImporterInfo
 * @version 5.0.1
 */
var ImporterInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ImporterInfo</code>.
   * @alias module:model/ImporterInfo
   */
  function ImporterInfo() {
    _classCallCheck(this, ImporterInfo);

    ImporterInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ImporterInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ImporterInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ImporterInfo} obj Optional instance to populate.
     * @return {module:model/ImporterInfo} The populated <code>ImporterInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ImporterInfo();

        if (data.hasOwnProperty('external_item_id')) {
          obj['external_item_id'] = _ApiClient["default"].convertToType(data['external_item_id'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ImporterInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ImporterInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['external_item_id'] && !(typeof data['external_item_id'] === 'string' || data['external_item_id'] instanceof String)) {
        throw new Error("Expected the field `external_item_id` to be a primitive type in the JSON string but got " + data['external_item_id']);
      }

      return true;
    }
  }]);

  return ImporterInfo;
}();
/**
 * @member {String} external_item_id
 */


ImporterInfo.prototype['external_item_id'] = undefined;
/**
 * @member {Number} version
 */

ImporterInfo.prototype['version'] = undefined;
var _default = ImporterInfo;
exports["default"] = _default;
