// akeyless/dist/model/ImportPasswords.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ImportPasswords model module.
 * @module model/ImportPasswords
 * @version 5.0.3
 */
var ImportPasswords = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ImportPasswords</code>.
   * importPasswords is a command that import passwords
   * @alias module:model/ImportPasswords
   * @param importPath {String} File path
   */
  function ImportPasswords(importPath) {
    _classCallCheck(this, ImportPasswords);

    ImportPasswords.initialize(this, importPath);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ImportPasswords, null, [{
    key: "initialize",
    value: function initialize(obj, importPath) {
      obj['accessibility'] = 'personal';
      obj['format'] = 'LastPass';
      obj['import-path'] = importPath;
      obj['json'] = false;
      obj['target-folder'] = '/';
    }
    /**
     * Constructs a <code>ImportPasswords</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ImportPasswords} obj Optional instance to populate.
     * @return {module:model/ImportPasswords} The populated <code>ImportPasswords</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ImportPasswords();

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'String');
        }

        if (data.hasOwnProperty('format')) {
          obj['format'] = _ApiClient["default"].convertToType(data['format'], 'String');
        }

        if (data.hasOwnProperty('import-path')) {
          obj['import-path'] = _ApiClient["default"].convertToType(data['import-path'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('target-folder')) {
          obj['target-folder'] = _ApiClient["default"].convertToType(data['target-folder'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-mode')) {
          obj['update-mode'] = _ApiClient["default"].convertToType(data['update-mode'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ImportPasswords</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ImportPasswords</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ImportPasswords.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['accessibility'] && !(typeof data['accessibility'] === 'string' || data['accessibility'] instanceof String)) {
        throw new Error("Expected the field `accessibility` to be a primitive type in the JSON string but got " + data['accessibility']);
      } // ensure the json data is a string


      if (data['format'] && !(typeof data['format'] === 'string' || data['format'] instanceof String)) {
        throw new Error("Expected the field `format` to be a primitive type in the JSON string but got " + data['format']);
      } // ensure the json data is a string


      if (data['import-path'] && !(typeof data['import-path'] === 'string' || data['import-path'] instanceof String)) {
        throw new Error("Expected the field `import-path` to be a primitive type in the JSON string but got " + data['import-path']);
      } // ensure the json data is a string


      if (data['protection_key'] && !(typeof data['protection_key'] === 'string' || data['protection_key'] instanceof String)) {
        throw new Error("Expected the field `protection_key` to be a primitive type in the JSON string but got " + data['protection_key']);
      } // ensure the json data is a string


      if (data['target-folder'] && !(typeof data['target-folder'] === 'string' || data['target-folder'] instanceof String)) {
        throw new Error("Expected the field `target-folder` to be a primitive type in the JSON string but got " + data['target-folder']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['update-mode'] && !(typeof data['update-mode'] === 'string' || data['update-mode'] instanceof String)) {
        throw new Error("Expected the field `update-mode` to be a primitive type in the JSON string but got " + data['update-mode']);
      }

      return true;
    }
  }]);

  return ImportPasswords;
}();

ImportPasswords.RequiredProperties = ["import-path"];
/**
 * for personal password manager
 * @member {String} accessibility
 * @default 'personal'
 */

ImportPasswords.prototype['accessibility'] = 'personal';
/**
 * Password format type [LastPass/Chrome/Firefox/1password/keeper/bitwarden/dashlane]
 * @member {String} format
 * @default 'LastPass'
 */

ImportPasswords.prototype['format'] = 'LastPass';
/**
 * File path
 * @member {String} import-path
 */

ImportPasswords.prototype['import-path'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

ImportPasswords.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

ImportPasswords.prototype['protection_key'] = undefined;
/**
 * Target folder for imported passwords
 * @member {String} target-folder
 * @default '/'
 */

ImportPasswords.prototype['target-folder'] = '/';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

ImportPasswords.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

ImportPasswords.prototype['uid-token'] = undefined;
/**
 * @member {String} update-mode
 */

ImportPasswords.prototype['update-mode'] = undefined;
var _default = ImportPasswords;
exports["default"] = _default;
