// akeyless/dist/model/ItemSraStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemSraStatus model module.
 * @module model/ItemSraStatus
 * @version 4.3.0
 */
var ItemSraStatus = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemSraStatus</code>.
   * @alias module:model/ItemSraStatus
   */
  function ItemSraStatus() {
    _classCallCheck(this, ItemSraStatus);

    ItemSraStatus.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemSraStatus, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemSraStatus</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemSraStatus} obj Optional instance to populate.
     * @return {module:model/ItemSraStatus} The populated <code>ItemSraStatus</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemSraStatus();

        if (data.hasOwnProperty('count_by_host_info')) {
          obj['count_by_host_info'] = _ApiClient["default"].convertToType(data['count_by_host_info'], {
            'String': 'Number'
          });
        }

        if (data.hasOwnProperty('count_info')) {
          obj['count_info'] = _ApiClient["default"].convertToType(data['count_info'], {
            'String': {
              'String': 'Number'
            }
          });
        }

        if (data.hasOwnProperty('hosts_in_use')) {
          obj['hosts_in_use'] = _ApiClient["default"].convertToType(data['hosts_in_use'], ['String']);
        }

        if (data.hasOwnProperty('is_in_use')) {
          obj['is_in_use'] = _ApiClient["default"].convertToType(data['is_in_use'], 'Boolean');
        }

        if (data.hasOwnProperty('last_used_item')) {
          obj['last_used_item'] = _ApiClient["default"].convertToType(data['last_used_item'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return ItemSraStatus;
}();
/**
 * @member {Object.<String, Number>} count_by_host_info
 */


ItemSraStatus.prototype['count_by_host_info'] = undefined;
/**
 * @member {Object.<String, Object.<String, Number>>} count_info
 */

ItemSraStatus.prototype['count_info'] = undefined;
/**
 * @member {Array.<String>} hosts_in_use
 */

ItemSraStatus.prototype['hosts_in_use'] = undefined;
/**
 * @member {Boolean} is_in_use
 */

ItemSraStatus.prototype['is_in_use'] = undefined;
/**
 * @member {Date} last_used_item
 */

ItemSraStatus.prototype['last_used_item'] = undefined;
var _default = ItemSraStatus;
exports["default"] = _default;
