// akeyless/dist/model/ItemUSCSyncAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RemoteSecretInfo = _interopRequireDefault(require("./RemoteSecretInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemUSCSyncAssociation model module.
 * @module model/ItemUSCSyncAssociation
 * @version 5.0.1
 */
var ItemUSCSyncAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemUSCSyncAssociation</code>.
   * ItemUSCSyncAssociation includes details of usc sync associations
   * @alias module:model/ItemUSCSyncAssociation
   */
  function ItemUSCSyncAssociation() {
    _classCallCheck(this, ItemUSCSyncAssociation);

    ItemUSCSyncAssociation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemUSCSyncAssociation, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemUSCSyncAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemUSCSyncAssociation} obj Optional instance to populate.
     * @return {module:model/ItemUSCSyncAssociation} The populated <code>ItemUSCSyncAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemUSCSyncAssociation();

        if (data.hasOwnProperty('assoc_id')) {
          obj['assoc_id'] = _ApiClient["default"].convertToType(data['assoc_id'], 'String');
        }

        if (data.hasOwnProperty('attributes')) {
          obj['attributes'] = _RemoteSecretInfo["default"].constructFromObject(data['attributes']);
        }

        if (data.hasOwnProperty('item_id')) {
          obj['item_id'] = _ApiClient["default"].convertToType(data['item_id'], 'Number');
        }

        if (data.hasOwnProperty('item_name')) {
          obj['item_name'] = _ApiClient["default"].convertToType(data['item_name'], 'String');
        }

        if (data.hasOwnProperty('item_type')) {
          obj['item_type'] = _ApiClient["default"].convertToType(data['item_type'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ItemUSCSyncAssociation</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ItemUSCSyncAssociation</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['assoc_id'] && !(typeof data['assoc_id'] === 'string' || data['assoc_id'] instanceof String)) {
        throw new Error("Expected the field `assoc_id` to be a primitive type in the JSON string but got " + data['assoc_id']);
      } // validate the optional field `attributes`


      if (data['attributes']) {
        // data not null
        _RemoteSecretInfo["default"].validateJSON(data['attributes']);
      } // ensure the json data is a string


      if (data['item_name'] && !(typeof data['item_name'] === 'string' || data['item_name'] instanceof String)) {
        throw new Error("Expected the field `item_name` to be a primitive type in the JSON string but got " + data['item_name']);
      } // ensure the json data is a string


      if (data['item_type'] && !(typeof data['item_type'] === 'string' || data['item_type'] instanceof String)) {
        throw new Error("Expected the field `item_type` to be a primitive type in the JSON string but got " + data['item_type']);
      }

      return true;
    }
  }]);

  return ItemUSCSyncAssociation;
}();
/**
 * @member {String} assoc_id
 */


ItemUSCSyncAssociation.prototype['assoc_id'] = undefined;
/**
 * @member {module:model/RemoteSecretInfo} attributes
 */

ItemUSCSyncAssociation.prototype['attributes'] = undefined;
/**
 * @member {Number} item_id
 */

ItemUSCSyncAssociation.prototype['item_id'] = undefined;
/**
 * @member {String} item_name
 */

ItemUSCSyncAssociation.prototype['item_name'] = undefined;
/**
 * @member {String} item_type
 */

ItemUSCSyncAssociation.prototype['item_type'] = undefined;
var _default = ItemUSCSyncAssociation;
exports["default"] = _default;
