// akeyless/dist/model/ItemSraStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemSraStatus model module.
 * @module model/ItemSraStatus
 * @version 3.6.3
 */
var ItemSraStatus = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemSraStatus</code>.
   * @alias module:model/ItemSraStatus
   */
  function ItemSraStatus() {
    _classCallCheck(this, ItemSraStatus);

    ItemSraStatus.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemSraStatus, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemSraStatus</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemSraStatus} obj Optional instance to populate.
     * @return {module:model/ItemSraStatus} The populated <code>ItemSraStatus</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemSraStatus();

        if (data.hasOwnProperty('bastion_access_id')) {
          obj['bastion_access_id'] = _ApiClient["default"].convertToType(data['bastion_access_id'], 'String');
        }

        if (data.hasOwnProperty('bastion_cluster_name')) {
          obj['bastion_cluster_name'] = _ApiClient["default"].convertToType(data['bastion_cluster_name'], 'String');
        }

        if (data.hasOwnProperty('bastion_instance_id')) {
          obj['bastion_instance_id'] = _ApiClient["default"].convertToType(data['bastion_instance_id'], 'String');
        }

        if (data.hasOwnProperty('is_in_used')) {
          obj['is_in_used'] = _ApiClient["default"].convertToType(data['is_in_used'], 'Boolean');
        }

        if (data.hasOwnProperty('session_id')) {
          obj['session_id'] = _ApiClient["default"].convertToType(data['session_id'], 'String');
        }

        if (data.hasOwnProperty('time')) {
          obj['time'] = _ApiClient["default"].convertToType(data['time'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return ItemSraStatus;
}();
/**
 * @member {String} bastion_access_id
 */


ItemSraStatus.prototype['bastion_access_id'] = undefined;
/**
 * @member {String} bastion_cluster_name
 */

ItemSraStatus.prototype['bastion_cluster_name'] = undefined;
/**
 * @member {String} bastion_instance_id
 */

ItemSraStatus.prototype['bastion_instance_id'] = undefined;
/**
 * @member {Boolean} is_in_used
 */

ItemSraStatus.prototype['is_in_used'] = undefined;
/**
 * @member {String} session_id
 */

ItemSraStatus.prototype['session_id'] = undefined;
/**
 * @member {Date} time
 */

ItemSraStatus.prototype['time'] = undefined;
var _default = ItemSraStatus;
exports["default"] = _default;
