// akeyless/dist/model/ItemGeneralInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateChainInfo = _interopRequireDefault(require("./CertificateChainInfo"));

var _CertificateExpirationEvent = _interopRequireDefault(require("./CertificateExpirationEvent"));

var _CertificateIssueInfo = _interopRequireDefault(require("./CertificateIssueInfo"));

var _CertificateTemplateInfo = _interopRequireDefault(require("./CertificateTemplateInfo"));

var _ClassicKeyDetailsInfo = _interopRequireDefault(require("./ClassicKeyDetailsInfo"));

var _DynamicSecretProducerInfo = _interopRequireDefault(require("./DynamicSecretProducerInfo"));

var _ImporterInfo = _interopRequireDefault(require("./ImporterInfo"));

var _NextAutoRotationEvent = _interopRequireDefault(require("./NextAutoRotationEvent"));

var _OidcClientInfo = _interopRequireDefault(require("./OidcClientInfo"));

var _PasswordPolicyInfo = _interopRequireDefault(require("./PasswordPolicyInfo"));

var _RotatedSecretDetailsInfo = _interopRequireDefault(require("./RotatedSecretDetailsInfo"));

var _SecureRemoteAccess = _interopRequireDefault(require("./SecureRemoteAccess"));

var _StaticSecretDetailsInfo = _interopRequireDefault(require("./StaticSecretDetailsInfo"));

var _TokenizerInfo = _interopRequireDefault(require("./TokenizerInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemGeneralInfo model module.
 * @module model/ItemGeneralInfo
 * @version 5.0.0
 */
var ItemGeneralInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemGeneralInfo</code>.
   * @alias module:model/ItemGeneralInfo
   */
  function ItemGeneralInfo() {
    _classCallCheck(this, ItemGeneralInfo);

    ItemGeneralInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemGeneralInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemGeneralInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemGeneralInfo} obj Optional instance to populate.
     * @return {module:model/ItemGeneralInfo} The populated <code>ItemGeneralInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemGeneralInfo();

        if (data.hasOwnProperty('cert_issue_details')) {
          obj['cert_issue_details'] = _CertificateIssueInfo["default"].constructFromObject(data['cert_issue_details']);
        }

        if (data.hasOwnProperty('certificate_chain_info')) {
          obj['certificate_chain_info'] = _CertificateChainInfo["default"].constructFromObject(data['certificate_chain_info']);
        }

        if (data.hasOwnProperty('certificate_format')) {
          obj['certificate_format'] = _ApiClient["default"].convertToType(data['certificate_format'], 'String');
        }

        if (data.hasOwnProperty('certificates_template_info')) {
          obj['certificates_template_info'] = _CertificateTemplateInfo["default"].constructFromObject(data['certificates_template_info']);
        }

        if (data.hasOwnProperty('classic_key_details')) {
          obj['classic_key_details'] = _ClassicKeyDetailsInfo["default"].constructFromObject(data['classic_key_details']);
        }

        if (data.hasOwnProperty('cluster_gw_url')) {
          obj['cluster_gw_url'] = _ApiClient["default"].convertToType(data['cluster_gw_url'], 'String');
        }

        if (data.hasOwnProperty('display_metadata')) {
          obj['display_metadata'] = _ApiClient["default"].convertToType(data['display_metadata'], 'String');
        }

        if (data.hasOwnProperty('dynamic_secret_producer_details')) {
          obj['dynamic_secret_producer_details'] = _DynamicSecretProducerInfo["default"].constructFromObject(data['dynamic_secret_producer_details']);
        }

        if (data.hasOwnProperty('expiration_events')) {
          obj['expiration_events'] = _ApiClient["default"].convertToType(data['expiration_events'], [_CertificateExpirationEvent["default"]]);
        }

        if (data.hasOwnProperty('importer_info')) {
          obj['importer_info'] = _ImporterInfo["default"].constructFromObject(data['importer_info']);
        }

        if (data.hasOwnProperty('next_rotation_events')) {
          obj['next_rotation_events'] = _ApiClient["default"].convertToType(data['next_rotation_events'], [_NextAutoRotationEvent["default"]]);
        }

        if (data.hasOwnProperty('oidc_client_info')) {
          obj['oidc_client_info'] = _OidcClientInfo["default"].constructFromObject(data['oidc_client_info']);
        }

        if (data.hasOwnProperty('password_policy')) {
          obj['password_policy'] = _PasswordPolicyInfo["default"].constructFromObject(data['password_policy']);
        }

        if (data.hasOwnProperty('rotated_secret_details')) {
          obj['rotated_secret_details'] = _RotatedSecretDetailsInfo["default"].constructFromObject(data['rotated_secret_details']);
        }

        if (data.hasOwnProperty('secure_remote_access_details')) {
          obj['secure_remote_access_details'] = _SecureRemoteAccess["default"].constructFromObject(data['secure_remote_access_details']);
        }

        if (data.hasOwnProperty('static_secret_info')) {
          obj['static_secret_info'] = _StaticSecretDetailsInfo["default"].constructFromObject(data['static_secret_info']);
        }

        if (data.hasOwnProperty('tokenizer_info')) {
          obj['tokenizer_info'] = _TokenizerInfo["default"].constructFromObject(data['tokenizer_info']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ItemGeneralInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ItemGeneralInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `cert_issue_details`
      if (data['cert_issue_details']) {
        // data not null
        _CertificateIssueInfo["default"].validateJSON(data['cert_issue_details']);
      } // validate the optional field `certificate_chain_info`


      if (data['certificate_chain_info']) {
        // data not null
        _CertificateChainInfo["default"].validateJSON(data['certificate_chain_info']);
      } // ensure the json data is a string


      if (data['certificate_format'] && !(typeof data['certificate_format'] === 'string' || data['certificate_format'] instanceof String)) {
        throw new Error("Expected the field `certificate_format` to be a primitive type in the JSON string but got " + data['certificate_format']);
      } // validate the optional field `certificates_template_info`


      if (data['certificates_template_info']) {
        // data not null
        _CertificateTemplateInfo["default"].validateJSON(data['certificates_template_info']);
      } // validate the optional field `classic_key_details`


      if (data['classic_key_details']) {
        // data not null
        _ClassicKeyDetailsInfo["default"].validateJSON(data['classic_key_details']);
      } // ensure the json data is a string


      if (data['cluster_gw_url'] && !(typeof data['cluster_gw_url'] === 'string' || data['cluster_gw_url'] instanceof String)) {
        throw new Error("Expected the field `cluster_gw_url` to be a primitive type in the JSON string but got " + data['cluster_gw_url']);
      } // ensure the json data is a string


      if (data['display_metadata'] && !(typeof data['display_metadata'] === 'string' || data['display_metadata'] instanceof String)) {
        throw new Error("Expected the field `display_metadata` to be a primitive type in the JSON string but got " + data['display_metadata']);
      } // validate the optional field `dynamic_secret_producer_details`


      if (data['dynamic_secret_producer_details']) {
        // data not null
        _DynamicSecretProducerInfo["default"].validateJSON(data['dynamic_secret_producer_details']);
      }

      if (data['expiration_events']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['expiration_events'])) {
          throw new Error("Expected the field `expiration_events` to be an array in the JSON data but got " + data['expiration_events']);
        } // validate the optional field `expiration_events` (array)


        var _iterator = _createForOfIteratorHelper(data['expiration_events']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _CertificateExpirationEvent["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // validate the optional field `importer_info`


      if (data['importer_info']) {
        // data not null
        _ImporterInfo["default"].validateJSON(data['importer_info']);
      }

      if (data['next_rotation_events']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['next_rotation_events'])) {
          throw new Error("Expected the field `next_rotation_events` to be an array in the JSON data but got " + data['next_rotation_events']);
        } // validate the optional field `next_rotation_events` (array)


        var _iterator2 = _createForOfIteratorHelper(data['next_rotation_events']),
            _step2;

        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;

            _NextAutoRotationEvent["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }

        ;
      } // validate the optional field `oidc_client_info`


      if (data['oidc_client_info']) {
        // data not null
        _OidcClientInfo["default"].validateJSON(data['oidc_client_info']);
      } // validate the optional field `password_policy`


      if (data['password_policy']) {
        // data not null
        _PasswordPolicyInfo["default"].validateJSON(data['password_policy']);
      } // validate the optional field `rotated_secret_details`


      if (data['rotated_secret_details']) {
        // data not null
        _RotatedSecretDetailsInfo["default"].validateJSON(data['rotated_secret_details']);
      } // validate the optional field `secure_remote_access_details`


      if (data['secure_remote_access_details']) {
        // data not null
        _SecureRemoteAccess["default"].validateJSON(data['secure_remote_access_details']);
      } // validate the optional field `static_secret_info`


      if (data['static_secret_info']) {
        // data not null
        _StaticSecretDetailsInfo["default"].validateJSON(data['static_secret_info']);
      } // validate the optional field `tokenizer_info`


      if (data['tokenizer_info']) {
        // data not null
        _TokenizerInfo["default"].validateJSON(data['tokenizer_info']);
      }

      return true;
    }
  }]);

  return ItemGeneralInfo;
}();
/**
 * @member {module:model/CertificateIssueInfo} cert_issue_details
 */


ItemGeneralInfo.prototype['cert_issue_details'] = undefined;
/**
 * @member {module:model/CertificateChainInfo} certificate_chain_info
 */

ItemGeneralInfo.prototype['certificate_chain_info'] = undefined;
/**
 * @member {String} certificate_format
 */

ItemGeneralInfo.prototype['certificate_format'] = undefined;
/**
 * @member {module:model/CertificateTemplateInfo} certificates_template_info
 */

ItemGeneralInfo.prototype['certificates_template_info'] = undefined;
/**
 * @member {module:model/ClassicKeyDetailsInfo} classic_key_details
 */

ItemGeneralInfo.prototype['classic_key_details'] = undefined;
/**
 * @member {String} cluster_gw_url
 */

ItemGeneralInfo.prototype['cluster_gw_url'] = undefined;
/**
 * @member {String} display_metadata
 */

ItemGeneralInfo.prototype['display_metadata'] = undefined;
/**
 * @member {module:model/DynamicSecretProducerInfo} dynamic_secret_producer_details
 */

ItemGeneralInfo.prototype['dynamic_secret_producer_details'] = undefined;
/**
 * @member {Array.<module:model/CertificateExpirationEvent>} expiration_events
 */

ItemGeneralInfo.prototype['expiration_events'] = undefined;
/**
 * @member {module:model/ImporterInfo} importer_info
 */

ItemGeneralInfo.prototype['importer_info'] = undefined;
/**
 * @member {Array.<module:model/NextAutoRotationEvent>} next_rotation_events
 */

ItemGeneralInfo.prototype['next_rotation_events'] = undefined;
/**
 * @member {module:model/OidcClientInfo} oidc_client_info
 */

ItemGeneralInfo.prototype['oidc_client_info'] = undefined;
/**
 * @member {module:model/PasswordPolicyInfo} password_policy
 */

ItemGeneralInfo.prototype['password_policy'] = undefined;
/**
 * @member {module:model/RotatedSecretDetailsInfo} rotated_secret_details
 */

ItemGeneralInfo.prototype['rotated_secret_details'] = undefined;
/**
 * @member {module:model/SecureRemoteAccess} secure_remote_access_details
 */

ItemGeneralInfo.prototype['secure_remote_access_details'] = undefined;
/**
 * @member {module:model/StaticSecretDetailsInfo} static_secret_info
 */

ItemGeneralInfo.prototype['static_secret_info'] = undefined;
/**
 * @member {module:model/TokenizerInfo} tokenizer_info
 */

ItemGeneralInfo.prototype['tokenizer_info'] = undefined;
var _default = ItemGeneralInfo;
exports["default"] = _default;
