// akeyless/dist/model/ItemVersion.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateVersionInfo = _interopRequireDefault(require("./CertificateVersionInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemVersion model module.
 * @module model/ItemVersion
 * @version 5.0.3
 */
var ItemVersion = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemVersion</code>.
   * @alias module:model/ItemVersion
   */
  function ItemVersion() {
    _classCallCheck(this, ItemVersion);

    ItemVersion.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemVersion, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemVersion</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemVersion} obj Optional instance to populate.
     * @return {module:model/ItemVersion} The populated <code>ItemVersion</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemVersion();

        if (data.hasOwnProperty('access_date')) {
          obj['access_date'] = _ApiClient["default"].convertToType(data['access_date'], 'Date');
        }

        if (data.hasOwnProperty('access_date_display')) {
          obj['access_date_display'] = _ApiClient["default"].convertToType(data['access_date_display'], 'String');
        }

        if (data.hasOwnProperty('certificate_version_info')) {
          obj['certificate_version_info'] = _CertificateVersionInfo["default"].constructFromObject(data['certificate_version_info']);
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('customer_fragment_id')) {
          obj['customer_fragment_id'] = _ApiClient["default"].convertToType(data['customer_fragment_id'], 'String');
        }

        if (data.hasOwnProperty('deletion_date')) {
          obj['deletion_date'] = _ApiClient["default"].convertToType(data['deletion_date'], 'Date');
        }

        if (data.hasOwnProperty('item_version_state')) {
          obj['item_version_state'] = _ApiClient["default"].convertToType(data['item_version_state'], 'String');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('protection_key_name')) {
          obj['protection_key_name'] = _ApiClient["default"].convertToType(data['protection_key_name'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('with_customer_fragment')) {
          obj['with_customer_fragment'] = _ApiClient["default"].convertToType(data['with_customer_fragment'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ItemVersion</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ItemVersion</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_date_display'] && !(typeof data['access_date_display'] === 'string' || data['access_date_display'] instanceof String)) {
        throw new Error("Expected the field `access_date_display` to be a primitive type in the JSON string but got " + data['access_date_display']);
      } // validate the optional field `certificate_version_info`


      if (data['certificate_version_info']) {
        // data not null
        _CertificateVersionInfo["default"].validateJSON(data['certificate_version_info']);
      } // ensure the json data is a string


      if (data['customer_fragment_id'] && !(typeof data['customer_fragment_id'] === 'string' || data['customer_fragment_id'] instanceof String)) {
        throw new Error("Expected the field `customer_fragment_id` to be a primitive type in the JSON string but got " + data['customer_fragment_id']);
      } // ensure the json data is a string


      if (data['item_version_state'] && !(typeof data['item_version_state'] === 'string' || data['item_version_state'] instanceof String)) {
        throw new Error("Expected the field `item_version_state` to be a primitive type in the JSON string but got " + data['item_version_state']);
      } // ensure the json data is a string


      if (data['protection_key_name'] && !(typeof data['protection_key_name'] === 'string' || data['protection_key_name'] instanceof String)) {
        throw new Error("Expected the field `protection_key_name` to be a primitive type in the JSON string but got " + data['protection_key_name']);
      }

      return true;
    }
  }]);

  return ItemVersion;
}();
/**
 * @member {Date} access_date
 */


ItemVersion.prototype['access_date'] = undefined;
/**
 * @member {String} access_date_display
 */

ItemVersion.prototype['access_date_display'] = undefined;
/**
 * @member {module:model/CertificateVersionInfo} certificate_version_info
 */

ItemVersion.prototype['certificate_version_info'] = undefined;
/**
 * @member {Date} creation_date
 */

ItemVersion.prototype['creation_date'] = undefined;
/**
 * @member {String} customer_fragment_id
 */

ItemVersion.prototype['customer_fragment_id'] = undefined;
/**
 * @member {Date} deletion_date
 */

ItemVersion.prototype['deletion_date'] = undefined;
/**
 * ItemState defines the different states an Item can be in
 * @member {String} item_version_state
 */

ItemVersion.prototype['item_version_state'] = undefined;
/**
 * @member {Date} modification_date
 */

ItemVersion.prototype['modification_date'] = undefined;
/**
 * @member {String} protection_key_name
 */

ItemVersion.prototype['protection_key_name'] = undefined;
/**
 * @member {Number} version
 */

ItemVersion.prototype['version'] = undefined;
/**
 * @member {Boolean} with_customer_fragment
 */

ItemVersion.prototype['with_customer_fragment'] = undefined;
var _default = ItemVersion;
exports["default"] = _default;
