// akeyless/dist/model/KerberosAuthMethodInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KerberosAuthMethodInfo model module.
 * @module model/KerberosAuthMethodInfo
 * @version 4.3.0
 */
var KerberosAuthMethodInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KerberosAuthMethodInfo</code>.
   * @alias module:model/KerberosAuthMethodInfo
   */
  function KerberosAuthMethodInfo() {
    _classCallCheck(this, KerberosAuthMethodInfo);

    KerberosAuthMethodInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KerberosAuthMethodInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KerberosAuthMethodInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KerberosAuthMethodInfo} obj Optional instance to populate.
     * @return {module:model/KerberosAuthMethodInfo} The populated <code>KerberosAuthMethodInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KerberosAuthMethodInfo();

        if (data.hasOwnProperty('kerberos_keytab')) {
          obj['kerberos_keytab'] = _ApiClient["default"].convertToType(data['kerberos_keytab'], 'String');
        }

        if (data.hasOwnProperty('kerberos_krb5_conf')) {
          obj['kerberos_krb5_conf'] = _ApiClient["default"].convertToType(data['kerberos_krb5_conf'], 'String');
        }

        if (data.hasOwnProperty('ldap_anonymous_search')) {
          obj['ldap_anonymous_search'] = _ApiClient["default"].convertToType(data['ldap_anonymous_search'], 'Boolean');
        }

        if (data.hasOwnProperty('ldap_bind_dn')) {
          obj['ldap_bind_dn'] = _ApiClient["default"].convertToType(data['ldap_bind_dn'], 'String');
        }

        if (data.hasOwnProperty('ldap_bind_password')) {
          obj['ldap_bind_password'] = _ApiClient["default"].convertToType(data['ldap_bind_password'], 'String');
        }

        if (data.hasOwnProperty('ldap_certificate')) {
          obj['ldap_certificate'] = _ApiClient["default"].convertToType(data['ldap_certificate'], 'String');
        }

        if (data.hasOwnProperty('ldap_group_attr')) {
          obj['ldap_group_attr'] = _ApiClient["default"].convertToType(data['ldap_group_attr'], 'String');
        }

        if (data.hasOwnProperty('ldap_group_dn')) {
          obj['ldap_group_dn'] = _ApiClient["default"].convertToType(data['ldap_group_dn'], 'String');
        }

        if (data.hasOwnProperty('ldap_group_filter')) {
          obj['ldap_group_filter'] = _ApiClient["default"].convertToType(data['ldap_group_filter'], 'String');
        }

        if (data.hasOwnProperty('ldap_url_address')) {
          obj['ldap_url_address'] = _ApiClient["default"].convertToType(data['ldap_url_address'], 'String');
        }

        if (data.hasOwnProperty('ldap_user_attr')) {
          obj['ldap_user_attr'] = _ApiClient["default"].convertToType(data['ldap_user_attr'], 'String');
        }

        if (data.hasOwnProperty('ldap_user_dn')) {
          obj['ldap_user_dn'] = _ApiClient["default"].convertToType(data['ldap_user_dn'], 'String');
        }
      }

      return obj;
    }
  }]);

  return KerberosAuthMethodInfo;
}();
/**
 * @member {String} kerberos_keytab
 */


KerberosAuthMethodInfo.prototype['kerberos_keytab'] = undefined;
/**
 * @member {String} kerberos_krb5_conf
 */

KerberosAuthMethodInfo.prototype['kerberos_krb5_conf'] = undefined;
/**
 * @member {Boolean} ldap_anonymous_search
 */

KerberosAuthMethodInfo.prototype['ldap_anonymous_search'] = undefined;
/**
 * @member {String} ldap_bind_dn
 */

KerberosAuthMethodInfo.prototype['ldap_bind_dn'] = undefined;
/**
 * @member {String} ldap_bind_password
 */

KerberosAuthMethodInfo.prototype['ldap_bind_password'] = undefined;
/**
 * @member {String} ldap_certificate
 */

KerberosAuthMethodInfo.prototype['ldap_certificate'] = undefined;
/**
 * @member {String} ldap_group_attr
 */

KerberosAuthMethodInfo.prototype['ldap_group_attr'] = undefined;
/**
 * @member {String} ldap_group_dn
 */

KerberosAuthMethodInfo.prototype['ldap_group_dn'] = undefined;
/**
 * @member {String} ldap_group_filter
 */

KerberosAuthMethodInfo.prototype['ldap_group_filter'] = undefined;
/**
 * @member {String} ldap_url_address
 */

KerberosAuthMethodInfo.prototype['ldap_url_address'] = undefined;
/**
 * @member {String} ldap_user_attr
 */

KerberosAuthMethodInfo.prototype['ldap_user_attr'] = undefined;
/**
 * @member {String} ldap_user_dn
 */

KerberosAuthMethodInfo.prototype['ldap_user_dn'] = undefined;
var _default = KerberosAuthMethodInfo;
exports["default"] = _default;
