// akeyless/dist/model/KmipMoveServerOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KmipMoveServerOutput model module.
 * @module model/KmipMoveServerOutput
 * @version 5.0.8
 */
var KmipMoveServerOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KmipMoveServerOutput</code>.
   * @alias module:model/KmipMoveServerOutput
   */
  function KmipMoveServerOutput() {
    _classCallCheck(this, KmipMoveServerOutput);

    KmipMoveServerOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KmipMoveServerOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KmipMoveServerOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KmipMoveServerOutput} obj Optional instance to populate.
     * @return {module:model/KmipMoveServerOutput} The populated <code>KmipMoveServerOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KmipMoveServerOutput();

        if (data.hasOwnProperty('new_root')) {
          obj['new_root'] = _ApiClient["default"].convertToType(data['new_root'], 'String');
        }

        if (data.hasOwnProperty('old_root')) {
          obj['old_root'] = _ApiClient["default"].convertToType(data['old_root'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>KmipMoveServerOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KmipMoveServerOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['new_root'] && !(typeof data['new_root'] === 'string' || data['new_root'] instanceof String)) {
        throw new Error("Expected the field `new_root` to be a primitive type in the JSON string but got " + data['new_root']);
      } // ensure the json data is a string


      if (data['old_root'] && !(typeof data['old_root'] === 'string' || data['old_root'] instanceof String)) {
        throw new Error("Expected the field `old_root` to be a primitive type in the JSON string but got " + data['old_root']);
      }

      return true;
    }
  }]);

  return KmipMoveServerOutput;
}();
/**
 * @member {String} new_root
 */


KmipMoveServerOutput.prototype['new_root'] = undefined;
/**
 * @member {String} old_root
 */

KmipMoveServerOutput.prototype['old_root'] = undefined;
var _default = KmipMoveServerOutput;
exports["default"] = _default;
