// akeyless/dist/model/KubeconfigNamedContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _KubeconfigContext = _interopRequireDefault(require("./KubeconfigContext"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KubeconfigNamedContext model module.
 * @module model/KubeconfigNamedContext
 * @version 5.0.1
 */
var KubeconfigNamedContext = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KubeconfigNamedContext</code>.
   * @alias module:model/KubeconfigNamedContext
   */
  function KubeconfigNamedContext() {
    _classCallCheck(this, KubeconfigNamedContext);

    KubeconfigNamedContext.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KubeconfigNamedContext, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KubeconfigNamedContext</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KubeconfigNamedContext} obj Optional instance to populate.
     * @return {module:model/KubeconfigNamedContext} The populated <code>KubeconfigNamedContext</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KubeconfigNamedContext();

        if (data.hasOwnProperty('context')) {
          obj['context'] = _KubeconfigContext["default"].constructFromObject(data['context']);
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>KubeconfigNamedContext</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KubeconfigNamedContext</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `context`
      if (data['context']) {
        // data not null
        _KubeconfigContext["default"].validateJSON(data['context']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }

      return true;
    }
  }]);

  return KubeconfigNamedContext;
}();
/**
 * @member {module:model/KubeconfigContext} context
 */


KubeconfigNamedContext.prototype['context'] = undefined;
/**
 * @member {String} name
 */

KubeconfigNamedContext.prototype['name'] = undefined;
var _default = KubeconfigNamedContext;
exports["default"] = _default;
