// akeyless/dist/model/KubeConfigValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _KubeconfigNamedCluster = _interopRequireDefault(require("./KubeconfigNamedCluster"));

var _KubeconfigNamedContext = _interopRequireDefault(require("./KubeconfigNamedContext"));

var _KubeconfigUser = _interopRequireDefault(require("./KubeconfigUser"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KubeConfigValue model module.
 * @module model/KubeConfigValue
 * @version 4.3.0
 */
var KubeConfigValue = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KubeConfigValue</code>.
   * @alias module:model/KubeConfigValue
   */
  function KubeConfigValue() {
    _classCallCheck(this, KubeConfigValue);

    KubeConfigValue.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KubeConfigValue, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KubeConfigValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KubeConfigValue} obj Optional instance to populate.
     * @return {module:model/KubeConfigValue} The populated <code>KubeConfigValue</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KubeConfigValue();

        if (data.hasOwnProperty('apiVersion')) {
          obj['apiVersion'] = _ApiClient["default"].convertToType(data['apiVersion'], 'String');
        }

        if (data.hasOwnProperty('clusters')) {
          obj['clusters'] = _ApiClient["default"].convertToType(data['clusters'], [_KubeconfigNamedCluster["default"]]);
        }

        if (data.hasOwnProperty('contexts')) {
          obj['contexts'] = _ApiClient["default"].convertToType(data['contexts'], [_KubeconfigNamedContext["default"]]);
        }

        if (data.hasOwnProperty('current-context')) {
          obj['current-context'] = _ApiClient["default"].convertToType(data['current-context'], 'String');
        }

        if (data.hasOwnProperty('kind')) {
          obj['kind'] = _ApiClient["default"].convertToType(data['kind'], 'String');
        }

        if (data.hasOwnProperty('users')) {
          obj['users'] = _ApiClient["default"].convertToType(data['users'], [_KubeconfigUser["default"]]);
        }
      }

      return obj;
    }
  }]);

  return KubeConfigValue;
}();
/**
 * @member {String} apiVersion
 */


KubeConfigValue.prototype['apiVersion'] = undefined;
/**
 * @member {Array.<module:model/KubeconfigNamedCluster>} clusters
 */

KubeConfigValue.prototype['clusters'] = undefined;
/**
 * @member {Array.<module:model/KubeconfigNamedContext>} contexts
 */

KubeConfigValue.prototype['contexts'] = undefined;
/**
 * @member {String} current-context
 */

KubeConfigValue.prototype['current-context'] = undefined;
/**
 * @member {String} kind
 */

KubeConfigValue.prototype['kind'] = undefined;
/**
 * @member {Array.<module:model/KubeconfigUser>} users
 */

KubeConfigValue.prototype['users'] = undefined;
var _default = KubeConfigValue;
exports["default"] = _default;
