// akeyless/dist/model/KubeconfigContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KubeconfigContext model module.
 * @module model/KubeconfigContext
 * @version 5.0.3
 */
var KubeconfigContext = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KubeconfigContext</code>.
   * @alias module:model/KubeconfigContext
   */
  function KubeconfigContext() {
    _classCallCheck(this, KubeconfigContext);

    KubeconfigContext.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KubeconfigContext, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KubeconfigContext</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KubeconfigContext} obj Optional instance to populate.
     * @return {module:model/KubeconfigContext} The populated <code>KubeconfigContext</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KubeconfigContext();

        if (data.hasOwnProperty('cluster')) {
          obj['cluster'] = _ApiClient["default"].convertToType(data['cluster'], 'String');
        }

        if (data.hasOwnProperty('namespace')) {
          obj['namespace'] = _ApiClient["default"].convertToType(data['namespace'], 'String');
        }

        if (data.hasOwnProperty('user')) {
          obj['user'] = _ApiClient["default"].convertToType(data['user'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>KubeconfigContext</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KubeconfigContext</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['cluster'] && !(typeof data['cluster'] === 'string' || data['cluster'] instanceof String)) {
        throw new Error("Expected the field `cluster` to be a primitive type in the JSON string but got " + data['cluster']);
      } // ensure the json data is a string


      if (data['namespace'] && !(typeof data['namespace'] === 'string' || data['namespace'] instanceof String)) {
        throw new Error("Expected the field `namespace` to be a primitive type in the JSON string but got " + data['namespace']);
      } // ensure the json data is a string


      if (data['user'] && !(typeof data['user'] === 'string' || data['user'] instanceof String)) {
        throw new Error("Expected the field `user` to be a primitive type in the JSON string but got " + data['user']);
      }

      return true;
    }
  }]);

  return KubeconfigContext;
}();
/**
 * @member {String} cluster
 */


KubeconfigContext.prototype['cluster'] = undefined;
/**
 * @member {String} namespace
 */

KubeconfigContext.prototype['namespace'] = undefined;
/**
 * @member {String} user
 */

KubeconfigContext.prototype['user'] = undefined;
var _default = KubeconfigContext;
exports["default"] = _default;
