// akeyless/dist/model/KubeconfigGenerateOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _KubeConfigValue = _interopRequireDefault(require("./KubeConfigValue"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KubeconfigGenerateOutput model module.
 * @module model/KubeconfigGenerateOutput
 * @version 5.0.6
 */
var KubeconfigGenerateOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KubeconfigGenerateOutput</code>.
   * @alias module:model/KubeconfigGenerateOutput
   */
  function KubeconfigGenerateOutput() {
    _classCallCheck(this, KubeconfigGenerateOutput);

    KubeconfigGenerateOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KubeconfigGenerateOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KubeconfigGenerateOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KubeconfigGenerateOutput} obj Optional instance to populate.
     * @return {module:model/KubeconfigGenerateOutput} The populated <code>KubeconfigGenerateOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KubeconfigGenerateOutput();

        if (data.hasOwnProperty('conflicted_clusters_names')) {
          obj['conflicted_clusters_names'] = _ApiClient["default"].convertToType(data['conflicted_clusters_names'], ['String']);
        }

        if (data.hasOwnProperty('data')) {
          obj['data'] = _KubeConfigValue["default"].constructFromObject(data['data']);
        }

        if (data.hasOwnProperty('errors')) {
          obj['errors'] = _ApiClient["default"].convertToType(data['errors'], ['String']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>KubeconfigGenerateOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>KubeconfigGenerateOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['conflicted_clusters_names'])) {
        throw new Error("Expected the field `conflicted_clusters_names` to be an array in the JSON data but got " + data['conflicted_clusters_names']);
      } // validate the optional field `data`


      if (data['data']) {
        // data not null
        _KubeConfigValue["default"].validateJSON(data['data']);
      } // ensure the json data is an array


      if (!Array.isArray(data['errors'])) {
        throw new Error("Expected the field `errors` to be an array in the JSON data but got " + data['errors']);
      }

      return true;
    }
  }]);

  return KubeconfigGenerateOutput;
}();
/**
 * @member {Array.<String>} conflicted_clusters_names
 */


KubeconfigGenerateOutput.prototype['conflicted_clusters_names'] = undefined;
/**
 * @member {module:model/KubeConfigValue} data
 */

KubeconfigGenerateOutput.prototype['data'] = undefined;
/**
 * @member {Array.<String>} errors
 */

KubeconfigGenerateOutput.prototype['errors'] = undefined;
var _default = KubeconfigGenerateOutput;
exports["default"] = _default;
