// akeyless/dist/model/LdapTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The LdapTargetDetails model module.
 * @module model/LdapTargetDetails
 * @version 5.0.5
 */
var LdapTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LdapTargetDetails</code>.
   * LdapTargetDetails
   * @alias module:model/LdapTargetDetails
   */
  function LdapTargetDetails() {
    _classCallCheck(this, LdapTargetDetails);

    LdapTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LdapTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>LdapTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LdapTargetDetails} obj Optional instance to populate.
     * @return {module:model/LdapTargetDetails} The populated <code>LdapTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LdapTargetDetails();

        if (data.hasOwnProperty('implementation_type')) {
          obj['implementation_type'] = _ApiClient["default"].convertToType(data['implementation_type'], 'String');
        }

        if (data.hasOwnProperty('ldap_audience')) {
          obj['ldap_audience'] = _ApiClient["default"].convertToType(data['ldap_audience'], 'String');
        }

        if (data.hasOwnProperty('ldap_bind_dn')) {
          obj['ldap_bind_dn'] = _ApiClient["default"].convertToType(data['ldap_bind_dn'], 'String');
        }

        if (data.hasOwnProperty('ldap_bind_password')) {
          obj['ldap_bind_password'] = _ApiClient["default"].convertToType(data['ldap_bind_password'], 'String');
        }

        if (data.hasOwnProperty('ldap_certificate')) {
          obj['ldap_certificate'] = _ApiClient["default"].convertToType(data['ldap_certificate'], 'String');
        }

        if (data.hasOwnProperty('ldap_token_expiration')) {
          obj['ldap_token_expiration'] = _ApiClient["default"].convertToType(data['ldap_token_expiration'], 'String');
        }

        if (data.hasOwnProperty('ldap_url')) {
          obj['ldap_url'] = _ApiClient["default"].convertToType(data['ldap_url'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>LdapTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>LdapTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['implementation_type'] && !(typeof data['implementation_type'] === 'string' || data['implementation_type'] instanceof String)) {
        throw new Error("Expected the field `implementation_type` to be a primitive type in the JSON string but got " + data['implementation_type']);
      } // ensure the json data is a string


      if (data['ldap_audience'] && !(typeof data['ldap_audience'] === 'string' || data['ldap_audience'] instanceof String)) {
        throw new Error("Expected the field `ldap_audience` to be a primitive type in the JSON string but got " + data['ldap_audience']);
      } // ensure the json data is a string


      if (data['ldap_bind_dn'] && !(typeof data['ldap_bind_dn'] === 'string' || data['ldap_bind_dn'] instanceof String)) {
        throw new Error("Expected the field `ldap_bind_dn` to be a primitive type in the JSON string but got " + data['ldap_bind_dn']);
      } // ensure the json data is a string


      if (data['ldap_bind_password'] && !(typeof data['ldap_bind_password'] === 'string' || data['ldap_bind_password'] instanceof String)) {
        throw new Error("Expected the field `ldap_bind_password` to be a primitive type in the JSON string but got " + data['ldap_bind_password']);
      } // ensure the json data is a string


      if (data['ldap_certificate'] && !(typeof data['ldap_certificate'] === 'string' || data['ldap_certificate'] instanceof String)) {
        throw new Error("Expected the field `ldap_certificate` to be a primitive type in the JSON string but got " + data['ldap_certificate']);
      } // ensure the json data is a string


      if (data['ldap_token_expiration'] && !(typeof data['ldap_token_expiration'] === 'string' || data['ldap_token_expiration'] instanceof String)) {
        throw new Error("Expected the field `ldap_token_expiration` to be a primitive type in the JSON string but got " + data['ldap_token_expiration']);
      } // ensure the json data is a string


      if (data['ldap_url'] && !(typeof data['ldap_url'] === 'string' || data['ldap_url'] instanceof String)) {
        throw new Error("Expected the field `ldap_url` to be a primitive type in the JSON string but got " + data['ldap_url']);
      }

      return true;
    }
  }]);

  return LdapTargetDetails;
}();
/**
 * @member {String} implementation_type
 */


LdapTargetDetails.prototype['implementation_type'] = undefined;
/**
 * @member {String} ldap_audience
 */

LdapTargetDetails.prototype['ldap_audience'] = undefined;
/**
 * @member {String} ldap_bind_dn
 */

LdapTargetDetails.prototype['ldap_bind_dn'] = undefined;
/**
 * @member {String} ldap_bind_password
 */

LdapTargetDetails.prototype['ldap_bind_password'] = undefined;
/**
 * @member {String} ldap_certificate
 */

LdapTargetDetails.prototype['ldap_certificate'] = undefined;
/**
 * @member {String} ldap_token_expiration
 */

LdapTargetDetails.prototype['ldap_token_expiration'] = undefined;
/**
 * @member {String} ldap_url
 */

LdapTargetDetails.prototype['ldap_url'] = undefined;
var _default = LdapTargetDetails;
exports["default"] = _default;
