// akeyless/dist/model/ListSRASessions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ListSRASessions model module.
 * @module model/ListSRASessions
 * @version 5.0.3
 */
var ListSRASessions = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListSRASessions</code>.
   * listSRASessions is a command that returns sra sessions of the given user
   * @alias module:model/ListSRASessions
   */
  function ListSRASessions() {
    _classCallCheck(this, ListSRASessions);

    ListSRASessions.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ListSRASessions, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
    }
    /**
     * Constructs a <code>ListSRASessions</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListSRASessions} obj Optional instance to populate.
     * @return {module:model/ListSRASessions} The populated <code>ListSRASessions</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListSRASessions();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('resource-type')) {
          obj['resource-type'] = _ApiClient["default"].convertToType(data['resource-type'], ['String']);
        }

        if (data.hasOwnProperty('status-type')) {
          obj['status-type'] = _ApiClient["default"].convertToType(data['status-type'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ListSRASessions</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ListSRASessions</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['resource-type'])) {
        throw new Error("Expected the field `resource-type` to be an array in the JSON data but got " + data['resource-type']);
      } // ensure the json data is an array


      if (!Array.isArray(data['status-type'])) {
        throw new Error("Expected the field `status-type` to be an array in the JSON data but got " + data['status-type']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return ListSRASessions;
}();
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */


ListSRASessions.prototype['json'] = false;
/**
 * session resource type. In case it is empty, all resources type will be returned. options: [mysql, k8s, ssh, mongodb, mssql, postgres, aws, eks, gke, rdp]
 * @member {Array.<String>} resource-type
 */

ListSRASessions.prototype['resource-type'] = undefined;
/**
 * session status type. In case it is empty, only active sessions will be returned. options: [connecting, connected, failed, completed, terminated]
 * @member {Array.<String>} status-type
 */

ListSRASessions.prototype['status-type'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

ListSRASessions.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

ListSRASessions.prototype['uid-token'] = undefined;
var _default = ListSRASessions;
exports["default"] = _default;
