// akeyless/dist/model/ListSraSessionsOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SraSessionEntryOut = _interopRequireDefault(require("./SraSessionEntryOut"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ListSraSessionsOutput model module.
 * @module model/ListSraSessionsOutput
 * @version 4.3.0
 */
var ListSraSessionsOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListSraSessionsOutput</code>.
   * @alias module:model/ListSraSessionsOutput
   */
  function ListSraSessionsOutput() {
    _classCallCheck(this, ListSraSessionsOutput);

    ListSraSessionsOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ListSraSessionsOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListSraSessionsOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListSraSessionsOutput} obj Optional instance to populate.
     * @return {module:model/ListSraSessionsOutput} The populated <code>ListSraSessionsOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListSraSessionsOutput();

        if (data.hasOwnProperty('allowed_gateways')) {
          obj['allowed_gateways'] = _ApiClient["default"].convertToType(data['allowed_gateways'], ['String']);
        }

        if (data.hasOwnProperty('next_page')) {
          obj['next_page'] = _ApiClient["default"].convertToType(data['next_page'], 'String');
        }

        if (data.hasOwnProperty('sessions')) {
          obj['sessions'] = _ApiClient["default"].convertToType(data['sessions'], [_SraSessionEntryOut["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ListSraSessionsOutput;
}();
/**
 * @member {Array.<String>} allowed_gateways
 */


ListSraSessionsOutput.prototype['allowed_gateways'] = undefined;
/**
 * @member {String} next_page
 */

ListSraSessionsOutput.prototype['next_page'] = undefined;
/**
 * @member {Array.<module:model/SraSessionEntryOut>} sessions
 */

ListSraSessionsOutput.prototype['sessions'] = undefined;
var _default = ListSraSessionsOutput;
exports["default"] = _default;
