// akeyless/dist/model/ListTargets.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ListTargets model module.
 * @module model/ListTargets
 * @version 5.0.14
 */
var ListTargets = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListTargets</code>.
   * listTargets is a command that returns a list of targets. [Deprecated: Use target-list command]
   * @alias module:model/ListTargets
   */
  function ListTargets() {
    _classCallCheck(this, ListTargets);

    ListTargets.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ListTargets, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
    }
    /**
     * Constructs a <code>ListTargets</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListTargets} obj Optional instance to populate.
     * @return {module:model/ListTargets} The populated <code>ListTargets</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListTargets();

        if (data.hasOwnProperty('filter')) {
          obj['filter'] = _ApiClient["default"].convertToType(data['filter'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('pagination-token')) {
          obj['pagination-token'] = _ApiClient["default"].convertToType(data['pagination-token'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], ['String']);
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ListTargets</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ListTargets</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['filter'] && !(typeof data['filter'] === 'string' || data['filter'] instanceof String)) {
        throw new Error("Expected the field `filter` to be a primitive type in the JSON string but got " + data['filter']);
      } // ensure the json data is a string


      if (data['pagination-token'] && !(typeof data['pagination-token'] === 'string' || data['pagination-token'] instanceof String)) {
        throw new Error("Expected the field `pagination-token` to be a primitive type in the JSON string but got " + data['pagination-token']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is an array


      if (!Array.isArray(data['type'])) {
        throw new Error("Expected the field `type` to be an array in the JSON data but got " + data['type']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return ListTargets;
}();
/**
 * Filter by auth method name or part of it
 * @member {String} filter
 */


ListTargets.prototype['filter'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

ListTargets.prototype['json'] = false;
/**
 * Next page reference
 * @member {String} pagination-token
 */

ListTargets.prototype['pagination-token'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

ListTargets.prototype['token'] = undefined;
/**
 * The target types list . In case it is empty, all types of targets will be returned. options: [hanadb cassandra aws ssh gke eks mysql mongodb snowflake mssql redshift artifactory azure rabbitmq k8s venafi gcp oracle dockerhub ldap github chef web salesforce postgres]
 * @member {Array.<String>} type
 */

ListTargets.prototype['type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

ListTargets.prototype['uid-token'] = undefined;
var _default = ListTargets;
exports["default"] = _default;
