// akeyless/dist/model/ListSraSessionsOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _GatewayNameInfo = _interopRequireDefault(require("./GatewayNameInfo"));

var _SraSessionEntryOut = _interopRequireDefault(require("./SraSessionEntryOut"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ListSraSessionsOutput model module.
 * @module model/ListSraSessionsOutput
 * @version 5.0.0
 */
var ListSraSessionsOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListSraSessionsOutput</code>.
   * @alias module:model/ListSraSessionsOutput
   */
  function ListSraSessionsOutput() {
    _classCallCheck(this, ListSraSessionsOutput);

    ListSraSessionsOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ListSraSessionsOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListSraSessionsOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListSraSessionsOutput} obj Optional instance to populate.
     * @return {module:model/ListSraSessionsOutput} The populated <code>ListSraSessionsOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListSraSessionsOutput();

        if (data.hasOwnProperty('allowed_gateways')) {
          obj['allowed_gateways'] = _ApiClient["default"].convertToType(data['allowed_gateways'], [_GatewayNameInfo["default"]]);
        }

        if (data.hasOwnProperty('next_page')) {
          obj['next_page'] = _ApiClient["default"].convertToType(data['next_page'], 'String');
        }

        if (data.hasOwnProperty('sessions')) {
          obj['sessions'] = _ApiClient["default"].convertToType(data['sessions'], [_SraSessionEntryOut["default"]]);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ListSraSessionsOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ListSraSessionsOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['allowed_gateways']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['allowed_gateways'])) {
          throw new Error("Expected the field `allowed_gateways` to be an array in the JSON data but got " + data['allowed_gateways']);
        } // validate the optional field `allowed_gateways` (array)


        var _iterator = _createForOfIteratorHelper(data['allowed_gateways']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _GatewayNameInfo["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is a string


      if (data['next_page'] && !(typeof data['next_page'] === 'string' || data['next_page'] instanceof String)) {
        throw new Error("Expected the field `next_page` to be a primitive type in the JSON string but got " + data['next_page']);
      }

      if (data['sessions']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['sessions'])) {
          throw new Error("Expected the field `sessions` to be an array in the JSON data but got " + data['sessions']);
        } // validate the optional field `sessions` (array)


        var _iterator2 = _createForOfIteratorHelper(data['sessions']),
            _step2;

        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;

            _SraSessionEntryOut["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }

        ;
      }

      return true;
    }
  }]);

  return ListSraSessionsOutput;
}();
/**
 * @member {Array.<module:model/GatewayNameInfo>} allowed_gateways
 */


ListSraSessionsOutput.prototype['allowed_gateways'] = undefined;
/**
 * @member {String} next_page
 */

ListSraSessionsOutput.prototype['next_page'] = undefined;
/**
 * @member {Array.<module:model/SraSessionEntryOut>} sessions
 */

ListSraSessionsOutput.prototype['sessions'] = undefined;
var _default = ListSraSessionsOutput;
exports["default"] = _default;
