// akeyless/dist/model/LogForwardingConfigPart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AwsS3LogForwardingConfig = _interopRequireDefault(require("./AwsS3LogForwardingConfig"));

var _AzureLogAnalyticsForwardingConfig = _interopRequireDefault(require("./AzureLogAnalyticsForwardingConfig"));

var _DatadogForwardingConfig = _interopRequireDefault(require("./DatadogForwardingConfig"));

var _ElasticsearchLogForwardingConfig = _interopRequireDefault(require("./ElasticsearchLogForwardingConfig"));

var _GoogleChronicleForwardingConfig = _interopRequireDefault(require("./GoogleChronicleForwardingConfig"));

var _LogstashLogForwardingConfig = _interopRequireDefault(require("./LogstashLogForwardingConfig"));

var _LogzIoLogForwardingConfig = _interopRequireDefault(require("./LogzIoLogForwardingConfig"));

var _SplunkLogForwardingConfig = _interopRequireDefault(require("./SplunkLogForwardingConfig"));

var _SumologicLogForwardingConfig = _interopRequireDefault(require("./SumologicLogForwardingConfig"));

var _SyslogLogForwardingConfig = _interopRequireDefault(require("./SyslogLogForwardingConfig"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The LogForwardingConfigPart model module.
 * @module model/LogForwardingConfigPart
 * @version 5.0.3
 */
var LogForwardingConfigPart = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LogForwardingConfigPart</code>.
   * @alias module:model/LogForwardingConfigPart
   */
  function LogForwardingConfigPart() {
    _classCallCheck(this, LogForwardingConfigPart);

    LogForwardingConfigPart.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LogForwardingConfigPart, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>LogForwardingConfigPart</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/LogForwardingConfigPart} obj Optional instance to populate.
     * @return {module:model/LogForwardingConfigPart} The populated <code>LogForwardingConfigPart</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LogForwardingConfigPart();

        if (data.hasOwnProperty('aws_s3_config')) {
          obj['aws_s3_config'] = _AwsS3LogForwardingConfig["default"].constructFromObject(data['aws_s3_config']);
        }

        if (data.hasOwnProperty('azure_analytics_config')) {
          obj['azure_analytics_config'] = _AzureLogAnalyticsForwardingConfig["default"].constructFromObject(data['azure_analytics_config']);
        }

        if (data.hasOwnProperty('datadog_config')) {
          obj['datadog_config'] = _DatadogForwardingConfig["default"].constructFromObject(data['datadog_config']);
        }

        if (data.hasOwnProperty('elasticsearch_config')) {
          obj['elasticsearch_config'] = _ElasticsearchLogForwardingConfig["default"].constructFromObject(data['elasticsearch_config']);
        }

        if (data.hasOwnProperty('google_chronicle_config')) {
          obj['google_chronicle_config'] = _GoogleChronicleForwardingConfig["default"].constructFromObject(data['google_chronicle_config']);
        }

        if (data.hasOwnProperty('json_output')) {
          obj['json_output'] = _ApiClient["default"].convertToType(data['json_output'], 'Boolean');
        }

        if (data.hasOwnProperty('logan_enable')) {
          obj['logan_enable'] = _ApiClient["default"].convertToType(data['logan_enable'], 'Boolean');
        }

        if (data.hasOwnProperty('logan_url')) {
          obj['logan_url'] = _ApiClient["default"].convertToType(data['logan_url'], 'String');
        }

        if (data.hasOwnProperty('logstash_config')) {
          obj['logstash_config'] = _LogstashLogForwardingConfig["default"].constructFromObject(data['logstash_config']);
        }

        if (data.hasOwnProperty('logz_io_config')) {
          obj['logz_io_config'] = _LogzIoLogForwardingConfig["default"].constructFromObject(data['logz_io_config']);
        }

        if (data.hasOwnProperty('pull_interval_sec')) {
          obj['pull_interval_sec'] = _ApiClient["default"].convertToType(data['pull_interval_sec'], 'String');
        }

        if (data.hasOwnProperty('splunk_config')) {
          obj['splunk_config'] = _SplunkLogForwardingConfig["default"].constructFromObject(data['splunk_config']);
        }

        if (data.hasOwnProperty('sumo_logic_config')) {
          obj['sumo_logic_config'] = _SumologicLogForwardingConfig["default"].constructFromObject(data['sumo_logic_config']);
        }

        if (data.hasOwnProperty('syslog_config')) {
          obj['syslog_config'] = _SyslogLogForwardingConfig["default"].constructFromObject(data['syslog_config']);
        }

        if (data.hasOwnProperty('target_log_type')) {
          obj['target_log_type'] = _ApiClient["default"].convertToType(data['target_log_type'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>LogForwardingConfigPart</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>LogForwardingConfigPart</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `aws_s3_config`
      if (data['aws_s3_config']) {
        // data not null
        _AwsS3LogForwardingConfig["default"].validateJSON(data['aws_s3_config']);
      } // validate the optional field `azure_analytics_config`


      if (data['azure_analytics_config']) {
        // data not null
        _AzureLogAnalyticsForwardingConfig["default"].validateJSON(data['azure_analytics_config']);
      } // validate the optional field `datadog_config`


      if (data['datadog_config']) {
        // data not null
        _DatadogForwardingConfig["default"].validateJSON(data['datadog_config']);
      } // validate the optional field `elasticsearch_config`


      if (data['elasticsearch_config']) {
        // data not null
        _ElasticsearchLogForwardingConfig["default"].validateJSON(data['elasticsearch_config']);
      } // validate the optional field `google_chronicle_config`


      if (data['google_chronicle_config']) {
        // data not null
        _GoogleChronicleForwardingConfig["default"].validateJSON(data['google_chronicle_config']);
      } // ensure the json data is a string


      if (data['logan_url'] && !(typeof data['logan_url'] === 'string' || data['logan_url'] instanceof String)) {
        throw new Error("Expected the field `logan_url` to be a primitive type in the JSON string but got " + data['logan_url']);
      } // validate the optional field `logstash_config`


      if (data['logstash_config']) {
        // data not null
        _LogstashLogForwardingConfig["default"].validateJSON(data['logstash_config']);
      } // validate the optional field `logz_io_config`


      if (data['logz_io_config']) {
        // data not null
        _LogzIoLogForwardingConfig["default"].validateJSON(data['logz_io_config']);
      } // ensure the json data is a string


      if (data['pull_interval_sec'] && !(typeof data['pull_interval_sec'] === 'string' || data['pull_interval_sec'] instanceof String)) {
        throw new Error("Expected the field `pull_interval_sec` to be a primitive type in the JSON string but got " + data['pull_interval_sec']);
      } // validate the optional field `splunk_config`


      if (data['splunk_config']) {
        // data not null
        _SplunkLogForwardingConfig["default"].validateJSON(data['splunk_config']);
      } // validate the optional field `sumo_logic_config`


      if (data['sumo_logic_config']) {
        // data not null
        _SumologicLogForwardingConfig["default"].validateJSON(data['sumo_logic_config']);
      } // validate the optional field `syslog_config`


      if (data['syslog_config']) {
        // data not null
        _SyslogLogForwardingConfig["default"].validateJSON(data['syslog_config']);
      } // ensure the json data is a string


      if (data['target_log_type'] && !(typeof data['target_log_type'] === 'string' || data['target_log_type'] instanceof String)) {
        throw new Error("Expected the field `target_log_type` to be a primitive type in the JSON string but got " + data['target_log_type']);
      }

      return true;
    }
  }]);

  return LogForwardingConfigPart;
}();
/**
 * @member {module:model/AwsS3LogForwardingConfig} aws_s3_config
 */


LogForwardingConfigPart.prototype['aws_s3_config'] = undefined;
/**
 * @member {module:model/AzureLogAnalyticsForwardingConfig} azure_analytics_config
 */

LogForwardingConfigPart.prototype['azure_analytics_config'] = undefined;
/**
 * @member {module:model/DatadogForwardingConfig} datadog_config
 */

LogForwardingConfigPart.prototype['datadog_config'] = undefined;
/**
 * @member {module:model/ElasticsearchLogForwardingConfig} elasticsearch_config
 */

LogForwardingConfigPart.prototype['elasticsearch_config'] = undefined;
/**
 * @member {module:model/GoogleChronicleForwardingConfig} google_chronicle_config
 */

LogForwardingConfigPart.prototype['google_chronicle_config'] = undefined;
/**
 * @member {Boolean} json_output
 */

LogForwardingConfigPart.prototype['json_output'] = undefined;
/**
 * @member {Boolean} logan_enable
 */

LogForwardingConfigPart.prototype['logan_enable'] = undefined;
/**
 * @member {String} logan_url
 */

LogForwardingConfigPart.prototype['logan_url'] = undefined;
/**
 * @member {module:model/LogstashLogForwardingConfig} logstash_config
 */

LogForwardingConfigPart.prototype['logstash_config'] = undefined;
/**
 * @member {module:model/LogzIoLogForwardingConfig} logz_io_config
 */

LogForwardingConfigPart.prototype['logz_io_config'] = undefined;
/**
 * @member {String} pull_interval_sec
 */

LogForwardingConfigPart.prototype['pull_interval_sec'] = undefined;
/**
 * @member {module:model/SplunkLogForwardingConfig} splunk_config
 */

LogForwardingConfigPart.prototype['splunk_config'] = undefined;
/**
 * @member {module:model/SumologicLogForwardingConfig} sumo_logic_config
 */

LogForwardingConfigPart.prototype['sumo_logic_config'] = undefined;
/**
 * @member {module:model/SyslogLogForwardingConfig} syslog_config
 */

LogForwardingConfigPart.prototype['syslog_config'] = undefined;
/**
 * @member {String} target_log_type
 */

LogForwardingConfigPart.prototype['target_log_type'] = undefined;
var _default = LogForwardingConfigPart;
exports["default"] = _default;
