// akeyless/dist/model/MigrationsConfigPart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AWSSecretsMigration = _interopRequireDefault(require("./AWSSecretsMigration"));

var _ActiveDirectoryMigration = _interopRequireDefault(require("./ActiveDirectoryMigration"));

var _AzureKeyVaultMigration = _interopRequireDefault(require("./AzureKeyVaultMigration"));

var _GCPSecretsMigration = _interopRequireDefault(require("./GCPSecretsMigration"));

var _HashiMigration = _interopRequireDefault(require("./HashiMigration"));

var _K8SMigration = _interopRequireDefault(require("./K8SMigration"));

var _MockMigration = _interopRequireDefault(require("./MockMigration"));

var _OnePasswordMigration = _interopRequireDefault(require("./OnePasswordMigration"));

var _ServerInventoryMigration = _interopRequireDefault(require("./ServerInventoryMigration"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MigrationsConfigPart model module.
 * @module model/MigrationsConfigPart
 * @version 5.0.8
 */
var MigrationsConfigPart = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MigrationsConfigPart</code>.
   * @alias module:model/MigrationsConfigPart
   */
  function MigrationsConfigPart() {
    _classCallCheck(this, MigrationsConfigPart);

    MigrationsConfigPart.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MigrationsConfigPart, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MigrationsConfigPart</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MigrationsConfigPart} obj Optional instance to populate.
     * @return {module:model/MigrationsConfigPart} The populated <code>MigrationsConfigPart</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MigrationsConfigPart();

        if (data.hasOwnProperty('active_directory_migrations')) {
          obj['active_directory_migrations'] = _ApiClient["default"].convertToType(data['active_directory_migrations'], [_ActiveDirectoryMigration["default"]]);
        }

        if (data.hasOwnProperty('aws_secrets_migrations')) {
          obj['aws_secrets_migrations'] = _ApiClient["default"].convertToType(data['aws_secrets_migrations'], [_AWSSecretsMigration["default"]]);
        }

        if (data.hasOwnProperty('azure_kv_migrations')) {
          obj['azure_kv_migrations'] = _ApiClient["default"].convertToType(data['azure_kv_migrations'], [_AzureKeyVaultMigration["default"]]);
        }

        if (data.hasOwnProperty('gcp_secrets_migrations')) {
          obj['gcp_secrets_migrations'] = _ApiClient["default"].convertToType(data['gcp_secrets_migrations'], [_GCPSecretsMigration["default"]]);
        }

        if (data.hasOwnProperty('hashi_migrations')) {
          obj['hashi_migrations'] = _ApiClient["default"].convertToType(data['hashi_migrations'], [_HashiMigration["default"]]);
        }

        if (data.hasOwnProperty('k8s_migrations')) {
          obj['k8s_migrations'] = _ApiClient["default"].convertToType(data['k8s_migrations'], [_K8SMigration["default"]]);
        }

        if (data.hasOwnProperty('mock_migrations')) {
          obj['mock_migrations'] = _ApiClient["default"].convertToType(data['mock_migrations'], [_MockMigration["default"]]);
        }

        if (data.hasOwnProperty('one_password_migrations')) {
          obj['one_password_migrations'] = _ApiClient["default"].convertToType(data['one_password_migrations'], [_OnePasswordMigration["default"]]);
        }

        if (data.hasOwnProperty('server_inventory_migrations')) {
          obj['server_inventory_migrations'] = _ApiClient["default"].convertToType(data['server_inventory_migrations'], [_ServerInventoryMigration["default"]]);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>MigrationsConfigPart</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>MigrationsConfigPart</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['active_directory_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['active_directory_migrations'])) {
          throw new Error("Expected the field `active_directory_migrations` to be an array in the JSON data but got " + data['active_directory_migrations']);
        } // validate the optional field `active_directory_migrations` (array)


        var _iterator = _createForOfIteratorHelper(data['active_directory_migrations']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _ActiveDirectoryMigration["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      }

      if (data['aws_secrets_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['aws_secrets_migrations'])) {
          throw new Error("Expected the field `aws_secrets_migrations` to be an array in the JSON data but got " + data['aws_secrets_migrations']);
        } // validate the optional field `aws_secrets_migrations` (array)


        var _iterator2 = _createForOfIteratorHelper(data['aws_secrets_migrations']),
            _step2;

        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;

            _AWSSecretsMigration["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }

        ;
      }

      if (data['azure_kv_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['azure_kv_migrations'])) {
          throw new Error("Expected the field `azure_kv_migrations` to be an array in the JSON data but got " + data['azure_kv_migrations']);
        } // validate the optional field `azure_kv_migrations` (array)


        var _iterator3 = _createForOfIteratorHelper(data['azure_kv_migrations']),
            _step3;

        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item2 = _step3.value;

            _AzureKeyVaultMigration["default"].validateJSON(_item2);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }

        ;
      }

      if (data['gcp_secrets_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['gcp_secrets_migrations'])) {
          throw new Error("Expected the field `gcp_secrets_migrations` to be an array in the JSON data but got " + data['gcp_secrets_migrations']);
        } // validate the optional field `gcp_secrets_migrations` (array)


        var _iterator4 = _createForOfIteratorHelper(data['gcp_secrets_migrations']),
            _step4;

        try {
          for (_iterator4.s(); !(_step4 = _iterator4.n()).done;) {
            var _item3 = _step4.value;

            _GCPSecretsMigration["default"].validateJSON(_item3);
          }
        } catch (err) {
          _iterator4.e(err);
        } finally {
          _iterator4.f();
        }

        ;
      }

      if (data['hashi_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['hashi_migrations'])) {
          throw new Error("Expected the field `hashi_migrations` to be an array in the JSON data but got " + data['hashi_migrations']);
        } // validate the optional field `hashi_migrations` (array)


        var _iterator5 = _createForOfIteratorHelper(data['hashi_migrations']),
            _step5;

        try {
          for (_iterator5.s(); !(_step5 = _iterator5.n()).done;) {
            var _item4 = _step5.value;

            _HashiMigration["default"].validateJSON(_item4);
          }
        } catch (err) {
          _iterator5.e(err);
        } finally {
          _iterator5.f();
        }

        ;
      }

      if (data['k8s_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['k8s_migrations'])) {
          throw new Error("Expected the field `k8s_migrations` to be an array in the JSON data but got " + data['k8s_migrations']);
        } // validate the optional field `k8s_migrations` (array)


        var _iterator6 = _createForOfIteratorHelper(data['k8s_migrations']),
            _step6;

        try {
          for (_iterator6.s(); !(_step6 = _iterator6.n()).done;) {
            var _item5 = _step6.value;

            _K8SMigration["default"].validateJSON(_item5);
          }
        } catch (err) {
          _iterator6.e(err);
        } finally {
          _iterator6.f();
        }

        ;
      }

      if (data['mock_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['mock_migrations'])) {
          throw new Error("Expected the field `mock_migrations` to be an array in the JSON data but got " + data['mock_migrations']);
        } // validate the optional field `mock_migrations` (array)


        var _iterator7 = _createForOfIteratorHelper(data['mock_migrations']),
            _step7;

        try {
          for (_iterator7.s(); !(_step7 = _iterator7.n()).done;) {
            var _item6 = _step7.value;

            _MockMigration["default"].validateJSON(_item6);
          }
        } catch (err) {
          _iterator7.e(err);
        } finally {
          _iterator7.f();
        }

        ;
      }

      if (data['one_password_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['one_password_migrations'])) {
          throw new Error("Expected the field `one_password_migrations` to be an array in the JSON data but got " + data['one_password_migrations']);
        } // validate the optional field `one_password_migrations` (array)


        var _iterator8 = _createForOfIteratorHelper(data['one_password_migrations']),
            _step8;

        try {
          for (_iterator8.s(); !(_step8 = _iterator8.n()).done;) {
            var _item7 = _step8.value;

            _OnePasswordMigration["default"].validateJSON(_item7);
          }
        } catch (err) {
          _iterator8.e(err);
        } finally {
          _iterator8.f();
        }

        ;
      }

      if (data['server_inventory_migrations']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['server_inventory_migrations'])) {
          throw new Error("Expected the field `server_inventory_migrations` to be an array in the JSON data but got " + data['server_inventory_migrations']);
        } // validate the optional field `server_inventory_migrations` (array)


        var _iterator9 = _createForOfIteratorHelper(data['server_inventory_migrations']),
            _step9;

        try {
          for (_iterator9.s(); !(_step9 = _iterator9.n()).done;) {
            var _item8 = _step9.value;

            _ServerInventoryMigration["default"].validateJSON(_item8);
          }
        } catch (err) {
          _iterator9.e(err);
        } finally {
          _iterator9.f();
        }

        ;
      }

      return true;
    }
  }]);

  return MigrationsConfigPart;
}();
/**
 * @member {Array.<module:model/ActiveDirectoryMigration>} active_directory_migrations
 */


MigrationsConfigPart.prototype['active_directory_migrations'] = undefined;
/**
 * @member {Array.<module:model/AWSSecretsMigration>} aws_secrets_migrations
 */

MigrationsConfigPart.prototype['aws_secrets_migrations'] = undefined;
/**
 * @member {Array.<module:model/AzureKeyVaultMigration>} azure_kv_migrations
 */

MigrationsConfigPart.prototype['azure_kv_migrations'] = undefined;
/**
 * @member {Array.<module:model/GCPSecretsMigration>} gcp_secrets_migrations
 */

MigrationsConfigPart.prototype['gcp_secrets_migrations'] = undefined;
/**
 * @member {Array.<module:model/HashiMigration>} hashi_migrations
 */

MigrationsConfigPart.prototype['hashi_migrations'] = undefined;
/**
 * @member {Array.<module:model/K8SMigration>} k8s_migrations
 */

MigrationsConfigPart.prototype['k8s_migrations'] = undefined;
/**
 * @member {Array.<module:model/MockMigration>} mock_migrations
 */

MigrationsConfigPart.prototype['mock_migrations'] = undefined;
/**
 * @member {Array.<module:model/OnePasswordMigration>} one_password_migrations
 */

MigrationsConfigPart.prototype['one_password_migrations'] = undefined;
/**
 * @member {Array.<module:model/ServerInventoryMigration>} server_inventory_migrations
 */

MigrationsConfigPart.prototype['server_inventory_migrations'] = undefined;
var _default = MigrationsConfigPart;
exports["default"] = _default;
