// akeyless/dist/model/Name.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AttributeTypeAndValue = _interopRequireDefault(require("./AttributeTypeAndValue"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Name model module.
 * @module model/Name
 * @version 5.0.5
 */
var Name = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Name</code>.
   * Name represents an X.509 distinguished name. This only includes the common elements of a DN. Note that Name is only an approximation of the X.509 structure. If an accurate representation is needed, asn1.Unmarshal the raw subject or issuer as an [RDNSequence].
   * @alias module:model/Name
   */
  function Name() {
    _classCallCheck(this, Name);

    Name.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Name, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Name</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Name} obj Optional instance to populate.
     * @return {module:model/Name} The populated <code>Name</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Name();

        if (data.hasOwnProperty('Country')) {
          obj['Country'] = _ApiClient["default"].convertToType(data['Country'], ['String']);
        }

        if (data.hasOwnProperty('ExtraNames')) {
          obj['ExtraNames'] = _ApiClient["default"].convertToType(data['ExtraNames'], [_AttributeTypeAndValue["default"]]);
        }

        if (data.hasOwnProperty('Locality')) {
          obj['Locality'] = _ApiClient["default"].convertToType(data['Locality'], ['String']);
        }

        if (data.hasOwnProperty('Names')) {
          obj['Names'] = _ApiClient["default"].convertToType(data['Names'], [_AttributeTypeAndValue["default"]]);
        }

        if (data.hasOwnProperty('SerialNumber')) {
          obj['SerialNumber'] = _ApiClient["default"].convertToType(data['SerialNumber'], 'String');
        }

        if (data.hasOwnProperty('StreetAddress')) {
          obj['StreetAddress'] = _ApiClient["default"].convertToType(data['StreetAddress'], ['String']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>Name</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Name</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['Country'])) {
        throw new Error("Expected the field `Country` to be an array in the JSON data but got " + data['Country']);
      }

      if (data['ExtraNames']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['ExtraNames'])) {
          throw new Error("Expected the field `ExtraNames` to be an array in the JSON data but got " + data['ExtraNames']);
        } // validate the optional field `ExtraNames` (array)


        var _iterator = _createForOfIteratorHelper(data['ExtraNames']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _AttributeTypeAndValue["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is an array


      if (!Array.isArray(data['Locality'])) {
        throw new Error("Expected the field `Locality` to be an array in the JSON data but got " + data['Locality']);
      }

      if (data['Names']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['Names'])) {
          throw new Error("Expected the field `Names` to be an array in the JSON data but got " + data['Names']);
        } // validate the optional field `Names` (array)


        var _iterator2 = _createForOfIteratorHelper(data['Names']),
            _step2;

        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;

            _AttributeTypeAndValue["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }

        ;
      } // ensure the json data is a string


      if (data['SerialNumber'] && !(typeof data['SerialNumber'] === 'string' || data['SerialNumber'] instanceof String)) {
        throw new Error("Expected the field `SerialNumber` to be a primitive type in the JSON string but got " + data['SerialNumber']);
      } // ensure the json data is an array


      if (!Array.isArray(data['StreetAddress'])) {
        throw new Error("Expected the field `StreetAddress` to be an array in the JSON data but got " + data['StreetAddress']);
      }

      return true;
    }
  }]);

  return Name;
}();
/**
 * @member {Array.<String>} Country
 */


Name.prototype['Country'] = undefined;
/**
 * ExtraNames contains attributes to be copied, raw, into any marshaled distinguished names. Values override any attributes with the same OID. The ExtraNames field is not populated when parsing, see Names.
 * @member {Array.<module:model/AttributeTypeAndValue>} ExtraNames
 */

Name.prototype['ExtraNames'] = undefined;
/**
 * @member {Array.<String>} Locality
 */

Name.prototype['Locality'] = undefined;
/**
 * Names contains all parsed attributes. When parsing distinguished names, this can be used to extract non-standard attributes that are not parsed by this package. When marshaling to RDNSequences, the Names field is ignored, see ExtraNames.
 * @member {Array.<module:model/AttributeTypeAndValue>} Names
 */

Name.prototype['Names'] = undefined;
/**
 * @member {String} SerialNumber
 */

Name.prototype['SerialNumber'] = undefined;
/**
 * @member {Array.<String>} StreetAddress
 */

Name.prototype['StreetAddress'] = undefined;
var _default = Name;
exports["default"] = _default;
