// akeyless/dist/model/NotiForwarderDetailsInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SlackNotiForwarderDetails = _interopRequireDefault(require("./SlackNotiForwarderDetails"));

var _TeamsNotiForwarderDetails = _interopRequireDefault(require("./TeamsNotiForwarderDetails"));

var _WebhookNotiForwarderDetails = _interopRequireDefault(require("./WebhookNotiForwarderDetails"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The NotiForwarderDetailsInput model module.
 * @module model/NotiForwarderDetailsInput
 * @version 5.0.16
 */
var NotiForwarderDetailsInput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NotiForwarderDetailsInput</code>.
   * @alias module:model/NotiForwarderDetailsInput
   */
  function NotiForwarderDetailsInput() {
    _classCallCheck(this, NotiForwarderDetailsInput);

    NotiForwarderDetailsInput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(NotiForwarderDetailsInput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NotiForwarderDetailsInput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NotiForwarderDetailsInput} obj Optional instance to populate.
     * @return {module:model/NotiForwarderDetailsInput} The populated <code>NotiForwarderDetailsInput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NotiForwarderDetailsInput();

        if (data.hasOwnProperty('app_private_key_pem_base64')) {
          obj['app_private_key_pem_base64'] = _ApiClient["default"].convertToType(data['app_private_key_pem_base64'], 'String');
        }

        if (data.hasOwnProperty('client_secret')) {
          obj['client_secret'] = _ApiClient["default"].convertToType(data['client_secret'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('slack_noti_forwarder_details')) {
          obj['slack_noti_forwarder_details'] = _SlackNotiForwarderDetails["default"].constructFromObject(data['slack_noti_forwarder_details']);
        }

        if (data.hasOwnProperty('teams_noti_forwarder_details')) {
          obj['teams_noti_forwarder_details'] = _TeamsNotiForwarderDetails["default"].constructFromObject(data['teams_noti_forwarder_details']);
        }

        if (data.hasOwnProperty('webhook_noti_forwarder_details')) {
          obj['webhook_noti_forwarder_details'] = _WebhookNotiForwarderDetails["default"].constructFromObject(data['webhook_noti_forwarder_details']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>NotiForwarderDetailsInput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>NotiForwarderDetailsInput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['app_private_key_pem_base64'] && !(typeof data['app_private_key_pem_base64'] === 'string' || data['app_private_key_pem_base64'] instanceof String)) {
        throw new Error("Expected the field `app_private_key_pem_base64` to be a primitive type in the JSON string but got " + data['app_private_key_pem_base64']);
      } // ensure the json data is a string


      if (data['client_secret'] && !(typeof data['client_secret'] === 'string' || data['client_secret'] instanceof String)) {
        throw new Error("Expected the field `client_secret` to be a primitive type in the JSON string but got " + data['client_secret']);
      } // ensure the json data is a string


      if (data['password'] && !(typeof data['password'] === 'string' || data['password'] instanceof String)) {
        throw new Error("Expected the field `password` to be a primitive type in the JSON string but got " + data['password']);
      } // validate the optional field `slack_noti_forwarder_details`


      if (data['slack_noti_forwarder_details']) {
        // data not null
        _SlackNotiForwarderDetails["default"].validateJSON(data['slack_noti_forwarder_details']);
      } // validate the optional field `teams_noti_forwarder_details`


      if (data['teams_noti_forwarder_details']) {
        // data not null
        _TeamsNotiForwarderDetails["default"].validateJSON(data['teams_noti_forwarder_details']);
      } // validate the optional field `webhook_noti_forwarder_details`


      if (data['webhook_noti_forwarder_details']) {
        // data not null
        _WebhookNotiForwarderDetails["default"].validateJSON(data['webhook_noti_forwarder_details']);
      }

      return true;
    }
  }]);

  return NotiForwarderDetailsInput;
}();
/**
 * @member {String} app_private_key_pem_base64
 */


NotiForwarderDetailsInput.prototype['app_private_key_pem_base64'] = undefined;
/**
 * Auth - JWT
 * @member {String} client_secret
 */

NotiForwarderDetailsInput.prototype['client_secret'] = undefined;
/**
 * Auth - User Password
 * @member {String} password
 */

NotiForwarderDetailsInput.prototype['password'] = undefined;
/**
 * @member {module:model/SlackNotiForwarderDetails} slack_noti_forwarder_details
 */

NotiForwarderDetailsInput.prototype['slack_noti_forwarder_details'] = undefined;
/**
 * @member {module:model/TeamsNotiForwarderDetails} teams_noti_forwarder_details
 */

NotiForwarderDetailsInput.prototype['teams_noti_forwarder_details'] = undefined;
/**
 * @member {module:model/WebhookNotiForwarderDetails} webhook_noti_forwarder_details
 */

NotiForwarderDetailsInput.prototype['webhook_noti_forwarder_details'] = undefined;
var _default = NotiForwarderDetailsInput;
exports["default"] = _default;
