// akeyless/dist/model/NullTime.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The NullTime model module.
 * @module model/NullTime
 * @version 5.0.14
 */
var NullTime = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NullTime</code>.
   * NullTime implements the [Scanner] interface so it can be used as a scan destination, similar to [NullString].
   * @alias module:model/NullTime
   */
  function NullTime() {
    _classCallCheck(this, NullTime);

    NullTime.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(NullTime, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NullTime</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NullTime} obj Optional instance to populate.
     * @return {module:model/NullTime} The populated <code>NullTime</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NullTime();

        if (data.hasOwnProperty('Time')) {
          obj['Time'] = _ApiClient["default"].convertToType(data['Time'], 'Date');
        }

        if (data.hasOwnProperty('Valid')) {
          obj['Valid'] = _ApiClient["default"].convertToType(data['Valid'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>NullTime</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>NullTime</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);

  return NullTime;
}();
/**
 * @member {Date} Time
 */


NullTime.prototype['Time'] = undefined;
/**
 * @member {Boolean} Valid
 */

NullTime.prototype['Valid'] = undefined;
var _default = NullTime;
exports["default"] = _default;
