// akeyless/dist/model/NullString.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The NullString model module.
 * @module model/NullString
 * @version 5.0.3
 */
var NullString = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NullString</code>.
   * var s NullString err :&#x3D; db.QueryRow(\&quot;SELECT name FROM foo WHERE id&#x3D;?\&quot;, id).Scan(&amp;s) ... if s.Valid { use s.String } else { NULL value }
   * @alias module:model/NullString
   */
  function NullString() {
    _classCallCheck(this, NullString);

    NullString.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(NullString, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>NullString</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NullString} obj Optional instance to populate.
     * @return {module:model/NullString} The populated <code>NullString</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NullString();

        if (data.hasOwnProperty('String')) {
          obj['String'] = _ApiClient["default"].convertToType(data['String'], 'String');
        }

        if (data.hasOwnProperty('Valid')) {
          obj['Valid'] = _ApiClient["default"].convertToType(data['Valid'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>NullString</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>NullString</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['String'] && !(typeof data['String'] === 'string' || data['String'] instanceof String)) {
        throw new Error("Expected the field `String` to be a primitive type in the JSON string but got " + data['String']);
      }

      return true;
    }
  }]);

  return NullString;
}();
/**
 * @member {String} String
 */


NullString.prototype['String'] = undefined;
/**
 * @member {Boolean} Valid
 */

NullString.prototype['Valid'] = undefined;
var _default = NullString;
exports["default"] = _default;
