// akeyless/dist/model/OAuth2AccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _OAuth2CustomClaim = _interopRequireDefault(require("./OAuth2CustomClaim"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The OAuth2AccessRules model module.
 * @module model/OAuth2AccessRules
 * @version 5.0.3
 */
var OAuth2AccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OAuth2AccessRules</code>.
   * OAuth2AccessRules contains access rules specific to OAuth2 authentication method.
   * @alias module:model/OAuth2AccessRules
   */
  function OAuth2AccessRules() {
    _classCallCheck(this, OAuth2AccessRules);

    OAuth2AccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(OAuth2AccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OAuth2AccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OAuth2AccessRules} obj Optional instance to populate.
     * @return {module:model/OAuth2AccessRules} The populated <code>OAuth2AccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OAuth2AccessRules();

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('authorized_gw_cluster_name')) {
          obj['authorized_gw_cluster_name'] = _ApiClient["default"].convertToType(data['authorized_gw_cluster_name'], 'String');
        }

        if (data.hasOwnProperty('bound_claims')) {
          obj['bound_claims'] = _ApiClient["default"].convertToType(data['bound_claims'], [_OAuth2CustomClaim["default"]]);
        }

        if (data.hasOwnProperty('bound_clients_id')) {
          obj['bound_clients_id'] = _ApiClient["default"].convertToType(data['bound_clients_id'], ['String']);
        }

        if (data.hasOwnProperty('certificate')) {
          obj['certificate'] = _ApiClient["default"].convertToType(data['certificate'], 'String');
        }

        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }

        if (data.hasOwnProperty('jwks_json_data')) {
          obj['jwks_json_data'] = _ApiClient["default"].convertToType(data['jwks_json_data'], 'String');
        }

        if (data.hasOwnProperty('jwks_uri')) {
          obj['jwks_uri'] = _ApiClient["default"].convertToType(data['jwks_uri'], 'String');
        }

        if (data.hasOwnProperty('unique_identifier')) {
          obj['unique_identifier'] = _ApiClient["default"].convertToType(data['unique_identifier'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>OAuth2AccessRules</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OAuth2AccessRules</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['audience'] && !(typeof data['audience'] === 'string' || data['audience'] instanceof String)) {
        throw new Error("Expected the field `audience` to be a primitive type in the JSON string but got " + data['audience']);
      } // ensure the json data is a string


      if (data['authorized_gw_cluster_name'] && !(typeof data['authorized_gw_cluster_name'] === 'string' || data['authorized_gw_cluster_name'] instanceof String)) {
        throw new Error("Expected the field `authorized_gw_cluster_name` to be a primitive type in the JSON string but got " + data['authorized_gw_cluster_name']);
      }

      if (data['bound_claims']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['bound_claims'])) {
          throw new Error("Expected the field `bound_claims` to be an array in the JSON data but got " + data['bound_claims']);
        } // validate the optional field `bound_claims` (array)


        var _iterator = _createForOfIteratorHelper(data['bound_claims']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _OAuth2CustomClaim["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is an array


      if (!Array.isArray(data['bound_clients_id'])) {
        throw new Error("Expected the field `bound_clients_id` to be an array in the JSON data but got " + data['bound_clients_id']);
      } // ensure the json data is a string


      if (data['certificate'] && !(typeof data['certificate'] === 'string' || data['certificate'] instanceof String)) {
        throw new Error("Expected the field `certificate` to be a primitive type in the JSON string but got " + data['certificate']);
      } // ensure the json data is a string


      if (data['issuer'] && !(typeof data['issuer'] === 'string' || data['issuer'] instanceof String)) {
        throw new Error("Expected the field `issuer` to be a primitive type in the JSON string but got " + data['issuer']);
      } // ensure the json data is a string


      if (data['jwks_json_data'] && !(typeof data['jwks_json_data'] === 'string' || data['jwks_json_data'] instanceof String)) {
        throw new Error("Expected the field `jwks_json_data` to be a primitive type in the JSON string but got " + data['jwks_json_data']);
      } // ensure the json data is a string


      if (data['jwks_uri'] && !(typeof data['jwks_uri'] === 'string' || data['jwks_uri'] instanceof String)) {
        throw new Error("Expected the field `jwks_uri` to be a primitive type in the JSON string but got " + data['jwks_uri']);
      } // ensure the json data is a string


      if (data['unique_identifier'] && !(typeof data['unique_identifier'] === 'string' || data['unique_identifier'] instanceof String)) {
        throw new Error("Expected the field `unique_identifier` to be a primitive type in the JSON string but got " + data['unique_identifier']);
      }

      return true;
    }
  }]);

  return OAuth2AccessRules;
}();
/**
 * The audience in the JWT.
 * @member {String} audience
 */


OAuth2AccessRules.prototype['audience'] = undefined;
/**
 * The gateway cluster name that is authorized to access JWKeySetURL
 * @member {String} authorized_gw_cluster_name
 */

OAuth2AccessRules.prototype['authorized_gw_cluster_name'] = undefined;
/**
 * The claims that login is restricted to.
 * @member {Array.<module:model/OAuth2CustomClaim>} bound_claims
 */

OAuth2AccessRules.prototype['bound_claims'] = undefined;
/**
 * The clients ids that login is restricted to.
 * @member {Array.<String>} bound_clients_id
 */

OAuth2AccessRules.prototype['bound_clients_id'] = undefined;
/**
 * Certificate to use when calling jwks_uri from the gateway. in PEM format
 * @member {String} certificate
 */

OAuth2AccessRules.prototype['certificate'] = undefined;
/**
 * Issuer URL
 * @member {String} issuer
 */

OAuth2AccessRules.prototype['issuer'] = undefined;
/**
 * The JSON Web Key Set (JWKS) that containing the public keys that should be used to verify any JSON Web Token (JWT) issued by the authorization server. base64 encoded string
 * @member {String} jwks_json_data
 */

OAuth2AccessRules.prototype['jwks_json_data'] = undefined;
/**
 * The URL to the JSON Web Key Set (JWKS) that containing the public keys that should be used to verify any JSON Web Token (JWT) issued by the authorization server.
 * @member {String} jwks_uri
 */

OAuth2AccessRules.prototype['jwks_uri'] = undefined;
/**
 * A unique identifier to distinguish different users
 * @member {String} unique_identifier
 */

OAuth2AccessRules.prototype['unique_identifier'] = undefined;
var _default = OAuth2AccessRules;
exports["default"] = _default;
