// akeyless/dist/model/OCIAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The OCIAccessRules model module.
 * @module model/OCIAccessRules
 * @version 5.0.14
 */
var OCIAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OCIAccessRules</code>.
   * OCIAccessRules contains access rules specific to Oracle cloud instance / user authentication
   * @alias module:model/OCIAccessRules
   */
  function OCIAccessRules() {
    _classCallCheck(this, OCIAccessRules);

    OCIAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(OCIAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OCIAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OCIAccessRules} obj Optional instance to populate.
     * @return {module:model/OCIAccessRules} The populated <code>OCIAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OCIAccessRules();

        if (data.hasOwnProperty('group_ocids')) {
          obj['group_ocids'] = _ApiClient["default"].convertToType(data['group_ocids'], ['String']);
        }

        if (data.hasOwnProperty('tenant_ocid')) {
          obj['tenant_ocid'] = _ApiClient["default"].convertToType(data['tenant_ocid'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>OCIAccessRules</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OCIAccessRules</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['group_ocids'])) {
        throw new Error("Expected the field `group_ocids` to be an array in the JSON data but got " + data['group_ocids']);
      } // ensure the json data is a string


      if (data['tenant_ocid'] && !(typeof data['tenant_ocid'] === 'string' || data['tenant_ocid'] instanceof String)) {
        throw new Error("Expected the field `tenant_ocid` to be a primitive type in the JSON string but got " + data['tenant_ocid']);
      }

      return true;
    }
  }]);

  return OCIAccessRules;
}();
/**
 * @member {Array.<String>} group_ocids
 */


OCIAccessRules.prototype['group_ocids'] = undefined;
/**
 * @member {String} tenant_ocid
 */

OCIAccessRules.prototype['tenant_ocid'] = undefined;
var _default = OCIAccessRules;
exports["default"] = _default;
