// akeyless/dist/model/OidcClientInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AccessOrGroupPermissionAssignment = _interopRequireDefault(require("./AccessOrGroupPermissionAssignment"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The OidcClientInfo model module.
 * @module model/OidcClientInfo
 * @version 5.0.6
 */
var OidcClientInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OidcClientInfo</code>.
   * @alias module:model/OidcClientInfo
   */
  function OidcClientInfo() {
    _classCallCheck(this, OidcClientInfo);

    OidcClientInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(OidcClientInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OidcClientInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OidcClientInfo} obj Optional instance to populate.
     * @return {module:model/OidcClientInfo} The populated <code>OidcClientInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OidcClientInfo();

        if (data.hasOwnProperty('access_permission_assignment')) {
          obj['access_permission_assignment'] = _ApiClient["default"].convertToType(data['access_permission_assignment'], [_AccessOrGroupPermissionAssignment["default"]]);
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], ['String']);
        }

        if (data.hasOwnProperty('client_id')) {
          obj['client_id'] = _ApiClient["default"].convertToType(data['client_id'], 'String');
        }

        if (data.hasOwnProperty('grant_types')) {
          obj['grant_types'] = _ApiClient["default"].convertToType(data['grant_types'], ['String']);
        }

        if (data.hasOwnProperty('issuer_url')) {
          obj['issuer_url'] = _ApiClient["default"].convertToType(data['issuer_url'], 'String');
        }

        if (data.hasOwnProperty('logout_uris')) {
          obj['logout_uris'] = _ApiClient["default"].convertToType(data['logout_uris'], ['String']);
        }

        if (data.hasOwnProperty('public')) {
          obj['public'] = _ApiClient["default"].convertToType(data['public'], 'Boolean');
        }

        if (data.hasOwnProperty('redirect_uris')) {
          obj['redirect_uris'] = _ApiClient["default"].convertToType(data['redirect_uris'], ['String']);
        }

        if (data.hasOwnProperty('response_types')) {
          obj['response_types'] = _ApiClient["default"].convertToType(data['response_types'], ['String']);
        }

        if (data.hasOwnProperty('scopes')) {
          obj['scopes'] = _ApiClient["default"].convertToType(data['scopes'], ['String']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>OidcClientInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>OidcClientInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['access_permission_assignment']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['access_permission_assignment'])) {
          throw new Error("Expected the field `access_permission_assignment` to be an array in the JSON data but got " + data['access_permission_assignment']);
        } // validate the optional field `access_permission_assignment` (array)


        var _iterator = _createForOfIteratorHelper(data['access_permission_assignment']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _AccessOrGroupPermissionAssignment["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is an array


      if (!Array.isArray(data['audience'])) {
        throw new Error("Expected the field `audience` to be an array in the JSON data but got " + data['audience']);
      } // ensure the json data is a string


      if (data['client_id'] && !(typeof data['client_id'] === 'string' || data['client_id'] instanceof String)) {
        throw new Error("Expected the field `client_id` to be a primitive type in the JSON string but got " + data['client_id']);
      } // ensure the json data is an array


      if (!Array.isArray(data['grant_types'])) {
        throw new Error("Expected the field `grant_types` to be an array in the JSON data but got " + data['grant_types']);
      } // ensure the json data is a string


      if (data['issuer_url'] && !(typeof data['issuer_url'] === 'string' || data['issuer_url'] instanceof String)) {
        throw new Error("Expected the field `issuer_url` to be a primitive type in the JSON string but got " + data['issuer_url']);
      } // ensure the json data is an array


      if (!Array.isArray(data['logout_uris'])) {
        throw new Error("Expected the field `logout_uris` to be an array in the JSON data but got " + data['logout_uris']);
      } // ensure the json data is an array


      if (!Array.isArray(data['redirect_uris'])) {
        throw new Error("Expected the field `redirect_uris` to be an array in the JSON data but got " + data['redirect_uris']);
      } // ensure the json data is an array


      if (!Array.isArray(data['response_types'])) {
        throw new Error("Expected the field `response_types` to be an array in the JSON data but got " + data['response_types']);
      } // ensure the json data is an array


      if (!Array.isArray(data['scopes'])) {
        throw new Error("Expected the field `scopes` to be an array in the JSON data but got " + data['scopes']);
      }

      return true;
    }
  }]);

  return OidcClientInfo;
}();
/**
 * @member {Array.<module:model/AccessOrGroupPermissionAssignment>} access_permission_assignment
 */


OidcClientInfo.prototype['access_permission_assignment'] = undefined;
/**
 * @member {Array.<String>} audience
 */

OidcClientInfo.prototype['audience'] = undefined;
/**
 * @member {String} client_id
 */

OidcClientInfo.prototype['client_id'] = undefined;
/**
 * @member {Array.<String>} grant_types
 */

OidcClientInfo.prototype['grant_types'] = undefined;
/**
 * @member {String} issuer_url
 */

OidcClientInfo.prototype['issuer_url'] = undefined;
/**
 * @member {Array.<String>} logout_uris
 */

OidcClientInfo.prototype['logout_uris'] = undefined;
/**
 * @member {Boolean} public
 */

OidcClientInfo.prototype['public'] = undefined;
/**
 * @member {Array.<String>} redirect_uris
 */

OidcClientInfo.prototype['redirect_uris'] = undefined;
/**
 * @member {Array.<String>} response_types
 */

OidcClientInfo.prototype['response_types'] = undefined;
/**
 * @member {Array.<String>} scopes
 */

OidcClientInfo.prototype['scopes'] = undefined;
var _default = OidcClientInfo;
exports["default"] = _default;
