// akeyless/dist/model/PasswordExpirationInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PasswordExpirationInfo model module.
 * @module model/PasswordExpirationInfo
 * @version 5.0.5
 */
var PasswordExpirationInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PasswordExpirationInfo</code>.
   * @alias module:model/PasswordExpirationInfo
   */
  function PasswordExpirationInfo() {
    _classCallCheck(this, PasswordExpirationInfo);

    PasswordExpirationInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PasswordExpirationInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PasswordExpirationInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PasswordExpirationInfo} obj Optional instance to populate.
     * @return {module:model/PasswordExpirationInfo} The populated <code>PasswordExpirationInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PasswordExpirationInfo();

        if (data.hasOwnProperty('days_before_expire')) {
          obj['days_before_expire'] = _ApiClient["default"].convertToType(data['days_before_expire'], 'Number');
        }

        if (data.hasOwnProperty('days_before_notification')) {
          obj['days_before_notification'] = _ApiClient["default"].convertToType(data['days_before_notification'], 'Number');
        }

        if (data.hasOwnProperty('days_until_expire')) {
          obj['days_until_expire'] = _ApiClient["default"].convertToType(data['days_until_expire'], 'Number');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>PasswordExpirationInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PasswordExpirationInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);

  return PasswordExpirationInfo;
}();
/**
 * @member {Number} days_before_expire
 */


PasswordExpirationInfo.prototype['days_before_expire'] = undefined;
/**
 * @member {Number} days_before_notification
 */

PasswordExpirationInfo.prototype['days_before_notification'] = undefined;
/**
 * r/o calculated parameter
 * @member {Number} days_until_expire
 */

PasswordExpirationInfo.prototype['days_until_expire'] = undefined;
/**
 * @member {Boolean} enable
 */

PasswordExpirationInfo.prototype['enable'] = undefined;
var _default = PasswordExpirationInfo;
exports["default"] = _default;
