// akeyless/dist/model/PasswordSecurityInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PasswordBreachInfo = _interopRequireDefault(require("./PasswordBreachInfo"));

var _PasswordScoreInfo = _interopRequireDefault(require("./PasswordScoreInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PasswordSecurityInfo model module.
 * @module model/PasswordSecurityInfo
 * @version 5.0.6
 */
var PasswordSecurityInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PasswordSecurityInfo</code>.
   * @alias module:model/PasswordSecurityInfo
   */
  function PasswordSecurityInfo() {
    _classCallCheck(this, PasswordSecurityInfo);

    PasswordSecurityInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PasswordSecurityInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PasswordSecurityInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PasswordSecurityInfo} obj Optional instance to populate.
     * @return {module:model/PasswordSecurityInfo} The populated <code>PasswordSecurityInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PasswordSecurityInfo();

        if (data.hasOwnProperty('breach_info')) {
          obj['breach_info'] = _PasswordBreachInfo["default"].constructFromObject(data['breach_info']);
        }

        if (data.hasOwnProperty('score_info')) {
          obj['score_info'] = _PasswordScoreInfo["default"].constructFromObject(data['score_info']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>PasswordSecurityInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PasswordSecurityInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `breach_info`
      if (data['breach_info']) {
        // data not null
        _PasswordBreachInfo["default"].validateJSON(data['breach_info']);
      } // validate the optional field `score_info`


      if (data['score_info']) {
        // data not null
        _PasswordScoreInfo["default"].validateJSON(data['score_info']);
      }

      return true;
    }
  }]);

  return PasswordSecurityInfo;
}();
/**
 * @member {module:model/PasswordBreachInfo} breach_info
 */


PasswordSecurityInfo.prototype['breach_info'] = undefined;
/**
 * @member {module:model/PasswordScoreInfo} score_info
 */

PasswordSecurityInfo.prototype['score_info'] = undefined;
var _default = PasswordSecurityInfo;
exports["default"] = _default;
