// akeyless/dist/model/PasswordScoreInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PasswordScoreInfo model module.
 * @module model/PasswordScoreInfo
 * @version 5.0.3
 */
var PasswordScoreInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PasswordScoreInfo</code>.
   * @alias module:model/PasswordScoreInfo
   */
  function PasswordScoreInfo() {
    _classCallCheck(this, PasswordScoreInfo);

    PasswordScoreInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PasswordScoreInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PasswordScoreInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PasswordScoreInfo} obj Optional instance to populate.
     * @return {module:model/PasswordScoreInfo} The populated <code>PasswordScoreInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PasswordScoreInfo();

        if (data.hasOwnProperty('score')) {
          obj['score'] = _ApiClient["default"].convertToType(data['score'], 'Number');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }

        if (data.hasOwnProperty('suggestions')) {
          obj['suggestions'] = _ApiClient["default"].convertToType(data['suggestions'], ['String']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>PasswordScoreInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PasswordScoreInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['status'] && !(typeof data['status'] === 'string' || data['status'] instanceof String)) {
        throw new Error("Expected the field `status` to be a primitive type in the JSON string but got " + data['status']);
      } // ensure the json data is an array


      if (!Array.isArray(data['suggestions'])) {
        throw new Error("Expected the field `suggestions` to be an array in the JSON data but got " + data['suggestions']);
      }

      return true;
    }
  }]);

  return PasswordScoreInfo;
}();
/**
 * @member {Number} score
 */


PasswordScoreInfo.prototype['score'] = undefined;
/**
 * @member {String} status
 */

PasswordScoreInfo.prototype['status'] = undefined;
/**
 * @member {Array.<String>} suggestions
 */

PasswordScoreInfo.prototype['suggestions'] = undefined;
var _default = PasswordScoreInfo;
exports["default"] = _default;
