// akeyless/dist/model/PingTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PingTargetDetails model module.
 * @module model/PingTargetDetails
 * @version 5.0.1
 */
var PingTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PingTargetDetails</code>.
   * PingTargetDetails
   * @alias module:model/PingTargetDetails
   */
  function PingTargetDetails() {
    _classCallCheck(this, PingTargetDetails);

    PingTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PingTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PingTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PingTargetDetails} obj Optional instance to populate.
     * @return {module:model/PingTargetDetails} The populated <code>PingTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PingTargetDetails();

        if (data.hasOwnProperty('administrative_port')) {
          obj['administrative_port'] = _ApiClient["default"].convertToType(data['administrative_port'], 'String');
        }

        if (data.hasOwnProperty('authorization_port')) {
          obj['authorization_port'] = _ApiClient["default"].convertToType(data['authorization_port'], 'String');
        }

        if (data.hasOwnProperty('ping_url')) {
          obj['ping_url'] = _ApiClient["default"].convertToType(data['ping_url'], 'String');
        }

        if (data.hasOwnProperty('privileged_user')) {
          obj['privileged_user'] = _ApiClient["default"].convertToType(data['privileged_user'], 'String');
        }

        if (data.hasOwnProperty('user_password')) {
          obj['user_password'] = _ApiClient["default"].convertToType(data['user_password'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>PingTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PingTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['administrative_port'] && !(typeof data['administrative_port'] === 'string' || data['administrative_port'] instanceof String)) {
        throw new Error("Expected the field `administrative_port` to be a primitive type in the JSON string but got " + data['administrative_port']);
      } // ensure the json data is a string


      if (data['authorization_port'] && !(typeof data['authorization_port'] === 'string' || data['authorization_port'] instanceof String)) {
        throw new Error("Expected the field `authorization_port` to be a primitive type in the JSON string but got " + data['authorization_port']);
      } // ensure the json data is a string


      if (data['ping_url'] && !(typeof data['ping_url'] === 'string' || data['ping_url'] instanceof String)) {
        throw new Error("Expected the field `ping_url` to be a primitive type in the JSON string but got " + data['ping_url']);
      } // ensure the json data is a string


      if (data['privileged_user'] && !(typeof data['privileged_user'] === 'string' || data['privileged_user'] instanceof String)) {
        throw new Error("Expected the field `privileged_user` to be a primitive type in the JSON string but got " + data['privileged_user']);
      } // ensure the json data is a string


      if (data['user_password'] && !(typeof data['user_password'] === 'string' || data['user_password'] instanceof String)) {
        throw new Error("Expected the field `user_password` to be a primitive type in the JSON string but got " + data['user_password']);
      }

      return true;
    }
  }]);

  return PingTargetDetails;
}();
/**
 * @member {String} administrative_port
 */


PingTargetDetails.prototype['administrative_port'] = undefined;
/**
 * @member {String} authorization_port
 */

PingTargetDetails.prototype['authorization_port'] = undefined;
/**
 * @member {String} ping_url
 */

PingTargetDetails.prototype['ping_url'] = undefined;
/**
 * @member {String} privileged_user
 */

PingTargetDetails.prototype['privileged_user'] = undefined;
/**
 * @member {String} user_password
 */

PingTargetDetails.prototype['user_password'] = undefined;
var _default = PingTargetDetails;
exports["default"] = _default;
