// akeyless/dist/model/ProvisionCertificateOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ProvisionCertificateOutput model module.
 * @module model/ProvisionCertificateOutput
 * @version 4.1.0
 */
var ProvisionCertificateOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProvisionCertificateOutput</code>.
   * @alias module:model/ProvisionCertificateOutput
   */
  function ProvisionCertificateOutput() {
    _classCallCheck(this, ProvisionCertificateOutput);

    ProvisionCertificateOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProvisionCertificateOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProvisionCertificateOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProvisionCertificateOutput} obj Optional instance to populate.
     * @return {module:model/ProvisionCertificateOutput} The populated <code>ProvisionCertificateOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProvisionCertificateOutput();

        if (data.hasOwnProperty('FailMessage')) {
          obj['FailMessage'] = _ApiClient["default"].convertToType(data['FailMessage'], 'String');
        }

        if (data.hasOwnProperty('SuccessMessage')) {
          obj['SuccessMessage'] = _ApiClient["default"].convertToType(data['SuccessMessage'], 'String');
        }

        if (data.hasOwnProperty('host_names')) {
          obj['host_names'] = _ApiClient["default"].convertToType(data['host_names'], ['String']);
        }

        if (data.hasOwnProperty('provisioned_at')) {
          obj['provisioned_at'] = _ApiClient["default"].convertToType(data['provisioned_at'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return ProvisionCertificateOutput;
}();
/**
 * @member {String} FailMessage
 */


ProvisionCertificateOutput.prototype['FailMessage'] = undefined;
/**
 * @member {String} SuccessMessage
 */

ProvisionCertificateOutput.prototype['SuccessMessage'] = undefined;
/**
 * @member {Array.<String>} host_names
 */

ProvisionCertificateOutput.prototype['host_names'] = undefined;
/**
 * @member {Date} provisioned_at
 */

ProvisionCertificateOutput.prototype['provisioned_at'] = undefined;
var _default = ProvisionCertificateOutput;
exports["default"] = _default;
