// akeyless/dist/model/RenewCertificateOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RenewCertificateOutput model module.
 * @module model/RenewCertificateOutput
 * @version 4.2.1
 */
var RenewCertificateOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RenewCertificateOutput</code>.
   * @alias module:model/RenewCertificateOutput
   */
  function RenewCertificateOutput() {
    _classCallCheck(this, RenewCertificateOutput);

    RenewCertificateOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RenewCertificateOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RenewCertificateOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RenewCertificateOutput} obj Optional instance to populate.
     * @return {module:model/RenewCertificateOutput} The populated <code>RenewCertificateOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RenewCertificateOutput();

        if (data.hasOwnProperty('cert')) {
          obj['cert'] = _ApiClient["default"].convertToType(data['cert'], 'String');
        }

        if (data.hasOwnProperty('cert_display_id')) {
          obj['cert_display_id'] = _ApiClient["default"].convertToType(data['cert_display_id'], 'String');
        }

        if (data.hasOwnProperty('item_id')) {
          obj['item_id'] = _ApiClient["default"].convertToType(data['item_id'], 'String');
        }

        if (data.hasOwnProperty('parent_cert')) {
          obj['parent_cert'] = _ApiClient["default"].convertToType(data['parent_cert'], 'String');
        }

        if (data.hasOwnProperty('private_key')) {
          obj['private_key'] = _ApiClient["default"].convertToType(data['private_key'], 'String');
        }

        if (data.hasOwnProperty('reading_token')) {
          obj['reading_token'] = _ApiClient["default"].convertToType(data['reading_token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RenewCertificateOutput;
}();
/**
 * @member {String} cert
 */


RenewCertificateOutput.prototype['cert'] = undefined;
/**
 * @member {String} cert_display_id
 */

RenewCertificateOutput.prototype['cert_display_id'] = undefined;
/**
 * @member {String} item_id
 */

RenewCertificateOutput.prototype['item_id'] = undefined;
/**
 * @member {String} parent_cert
 */

RenewCertificateOutput.prototype['parent_cert'] = undefined;
/**
 * @member {String} private_key
 */

RenewCertificateOutput.prototype['private_key'] = undefined;
/**
 * @member {String} reading_token
 */

RenewCertificateOutput.prototype['reading_token'] = undefined;
var _default = RenewCertificateOutput;
exports["default"] = _default;
