// akeyless/dist/model/RenewCertificate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RenewCertificate model module.
 * @module model/RenewCertificate
 * @version 4.2.1
 */
var RenewCertificate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RenewCertificate</code>.
   * @alias module:model/RenewCertificate
   */
  function RenewCertificate() {
    _classCallCheck(this, RenewCertificate);

    RenewCertificate.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RenewCertificate, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RenewCertificate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RenewCertificate} obj Optional instance to populate.
     * @return {module:model/RenewCertificate} The populated <code>RenewCertificate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RenewCertificate();

        if (data.hasOwnProperty('cert-issuer-name')) {
          obj['cert-issuer-name'] = _ApiClient["default"].convertToType(data['cert-issuer-name'], 'String');
        }

        if (data.hasOwnProperty('generate-key')) {
          obj['generate-key'] = _ApiClient["default"].convertToType(data['generate-key'], 'Boolean');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RenewCertificate;
}();
/**
 * The name of the PKI certificate issuer
 * @member {String} cert-issuer-name
 */


RenewCertificate.prototype['cert-issuer-name'] = undefined;
/**
 * Generate a new key as part of the certificate renewal
 * @member {Boolean} generate-key
 */

RenewCertificate.prototype['generate-key'] = undefined;
/**
 * Certificate item id
 * @member {Number} item-id
 */

RenewCertificate.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

RenewCertificate.prototype['json'] = false;
/**
 * Certificate name
 * @member {String} name
 */

RenewCertificate.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

RenewCertificate.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

RenewCertificate.prototype['uid-token'] = undefined;
var _default = RenewCertificate;
exports["default"] = _default;
