// akeyless/dist/model/RevokeCertificate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RevokeCertificate model module.
 * @module model/RevokeCertificate
 * @version 5.0.1
 */
var RevokeCertificate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RevokeCertificate</code>.
   * RevokeCertificate is a command that revokes a certificate and adds it to the crl
   * @alias module:model/RevokeCertificate
   */
  function RevokeCertificate() {
    _classCallCheck(this, RevokeCertificate);

    RevokeCertificate.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RevokeCertificate, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
    }
    /**
     * Constructs a <code>RevokeCertificate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RevokeCertificate} obj Optional instance to populate.
     * @return {module:model/RevokeCertificate} The populated <code>RevokeCertificate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RevokeCertificate();

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('serial-number')) {
          obj['serial-number'] = _ApiClient["default"].convertToType(data['serial-number'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>RevokeCertificate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>RevokeCertificate</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['serial-number'] && !(typeof data['serial-number'] === 'string' || data['serial-number'] instanceof String)) {
        throw new Error("Expected the field `serial-number` to be a primitive type in the JSON string but got " + data['serial-number']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return RevokeCertificate;
}();
/**
 * The item id of the certificate to revoke
 * @member {Number} item-id
 */


RevokeCertificate.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

RevokeCertificate.prototype['json'] = false;
/**
 * Certificate item name to revoke
 * @member {String} name
 */

RevokeCertificate.prototype['name'] = undefined;
/**
 * The serial number of the certificate to revoke
 * @member {String} serial-number
 */

RevokeCertificate.prototype['serial-number'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

RevokeCertificate.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

RevokeCertificate.prototype['uid-token'] = undefined;
/**
 * Certificate version to revoke. Required if item-id or name are used.
 * @member {Number} version
 */

RevokeCertificate.prototype['version'] = undefined;
var _default = RevokeCertificate;
exports["default"] = _default;
