// akeyless/dist/model/Role.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RoleAuthMethodAssociation = _interopRequireDefault(require("./RoleAuthMethodAssociation"));

var _Rules = _interopRequireDefault(require("./Rules"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Role model module.
 * @module model/Role
 * @version 5.0.14
 */
var Role = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Role</code>.
   * @alias module:model/Role
   */
  function Role() {
    _classCallCheck(this, Role);

    Role.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Role, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Role</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Role} obj Optional instance to populate.
     * @return {module:model/Role} The populated <code>Role</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Role();

        if (data.hasOwnProperty('access_date')) {
          obj['access_date'] = _ApiClient["default"].convertToType(data['access_date'], 'Date');
        }

        if (data.hasOwnProperty('access_date_display')) {
          obj['access_date_display'] = _ApiClient["default"].convertToType(data['access_date_display'], 'String');
        }

        if (data.hasOwnProperty('client_permissions')) {
          obj['client_permissions'] = _ApiClient["default"].convertToType(data['client_permissions'], ['String']);
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'Boolean');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('role_auth_methods_assoc')) {
          obj['role_auth_methods_assoc'] = _ApiClient["default"].convertToType(data['role_auth_methods_assoc'], [_RoleAuthMethodAssociation["default"]]);
        }

        if (data.hasOwnProperty('role_id')) {
          obj['role_id'] = _ApiClient["default"].convertToType(data['role_id'], 'Number');
        }

        if (data.hasOwnProperty('role_name')) {
          obj['role_name'] = _ApiClient["default"].convertToType(data['role_name'], 'String');
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _Rules["default"].constructFromObject(data['rules']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>Role</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Role</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_date_display'] && !(typeof data['access_date_display'] === 'string' || data['access_date_display'] instanceof String)) {
        throw new Error("Expected the field `access_date_display` to be a primitive type in the JSON string but got " + data['access_date_display']);
      } // ensure the json data is an array


      if (!Array.isArray(data['client_permissions'])) {
        throw new Error("Expected the field `client_permissions` to be an array in the JSON data but got " + data['client_permissions']);
      } // ensure the json data is a string


      if (data['comment'] && !(typeof data['comment'] === 'string' || data['comment'] instanceof String)) {
        throw new Error("Expected the field `comment` to be a primitive type in the JSON string but got " + data['comment']);
      }

      if (data['role_auth_methods_assoc']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['role_auth_methods_assoc'])) {
          throw new Error("Expected the field `role_auth_methods_assoc` to be an array in the JSON data but got " + data['role_auth_methods_assoc']);
        } // validate the optional field `role_auth_methods_assoc` (array)


        var _iterator = _createForOfIteratorHelper(data['role_auth_methods_assoc']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _RoleAuthMethodAssociation["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is a string


      if (data['role_name'] && !(typeof data['role_name'] === 'string' || data['role_name'] instanceof String)) {
        throw new Error("Expected the field `role_name` to be a primitive type in the JSON string but got " + data['role_name']);
      } // validate the optional field `rules`


      if (data['rules']) {
        // data not null
        _Rules["default"].validateJSON(data['rules']);
      }

      return true;
    }
  }]);

  return Role;
}();
/**
 * @member {Date} access_date
 */


Role.prototype['access_date'] = undefined;
/**
 * @member {String} access_date_display
 */

Role.prototype['access_date_display'] = undefined;
/**
 * @member {Array.<String>} client_permissions
 */

Role.prototype['client_permissions'] = undefined;
/**
 * @member {String} comment
 */

Role.prototype['comment'] = undefined;
/**
 * @member {Date} creation_date
 */

Role.prototype['creation_date'] = undefined;
/**
 * @member {Boolean} delete_protection
 */

Role.prototype['delete_protection'] = undefined;
/**
 * @member {Date} modification_date
 */

Role.prototype['modification_date'] = undefined;
/**
 * @member {Array.<module:model/RoleAuthMethodAssociation>} role_auth_methods_assoc
 */

Role.prototype['role_auth_methods_assoc'] = undefined;
/**
 * @member {Number} role_id
 */

Role.prototype['role_id'] = undefined;
/**
 * @member {String} role_name
 */

Role.prototype['role_name'] = undefined;
/**
 * @member {module:model/Rules} rules
 */

Role.prototype['rules'] = undefined;
var _default = Role;
exports["default"] = _default;
