// akeyless/dist/model/RotatedSecretUpdateAzure.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RotatedSecretUpdateAzure model module.
 * @module model/RotatedSecretUpdateAzure
 * @version 5.0.14
 */
var RotatedSecretUpdateAzure = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RotatedSecretUpdateAzure</code>.
   * @alias module:model/RotatedSecretUpdateAzure
   * @param name {String} Rotated secret name
   */
  function RotatedSecretUpdateAzure(name) {
    _classCallCheck(this, RotatedSecretUpdateAzure);

    RotatedSecretUpdateAzure.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RotatedSecretUpdateAzure, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['authentication-credentials'] = 'use-user-creds';
      obj['description'] = 'default_metadata';
      obj['explicitly-set-sa'] = 'false';
      obj['json'] = false;
      obj['name'] = name;
      obj['rotate-after-disconnect'] = 'false';
      obj['secure-access-web'] = false;
      obj['secure-access-web-browsing'] = false;
      obj['secure-access-web-proxy'] = false;
    }
    /**
     * Constructs a <code>RotatedSecretUpdateAzure</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RotatedSecretUpdateAzure} obj Optional instance to populate.
     * @return {module:model/RotatedSecretUpdateAzure} The populated <code>RotatedSecretUpdateAzure</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RotatedSecretUpdateAzure();

        if (data.hasOwnProperty('add-tag')) {
          obj['add-tag'] = _ApiClient["default"].convertToType(data['add-tag'], ['String']);
        }

        if (data.hasOwnProperty('api-id')) {
          obj['api-id'] = _ApiClient["default"].convertToType(data['api-id'], 'String');
        }

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('application-id')) {
          obj['application-id'] = _ApiClient["default"].convertToType(data['application-id'], 'String');
        }

        if (data.hasOwnProperty('authentication-credentials')) {
          obj['authentication-credentials'] = _ApiClient["default"].convertToType(data['authentication-credentials'], 'String');
        }

        if (data.hasOwnProperty('auto-rotate')) {
          obj['auto-rotate'] = _ApiClient["default"].convertToType(data['auto-rotate'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('explicitly-set-sa')) {
          obj['explicitly-set-sa'] = _ApiClient["default"].convertToType(data['explicitly-set-sa'], 'String');
        }

        if (data.hasOwnProperty('grace-rotation')) {
          obj['grace-rotation'] = _ApiClient["default"].convertToType(data['grace-rotation'], 'String');
        }

        if (data.hasOwnProperty('grace-rotation-hour')) {
          obj['grace-rotation-hour'] = _ApiClient["default"].convertToType(data['grace-rotation-hour'], 'Number');
        }

        if (data.hasOwnProperty('grace-rotation-interval')) {
          obj['grace-rotation-interval'] = _ApiClient["default"].convertToType(data['grace-rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('item-custom-fields')) {
          obj['item-custom-fields'] = _ApiClient["default"].convertToType(data['item-custom-fields'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password-length')) {
          obj['password-length'] = _ApiClient["default"].convertToType(data['password-length'], 'String');
        }

        if (data.hasOwnProperty('resource-group-name')) {
          obj['resource-group-name'] = _ApiClient["default"].convertToType(data['resource-group-name'], 'String');
        }

        if (data.hasOwnProperty('resource-name')) {
          obj['resource-name'] = _ApiClient["default"].convertToType(data['resource-name'], 'String');
        }

        if (data.hasOwnProperty('rm-tag')) {
          obj['rm-tag'] = _ApiClient["default"].convertToType(data['rm-tag'], ['String']);
        }

        if (data.hasOwnProperty('rotate-after-disconnect')) {
          obj['rotate-after-disconnect'] = _ApiClient["default"].convertToType(data['rotate-after-disconnect'], 'String');
        }

        if (data.hasOwnProperty('rotation-event-in')) {
          obj['rotation-event-in'] = _ApiClient["default"].convertToType(data['rotation-event-in'], ['String']);
        }

        if (data.hasOwnProperty('rotation-hour')) {
          obj['rotation-hour'] = _ApiClient["default"].convertToType(data['rotation-hour'], 'Number');
        }

        if (data.hasOwnProperty('rotation-interval')) {
          obj['rotation-interval'] = _ApiClient["default"].convertToType(data['rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('secure-access-disable-concurrent-connections')) {
          obj['secure-access-disable-concurrent-connections'] = _ApiClient["default"].convertToType(data['secure-access-disable-concurrent-connections'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-url')) {
          obj['secure-access-url'] = _ApiClient["default"].convertToType(data['secure-access-url'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-browsing')) {
          obj['secure-access-web-browsing'] = _ApiClient["default"].convertToType(data['secure-access-web-browsing'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-proxy')) {
          obj['secure-access-web-proxy'] = _ApiClient["default"].convertToType(data['secure-access-web-proxy'], 'Boolean');
        }

        if (data.hasOwnProperty('storage-account-key-name')) {
          obj['storage-account-key-name'] = _ApiClient["default"].convertToType(data['storage-account-key-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>RotatedSecretUpdateAzure</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>RotatedSecretUpdateAzure</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(RotatedSecretUpdateAzure.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is an array

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (!Array.isArray(data['add-tag'])) {
        throw new Error("Expected the field `add-tag` to be an array in the JSON data but got " + data['add-tag']);
      } // ensure the json data is a string


      if (data['api-id'] && !(typeof data['api-id'] === 'string' || data['api-id'] instanceof String)) {
        throw new Error("Expected the field `api-id` to be a primitive type in the JSON string but got " + data['api-id']);
      } // ensure the json data is a string


      if (data['api-key'] && !(typeof data['api-key'] === 'string' || data['api-key'] instanceof String)) {
        throw new Error("Expected the field `api-key` to be a primitive type in the JSON string but got " + data['api-key']);
      } // ensure the json data is a string


      if (data['application-id'] && !(typeof data['application-id'] === 'string' || data['application-id'] instanceof String)) {
        throw new Error("Expected the field `application-id` to be a primitive type in the JSON string but got " + data['application-id']);
      } // ensure the json data is a string


      if (data['authentication-credentials'] && !(typeof data['authentication-credentials'] === 'string' || data['authentication-credentials'] instanceof String)) {
        throw new Error("Expected the field `authentication-credentials` to be a primitive type in the JSON string but got " + data['authentication-credentials']);
      } // ensure the json data is a string


      if (data['auto-rotate'] && !(typeof data['auto-rotate'] === 'string' || data['auto-rotate'] instanceof String)) {
        throw new Error("Expected the field `auto-rotate` to be a primitive type in the JSON string but got " + data['auto-rotate']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['explicitly-set-sa'] && !(typeof data['explicitly-set-sa'] === 'string' || data['explicitly-set-sa'] instanceof String)) {
        throw new Error("Expected the field `explicitly-set-sa` to be a primitive type in the JSON string but got " + data['explicitly-set-sa']);
      } // ensure the json data is a string


      if (data['grace-rotation'] && !(typeof data['grace-rotation'] === 'string' || data['grace-rotation'] instanceof String)) {
        throw new Error("Expected the field `grace-rotation` to be a primitive type in the JSON string but got " + data['grace-rotation']);
      } // ensure the json data is a string


      if (data['grace-rotation-interval'] && !(typeof data['grace-rotation-interval'] === 'string' || data['grace-rotation-interval'] instanceof String)) {
        throw new Error("Expected the field `grace-rotation-interval` to be a primitive type in the JSON string but got " + data['grace-rotation-interval']);
      } // ensure the json data is a string


      if (data['keep-prev-version'] && !(typeof data['keep-prev-version'] === 'string' || data['keep-prev-version'] instanceof String)) {
        throw new Error("Expected the field `keep-prev-version` to be a primitive type in the JSON string but got " + data['keep-prev-version']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['password-length'] && !(typeof data['password-length'] === 'string' || data['password-length'] instanceof String)) {
        throw new Error("Expected the field `password-length` to be a primitive type in the JSON string but got " + data['password-length']);
      } // ensure the json data is a string


      if (data['resource-group-name'] && !(typeof data['resource-group-name'] === 'string' || data['resource-group-name'] instanceof String)) {
        throw new Error("Expected the field `resource-group-name` to be a primitive type in the JSON string but got " + data['resource-group-name']);
      } // ensure the json data is a string


      if (data['resource-name'] && !(typeof data['resource-name'] === 'string' || data['resource-name'] instanceof String)) {
        throw new Error("Expected the field `resource-name` to be a primitive type in the JSON string but got " + data['resource-name']);
      } // ensure the json data is an array


      if (!Array.isArray(data['rm-tag'])) {
        throw new Error("Expected the field `rm-tag` to be an array in the JSON data but got " + data['rm-tag']);
      } // ensure the json data is a string


      if (data['rotate-after-disconnect'] && !(typeof data['rotate-after-disconnect'] === 'string' || data['rotate-after-disconnect'] instanceof String)) {
        throw new Error("Expected the field `rotate-after-disconnect` to be a primitive type in the JSON string but got " + data['rotate-after-disconnect']);
      } // ensure the json data is an array


      if (!Array.isArray(data['rotation-event-in'])) {
        throw new Error("Expected the field `rotation-event-in` to be an array in the JSON data but got " + data['rotation-event-in']);
      } // ensure the json data is a string


      if (data['rotation-interval'] && !(typeof data['rotation-interval'] === 'string' || data['rotation-interval'] instanceof String)) {
        throw new Error("Expected the field `rotation-interval` to be a primitive type in the JSON string but got " + data['rotation-interval']);
      } // ensure the json data is a string


      if (data['secure-access-enable'] && !(typeof data['secure-access-enable'] === 'string' || data['secure-access-enable'] instanceof String)) {
        throw new Error("Expected the field `secure-access-enable` to be a primitive type in the JSON string but got " + data['secure-access-enable']);
      } // ensure the json data is a string


      if (data['secure-access-url'] && !(typeof data['secure-access-url'] === 'string' || data['secure-access-url'] instanceof String)) {
        throw new Error("Expected the field `secure-access-url` to be a primitive type in the JSON string but got " + data['secure-access-url']);
      } // ensure the json data is a string


      if (data['storage-account-key-name'] && !(typeof data['storage-account-key-name'] === 'string' || data['storage-account-key-name'] instanceof String)) {
        throw new Error("Expected the field `storage-account-key-name` to be a primitive type in the JSON string but got " + data['storage-account-key-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['username'] && !(typeof data['username'] === 'string' || data['username'] instanceof String)) {
        throw new Error("Expected the field `username` to be a primitive type in the JSON string but got " + data['username']);
      }

      return true;
    }
  }]);

  return RotatedSecretUpdateAzure;
}();

RotatedSecretUpdateAzure.RequiredProperties = ["name"];
/**
 * List of the new tags that will be attached to this item
 * @member {Array.<String>} add-tag
 */

RotatedSecretUpdateAzure.prototype['add-tag'] = undefined;
/**
 * API ID to rotate (relevant only for rotator-type=api-key)
 * @member {String} api-id
 */

RotatedSecretUpdateAzure.prototype['api-id'] = undefined;
/**
 * API key to rotate (relevant only for rotator-type=api-key)
 * @member {String} api-key
 */

RotatedSecretUpdateAzure.prototype['api-key'] = undefined;
/**
 * Id of the azure app that hold the serect to be rotated (relevant only for rotator-type=api-key & authentication-credentials=use-target-creds)
 * @member {String} application-id
 */

RotatedSecretUpdateAzure.prototype['application-id'] = undefined;
/**
 * The credentials to connect with use-user-creds/use-target-creds
 * @member {String} authentication-credentials
 * @default 'use-user-creds'
 */

RotatedSecretUpdateAzure.prototype['authentication-credentials'] = 'use-user-creds';
/**
 * Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation [true/false]
 * @member {String} auto-rotate
 */

RotatedSecretUpdateAzure.prototype['auto-rotate'] = undefined;
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

RotatedSecretUpdateAzure.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 * @default 'default_metadata'
 */

RotatedSecretUpdateAzure.prototype['description'] = 'default_metadata';
/**
 * If set, explicitly provide the storage account details [true/false]
 * @member {String} explicitly-set-sa
 * @default 'false'
 */

RotatedSecretUpdateAzure.prototype['explicitly-set-sa'] = 'false';
/**
 * Create a new access key without deleting the old key from AWS/Azure/GCP for backup (relevant only for AWS/Azure/GCP) [true/false]
 * @member {String} grace-rotation
 */

RotatedSecretUpdateAzure.prototype['grace-rotation'] = undefined;
/**
 * The Hour of the grace rotation in UTC
 * @member {Number} grace-rotation-hour
 */

RotatedSecretUpdateAzure.prototype['grace-rotation-hour'] = undefined;
/**
 * The number of days to wait before deleting the old key (must be bigger than rotation-interval)
 * @member {String} grace-rotation-interval
 */

RotatedSecretUpdateAzure.prototype['grace-rotation-interval'] = undefined;
/**
 * Additional custom fields to associate with the item
 * @member {Object.<String, String>} item-custom-fields
 */

RotatedSecretUpdateAzure.prototype['item-custom-fields'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

RotatedSecretUpdateAzure.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

RotatedSecretUpdateAzure.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

RotatedSecretUpdateAzure.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

RotatedSecretUpdateAzure.prototype['max-versions'] = undefined;
/**
 * Rotated secret name
 * @member {String} name
 */

RotatedSecretUpdateAzure.prototype['name'] = undefined;
/**
 * New item name
 * @member {String} new-name
 */

RotatedSecretUpdateAzure.prototype['new-name'] = undefined;
/**
 * The length of the password to be generated
 * @member {String} password-length
 */

RotatedSecretUpdateAzure.prototype['password-length'] = undefined;
/**
 * The resource group name (only relevant when explicitly-set-sa=true)
 * @member {String} resource-group-name
 */

RotatedSecretUpdateAzure.prototype['resource-group-name'] = undefined;
/**
 * The name of the storage account (only relevant when explicitly-set-sa=true)
 * @member {String} resource-name
 */

RotatedSecretUpdateAzure.prototype['resource-name'] = undefined;
/**
 * List of the existent tags that will be removed from this item
 * @member {Array.<String>} rm-tag
 */

RotatedSecretUpdateAzure.prototype['rm-tag'] = undefined;
/**
 * Rotate the value of the secret after SRA session ends [true/false]
 * @member {String} rotate-after-disconnect
 * @default 'false'
 */

RotatedSecretUpdateAzure.prototype['rotate-after-disconnect'] = 'false';
/**
 * How many days before the rotation of the item would you like to be notified
 * @member {Array.<String>} rotation-event-in
 */

RotatedSecretUpdateAzure.prototype['rotation-event-in'] = undefined;
/**
 * The Hour of the rotation in UTC
 * @member {Number} rotation-hour
 */

RotatedSecretUpdateAzure.prototype['rotation-hour'] = undefined;
/**
 * The number of days to wait between every automatic key rotation (1-365)
 * @member {String} rotation-interval
 */

RotatedSecretUpdateAzure.prototype['rotation-interval'] = undefined;
/**
 * Enable this flag to prevent simultaneous use of the same secret
 * @member {Boolean} secure-access-disable-concurrent-connections
 */

RotatedSecretUpdateAzure.prototype['secure-access-disable-concurrent-connections'] = undefined;
/**
 * Enable/Disable secure remote access [true/false]
 * @member {String} secure-access-enable
 */

RotatedSecretUpdateAzure.prototype['secure-access-enable'] = undefined;
/**
 * Destination URL to inject secrets
 * @member {String} secure-access-url
 */

RotatedSecretUpdateAzure.prototype['secure-access-url'] = undefined;
/**
 * Enable Web Secure Remote Access
 * @member {Boolean} secure-access-web
 * @default false
 */

RotatedSecretUpdateAzure.prototype['secure-access-web'] = false;
/**
 * Secure browser via Akeyless's Secure Remote Access (SRA)
 * @member {Boolean} secure-access-web-browsing
 * @default false
 */

RotatedSecretUpdateAzure.prototype['secure-access-web-browsing'] = false;
/**
 * Web-Proxy via Akeyless's Secure Remote Access (SRA)
 * @member {Boolean} secure-access-web-proxy
 * @default false
 */

RotatedSecretUpdateAzure.prototype['secure-access-web-proxy'] = false;
/**
 * The name of the storage account key to rotate [key1/key2/kerb1/kerb2] (relevat to azure-storage-account)
 * @member {String} storage-account-key-name
 */

RotatedSecretUpdateAzure.prototype['storage-account-key-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

RotatedSecretUpdateAzure.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

RotatedSecretUpdateAzure.prototype['uid-token'] = undefined;
/**
 * The user principal name to rotate his password (relevant only for rotator-type=password)
 * @member {String} username
 */

RotatedSecretUpdateAzure.prototype['username'] = undefined;
var _default = RotatedSecretUpdateAzure;
exports["default"] = _default;
