// akeyless/dist/model/RotatedSecretDetailsInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _WindowsService = _interopRequireDefault(require("./WindowsService"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RotatedSecretDetailsInfo model module.
 * @module model/RotatedSecretDetailsInfo
 * @version 5.0.0
 */
var RotatedSecretDetailsInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RotatedSecretDetailsInfo</code>.
   * RotatedSecretDetailsInfo The rotated secret rotator info
   * @alias module:model/RotatedSecretDetailsInfo
   */
  function RotatedSecretDetailsInfo() {
    _classCallCheck(this, RotatedSecretDetailsInfo);

    RotatedSecretDetailsInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RotatedSecretDetailsInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RotatedSecretDetailsInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RotatedSecretDetailsInfo} obj Optional instance to populate.
     * @return {module:model/RotatedSecretDetailsInfo} The populated <code>RotatedSecretDetailsInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RotatedSecretDetailsInfo();

        if (data.hasOwnProperty('delete_previous_version_in_days')) {
          obj['delete_previous_version_in_days'] = _ApiClient["default"].convertToType(data['delete_previous_version_in_days'], 'Number');
        }

        if (data.hasOwnProperty('grace_rotation')) {
          obj['grace_rotation'] = _ApiClient["default"].convertToType(data['grace_rotation'], 'Boolean');
        }

        if (data.hasOwnProperty('gw_cluster_id')) {
          obj['gw_cluster_id'] = _ApiClient["default"].convertToType(data['gw_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('last_rotation_error')) {
          obj['last_rotation_error'] = _ApiClient["default"].convertToType(data['last_rotation_error'], 'String');
        }

        if (data.hasOwnProperty('managed_by_akeyless')) {
          obj['managed_by_akeyless'] = _ApiClient["default"].convertToType(data['managed_by_akeyless'], 'Boolean');
        }

        if (data.hasOwnProperty('max_versions')) {
          obj['max_versions'] = _ApiClient["default"].convertToType(data['max_versions'], 'Number');
        }

        if (data.hasOwnProperty('number_of_versions_to_save')) {
          obj['number_of_versions_to_save'] = _ApiClient["default"].convertToType(data['number_of_versions_to_save'], 'Number');
        }

        if (data.hasOwnProperty('rotation_hour')) {
          obj['rotation_hour'] = _ApiClient["default"].convertToType(data['rotation_hour'], 'Number');
        }

        if (data.hasOwnProperty('rotation_interval_min')) {
          obj['rotation_interval_min'] = _ApiClient["default"].convertToType(data['rotation_interval_min'], 'Boolean');
        }

        if (data.hasOwnProperty('rotation_statement')) {
          obj['rotation_statement'] = _ApiClient["default"].convertToType(data['rotation_statement'], 'String');
        }

        if (data.hasOwnProperty('rotator_creds_type')) {
          obj['rotator_creds_type'] = _ApiClient["default"].convertToType(data['rotator_creds_type'], 'String');
        }

        if (data.hasOwnProperty('rotator_status')) {
          obj['rotator_status'] = _ApiClient["default"].convertToType(data['rotator_status'], 'String');
        }

        if (data.hasOwnProperty('rotator_type')) {
          obj['rotator_type'] = _ApiClient["default"].convertToType(data['rotator_type'], 'String');
        }

        if (data.hasOwnProperty('same_password')) {
          obj['same_password'] = _ApiClient["default"].convertToType(data['same_password'], 'Boolean');
        }

        if (data.hasOwnProperty('services_details')) {
          obj['services_details'] = _ApiClient["default"].convertToType(data['services_details'], [_WindowsService["default"]]);
        }

        if (data.hasOwnProperty('timeout_seconds')) {
          obj['timeout_seconds'] = _ApiClient["default"].convertToType(data['timeout_seconds'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>RotatedSecretDetailsInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>RotatedSecretDetailsInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['last_rotation_error'] && !(typeof data['last_rotation_error'] === 'string' || data['last_rotation_error'] instanceof String)) {
        throw new Error("Expected the field `last_rotation_error` to be a primitive type in the JSON string but got " + data['last_rotation_error']);
      } // ensure the json data is a string


      if (data['rotation_statement'] && !(typeof data['rotation_statement'] === 'string' || data['rotation_statement'] instanceof String)) {
        throw new Error("Expected the field `rotation_statement` to be a primitive type in the JSON string but got " + data['rotation_statement']);
      } // ensure the json data is a string


      if (data['rotator_creds_type'] && !(typeof data['rotator_creds_type'] === 'string' || data['rotator_creds_type'] instanceof String)) {
        throw new Error("Expected the field `rotator_creds_type` to be a primitive type in the JSON string but got " + data['rotator_creds_type']);
      } // ensure the json data is a string


      if (data['rotator_status'] && !(typeof data['rotator_status'] === 'string' || data['rotator_status'] instanceof String)) {
        throw new Error("Expected the field `rotator_status` to be a primitive type in the JSON string but got " + data['rotator_status']);
      } // ensure the json data is a string


      if (data['rotator_type'] && !(typeof data['rotator_type'] === 'string' || data['rotator_type'] instanceof String)) {
        throw new Error("Expected the field `rotator_type` to be a primitive type in the JSON string but got " + data['rotator_type']);
      }

      if (data['services_details']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['services_details'])) {
          throw new Error("Expected the field `services_details` to be an array in the JSON data but got " + data['services_details']);
        } // validate the optional field `services_details` (array)


        var _iterator = _createForOfIteratorHelper(data['services_details']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _WindowsService["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      }

      return true;
    }
  }]);

  return RotatedSecretDetailsInfo;
}();
/**
 * @member {Number} delete_previous_version_in_days
 */


RotatedSecretDetailsInfo.prototype['delete_previous_version_in_days'] = undefined;
/**
 * @member {Boolean} grace_rotation
 */

RotatedSecretDetailsInfo.prototype['grace_rotation'] = undefined;
/**
 * @member {Number} gw_cluster_id
 */

RotatedSecretDetailsInfo.prototype['gw_cluster_id'] = undefined;
/**
 * @member {String} last_rotation_error
 */

RotatedSecretDetailsInfo.prototype['last_rotation_error'] = undefined;
/**
 * @member {Boolean} managed_by_akeyless
 */

RotatedSecretDetailsInfo.prototype['managed_by_akeyless'] = undefined;
/**
 * @member {Number} max_versions
 */

RotatedSecretDetailsInfo.prototype['max_versions'] = undefined;
/**
 * @member {Number} number_of_versions_to_save
 */

RotatedSecretDetailsInfo.prototype['number_of_versions_to_save'] = undefined;
/**
 * @member {Number} rotation_hour
 */

RotatedSecretDetailsInfo.prototype['rotation_hour'] = undefined;
/**
 * @member {Boolean} rotation_interval_min
 */

RotatedSecretDetailsInfo.prototype['rotation_interval_min'] = undefined;
/**
 * @member {String} rotation_statement
 */

RotatedSecretDetailsInfo.prototype['rotation_statement'] = undefined;
/**
 * @member {String} rotator_creds_type
 */

RotatedSecretDetailsInfo.prototype['rotator_creds_type'] = undefined;
/**
 * RotationStatus defines types of rotation Status
 * @member {String} rotator_status
 */

RotatedSecretDetailsInfo.prototype['rotator_status'] = undefined;
/**
 * @member {String} rotator_type
 */

RotatedSecretDetailsInfo.prototype['rotator_type'] = undefined;
/**
 * @member {Boolean} same_password
 */

RotatedSecretDetailsInfo.prototype['same_password'] = undefined;
/**
 * @member {Array.<module:model/WindowsService>} services_details
 */

RotatedSecretDetailsInfo.prototype['services_details'] = undefined;
/**
 * @member {Number} timeout_seconds
 */

RotatedSecretDetailsInfo.prototype['timeout_seconds'] = undefined;
var _default = RotatedSecretDetailsInfo;
exports["default"] = _default;
