// akeyless/dist/model/RotatedSecretGetValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RotatedSecretGetValue model module.
 * @module model/RotatedSecretGetValue
 * @version 4.2.2
 */
var RotatedSecretGetValue = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RotatedSecretGetValue</code>.
   * @alias module:model/RotatedSecretGetValue
   * @param name {String} Secret name
   */
  function RotatedSecretGetValue(name) {
    _classCallCheck(this, RotatedSecretGetValue);

    RotatedSecretGetValue.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RotatedSecretGetValue, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>RotatedSecretGetValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RotatedSecretGetValue} obj Optional instance to populate.
     * @return {module:model/RotatedSecretGetValue} The populated <code>RotatedSecretGetValue</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RotatedSecretGetValue();

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('ignore-cache')) {
          obj['ignore-cache'] = _ApiClient["default"].convertToType(data['ignore-cache'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return RotatedSecretGetValue;
}();
/**
 * Get rotated secret value of specific Host (relevant only for Linked Target)
 * @member {String} host
 */


RotatedSecretGetValue.prototype['host'] = undefined;
/**
 * Retrieve the Secret value without checking the Gateway's cache [true/false]. This flag is only relevant when using the RestAPI
 * @member {String} ignore-cache
 * @default 'false'
 */

RotatedSecretGetValue.prototype['ignore-cache'] = 'false';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

RotatedSecretGetValue.prototype['json'] = false;
/**
 * Secret name
 * @member {String} name
 */

RotatedSecretGetValue.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

RotatedSecretGetValue.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

RotatedSecretGetValue.prototype['uid-token'] = undefined;
/**
 * Secret version
 * @member {Number} version
 */

RotatedSecretGetValue.prototype['version'] = undefined;
var _default = RotatedSecretGetValue;
exports["default"] = _default;
