// akeyless/dist/model/RotatedSecretCreateAws.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RotatedSecretCreateAws model module.
 * @module model/RotatedSecretCreateAws
 * @version 5.0.8
 */
var RotatedSecretCreateAws = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RotatedSecretCreateAws</code>.
   * @alias module:model/RotatedSecretCreateAws
   * @param name {String} Rotated secret name
   * @param rotatorType {String} The rotator type. options: [target/api-key]
   * @param targetName {String} Target name
   */
  function RotatedSecretCreateAws(name, rotatorType, targetName) {
    _classCallCheck(this, RotatedSecretCreateAws);

    RotatedSecretCreateAws.initialize(this, name, rotatorType, targetName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RotatedSecretCreateAws, null, [{
    key: "initialize",
    value: function initialize(obj, name, rotatorType, targetName) {
      obj['authentication-credentials'] = 'use-user-creds';
      obj['aws-region'] = 'us-east-2';
      obj['json'] = false;
      obj['name'] = name;
      obj['rotate-after-disconnect'] = 'false';
      obj['rotator-type'] = rotatorType;
      obj['target-name'] = targetName;
    }
    /**
     * Constructs a <code>RotatedSecretCreateAws</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RotatedSecretCreateAws} obj Optional instance to populate.
     * @return {module:model/RotatedSecretCreateAws} The populated <code>RotatedSecretCreateAws</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RotatedSecretCreateAws();

        if (data.hasOwnProperty('api-id')) {
          obj['api-id'] = _ApiClient["default"].convertToType(data['api-id'], 'String');
        }

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('authentication-credentials')) {
          obj['authentication-credentials'] = _ApiClient["default"].convertToType(data['authentication-credentials'], 'String');
        }

        if (data.hasOwnProperty('auto-rotate')) {
          obj['auto-rotate'] = _ApiClient["default"].convertToType(data['auto-rotate'], 'String');
        }

        if (data.hasOwnProperty('aws-region')) {
          obj['aws-region'] = _ApiClient["default"].convertToType(data['aws-region'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('grace-rotation')) {
          obj['grace-rotation'] = _ApiClient["default"].convertToType(data['grace-rotation'], 'String');
        }

        if (data.hasOwnProperty('grace-rotation-hour')) {
          obj['grace-rotation-hour'] = _ApiClient["default"].convertToType(data['grace-rotation-hour'], 'Number');
        }

        if (data.hasOwnProperty('grace-rotation-interval')) {
          obj['grace-rotation-interval'] = _ApiClient["default"].convertToType(data['grace-rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password-length')) {
          obj['password-length'] = _ApiClient["default"].convertToType(data['password-length'], 'String');
        }

        if (data.hasOwnProperty('rotate-after-disconnect')) {
          obj['rotate-after-disconnect'] = _ApiClient["default"].convertToType(data['rotate-after-disconnect'], 'String');
        }

        if (data.hasOwnProperty('rotation-event-in')) {
          obj['rotation-event-in'] = _ApiClient["default"].convertToType(data['rotation-event-in'], ['String']);
        }

        if (data.hasOwnProperty('rotation-hour')) {
          obj['rotation-hour'] = _ApiClient["default"].convertToType(data['rotation-hour'], 'Number');
        }

        if (data.hasOwnProperty('rotation-interval')) {
          obj['rotation-interval'] = _ApiClient["default"].convertToType(data['rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('rotator-type')) {
          obj['rotator-type'] = _ApiClient["default"].convertToType(data['rotator-type'], 'String');
        }

        if (data.hasOwnProperty('secure-access-aws-account-id')) {
          obj['secure-access-aws-account-id'] = _ApiClient["default"].convertToType(data['secure-access-aws-account-id'], 'String');
        }

        if (data.hasOwnProperty('secure-access-aws-native-cli')) {
          obj['secure-access-aws-native-cli'] = _ApiClient["default"].convertToType(data['secure-access-aws-native-cli'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-certificate-issuer')) {
          obj['secure-access-certificate-issuer'] = _ApiClient["default"].convertToType(data['secure-access-certificate-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>RotatedSecretCreateAws</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>RotatedSecretCreateAws</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(RotatedSecretCreateAws.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['api-id'] && !(typeof data['api-id'] === 'string' || data['api-id'] instanceof String)) {
        throw new Error("Expected the field `api-id` to be a primitive type in the JSON string but got " + data['api-id']);
      } // ensure the json data is a string


      if (data['api-key'] && !(typeof data['api-key'] === 'string' || data['api-key'] instanceof String)) {
        throw new Error("Expected the field `api-key` to be a primitive type in the JSON string but got " + data['api-key']);
      } // ensure the json data is a string


      if (data['authentication-credentials'] && !(typeof data['authentication-credentials'] === 'string' || data['authentication-credentials'] instanceof String)) {
        throw new Error("Expected the field `authentication-credentials` to be a primitive type in the JSON string but got " + data['authentication-credentials']);
      } // ensure the json data is a string


      if (data['auto-rotate'] && !(typeof data['auto-rotate'] === 'string' || data['auto-rotate'] instanceof String)) {
        throw new Error("Expected the field `auto-rotate` to be a primitive type in the JSON string but got " + data['auto-rotate']);
      } // ensure the json data is a string


      if (data['aws-region'] && !(typeof data['aws-region'] === 'string' || data['aws-region'] instanceof String)) {
        throw new Error("Expected the field `aws-region` to be a primitive type in the JSON string but got " + data['aws-region']);
      } // ensure the json data is a string


      if (data['delete_protection'] && !(typeof data['delete_protection'] === 'string' || data['delete_protection'] instanceof String)) {
        throw new Error("Expected the field `delete_protection` to be a primitive type in the JSON string but got " + data['delete_protection']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['grace-rotation'] && !(typeof data['grace-rotation'] === 'string' || data['grace-rotation'] instanceof String)) {
        throw new Error("Expected the field `grace-rotation` to be a primitive type in the JSON string but got " + data['grace-rotation']);
      } // ensure the json data is a string


      if (data['grace-rotation-interval'] && !(typeof data['grace-rotation-interval'] === 'string' || data['grace-rotation-interval'] instanceof String)) {
        throw new Error("Expected the field `grace-rotation-interval` to be a primitive type in the JSON string but got " + data['grace-rotation-interval']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['password-length'] && !(typeof data['password-length'] === 'string' || data['password-length'] instanceof String)) {
        throw new Error("Expected the field `password-length` to be a primitive type in the JSON string but got " + data['password-length']);
      } // ensure the json data is a string


      if (data['rotate-after-disconnect'] && !(typeof data['rotate-after-disconnect'] === 'string' || data['rotate-after-disconnect'] instanceof String)) {
        throw new Error("Expected the field `rotate-after-disconnect` to be a primitive type in the JSON string but got " + data['rotate-after-disconnect']);
      } // ensure the json data is an array


      if (!Array.isArray(data['rotation-event-in'])) {
        throw new Error("Expected the field `rotation-event-in` to be an array in the JSON data but got " + data['rotation-event-in']);
      } // ensure the json data is a string


      if (data['rotation-interval'] && !(typeof data['rotation-interval'] === 'string' || data['rotation-interval'] instanceof String)) {
        throw new Error("Expected the field `rotation-interval` to be a primitive type in the JSON string but got " + data['rotation-interval']);
      } // ensure the json data is a string


      if (data['rotator-type'] && !(typeof data['rotator-type'] === 'string' || data['rotator-type'] instanceof String)) {
        throw new Error("Expected the field `rotator-type` to be a primitive type in the JSON string but got " + data['rotator-type']);
      } // ensure the json data is a string


      if (data['secure-access-aws-account-id'] && !(typeof data['secure-access-aws-account-id'] === 'string' || data['secure-access-aws-account-id'] instanceof String)) {
        throw new Error("Expected the field `secure-access-aws-account-id` to be a primitive type in the JSON string but got " + data['secure-access-aws-account-id']);
      } // ensure the json data is a string


      if (data['secure-access-bastion-issuer'] && !(typeof data['secure-access-bastion-issuer'] === 'string' || data['secure-access-bastion-issuer'] instanceof String)) {
        throw new Error("Expected the field `secure-access-bastion-issuer` to be a primitive type in the JSON string but got " + data['secure-access-bastion-issuer']);
      } // ensure the json data is a string


      if (data['secure-access-certificate-issuer'] && !(typeof data['secure-access-certificate-issuer'] === 'string' || data['secure-access-certificate-issuer'] instanceof String)) {
        throw new Error("Expected the field `secure-access-certificate-issuer` to be a primitive type in the JSON string but got " + data['secure-access-certificate-issuer']);
      } // ensure the json data is a string


      if (data['secure-access-enable'] && !(typeof data['secure-access-enable'] === 'string' || data['secure-access-enable'] instanceof String)) {
        throw new Error("Expected the field `secure-access-enable` to be a primitive type in the JSON string but got " + data['secure-access-enable']);
      } // ensure the json data is an array


      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      } // ensure the json data is a string


      if (data['target-name'] && !(typeof data['target-name'] === 'string' || data['target-name'] instanceof String)) {
        throw new Error("Expected the field `target-name` to be a primitive type in the JSON string but got " + data['target-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return RotatedSecretCreateAws;
}();

RotatedSecretCreateAws.RequiredProperties = ["name", "rotator-type", "target-name"];
/**
 * API ID to rotate (relevant only for rotator-type=api-key)
 * @member {String} api-id
 */

RotatedSecretCreateAws.prototype['api-id'] = undefined;
/**
 * API key to rotate (relevant only for rotator-type=api-key)
 * @member {String} api-key
 */

RotatedSecretCreateAws.prototype['api-key'] = undefined;
/**
 * The credentials to connect with use-user-creds/use-target-creds
 * @member {String} authentication-credentials
 * @default 'use-user-creds'
 */

RotatedSecretCreateAws.prototype['authentication-credentials'] = 'use-user-creds';
/**
 * Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation [true/false]
 * @member {String} auto-rotate
 */

RotatedSecretCreateAws.prototype['auto-rotate'] = undefined;
/**
 * Aws Region
 * @member {String} aws-region
 * @default 'us-east-2'
 */

RotatedSecretCreateAws.prototype['aws-region'] = 'us-east-2';
/**
 * Protection from accidental deletion of this object [true/false]
 * @member {String} delete_protection
 */

RotatedSecretCreateAws.prototype['delete_protection'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

RotatedSecretCreateAws.prototype['description'] = undefined;
/**
 * Create a new access key without deleting the old key from AWS/Azure/GCP for backup (relevant only for AWS/Azure/GCP) [true/false]
 * @member {String} grace-rotation
 */

RotatedSecretCreateAws.prototype['grace-rotation'] = undefined;
/**
 * The Hour of the grace rotation in UTC
 * @member {Number} grace-rotation-hour
 */

RotatedSecretCreateAws.prototype['grace-rotation-hour'] = undefined;
/**
 * The number of days to wait before deleting the old key (must be bigger than rotation-interval)
 * @member {String} grace-rotation-interval
 */

RotatedSecretCreateAws.prototype['grace-rotation-interval'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

RotatedSecretCreateAws.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

RotatedSecretCreateAws.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

RotatedSecretCreateAws.prototype['max-versions'] = undefined;
/**
 * Rotated secret name
 * @member {String} name
 */

RotatedSecretCreateAws.prototype['name'] = undefined;
/**
 * The length of the password to be generated
 * @member {String} password-length
 */

RotatedSecretCreateAws.prototype['password-length'] = undefined;
/**
 * Rotate the value of the secret after SRA session ends [true/false]
 * @member {String} rotate-after-disconnect
 * @default 'false'
 */

RotatedSecretCreateAws.prototype['rotate-after-disconnect'] = 'false';
/**
 * How many days before the rotation of the item would you like to be notified
 * @member {Array.<String>} rotation-event-in
 */

RotatedSecretCreateAws.prototype['rotation-event-in'] = undefined;
/**
 * The Hour of the rotation in UTC
 * @member {Number} rotation-hour
 */

RotatedSecretCreateAws.prototype['rotation-hour'] = undefined;
/**
 * The number of days to wait between every automatic key rotation (1-365)
 * @member {String} rotation-interval
 */

RotatedSecretCreateAws.prototype['rotation-interval'] = undefined;
/**
 * The rotator type. options: [target/api-key]
 * @member {String} rotator-type
 */

RotatedSecretCreateAws.prototype['rotator-type'] = undefined;
/**
 * The AWS account id
 * @member {String} secure-access-aws-account-id
 */

RotatedSecretCreateAws.prototype['secure-access-aws-account-id'] = undefined;
/**
 * The AWS native cli
 * @member {Boolean} secure-access-aws-native-cli
 */

RotatedSecretCreateAws.prototype['secure-access-aws-native-cli'] = undefined;
/**
 * Deprecated. use secure-access-certificate-issuer
 * @member {String} secure-access-bastion-issuer
 */

RotatedSecretCreateAws.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * Path to the SSH Certificate Issuer for your Akeyless Secure Access
 * @member {String} secure-access-certificate-issuer
 */

RotatedSecretCreateAws.prototype['secure-access-certificate-issuer'] = undefined;
/**
 * Enable/Disable secure remote access [true/false]
 * @member {String} secure-access-enable
 */

RotatedSecretCreateAws.prototype['secure-access-enable'] = undefined;
/**
 * Add tags attached to this object
 * @member {Array.<String>} tags
 */

RotatedSecretCreateAws.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

RotatedSecretCreateAws.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

RotatedSecretCreateAws.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

RotatedSecretCreateAws.prototype['uid-token'] = undefined;
var _default = RotatedSecretCreateAws;
exports["default"] = _default;
