// akeyless/dist/model/SAMLAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SAMLAttribute = _interopRequireDefault(require("./SAMLAttribute"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SAMLAccessRules model module.
 * @module model/SAMLAccessRules
 * @version 5.0.1
 */
var SAMLAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SAMLAccessRules</code>.
   * @alias module:model/SAMLAccessRules
   */
  function SAMLAccessRules() {
    _classCallCheck(this, SAMLAccessRules);

    SAMLAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SAMLAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SAMLAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SAMLAccessRules} obj Optional instance to populate.
     * @return {module:model/SAMLAccessRules} The populated <code>SAMLAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SAMLAccessRules();

        if (data.hasOwnProperty('allowed_redirect_URIs')) {
          obj['allowed_redirect_URIs'] = _ApiClient["default"].convertToType(data['allowed_redirect_URIs'], ['String']);
        }

        if (data.hasOwnProperty('bound_attributes')) {
          obj['bound_attributes'] = _ApiClient["default"].convertToType(data['bound_attributes'], [_SAMLAttribute["default"]]);
        }

        if (data.hasOwnProperty('idp_metadata_url')) {
          obj['idp_metadata_url'] = _ApiClient["default"].convertToType(data['idp_metadata_url'], 'String');
        }

        if (data.hasOwnProperty('idp_metadata_xml')) {
          obj['idp_metadata_xml'] = _ApiClient["default"].convertToType(data['idp_metadata_xml'], 'String');
        }

        if (data.hasOwnProperty('unique_identifier')) {
          obj['unique_identifier'] = _ApiClient["default"].convertToType(data['unique_identifier'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SAMLAccessRules</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SAMLAccessRules</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['allowed_redirect_URIs'])) {
        throw new Error("Expected the field `allowed_redirect_URIs` to be an array in the JSON data but got " + data['allowed_redirect_URIs']);
      }

      if (data['bound_attributes']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['bound_attributes'])) {
          throw new Error("Expected the field `bound_attributes` to be an array in the JSON data but got " + data['bound_attributes']);
        } // validate the optional field `bound_attributes` (array)


        var _iterator = _createForOfIteratorHelper(data['bound_attributes']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _SAMLAttribute["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is a string


      if (data['idp_metadata_url'] && !(typeof data['idp_metadata_url'] === 'string' || data['idp_metadata_url'] instanceof String)) {
        throw new Error("Expected the field `idp_metadata_url` to be a primitive type in the JSON string but got " + data['idp_metadata_url']);
      } // ensure the json data is a string


      if (data['idp_metadata_xml'] && !(typeof data['idp_metadata_xml'] === 'string' || data['idp_metadata_xml'] instanceof String)) {
        throw new Error("Expected the field `idp_metadata_xml` to be a primitive type in the JSON string but got " + data['idp_metadata_xml']);
      } // ensure the json data is a string


      if (data['unique_identifier'] && !(typeof data['unique_identifier'] === 'string' || data['unique_identifier'] instanceof String)) {
        throw new Error("Expected the field `unique_identifier` to be a primitive type in the JSON string but got " + data['unique_identifier']);
      }

      return true;
    }
  }]);

  return SAMLAccessRules;
}();
/**
 * Allowed redirect URIs after the authentication
 * @member {Array.<String>} allowed_redirect_URIs
 */


SAMLAccessRules.prototype['allowed_redirect_URIs'] = undefined;
/**
 * The attributes that login is restricted to.
 * @member {Array.<module:model/SAMLAttribute>} bound_attributes
 */

SAMLAccessRules.prototype['bound_attributes'] = undefined;
/**
 * IDP metadata url
 * @member {String} idp_metadata_url
 */

SAMLAccessRules.prototype['idp_metadata_url'] = undefined;
/**
 * IDP metadata XML
 * @member {String} idp_metadata_xml
 */

SAMLAccessRules.prototype['idp_metadata_xml'] = undefined;
/**
 * A unique identifier to distinguish different users
 * @member {String} unique_identifier
 */

SAMLAccessRules.prototype['unique_identifier'] = undefined;
var _default = SAMLAccessRules;
exports["default"] = _default;
