// akeyless/dist/model/ScanResults.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _TargetError = _interopRequireDefault(require("./TargetError"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ScanResults model module.
 * @module model/ScanResults
 * @version 5.0.20
 */
var ScanResults = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ScanResults</code>.
   * ScanResults contains detailed results from a certificate scan
   * @alias module:model/ScanResults
   */
  function ScanResults() {
    _classCallCheck(this, ScanResults);

    ScanResults.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ScanResults, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ScanResults</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ScanResults} obj Optional instance to populate.
     * @return {module:model/ScanResults} The populated <code>ScanResults</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ScanResults();

        if (data.hasOwnProperty('CountExisting')) {
          obj['CountExisting'] = _ApiClient["default"].convertToType(data['CountExisting'], 'Number');
        }

        if (data.hasOwnProperty('CountFailed')) {
          obj['CountFailed'] = _ApiClient["default"].convertToType(data['CountFailed'], 'Number');
        }

        if (data.hasOwnProperty('CountHosts')) {
          obj['CountHosts'] = _ApiClient["default"].convertToType(data['CountHosts'], 'Number');
        }

        if (data.hasOwnProperty('CountNew')) {
          obj['CountNew'] = _ApiClient["default"].convertToType(data['CountNew'], 'Number');
        }

        if (data.hasOwnProperty('CountSubdomains')) {
          obj['CountSubdomains'] = _ApiClient["default"].convertToType(data['CountSubdomains'], 'Number');
        }

        if (data.hasOwnProperty('Error')) {
          obj['Error'] = _ApiClient["default"].convertToType(data['Error'], 'String');
        }

        if (data.hasOwnProperty('FailedTargets')) {
          obj['FailedTargets'] = _ApiClient["default"].convertToType(data['FailedTargets'], [_TargetError["default"]]);
        }

        if (data.hasOwnProperty('ItemNames')) {
          obj['ItemNames'] = _ApiClient["default"].convertToType(data['ItemNames'], ['String']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ScanResults</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ScanResults</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['Error'] && !(typeof data['Error'] === 'string' || data['Error'] instanceof String)) {
        throw new Error("Expected the field `Error` to be a primitive type in the JSON string but got " + data['Error']);
      }

      if (data['FailedTargets']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['FailedTargets'])) {
          throw new Error("Expected the field `FailedTargets` to be an array in the JSON data but got " + data['FailedTargets']);
        } // validate the optional field `FailedTargets` (array)


        var _iterator = _createForOfIteratorHelper(data['FailedTargets']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _TargetError["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is an array


      if (!Array.isArray(data['ItemNames'])) {
        throw new Error("Expected the field `ItemNames` to be an array in the JSON data but got " + data['ItemNames']);
      }

      return true;
    }
  }]);

  return ScanResults;
}();
/**
 * @member {Number} CountExisting
 */


ScanResults.prototype['CountExisting'] = undefined;
/**
 * @member {Number} CountFailed
 */

ScanResults.prototype['CountFailed'] = undefined;
/**
 * @member {Number} CountHosts
 */

ScanResults.prototype['CountHosts'] = undefined;
/**
 * @member {Number} CountNew
 */

ScanResults.prototype['CountNew'] = undefined;
/**
 * @member {Number} CountSubdomains
 */

ScanResults.prototype['CountSubdomains'] = undefined;
/**
 * @member {String} Error
 */

ScanResults.prototype['Error'] = undefined;
/**
 * @member {Array.<module:model/TargetError>} FailedTargets
 */

ScanResults.prototype['FailedTargets'] = undefined;
/**
 * @member {Array.<String>} ItemNames
 */

ScanResults.prototype['ItemNames'] = undefined;
var _default = ScanResults;
exports["default"] = _default;
