// akeyless/dist/model/SecureRemoteAccess.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ItemSraStatus = _interopRequireDefault(require("./ItemSraStatus"));

var _TargetNameWithHosts = _interopRequireDefault(require("./TargetNameWithHosts"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SecureRemoteAccess model module.
 * @module model/SecureRemoteAccess
 * @version 5.0.1
 */
var SecureRemoteAccess = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SecureRemoteAccess</code>.
   * @alias module:model/SecureRemoteAccess
   */
  function SecureRemoteAccess() {
    _classCallCheck(this, SecureRemoteAccess);

    SecureRemoteAccess.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SecureRemoteAccess, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SecureRemoteAccess</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SecureRemoteAccess} obj Optional instance to populate.
     * @return {module:model/SecureRemoteAccess} The populated <code>SecureRemoteAccess</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SecureRemoteAccess();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('allow_port_forwarding')) {
          obj['allow_port_forwarding'] = _ApiClient["default"].convertToType(data['allow_port_forwarding'], 'Boolean');
        }

        if (data.hasOwnProperty('allow_providing_external_username')) {
          obj['allow_providing_external_username'] = _ApiClient["default"].convertToType(data['allow_providing_external_username'], 'Boolean');
        }

        if (data.hasOwnProperty('bastion_api')) {
          obj['bastion_api'] = _ApiClient["default"].convertToType(data['bastion_api'], 'String');
        }

        if (data.hasOwnProperty('bastion_issuer')) {
          obj['bastion_issuer'] = _ApiClient["default"].convertToType(data['bastion_issuer'], 'String');
        }

        if (data.hasOwnProperty('bastion_issuer_id')) {
          obj['bastion_issuer_id'] = _ApiClient["default"].convertToType(data['bastion_issuer_id'], 'Number');
        }

        if (data.hasOwnProperty('bastion_ssh')) {
          obj['bastion_ssh'] = _ApiClient["default"].convertToType(data['bastion_ssh'], 'String');
        }

        if (data.hasOwnProperty('block_concurrent_connections')) {
          obj['block_concurrent_connections'] = _ApiClient["default"].convertToType(data['block_concurrent_connections'], 'Boolean');
        }

        if (data.hasOwnProperty('block_concurrent_connections_level')) {
          obj['block_concurrent_connections_level'] = _ApiClient["default"].convertToType(data['block_concurrent_connections_level'], 'String');
        }

        if (data.hasOwnProperty('category')) {
          obj['category'] = _ApiClient["default"].convertToType(data['category'], 'String');
        }

        if (data.hasOwnProperty('connection_delay_seconds')) {
          obj['connection_delay_seconds'] = _ApiClient["default"].convertToType(data['connection_delay_seconds'], 'Number');
        }

        if (data.hasOwnProperty('dashboard_url')) {
          obj['dashboard_url'] = _ApiClient["default"].convertToType(data['dashboard_url'], 'String');
        }

        if (data.hasOwnProperty('db_name')) {
          obj['db_name'] = _ApiClient["default"].convertToType(data['db_name'], 'String');
        }

        if (data.hasOwnProperty('domain')) {
          obj['domain'] = _ApiClient["default"].convertToType(data['domain'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'Boolean');
        }

        if (data.hasOwnProperty('endpoint')) {
          obj['endpoint'] = _ApiClient["default"].convertToType(data['endpoint'], 'String');
        }

        if (data.hasOwnProperty('enforce_hosts_restriction')) {
          obj['enforce_hosts_restriction'] = _ApiClient["default"].convertToType(data['enforce_hosts_restriction'], 'Boolean');
        }

        if (data.hasOwnProperty('gw_cluster_id')) {
          obj['gw_cluster_id'] = _ApiClient["default"].convertToType(data['gw_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], ['String']);
        }

        if (data.hasOwnProperty('host_provider_type')) {
          obj['host_provider_type'] = _ApiClient["default"].convertToType(data['host_provider_type'], 'String');
        }

        if (data.hasOwnProperty('is_cli')) {
          obj['is_cli'] = _ApiClient["default"].convertToType(data['is_cli'], 'Boolean');
        }

        if (data.hasOwnProperty('is_web')) {
          obj['is_web'] = _ApiClient["default"].convertToType(data['is_web'], 'Boolean');
        }

        if (data.hasOwnProperty('isolated')) {
          obj['isolated'] = _ApiClient["default"].convertToType(data['isolated'], 'Boolean');
        }

        if (data.hasOwnProperty('native')) {
          obj['native'] = _ApiClient["default"].convertToType(data['native'], 'Boolean');
        }

        if (data.hasOwnProperty('rd_gateway_server')) {
          obj['rd_gateway_server'] = _ApiClient["default"].convertToType(data['rd_gateway_server'], 'String');
        }

        if (data.hasOwnProperty('rdp_user')) {
          obj['rdp_user'] = _ApiClient["default"].convertToType(data['rdp_user'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('rotate_after_disconnect')) {
          obj['rotate_after_disconnect'] = _ApiClient["default"].convertToType(data['rotate_after_disconnect'], 'Boolean');
        }

        if (data.hasOwnProperty('schema')) {
          obj['schema'] = _ApiClient["default"].convertToType(data['schema'], 'String');
        }

        if (data.hasOwnProperty('ssh_password')) {
          obj['ssh_password'] = _ApiClient["default"].convertToType(data['ssh_password'], 'Boolean');
        }

        if (data.hasOwnProperty('ssh_private_key')) {
          obj['ssh_private_key'] = _ApiClient["default"].convertToType(data['ssh_private_key'], 'Boolean');
        }

        if (data.hasOwnProperty('ssh_user')) {
          obj['ssh_user'] = _ApiClient["default"].convertToType(data['ssh_user'], 'String');
        }

        if (data.hasOwnProperty('status_info')) {
          obj['status_info'] = _ItemSraStatus["default"].constructFromObject(data['status_info']);
        }

        if (data.hasOwnProperty('target_hosts')) {
          obj['target_hosts'] = _ApiClient["default"].convertToType(data['target_hosts'], [_TargetNameWithHosts["default"]]);
        }

        if (data.hasOwnProperty('targets')) {
          obj['targets'] = _ApiClient["default"].convertToType(data['targets'], ['String']);
        }

        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }

        if (data.hasOwnProperty('use_internal_bastion')) {
          obj['use_internal_bastion'] = _ApiClient["default"].convertToType(data['use_internal_bastion'], 'Boolean');
        }

        if (data.hasOwnProperty('web_proxy')) {
          obj['web_proxy'] = _ApiClient["default"].convertToType(data['web_proxy'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SecureRemoteAccess</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SecureRemoteAccess</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['account_id'] && !(typeof data['account_id'] === 'string' || data['account_id'] instanceof String)) {
        throw new Error("Expected the field `account_id` to be a primitive type in the JSON string but got " + data['account_id']);
      } // ensure the json data is a string


      if (data['bastion_api'] && !(typeof data['bastion_api'] === 'string' || data['bastion_api'] instanceof String)) {
        throw new Error("Expected the field `bastion_api` to be a primitive type in the JSON string but got " + data['bastion_api']);
      } // ensure the json data is a string


      if (data['bastion_issuer'] && !(typeof data['bastion_issuer'] === 'string' || data['bastion_issuer'] instanceof String)) {
        throw new Error("Expected the field `bastion_issuer` to be a primitive type in the JSON string but got " + data['bastion_issuer']);
      } // ensure the json data is a string


      if (data['bastion_ssh'] && !(typeof data['bastion_ssh'] === 'string' || data['bastion_ssh'] instanceof String)) {
        throw new Error("Expected the field `bastion_ssh` to be a primitive type in the JSON string but got " + data['bastion_ssh']);
      } // ensure the json data is a string


      if (data['block_concurrent_connections_level'] && !(typeof data['block_concurrent_connections_level'] === 'string' || data['block_concurrent_connections_level'] instanceof String)) {
        throw new Error("Expected the field `block_concurrent_connections_level` to be a primitive type in the JSON string but got " + data['block_concurrent_connections_level']);
      } // ensure the json data is a string


      if (data['category'] && !(typeof data['category'] === 'string' || data['category'] instanceof String)) {
        throw new Error("Expected the field `category` to be a primitive type in the JSON string but got " + data['category']);
      } // ensure the json data is a string


      if (data['dashboard_url'] && !(typeof data['dashboard_url'] === 'string' || data['dashboard_url'] instanceof String)) {
        throw new Error("Expected the field `dashboard_url` to be a primitive type in the JSON string but got " + data['dashboard_url']);
      } // ensure the json data is a string


      if (data['db_name'] && !(typeof data['db_name'] === 'string' || data['db_name'] instanceof String)) {
        throw new Error("Expected the field `db_name` to be a primitive type in the JSON string but got " + data['db_name']);
      } // ensure the json data is a string


      if (data['domain'] && !(typeof data['domain'] === 'string' || data['domain'] instanceof String)) {
        throw new Error("Expected the field `domain` to be a primitive type in the JSON string but got " + data['domain']);
      } // ensure the json data is a string


      if (data['endpoint'] && !(typeof data['endpoint'] === 'string' || data['endpoint'] instanceof String)) {
        throw new Error("Expected the field `endpoint` to be a primitive type in the JSON string but got " + data['endpoint']);
      } // ensure the json data is an array


      if (!Array.isArray(data['host'])) {
        throw new Error("Expected the field `host` to be an array in the JSON data but got " + data['host']);
      } // ensure the json data is a string


      if (data['host_provider_type'] && !(typeof data['host_provider_type'] === 'string' || data['host_provider_type'] instanceof String)) {
        throw new Error("Expected the field `host_provider_type` to be a primitive type in the JSON string but got " + data['host_provider_type']);
      } // ensure the json data is a string


      if (data['rd_gateway_server'] && !(typeof data['rd_gateway_server'] === 'string' || data['rd_gateway_server'] instanceof String)) {
        throw new Error("Expected the field `rd_gateway_server` to be a primitive type in the JSON string but got " + data['rd_gateway_server']);
      } // ensure the json data is a string


      if (data['rdp_user'] && !(typeof data['rdp_user'] === 'string' || data['rdp_user'] instanceof String)) {
        throw new Error("Expected the field `rdp_user` to be a primitive type in the JSON string but got " + data['rdp_user']);
      } // ensure the json data is a string


      if (data['region'] && !(typeof data['region'] === 'string' || data['region'] instanceof String)) {
        throw new Error("Expected the field `region` to be a primitive type in the JSON string but got " + data['region']);
      } // ensure the json data is a string


      if (data['schema'] && !(typeof data['schema'] === 'string' || data['schema'] instanceof String)) {
        throw new Error("Expected the field `schema` to be a primitive type in the JSON string but got " + data['schema']);
      } // ensure the json data is a string


      if (data['ssh_user'] && !(typeof data['ssh_user'] === 'string' || data['ssh_user'] instanceof String)) {
        throw new Error("Expected the field `ssh_user` to be a primitive type in the JSON string but got " + data['ssh_user']);
      } // validate the optional field `status_info`


      if (data['status_info']) {
        // data not null
        _ItemSraStatus["default"].validateJSON(data['status_info']);
      }

      if (data['target_hosts']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['target_hosts'])) {
          throw new Error("Expected the field `target_hosts` to be an array in the JSON data but got " + data['target_hosts']);
        } // validate the optional field `target_hosts` (array)


        var _iterator = _createForOfIteratorHelper(data['target_hosts']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _TargetNameWithHosts["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is an array


      if (!Array.isArray(data['targets'])) {
        throw new Error("Expected the field `targets` to be an array in the JSON data but got " + data['targets']);
      } // ensure the json data is a string


      if (data['url'] && !(typeof data['url'] === 'string' || data['url'] instanceof String)) {
        throw new Error("Expected the field `url` to be a primitive type in the JSON string but got " + data['url']);
      }

      return true;
    }
  }]);

  return SecureRemoteAccess;
}();
/**
 * @member {String} account_id
 */


SecureRemoteAccess.prototype['account_id'] = undefined;
/**
 * @member {Boolean} allow_port_forwarding
 */

SecureRemoteAccess.prototype['allow_port_forwarding'] = undefined;
/**
 * @member {Boolean} allow_providing_external_username
 */

SecureRemoteAccess.prototype['allow_providing_external_username'] = undefined;
/**
 * @member {String} bastion_api
 */

SecureRemoteAccess.prototype['bastion_api'] = undefined;
/**
 * @member {String} bastion_issuer
 */

SecureRemoteAccess.prototype['bastion_issuer'] = undefined;
/**
 * @member {Number} bastion_issuer_id
 */

SecureRemoteAccess.prototype['bastion_issuer_id'] = undefined;
/**
 * @member {String} bastion_ssh
 */

SecureRemoteAccess.prototype['bastion_ssh'] = undefined;
/**
 * @member {Boolean} block_concurrent_connections
 */

SecureRemoteAccess.prototype['block_concurrent_connections'] = undefined;
/**
 * @member {String} block_concurrent_connections_level
 */

SecureRemoteAccess.prototype['block_concurrent_connections_level'] = undefined;
/**
 * @member {String} category
 */

SecureRemoteAccess.prototype['category'] = undefined;
/**
 * @member {Number} connection_delay_seconds
 */

SecureRemoteAccess.prototype['connection_delay_seconds'] = undefined;
/**
 * @member {String} dashboard_url
 */

SecureRemoteAccess.prototype['dashboard_url'] = undefined;
/**
 * @member {String} db_name
 */

SecureRemoteAccess.prototype['db_name'] = undefined;
/**
 * @member {String} domain
 */

SecureRemoteAccess.prototype['domain'] = undefined;
/**
 * @member {Boolean} enable
 */

SecureRemoteAccess.prototype['enable'] = undefined;
/**
 * @member {String} endpoint
 */

SecureRemoteAccess.prototype['endpoint'] = undefined;
/**
 * @member {Boolean} enforce_hosts_restriction
 */

SecureRemoteAccess.prototype['enforce_hosts_restriction'] = undefined;
/**
 * @member {Number} gw_cluster_id
 */

SecureRemoteAccess.prototype['gw_cluster_id'] = undefined;
/**
 * @member {Array.<String>} host
 */

SecureRemoteAccess.prototype['host'] = undefined;
/**
 * @member {String} host_provider_type
 */

SecureRemoteAccess.prototype['host_provider_type'] = undefined;
/**
 * @member {Boolean} is_cli
 */

SecureRemoteAccess.prototype['is_cli'] = undefined;
/**
 * @member {Boolean} is_web
 */

SecureRemoteAccess.prototype['is_web'] = undefined;
/**
 * @member {Boolean} isolated
 */

SecureRemoteAccess.prototype['isolated'] = undefined;
/**
 * @member {Boolean} native
 */

SecureRemoteAccess.prototype['native'] = undefined;
/**
 * @member {String} rd_gateway_server
 */

SecureRemoteAccess.prototype['rd_gateway_server'] = undefined;
/**
 * @member {String} rdp_user
 */

SecureRemoteAccess.prototype['rdp_user'] = undefined;
/**
 * @member {String} region
 */

SecureRemoteAccess.prototype['region'] = undefined;
/**
 * @member {Boolean} rotate_after_disconnect
 */

SecureRemoteAccess.prototype['rotate_after_disconnect'] = undefined;
/**
 * @member {String} schema
 */

SecureRemoteAccess.prototype['schema'] = undefined;
/**
 * @member {Boolean} ssh_password
 */

SecureRemoteAccess.prototype['ssh_password'] = undefined;
/**
 * @member {Boolean} ssh_private_key
 */

SecureRemoteAccess.prototype['ssh_private_key'] = undefined;
/**
 * @member {String} ssh_user
 */

SecureRemoteAccess.prototype['ssh_user'] = undefined;
/**
 * @member {module:model/ItemSraStatus} status_info
 */

SecureRemoteAccess.prototype['status_info'] = undefined;
/**
 * @member {Array.<module:model/TargetNameWithHosts>} target_hosts
 */

SecureRemoteAccess.prototype['target_hosts'] = undefined;
/**
 * @member {Array.<String>} targets
 */

SecureRemoteAccess.prototype['targets'] = undefined;
/**
 * @member {String} url
 */

SecureRemoteAccess.prototype['url'] = undefined;
/**
 * @member {Boolean} use_internal_bastion
 */

SecureRemoteAccess.prototype['use_internal_bastion'] = undefined;
/**
 * @member {Boolean} web_proxy
 */

SecureRemoteAccess.prototype['web_proxy'] = undefined;
var _default = SecureRemoteAccess;
exports["default"] = _default;
