// akeyless/dist/model/ShareItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ShareItem model module.
 * @module model/ShareItem
 * @version 5.0.0
 */
var ShareItem = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ShareItem</code>.
   * @alias module:model/ShareItem
   * @param action {String} Action to be performed on the item [start/stop/describe]
   * @param itemName {String} Item name
   */
  function ShareItem(action, itemName) {
    _classCallCheck(this, ShareItem);

    ShareItem.initialize(this, action, itemName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ShareItem, null, [{
    key: "initialize",
    value: function initialize(obj, action, itemName) {
      obj['accessibility'] = 'regular';
      obj['action'] = action;
      obj['item-name'] = itemName;
      obj['json'] = false;
      obj['share-type'] = 'email';
      obj['view-once'] = false;
    }
    /**
     * Constructs a <code>ShareItem</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ShareItem} obj Optional instance to populate.
     * @return {module:model/ShareItem} The populated <code>ShareItem</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ShareItem();

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'String');
        }

        if (data.hasOwnProperty('action')) {
          obj['action'] = _ApiClient["default"].convertToType(data['action'], 'String');
        }

        if (data.hasOwnProperty('emails')) {
          obj['emails'] = _ApiClient["default"].convertToType(data['emails'], ['String']);
        }

        if (data.hasOwnProperty('item-name')) {
          obj['item-name'] = _ApiClient["default"].convertToType(data['item-name'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('share-type')) {
          obj['share-type'] = _ApiClient["default"].convertToType(data['share-type'], 'String');
        }

        if (data.hasOwnProperty('shared-token-id')) {
          obj['shared-token-id'] = _ApiClient["default"].convertToType(data['shared-token-id'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('view-once')) {
          obj['view-once'] = _ApiClient["default"].convertToType(data['view-once'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ShareItem</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ShareItem</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(ShareItem.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['accessibility'] && !(typeof data['accessibility'] === 'string' || data['accessibility'] instanceof String)) {
        throw new Error("Expected the field `accessibility` to be a primitive type in the JSON string but got " + data['accessibility']);
      } // ensure the json data is a string


      if (data['action'] && !(typeof data['action'] === 'string' || data['action'] instanceof String)) {
        throw new Error("Expected the field `action` to be a primitive type in the JSON string but got " + data['action']);
      } // ensure the json data is an array


      if (!Array.isArray(data['emails'])) {
        throw new Error("Expected the field `emails` to be an array in the JSON data but got " + data['emails']);
      } // ensure the json data is a string


      if (data['item-name'] && !(typeof data['item-name'] === 'string' || data['item-name'] instanceof String)) {
        throw new Error("Expected the field `item-name` to be a primitive type in the JSON string but got " + data['item-name']);
      } // ensure the json data is a string


      if (data['share-type'] && !(typeof data['share-type'] === 'string' || data['share-type'] instanceof String)) {
        throw new Error("Expected the field `share-type` to be a primitive type in the JSON string but got " + data['share-type']);
      } // ensure the json data is an array


      if (!Array.isArray(data['shared-token-id'])) {
        throw new Error("Expected the field `shared-token-id` to be an array in the JSON data but got " + data['shared-token-id']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return ShareItem;
}();

ShareItem.RequiredProperties = ["action", "item-name"];
/**
 * for personal password manager
 * @member {String} accessibility
 * @default 'regular'
 */

ShareItem.prototype['accessibility'] = 'regular';
/**
 * Action to be performed on the item [start/stop/describe]
 * @member {String} action
 */

ShareItem.prototype['action'] = undefined;
/**
 * List of emails to start/stop sharing the secret with
 * @member {Array.<String>} emails
 */

ShareItem.prototype['emails'] = undefined;
/**
 * Item name
 * @member {String} item-name
 */

ShareItem.prototype['item-name'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

ShareItem.prototype['json'] = false;
/**
 * Share type [email/token]
 * @member {String} share-type
 * @default 'email'
 */

ShareItem.prototype['share-type'] = 'email';
/**
 * Shared token ids in order to stop sharing a secret
 * @member {Array.<String>} shared-token-id
 */

ShareItem.prototype['shared-token-id'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

ShareItem.prototype['token'] = undefined;
/**
 * TTL of the Availability of the shared secret in seconds
 * @member {Number} ttl
 */

ShareItem.prototype['ttl'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

ShareItem.prototype['uid-token'] = undefined;
/**
 * ViewOnlyOnce Shared secrets can only be viewed once [true/false]
 * @member {Boolean} view-once
 * @default false
 */

ShareItem.prototype['view-once'] = false;
var _default = ShareItem;
exports["default"] = _default;
