// akeyless/dist/model/ShareItemOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ResponseStopShareItem = _interopRequireDefault(require("./ResponseStopShareItem"));

var _SharingItemFullInfo = _interopRequireDefault(require("./SharingItemFullInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ShareItemOutput model module.
 * @module model/ShareItemOutput
 * @version 5.0.6
 */
var ShareItemOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ShareItemOutput</code>.
   * @alias module:model/ShareItemOutput
   */
  function ShareItemOutput() {
    _classCallCheck(this, ShareItemOutput);

    ShareItemOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ShareItemOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ShareItemOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ShareItemOutput} obj Optional instance to populate.
     * @return {module:model/ShareItemOutput} The populated <code>ShareItemOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ShareItemOutput();

        if (data.hasOwnProperty('email_error')) {
          obj['email_error'] = _ApiClient["default"].convertToType(data['email_error'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('items_error')) {
          obj['items_error'] = _ApiClient["default"].convertToType(data['items_error'], [_ResponseStopShareItem["default"]]);
        }

        if (data.hasOwnProperty('s_token')) {
          obj['s_token'] = _ApiClient["default"].convertToType(data['s_token'], 'String');
        }

        if (data.hasOwnProperty('shared_users')) {
          obj['shared_users'] = _ApiClient["default"].convertToType(data['shared_users'], ['String']);
        }

        if (data.hasOwnProperty('shared_users_full_info')) {
          obj['shared_users_full_info'] = _ApiClient["default"].convertToType(data['shared_users_full_info'], [_SharingItemFullInfo["default"]]);
        }

        if (data.hasOwnProperty('sharing_url')) {
          obj['sharing_url'] = _ApiClient["default"].convertToType(data['sharing_url'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ShareItemOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ShareItemOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['items_error']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['items_error'])) {
          throw new Error("Expected the field `items_error` to be an array in the JSON data but got " + data['items_error']);
        } // validate the optional field `items_error` (array)


        var _iterator = _createForOfIteratorHelper(data['items_error']),
            _step;

        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;

            _ResponseStopShareItem["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }

        ;
      } // ensure the json data is a string


      if (data['s_token'] && !(typeof data['s_token'] === 'string' || data['s_token'] instanceof String)) {
        throw new Error("Expected the field `s_token` to be a primitive type in the JSON string but got " + data['s_token']);
      } // ensure the json data is an array


      if (!Array.isArray(data['shared_users'])) {
        throw new Error("Expected the field `shared_users` to be an array in the JSON data but got " + data['shared_users']);
      }

      if (data['shared_users_full_info']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['shared_users_full_info'])) {
          throw new Error("Expected the field `shared_users_full_info` to be an array in the JSON data but got " + data['shared_users_full_info']);
        } // validate the optional field `shared_users_full_info` (array)


        var _iterator2 = _createForOfIteratorHelper(data['shared_users_full_info']),
            _step2;

        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;

            _SharingItemFullInfo["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }

        ;
      } // ensure the json data is a string


      if (data['sharing_url'] && !(typeof data['sharing_url'] === 'string' || data['sharing_url'] instanceof String)) {
        throw new Error("Expected the field `sharing_url` to be a primitive type in the JSON string but got " + data['sharing_url']);
      }

      return true;
    }
  }]);

  return ShareItemOutput;
}();
/**
 * @member {Object.<String, String>} email_error
 */


ShareItemOutput.prototype['email_error'] = undefined;
/**
 * @member {Array.<module:model/ResponseStopShareItem>} items_error
 */

ShareItemOutput.prototype['items_error'] = undefined;
/**
 * @member {String} s_token
 */

ShareItemOutput.prototype['s_token'] = undefined;
/**
 * @member {Array.<String>} shared_users
 */

ShareItemOutput.prototype['shared_users'] = undefined;
/**
 * @member {Array.<module:model/SharingItemFullInfo>} shared_users_full_info
 */

ShareItemOutput.prototype['shared_users_full_info'] = undefined;
/**
 * @member {String} sharing_url
 */

ShareItemOutput.prototype['sharing_url'] = undefined;
var _default = ShareItemOutput;
exports["default"] = _default;
