// akeyless/dist/model/SignPKCS1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignPKCS1 model module.
 * @module model/SignPKCS1
 * @version 3.6.3
 */
var SignPKCS1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignPKCS1</code>.
   * signPKCS1 is a command that calculates the signature of hashed data using RSASSA-PKCS1-V1_5-SIGN from RSA PKCS#1 v1.5.
   * @alias module:model/SignPKCS1
   * @param message {String} The message to be signed
   */
  function SignPKCS1(message) {
    _classCallCheck(this, SignPKCS1);

    SignPKCS1.initialize(this, message);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignPKCS1, null, [{
    key: "initialize",
    value: function initialize(obj, message) {
      obj['message'] = message;
    }
    /**
     * Constructs a <code>SignPKCS1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignPKCS1} obj Optional instance to populate.
     * @return {module:model/SignPKCS1} The populated <code>SignPKCS1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignPKCS1();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('hash-function')) {
          obj['hash-function'] = _ApiClient["default"].convertToType(data['hash-function'], 'String');
        }

        if (data.hasOwnProperty('input-format')) {
          obj['input-format'] = _ApiClient["default"].convertToType(data['input-format'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('prehashed')) {
          obj['prehashed'] = _ApiClient["default"].convertToType(data['prehashed'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return SignPKCS1;
}();
/**
 * The display id of the key to use in the signing process
 * @member {String} display-id
 */


SignPKCS1.prototype['display-id'] = undefined;
/**
 * HashFunction defines the hash function (e.g. sha-256)
 * @member {String} hash-function
 */

SignPKCS1.prototype['hash-function'] = undefined;
/**
 * Select default assumed format for the plaintext message. Currently supported options: [base64]
 * @member {String} input-format
 */

SignPKCS1.prototype['input-format'] = undefined;
/**
 * The item id of the key to use in the signing process
 * @member {Number} item-id
 */

SignPKCS1.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

SignPKCS1.prototype['json'] = false;
/**
 * The name of the RSA key to use in the signing process
 * @member {String} key-name
 */

SignPKCS1.prototype['key-name'] = undefined;
/**
 * The message to be signed
 * @member {String} message
 */

SignPKCS1.prototype['message'] = undefined;
/**
 * Markes that the message is already hashed
 * @member {Boolean} prehashed
 */

SignPKCS1.prototype['prehashed'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignPKCS1.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignPKCS1.prototype['uid-token'] = undefined;
/**
 * The version of the key to use for signing
 * @member {Number} version
 */

SignPKCS1.prototype['version'] = undefined;
var _default = SignPKCS1;
exports["default"] = _default;
