// akeyless/dist/model/SignDataWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignDataWithClassicKey model module.
 * @module model/SignDataWithClassicKey
 * @version 5.0.8
 */
var SignDataWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignDataWithClassicKey</code>.
   * @alias module:model/SignDataWithClassicKey
   * @param data {String} Data
   * @param displayId {String} The name of the key to use in the sign data process
   * @param name {String} ClassicKey name
   * @param version {Number} classic key version
   */
  function SignDataWithClassicKey(data, displayId, name, version) {
    _classCallCheck(this, SignDataWithClassicKey);

    SignDataWithClassicKey.initialize(this, data, displayId, name, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignDataWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, data, displayId, name, version) {
      obj['data'] = data;
      obj['display-id'] = displayId;
      obj['hashed'] = false;
      obj['hashing-method'] = 'SHA256';
      obj['ignore-cache'] = 'false';
      obj['json'] = false;
      obj['name'] = name;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>SignDataWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignDataWithClassicKey} obj Optional instance to populate.
     * @return {module:model/SignDataWithClassicKey} The populated <code>SignDataWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignDataWithClassicKey();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('hashed')) {
          obj['hashed'] = _ApiClient["default"].convertToType(data['hashed'], 'Boolean');
        }

        if (data.hasOwnProperty('hashing-method')) {
          obj['hashing-method'] = _ApiClient["default"].convertToType(data['hashing-method'], 'String');
        }

        if (data.hasOwnProperty('ignore-cache')) {
          obj['ignore-cache'] = _ApiClient["default"].convertToType(data['ignore-cache'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SignDataWithClassicKey</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SignDataWithClassicKey</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(SignDataWithClassicKey.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['data'] && !(typeof data['data'] === 'string' || data['data'] instanceof String)) {
        throw new Error("Expected the field `data` to be a primitive type in the JSON string but got " + data['data']);
      } // ensure the json data is a string


      if (data['display-id'] && !(typeof data['display-id'] === 'string' || data['display-id'] instanceof String)) {
        throw new Error("Expected the field `display-id` to be a primitive type in the JSON string but got " + data['display-id']);
      } // ensure the json data is a string


      if (data['hashing-method'] && !(typeof data['hashing-method'] === 'string' || data['hashing-method'] instanceof String)) {
        throw new Error("Expected the field `hashing-method` to be a primitive type in the JSON string but got " + data['hashing-method']);
      } // ensure the json data is a string


      if (data['ignore-cache'] && !(typeof data['ignore-cache'] === 'string' || data['ignore-cache'] instanceof String)) {
        throw new Error("Expected the field `ignore-cache` to be a primitive type in the JSON string but got " + data['ignore-cache']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return SignDataWithClassicKey;
}();

SignDataWithClassicKey.RequiredProperties = ["data", "display-id", "name", "version"];
/**
 * Data
 * @member {String} data
 */

SignDataWithClassicKey.prototype['data'] = undefined;
/**
 * The name of the key to use in the sign data process
 * @member {String} display-id
 */

SignDataWithClassicKey.prototype['display-id'] = undefined;
/**
 * Defines whether the data should be hashed as part of the signing. If true, the data will not be hashed
 * @member {Boolean} hashed
 * @default false
 */

SignDataWithClassicKey.prototype['hashed'] = false;
/**
 * HashingMethod
 * @member {String} hashing-method
 * @default 'SHA256'
 */

SignDataWithClassicKey.prototype['hashing-method'] = 'SHA256';
/**
 * Retrieve the Secret value without checking the Gateway's cache [true/false]. This flag is only relevant when using the RestAPI
 * @member {String} ignore-cache
 * @default 'false'
 */

SignDataWithClassicKey.prototype['ignore-cache'] = 'false';
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

SignDataWithClassicKey.prototype['json'] = false;
/**
 * ClassicKey name
 * @member {String} name
 */

SignDataWithClassicKey.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignDataWithClassicKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignDataWithClassicKey.prototype['uid-token'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

SignDataWithClassicKey.prototype['version'] = undefined;
var _default = SignDataWithClassicKey;
exports["default"] = _default;
