// akeyless/dist/model/SraDesktopAppConf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SraDesktopAppConf model module.
 * @module model/SraDesktopAppConf
 * @version 5.0.14
 */
var SraDesktopAppConf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SraDesktopAppConf</code>.
   * @alias module:model/SraDesktopAppConf
   */
  function SraDesktopAppConf() {
    _classCallCheck(this, SraDesktopAppConf);

    SraDesktopAppConf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SraDesktopAppConf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SraDesktopAppConf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SraDesktopAppConf} obj Optional instance to populate.
     * @return {module:model/SraDesktopAppConf} The populated <code>SraDesktopAppConf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SraDesktopAppConf();

        if (data.hasOwnProperty('default_cert_issuer_id')) {
          obj['default_cert_issuer_id'] = _ApiClient["default"].convertToType(data['default_cert_issuer_id'], 'Number');
        }

        if (data.hasOwnProperty('secure_web_access_url')) {
          obj['secure_web_access_url'] = _ApiClient["default"].convertToType(data['secure_web_access_url'], 'String');
        }

        if (data.hasOwnProperty('secure_web_proxy_url')) {
          obj['secure_web_proxy_url'] = _ApiClient["default"].convertToType(data['secure_web_proxy_url'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SraDesktopAppConf</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SraDesktopAppConf</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['secure_web_access_url'] && !(typeof data['secure_web_access_url'] === 'string' || data['secure_web_access_url'] instanceof String)) {
        throw new Error("Expected the field `secure_web_access_url` to be a primitive type in the JSON string but got " + data['secure_web_access_url']);
      } // ensure the json data is a string


      if (data['secure_web_proxy_url'] && !(typeof data['secure_web_proxy_url'] === 'string' || data['secure_web_proxy_url'] instanceof String)) {
        throw new Error("Expected the field `secure_web_proxy_url` to be a primitive type in the JSON string but got " + data['secure_web_proxy_url']);
      }

      return true;
    }
  }]);

  return SraDesktopAppConf;
}();
/**
 * @member {Number} default_cert_issuer_id
 */


SraDesktopAppConf.prototype['default_cert_issuer_id'] = undefined;
/**
 * @member {String} secure_web_access_url
 */

SraDesktopAppConf.prototype['secure_web_access_url'] = undefined;
/**
 * @member {String} secure_web_proxy_url
 */

SraDesktopAppConf.prototype['secure_web_proxy_url'] = undefined;
var _default = SraDesktopAppConf;
exports["default"] = _default;
