// akeyless/dist/model/SraInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SraInfo model module.
 * @module model/SraInfo
 * @version 5.0.8
 */
var SraInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SraInfo</code>.
   * @alias module:model/SraInfo
   */
  function SraInfo() {
    _classCallCheck(this, SraInfo);

    SraInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SraInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SraInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SraInfo} obj Optional instance to populate.
     * @return {module:model/SraInfo} The populated <code>SraInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SraInfo();

        if (data.hasOwnProperty('sla')) {
          obj['sla'] = _ApiClient["default"].convertToType(data['sla'], 'String');
        }

        if (data.hasOwnProperty('tier')) {
          obj['tier'] = _ApiClient["default"].convertToType(data['tier'], 'String');
        }

        if (data.hasOwnProperty('user_type')) {
          obj['user_type'] = _ApiClient["default"].convertToType(data['user_type'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SraInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SraInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['sla'] && !(typeof data['sla'] === 'string' || data['sla'] instanceof String)) {
        throw new Error("Expected the field `sla` to be a primitive type in the JSON string but got " + data['sla']);
      } // ensure the json data is a string


      if (data['tier'] && !(typeof data['tier'] === 'string' || data['tier'] instanceof String)) {
        throw new Error("Expected the field `tier` to be a primitive type in the JSON string but got " + data['tier']);
      } // ensure the json data is a string


      if (data['user_type'] && !(typeof data['user_type'] === 'string' || data['user_type'] instanceof String)) {
        throw new Error("Expected the field `user_type` to be a primitive type in the JSON string but got " + data['user_type']);
      }

      return true;
    }
  }]);

  return SraInfo;
}();
/**
 * @member {String} sla
 */


SraInfo.prototype['sla'] = undefined;
/**
 * Tier represents a level of extensibility the account will have, defined by various limits for different resources of Akeyless e.g - A StarterTier may have a limit of 3 Client resources and 50 Secret resources
 * @member {String} tier
 */

SraInfo.prototype['tier'] = undefined;
/**
 * @member {String} user_type
 */

SraInfo.prototype['user_type'] = undefined;
var _default = SraInfo;
exports["default"] = _default;
