// akeyless/dist/model/SshBastionSessionTermination.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SshBastionSessionTermination model module.
 * @module model/SshBastionSessionTermination
 * @version 5.0.5
 */
var SshBastionSessionTermination = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SshBastionSessionTermination</code>.
   * @alias module:model/SshBastionSessionTermination
   */
  function SshBastionSessionTermination() {
    _classCallCheck(this, SshBastionSessionTermination);

    SshBastionSessionTermination.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SshBastionSessionTermination, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SshBastionSessionTermination</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SshBastionSessionTermination} obj Optional instance to populate.
     * @return {module:model/SshBastionSessionTermination} The populated <code>SshBastionSessionTermination</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SshBastionSessionTermination();

        if (data.hasOwnProperty('api_password')) {
          obj['api_password'] = _ApiClient["default"].convertToType(data['api_password'], 'String');
        }

        if (data.hasOwnProperty('api_token')) {
          obj['api_token'] = _ApiClient["default"].convertToType(data['api_token'], 'String');
        }

        if (data.hasOwnProperty('api_url')) {
          obj['api_url'] = _ApiClient["default"].convertToType(data['api_url'], 'String');
        }

        if (data.hasOwnProperty('api_username')) {
          obj['api_username'] = _ApiClient["default"].convertToType(data['api_username'], 'String');
        }

        if (data.hasOwnProperty('enabled')) {
          obj['enabled'] = _ApiClient["default"].convertToType(data['enabled'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SshBastionSessionTermination</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SshBastionSessionTermination</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['api_password'] && !(typeof data['api_password'] === 'string' || data['api_password'] instanceof String)) {
        throw new Error("Expected the field `api_password` to be a primitive type in the JSON string but got " + data['api_password']);
      } // ensure the json data is a string


      if (data['api_token'] && !(typeof data['api_token'] === 'string' || data['api_token'] instanceof String)) {
        throw new Error("Expected the field `api_token` to be a primitive type in the JSON string but got " + data['api_token']);
      } // ensure the json data is a string


      if (data['api_url'] && !(typeof data['api_url'] === 'string' || data['api_url'] instanceof String)) {
        throw new Error("Expected the field `api_url` to be a primitive type in the JSON string but got " + data['api_url']);
      } // ensure the json data is a string


      if (data['api_username'] && !(typeof data['api_username'] === 'string' || data['api_username'] instanceof String)) {
        throw new Error("Expected the field `api_username` to be a primitive type in the JSON string but got " + data['api_username']);
      }

      return true;
    }
  }]);

  return SshBastionSessionTermination;
}();
/**
 * @member {String} api_password
 */


SshBastionSessionTermination.prototype['api_password'] = undefined;
/**
 * @member {String} api_token
 */

SshBastionSessionTermination.prototype['api_token'] = undefined;
/**
 * @member {String} api_url
 */

SshBastionSessionTermination.prototype['api_url'] = undefined;
/**
 * @member {String} api_username
 */

SshBastionSessionTermination.prototype['api_username'] = undefined;
/**
 * @member {Boolean} enabled
 */

SshBastionSessionTermination.prototype['enabled'] = undefined;
var _default = SshBastionSessionTermination;
exports["default"] = _default;
