// akeyless/dist/model/SshBastionConf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _LogForwardingConfigPart = _interopRequireDefault(require("./LogForwardingConfigPart"));

var _SshBastionSessionTermination = _interopRequireDefault(require("./SshBastionSessionTermination"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SshBastionConf model module.
 * @module model/SshBastionConf
 * @version 5.0.1
 */
var SshBastionConf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SshBastionConf</code>.
   * @alias module:model/SshBastionConf
   */
  function SshBastionConf() {
    _classCallCheck(this, SshBastionConf);

    SshBastionConf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SshBastionConf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SshBastionConf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SshBastionConf} obj Optional instance to populate.
     * @return {module:model/SshBastionConf} The populated <code>SshBastionConf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SshBastionConf();

        if (data.hasOwnProperty('hide_session_recording')) {
          obj['hide_session_recording'] = _ApiClient["default"].convertToType(data['hide_session_recording'], 'Boolean');
        }

        if (data.hasOwnProperty('kexalgs')) {
          obj['kexalgs'] = _ApiClient["default"].convertToType(data['kexalgs'], 'String');
        }

        if (data.hasOwnProperty('log_forwarding')) {
          obj['log_forwarding'] = _LogForwardingConfigPart["default"].constructFromObject(data['log_forwarding']);
        }

        if (data.hasOwnProperty('session_termination')) {
          obj['session_termination'] = _SshBastionSessionTermination["default"].constructFromObject(data['session_termination']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SshBastionConf</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SshBastionConf</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['kexalgs'] && !(typeof data['kexalgs'] === 'string' || data['kexalgs'] instanceof String)) {
        throw new Error("Expected the field `kexalgs` to be a primitive type in the JSON string but got " + data['kexalgs']);
      } // validate the optional field `log_forwarding`


      if (data['log_forwarding']) {
        // data not null
        _LogForwardingConfigPart["default"].validateJSON(data['log_forwarding']);
      } // validate the optional field `session_termination`


      if (data['session_termination']) {
        // data not null
        _SshBastionSessionTermination["default"].validateJSON(data['session_termination']);
      }

      return true;
    }
  }]);

  return SshBastionConf;
}();
/**
 * @member {Boolean} hide_session_recording
 */


SshBastionConf.prototype['hide_session_recording'] = undefined;
/**
 * @member {String} kexalgs
 */

SshBastionConf.prototype['kexalgs'] = undefined;
/**
 * @member {module:model/LogForwardingConfigPart} log_forwarding
 */

SshBastionConf.prototype['log_forwarding'] = undefined;
/**
 * @member {module:model/SshBastionSessionTermination} session_termination
 */

SshBastionConf.prototype['session_termination'] = undefined;
var _default = SshBastionConf;
exports["default"] = _default;
