// akeyless/dist/model/SystemAccessCredentialsReplyObj.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SystemAccessCredentialsReplyObj model module.
 * @module model/SystemAccessCredentialsReplyObj
 * @version 5.0.3
 */
var SystemAccessCredentialsReplyObj = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SystemAccessCredentialsReplyObj</code>.
   * @alias module:model/SystemAccessCredentialsReplyObj
   */
  function SystemAccessCredentialsReplyObj() {
    _classCallCheck(this, SystemAccessCredentialsReplyObj);

    SystemAccessCredentialsReplyObj.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SystemAccessCredentialsReplyObj, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SystemAccessCredentialsReplyObj</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SystemAccessCredentialsReplyObj} obj Optional instance to populate.
     * @return {module:model/SystemAccessCredentialsReplyObj} The populated <code>SystemAccessCredentialsReplyObj</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SystemAccessCredentialsReplyObj();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('auth_creds')) {
          obj['auth_creds'] = _ApiClient["default"].convertToType(data['auth_creds'], 'String');
        }

        if (data.hasOwnProperty('expiry')) {
          obj['expiry'] = _ApiClient["default"].convertToType(data['expiry'], 'Number');
        }

        if (data.hasOwnProperty('kfm_creds')) {
          obj['kfm_creds'] = _ApiClient["default"].convertToType(data['kfm_creds'], 'String');
        }

        if (data.hasOwnProperty('need_mfa_app_first_config')) {
          obj['need_mfa_app_first_config'] = _ApiClient["default"].convertToType(data['need_mfa_app_first_config'], 'Boolean');
        }

        if (data.hasOwnProperty('required_mfa')) {
          obj['required_mfa'] = _ApiClient["default"].convertToType(data['required_mfa'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uam_creds')) {
          obj['uam_creds'] = _ApiClient["default"].convertToType(data['uam_creds'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>SystemAccessCredentialsReplyObj</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SystemAccessCredentialsReplyObj</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_id'] && !(typeof data['access_id'] === 'string' || data['access_id'] instanceof String)) {
        throw new Error("Expected the field `access_id` to be a primitive type in the JSON string but got " + data['access_id']);
      } // ensure the json data is a string


      if (data['auth_creds'] && !(typeof data['auth_creds'] === 'string' || data['auth_creds'] instanceof String)) {
        throw new Error("Expected the field `auth_creds` to be a primitive type in the JSON string but got " + data['auth_creds']);
      } // ensure the json data is a string


      if (data['kfm_creds'] && !(typeof data['kfm_creds'] === 'string' || data['kfm_creds'] instanceof String)) {
        throw new Error("Expected the field `kfm_creds` to be a primitive type in the JSON string but got " + data['kfm_creds']);
      } // ensure the json data is a string


      if (data['required_mfa'] && !(typeof data['required_mfa'] === 'string' || data['required_mfa'] instanceof String)) {
        throw new Error("Expected the field `required_mfa` to be a primitive type in the JSON string but got " + data['required_mfa']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uam_creds'] && !(typeof data['uam_creds'] === 'string' || data['uam_creds'] instanceof String)) {
        throw new Error("Expected the field `uam_creds` to be a primitive type in the JSON string but got " + data['uam_creds']);
      }

      return true;
    }
  }]);

  return SystemAccessCredentialsReplyObj;
}();
/**
 * @member {String} access_id
 */


SystemAccessCredentialsReplyObj.prototype['access_id'] = undefined;
/**
 * Temporary credentials for accessing Auth
 * @member {String} auth_creds
 */

SystemAccessCredentialsReplyObj.prototype['auth_creds'] = undefined;
/**
 * Credentials expiration date
 * @member {Number} expiry
 */

SystemAccessCredentialsReplyObj.prototype['expiry'] = undefined;
/**
 * Temporary credentials for accessing the KFMs instances
 * @member {String} kfm_creds
 */

SystemAccessCredentialsReplyObj.prototype['kfm_creds'] = undefined;
/**
 * If the user didn't complete to configure the MFA app
 * @member {Boolean} need_mfa_app_first_config
 */

SystemAccessCredentialsReplyObj.prototype['need_mfa_app_first_config'] = undefined;
/**
 * @member {String} required_mfa
 */

SystemAccessCredentialsReplyObj.prototype['required_mfa'] = undefined;
/**
 * Credentials tmp token
 * @member {String} token
 */

SystemAccessCredentialsReplyObj.prototype['token'] = undefined;
/**
 * Temporary credentials for accessing the UAM service
 * @member {String} uam_creds
 */

SystemAccessCredentialsReplyObj.prototype['uam_creds'] = undefined;
var _default = SystemAccessCredentialsReplyObj;
exports["default"] = _default;
