// akeyless/dist/model/TargetUpdateLinked.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateLinked model module.
 * @module model/TargetUpdateLinked
 * @version 4.2.1
 */
var TargetUpdateLinked = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateLinked</code>.
   * @alias module:model/TargetUpdateLinked
   * @param name {String} Linked Target name
   */
  function TargetUpdateLinked(name) {
    _classCallCheck(this, TargetUpdateLinked);

    TargetUpdateLinked.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateLinked, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetUpdateLinked</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateLinked} obj Optional instance to populate.
     * @return {module:model/TargetUpdateLinked} The populated <code>TargetUpdateLinked</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateLinked();

        if (data.hasOwnProperty('add-hosts')) {
          obj['add-hosts'] = _ApiClient["default"].convertToType(data['add-hosts'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('hosts')) {
          obj['hosts'] = _ApiClient["default"].convertToType(data['hosts'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('parent-target-name')) {
          obj['parent-target-name'] = _ApiClient["default"].convertToType(data['parent-target-name'], 'String');
        }

        if (data.hasOwnProperty('rm-hosts')) {
          obj['rm-hosts'] = _ApiClient["default"].convertToType(data['rm-hosts'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetUpdateLinked;
}();
/**
 * A comma seperated list of new server hosts and server descriptions joined by semicolon ';' that will be added to the Linked Target hosts.
 * @member {String} add-hosts
 */


TargetUpdateLinked.prototype['add-hosts'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateLinked.prototype['description'] = undefined;
/**
 * A comma seperated list of server hosts and server descriptions joined by semicolon ';' (i.e. 'server-dev.com;My Dev server,server-prod.com;My Prod server description')
 * @member {String} hosts
 */

TargetUpdateLinked.prototype['hosts'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateLinked.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateLinked.prototype['keep-prev-version'] = undefined;
/**
 * Linked Target name
 * @member {String} name
 */

TargetUpdateLinked.prototype['name'] = undefined;
/**
 * New Linked Target name
 * @member {String} new-name
 */

TargetUpdateLinked.prototype['new-name'] = undefined;
/**
 * The parent Target name
 * @member {String} parent-target-name
 */

TargetUpdateLinked.prototype['parent-target-name'] = undefined;
/**
 * Comma separated list of existing hosts that will be removed from Linked Target hosts.
 * @member {String} rm-hosts
 */

TargetUpdateLinked.prototype['rm-hosts'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateLinked.prototype['token'] = undefined;
/**
 * Specifies the hosts type, relevant only when working without parent target
 * @member {String} type
 */

TargetUpdateLinked.prototype['type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateLinked.prototype['uid-token'] = undefined;
var _default = TargetUpdateLinked;
exports["default"] = _default;
