// akeyless/dist/model/TargetCreateEks.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateEks model module.
 * @module model/TargetCreateEks
 * @version 4.3.0
 */
var TargetCreateEks = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateEks</code>.
   * @alias module:model/TargetCreateEks
   * @param eksAccessKeyId {String} Access Key ID
   * @param eksClusterCaCert {String} EKS cluster CA certificate
   * @param eksClusterEndpoint {String} EKS cluster URL endpoint
   * @param eksClusterName {String} EKS cluster name
   * @param eksSecretAccessKey {String} Secret Access Key
   * @param name {String} Target name
   */
  function TargetCreateEks(eksAccessKeyId, eksClusterCaCert, eksClusterEndpoint, eksClusterName, eksSecretAccessKey, name) {
    _classCallCheck(this, TargetCreateEks);

    TargetCreateEks.initialize(this, eksAccessKeyId, eksClusterCaCert, eksClusterEndpoint, eksClusterName, eksSecretAccessKey, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateEks, null, [{
    key: "initialize",
    value: function initialize(obj, eksAccessKeyId, eksClusterCaCert, eksClusterEndpoint, eksClusterName, eksSecretAccessKey, name) {
      obj['eks-access-key-id'] = eksAccessKeyId;
      obj['eks-cluster-ca-cert'] = eksClusterCaCert;
      obj['eks-cluster-endpoint'] = eksClusterEndpoint;
      obj['eks-cluster-name'] = eksClusterName;
      obj['eks-secret-access-key'] = eksSecretAccessKey;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetCreateEks</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateEks} obj Optional instance to populate.
     * @return {module:model/TargetCreateEks} The populated <code>TargetCreateEks</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateEks();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('eks-access-key-id')) {
          obj['eks-access-key-id'] = _ApiClient["default"].convertToType(data['eks-access-key-id'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-ca-cert')) {
          obj['eks-cluster-ca-cert'] = _ApiClient["default"].convertToType(data['eks-cluster-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-endpoint')) {
          obj['eks-cluster-endpoint'] = _ApiClient["default"].convertToType(data['eks-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-name')) {
          obj['eks-cluster-name'] = _ApiClient["default"].convertToType(data['eks-cluster-name'], 'String');
        }

        if (data.hasOwnProperty('eks-region')) {
          obj['eks-region'] = _ApiClient["default"].convertToType(data['eks-region'], 'String');
        }

        if (data.hasOwnProperty('eks-secret-access-key')) {
          obj['eks-secret-access-key'] = _ApiClient["default"].convertToType(data['eks-secret-access-key'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateEks;
}();
/**
 * Description of the object
 * @member {String} description
 */


TargetCreateEks.prototype['description'] = undefined;
/**
 * Access Key ID
 * @member {String} eks-access-key-id
 */

TargetCreateEks.prototype['eks-access-key-id'] = undefined;
/**
 * EKS cluster CA certificate
 * @member {String} eks-cluster-ca-cert
 */

TargetCreateEks.prototype['eks-cluster-ca-cert'] = undefined;
/**
 * EKS cluster URL endpoint
 * @member {String} eks-cluster-endpoint
 */

TargetCreateEks.prototype['eks-cluster-endpoint'] = undefined;
/**
 * EKS cluster name
 * @member {String} eks-cluster-name
 */

TargetCreateEks.prototype['eks-cluster-name'] = undefined;
/**
 * Region
 * @member {String} eks-region
 * @default 'us-east-2'
 */

TargetCreateEks.prototype['eks-region'] = 'us-east-2';
/**
 * Secret Access Key
 * @member {String} eks-secret-access-key
 */

TargetCreateEks.prototype['eks-secret-access-key'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateEks.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateEks.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateEks.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateEks.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateEks.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateEks.prototype['uid-token'] = undefined;
/**
 * @member {Boolean} use-gw-cloud-identity
 */

TargetCreateEks.prototype['use-gw-cloud-identity'] = undefined;
var _default = TargetCreateEks;
exports["default"] = _default;
