// akeyless/dist/model/TargetCreateWindows.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateWindows model module.
 * @module model/TargetCreateWindows
 * @version 4.3.0
 */
var TargetCreateWindows = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateWindows</code>.
   * GatewayCreateProducerWindowsCmd is a command that creates a new windows target
   * @alias module:model/TargetCreateWindows
   * @param hostname {String} Server hostname
   * @param name {String} Target name
   * @param password {String} Privileged user password
   * @param username {String} Privileged username
   */
  function TargetCreateWindows(hostname, name, password, username) {
    _classCallCheck(this, TargetCreateWindows);

    TargetCreateWindows.initialize(this, hostname, name, password, username);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateWindows, null, [{
    key: "initialize",
    value: function initialize(obj, hostname, name, password, username) {
      obj['hostname'] = hostname;
      obj['name'] = name;
      obj['password'] = password;
      obj['username'] = username;
    }
    /**
     * Constructs a <code>TargetCreateWindows</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateWindows} obj Optional instance to populate.
     * @return {module:model/TargetCreateWindows} The populated <code>TargetCreateWindows</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateWindows();

        if (data.hasOwnProperty('certificate')) {
          obj['certificate'] = _ApiClient["default"].convertToType(data['certificate'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('domain')) {
          obj['domain'] = _ApiClient["default"].convertToType(data['domain'], 'String');
        }

        if (data.hasOwnProperty('hostname')) {
          obj['hostname'] = _ApiClient["default"].convertToType(data['hostname'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-tls')) {
          obj['use-tls'] = _ApiClient["default"].convertToType(data['use-tls'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateWindows;
}();
/**
 * SSL CA certificate in base64 encoding generated from a trusted Certificate Authority (CA)
 * @member {String} certificate
 */


TargetCreateWindows.prototype['certificate'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetCreateWindows.prototype['description'] = undefined;
/**
 * User domain name
 * @member {String} domain
 */

TargetCreateWindows.prototype['domain'] = undefined;
/**
 * Server hostname
 * @member {String} hostname
 */

TargetCreateWindows.prototype['hostname'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateWindows.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateWindows.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateWindows.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateWindows.prototype['name'] = undefined;
/**
 * Privileged user password
 * @member {String} password
 */

TargetCreateWindows.prototype['password'] = undefined;
/**
 * Server WinRM port
 * @member {String} port
 * @default '5986'
 */

TargetCreateWindows.prototype['port'] = '5986';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateWindows.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateWindows.prototype['uid-token'] = undefined;
/**
 * Enable/Disable TLS for WinRM over HTTPS [true/false]
 * @member {String} use-tls
 * @default 'true'
 */

TargetCreateWindows.prototype['use-tls'] = 'true';
/**
 * Privileged username
 * @member {String} username
 */

TargetCreateWindows.prototype['username'] = undefined;
var _default = TargetCreateWindows;
exports["default"] = _default;
