// akeyless/dist/model/TargetCreateSsh.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateSsh model module.
 * @module model/TargetCreateSsh
 * @version 5.0.0
 */
var TargetCreateSsh = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateSsh</code>.
   * targetCreateSsh is a command that creates a new ssh target
   * @alias module:model/TargetCreateSsh
   * @param name {String} Target name
   */
  function TargetCreateSsh(name) {
    _classCallCheck(this, TargetCreateSsh);

    TargetCreateSsh.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateSsh, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['json'] = false;
      obj['name'] = name;
      obj['port'] = '22';
    }
    /**
     * Constructs a <code>TargetCreateSsh</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateSsh} obj Optional instance to populate.
     * @return {module:model/TargetCreateSsh} The populated <code>TargetCreateSsh</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateSsh();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('private-key')) {
          obj['private-key'] = _ApiClient["default"].convertToType(data['private-key'], 'String');
        }

        if (data.hasOwnProperty('private-key-password')) {
          obj['private-key-password'] = _ApiClient["default"].convertToType(data['private-key-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TargetCreateSsh</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TargetCreateSsh</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(TargetCreateSsh.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['host'] && !(typeof data['host'] === 'string' || data['host'] instanceof String)) {
        throw new Error("Expected the field `host` to be a primitive type in the JSON string but got " + data['host']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['port'] && !(typeof data['port'] === 'string' || data['port'] instanceof String)) {
        throw new Error("Expected the field `port` to be a primitive type in the JSON string but got " + data['port']);
      } // ensure the json data is a string


      if (data['private-key'] && !(typeof data['private-key'] === 'string' || data['private-key'] instanceof String)) {
        throw new Error("Expected the field `private-key` to be a primitive type in the JSON string but got " + data['private-key']);
      } // ensure the json data is a string


      if (data['private-key-password'] && !(typeof data['private-key-password'] === 'string' || data['private-key-password'] instanceof String)) {
        throw new Error("Expected the field `private-key-password` to be a primitive type in the JSON string but got " + data['private-key-password']);
      } // ensure the json data is a string


      if (data['ssh-password'] && !(typeof data['ssh-password'] === 'string' || data['ssh-password'] instanceof String)) {
        throw new Error("Expected the field `ssh-password` to be a primitive type in the JSON string but got " + data['ssh-password']);
      } // ensure the json data is a string


      if (data['ssh-username'] && !(typeof data['ssh-username'] === 'string' || data['ssh-username'] instanceof String)) {
        throw new Error("Expected the field `ssh-username` to be a primitive type in the JSON string but got " + data['ssh-username']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return TargetCreateSsh;
}();

TargetCreateSsh.RequiredProperties = ["name"];
/**
 * Description of the object
 * @member {String} description
 */

TargetCreateSsh.prototype['description'] = undefined;
/**
 * SSH host name
 * @member {String} host
 */

TargetCreateSsh.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateSsh.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateSsh.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateSsh.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateSsh.prototype['name'] = undefined;
/**
 * SSH port
 * @member {String} port
 * @default '22'
 */

TargetCreateSsh.prototype['port'] = '22';
/**
 * SSH private key
 * @member {String} private-key
 */

TargetCreateSsh.prototype['private-key'] = undefined;
/**
 * SSH private key password
 * @member {String} private-key-password
 */

TargetCreateSsh.prototype['private-key-password'] = undefined;
/**
 * SSH password to rotate
 * @member {String} ssh-password
 */

TargetCreateSsh.prototype['ssh-password'] = undefined;
/**
 * SSH username
 * @member {String} ssh-username
 */

TargetCreateSsh.prototype['ssh-username'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateSsh.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateSsh.prototype['uid-token'] = undefined;
var _default = TargetCreateSsh;
exports["default"] = _default;
