// akeyless/dist/model/TargetCreateSectigo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateSectigo model module.
 * @module model/TargetCreateSectigo
 * @version 5.0.6
 */
var TargetCreateSectigo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateSectigo</code>.
   * targetCreateSectigo is a command that creates a new Sectigo target
   * @alias module:model/TargetCreateSectigo
   * @param certificateProfileId {Number} Certificate Profile ID in Sectigo account
   * @param customerUri {String} Customer Uri of the Sectigo account
   * @param externalRequester {String} External Requester - a comma separated list of emails
   * @param name {String} Target name
   * @param organizationId {Number} Organization ID in Sectigo account
   * @param password {String} Password of the Sectigo account user
   * @param username {String} Username of the Sectigo account
   */
  function TargetCreateSectigo(certificateProfileId, customerUri, externalRequester, name, organizationId, password, username) {
    _classCallCheck(this, TargetCreateSectigo);

    TargetCreateSectigo.initialize(this, certificateProfileId, customerUri, externalRequester, name, organizationId, password, username);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateSectigo, null, [{
    key: "initialize",
    value: function initialize(obj, certificateProfileId, customerUri, externalRequester, name, organizationId, password, username) {
      obj['certificate-profile-id'] = certificateProfileId;
      obj['customer-uri'] = customerUri;
      obj['external-requester'] = externalRequester;
      obj['json'] = false;
      obj['name'] = name;
      obj['organization-id'] = organizationId;
      obj['password'] = password;
      obj['timeout'] = '5m';
      obj['username'] = username;
    }
    /**
     * Constructs a <code>TargetCreateSectigo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateSectigo} obj Optional instance to populate.
     * @return {module:model/TargetCreateSectigo} The populated <code>TargetCreateSectigo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateSectigo();

        if (data.hasOwnProperty('certificate-profile-id')) {
          obj['certificate-profile-id'] = _ApiClient["default"].convertToType(data['certificate-profile-id'], 'Number');
        }

        if (data.hasOwnProperty('customer-uri')) {
          obj['customer-uri'] = _ApiClient["default"].convertToType(data['customer-uri'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('external-requester')) {
          obj['external-requester'] = _ApiClient["default"].convertToType(data['external-requester'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('organization-id')) {
          obj['organization-id'] = _ApiClient["default"].convertToType(data['organization-id'], 'Number');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TargetCreateSectigo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TargetCreateSectigo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(TargetCreateSectigo.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['customer-uri'] && !(typeof data['customer-uri'] === 'string' || data['customer-uri'] instanceof String)) {
        throw new Error("Expected the field `customer-uri` to be a primitive type in the JSON string but got " + data['customer-uri']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['external-requester'] && !(typeof data['external-requester'] === 'string' || data['external-requester'] instanceof String)) {
        throw new Error("Expected the field `external-requester` to be a primitive type in the JSON string but got " + data['external-requester']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['password'] && !(typeof data['password'] === 'string' || data['password'] instanceof String)) {
        throw new Error("Expected the field `password` to be a primitive type in the JSON string but got " + data['password']);
      } // ensure the json data is a string


      if (data['timeout'] && !(typeof data['timeout'] === 'string' || data['timeout'] instanceof String)) {
        throw new Error("Expected the field `timeout` to be a primitive type in the JSON string but got " + data['timeout']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['username'] && !(typeof data['username'] === 'string' || data['username'] instanceof String)) {
        throw new Error("Expected the field `username` to be a primitive type in the JSON string but got " + data['username']);
      }

      return true;
    }
  }]);

  return TargetCreateSectigo;
}();

TargetCreateSectigo.RequiredProperties = ["certificate-profile-id", "customer-uri", "external-requester", "name", "organization-id", "password", "username"];
/**
 * Certificate Profile ID in Sectigo account
 * @member {Number} certificate-profile-id
 */

TargetCreateSectigo.prototype['certificate-profile-id'] = undefined;
/**
 * Customer Uri of the Sectigo account
 * @member {String} customer-uri
 */

TargetCreateSectigo.prototype['customer-uri'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetCreateSectigo.prototype['description'] = undefined;
/**
 * External Requester - a comma separated list of emails
 * @member {String} external-requester
 */

TargetCreateSectigo.prototype['external-requester'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateSectigo.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateSectigo.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateSectigo.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateSectigo.prototype['name'] = undefined;
/**
 * Organization ID in Sectigo account
 * @member {Number} organization-id
 */

TargetCreateSectigo.prototype['organization-id'] = undefined;
/**
 * Password of the Sectigo account user
 * @member {String} password
 */

TargetCreateSectigo.prototype['password'] = undefined;
/**
 * Timeout waiting for certificate validation in Duration format (1h - 1 Hour, 20m - 20 Minutes, 33m3s - 33 Minutes and 3 Seconds), maximum 1h.
 * @member {String} timeout
 * @default '5m'
 */

TargetCreateSectigo.prototype['timeout'] = '5m';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateSectigo.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateSectigo.prototype['uid-token'] = undefined;
/**
 * Username of the Sectigo account
 * @member {String} username
 */

TargetCreateSectigo.prototype['username'] = undefined;
var _default = TargetCreateSectigo;
exports["default"] = _default;
