// akeyless/dist/model/TargetCreateGodaddy.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateGodaddy model module.
 * @module model/TargetCreateGodaddy
 * @version 4.2.0
 */
var TargetCreateGodaddy = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateGodaddy</code>.
   * @alias module:model/TargetCreateGodaddy
   * @param apiKey {String} Key of the api credentials to the Godaddy account
   * @param imapFqdn {String} ImapFQDN of the IMAP service, FQDN or IPv4 address. Must be FQDN if the IMAP is using TLS
   * @param imapPassword {String} ImapPassword to access the IMAP service
   * @param imapUsername {String} ImapUsername to access the IMAP service
   * @param name {String} Target name
   * @param secret {String} Secret of the api credentials to the Godaddy account
   */
  function TargetCreateGodaddy(apiKey, imapFqdn, imapPassword, imapUsername, name, secret) {
    _classCallCheck(this, TargetCreateGodaddy);

    TargetCreateGodaddy.initialize(this, apiKey, imapFqdn, imapPassword, imapUsername, name, secret);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateGodaddy, null, [{
    key: "initialize",
    value: function initialize(obj, apiKey, imapFqdn, imapPassword, imapUsername, name, secret) {
      obj['api-key'] = apiKey;
      obj['imap-fqdn'] = imapFqdn;
      obj['imap-password'] = imapPassword;
      obj['imap-username'] = imapUsername;
      obj['name'] = name;
      obj['secret'] = secret;
    }
    /**
     * Constructs a <code>TargetCreateGodaddy</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateGodaddy} obj Optional instance to populate.
     * @return {module:model/TargetCreateGodaddy} The populated <code>TargetCreateGodaddy</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateGodaddy();

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('imap-fqdn')) {
          obj['imap-fqdn'] = _ApiClient["default"].convertToType(data['imap-fqdn'], 'String');
        }

        if (data.hasOwnProperty('imap-password')) {
          obj['imap-password'] = _ApiClient["default"].convertToType(data['imap-password'], 'String');
        }

        if (data.hasOwnProperty('imap-port')) {
          obj['imap-port'] = _ApiClient["default"].convertToType(data['imap-port'], 'String');
        }

        if (data.hasOwnProperty('imap-username')) {
          obj['imap-username'] = _ApiClient["default"].convertToType(data['imap-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('secret')) {
          obj['secret'] = _ApiClient["default"].convertToType(data['secret'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateGodaddy;
}();
/**
 * Key of the api credentials to the Godaddy account
 * @member {String} api-key
 */


TargetCreateGodaddy.prototype['api-key'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetCreateGodaddy.prototype['description'] = undefined;
/**
 * ImapFQDN of the IMAP service, FQDN or IPv4 address. Must be FQDN if the IMAP is using TLS
 * @member {String} imap-fqdn
 */

TargetCreateGodaddy.prototype['imap-fqdn'] = undefined;
/**
 * ImapPassword to access the IMAP service
 * @member {String} imap-password
 */

TargetCreateGodaddy.prototype['imap-password'] = undefined;
/**
 * ImapPort of the IMAP service
 * @member {String} imap-port
 * @default '993'
 */

TargetCreateGodaddy.prototype['imap-port'] = '993';
/**
 * ImapUsername to access the IMAP service
 * @member {String} imap-username
 */

TargetCreateGodaddy.prototype['imap-username'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateGodaddy.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateGodaddy.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateGodaddy.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateGodaddy.prototype['name'] = undefined;
/**
 * Secret of the api credentials to the Godaddy account
 * @member {String} secret
 */

TargetCreateGodaddy.prototype['secret'] = undefined;
/**
 * Timeout waiting for certificate validation in Duration format (1h - 1 Hour, 20m - 20 Minutes, 33m3s - 33 Minutes and 3 Seconds), maximum 1h.
 * @member {String} timeout
 * @default '5m'
 */

TargetCreateGodaddy.prototype['timeout'] = '5m';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateGodaddy.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateGodaddy.prototype['uid-token'] = undefined;
var _default = TargetCreateGodaddy;
exports["default"] = _default;
