// akeyless/dist/model/TargetCreateArtifactory.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateArtifactory model module.
 * @module model/TargetCreateArtifactory
 * @version 4.2.1
 */
var TargetCreateArtifactory = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateArtifactory</code>.
   * @alias module:model/TargetCreateArtifactory
   * @param artifactoryAdminName {String} Artifactory Admin Name
   * @param artifactoryAdminPwd {String} Artifactory Admin password
   * @param baseUrl {String} Base URL
   * @param name {String} Target name
   */
  function TargetCreateArtifactory(artifactoryAdminName, artifactoryAdminPwd, baseUrl, name) {
    _classCallCheck(this, TargetCreateArtifactory);

    TargetCreateArtifactory.initialize(this, artifactoryAdminName, artifactoryAdminPwd, baseUrl, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateArtifactory, null, [{
    key: "initialize",
    value: function initialize(obj, artifactoryAdminName, artifactoryAdminPwd, baseUrl, name) {
      obj['artifactory-admin-name'] = artifactoryAdminName;
      obj['artifactory-admin-pwd'] = artifactoryAdminPwd;
      obj['base-url'] = baseUrl;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetCreateArtifactory</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateArtifactory} obj Optional instance to populate.
     * @return {module:model/TargetCreateArtifactory} The populated <code>TargetCreateArtifactory</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateArtifactory();

        if (data.hasOwnProperty('artifactory-admin-name')) {
          obj['artifactory-admin-name'] = _ApiClient["default"].convertToType(data['artifactory-admin-name'], 'String');
        }

        if (data.hasOwnProperty('artifactory-admin-pwd')) {
          obj['artifactory-admin-pwd'] = _ApiClient["default"].convertToType(data['artifactory-admin-pwd'], 'String');
        }

        if (data.hasOwnProperty('base-url')) {
          obj['base-url'] = _ApiClient["default"].convertToType(data['base-url'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateArtifactory;
}();
/**
 * Artifactory Admin Name
 * @member {String} artifactory-admin-name
 */


TargetCreateArtifactory.prototype['artifactory-admin-name'] = undefined;
/**
 * Artifactory Admin password
 * @member {String} artifactory-admin-pwd
 */

TargetCreateArtifactory.prototype['artifactory-admin-pwd'] = undefined;
/**
 * Base URL
 * @member {String} base-url
 */

TargetCreateArtifactory.prototype['base-url'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetCreateArtifactory.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateArtifactory.prototype['json'] = false;
/**
 * The name of a key used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateArtifactory.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateArtifactory.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateArtifactory.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateArtifactory.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateArtifactory.prototype['uid-token'] = undefined;
var _default = TargetCreateArtifactory;
exports["default"] = _default;
