// akeyless/dist/model/TargetUpdateAzure.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateAzure model module.
 * @module model/TargetUpdateAzure
 * @version 4.2.2
 */
var TargetUpdateAzure = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateAzure</code>.
   * @alias module:model/TargetUpdateAzure
   * @param name {String} Target name
   */
  function TargetUpdateAzure(name) {
    _classCallCheck(this, TargetUpdateAzure);

    TargetUpdateAzure.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateAzure, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetUpdateAzure</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateAzure} obj Optional instance to populate.
     * @return {module:model/TargetUpdateAzure} The populated <code>TargetUpdateAzure</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateAzure();

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('client-secret')) {
          obj['client-secret'] = _ApiClient["default"].convertToType(data['client-secret'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('resource-group-name')) {
          obj['resource-group-name'] = _ApiClient["default"].convertToType(data['resource-group-name'], 'String');
        }

        if (data.hasOwnProperty('resource-name')) {
          obj['resource-name'] = _ApiClient["default"].convertToType(data['resource-name'], 'String');
        }

        if (data.hasOwnProperty('subscription-id')) {
          obj['subscription-id'] = _ApiClient["default"].convertToType(data['subscription-id'], 'String');
        }

        if (data.hasOwnProperty('tenant-id')) {
          obj['tenant-id'] = _ApiClient["default"].convertToType(data['tenant-id'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TargetUpdateAzure;
}();
/**
 * @member {String} client-id
 */


TargetUpdateAzure.prototype['client-id'] = undefined;
/**
 * @member {String} client-secret
 */

TargetUpdateAzure.prototype['client-secret'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateAzure.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateAzure.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateAzure.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateAzure.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateAzure.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateAzure.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateAzure.prototype['new-name'] = undefined;
/**
 * The Resource Group name in your Azure subscription
 * @member {String} resource-group-name
 */

TargetUpdateAzure.prototype['resource-group-name'] = undefined;
/**
 * The name of the relevant Resource
 * @member {String} resource-name
 */

TargetUpdateAzure.prototype['resource-name'] = undefined;
/**
 * Azure Subscription Id
 * @member {String} subscription-id
 */

TargetUpdateAzure.prototype['subscription-id'] = undefined;
/**
 * @member {String} tenant-id
 */

TargetUpdateAzure.prototype['tenant-id'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateAzure.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateAzure.prototype['uid-token'] = undefined;
/**
 * @member {Boolean} use-gw-cloud-identity
 */

TargetUpdateAzure.prototype['use-gw-cloud-identity'] = undefined;
var _default = TargetUpdateAzure;
exports["default"] = _default;
