// akeyless/dist/model/TargetCreateLinked.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateLinked model module.
 * @module model/TargetCreateLinked
 * @version 4.2.0
 */
var TargetCreateLinked = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateLinked</code>.
   * @alias module:model/TargetCreateLinked
   * @param name {String} Target name
   */
  function TargetCreateLinked(name) {
    _classCallCheck(this, TargetCreateLinked);

    TargetCreateLinked.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateLinked, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetCreateLinked</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateLinked} obj Optional instance to populate.
     * @return {module:model/TargetCreateLinked} The populated <code>TargetCreateLinked</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateLinked();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('hosts')) {
          obj['hosts'] = _ApiClient["default"].convertToType(data['hosts'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('parent-target-name')) {
          obj['parent-target-name'] = _ApiClient["default"].convertToType(data['parent-target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateLinked;
}();
/**
 * Description of the object
 * @member {String} description
 */


TargetCreateLinked.prototype['description'] = undefined;
/**
 * A comma seperated list of server hosts and server descriptions joined by semicolon ';' (i.e. 'server-dev.com;My Dev server,server-prod.com;My Prod server description')
 * @member {String} hosts
 */

TargetCreateLinked.prototype['hosts'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateLinked.prototype['json'] = false;
/**
 * Target name
 * @member {String} name
 */

TargetCreateLinked.prototype['name'] = undefined;
/**
 * The parent Target name
 * @member {String} parent-target-name
 */

TargetCreateLinked.prototype['parent-target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateLinked.prototype['token'] = undefined;
/**
 * Specifies the hosts type, relevant only when working without parent target
 * @member {String} type
 */

TargetCreateLinked.prototype['type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateLinked.prototype['uid-token'] = undefined;
var _default = TargetCreateLinked;
exports["default"] = _default;
