// akeyless/dist/model/TargetUpdateArtifactory.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateArtifactory model module.
 * @module model/TargetUpdateArtifactory
 * @version 4.2.0
 */
var TargetUpdateArtifactory = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateArtifactory</code>.
   * @alias module:model/TargetUpdateArtifactory
   * @param artifactoryAdminName {String} Artifactory Admin Name
   * @param artifactoryAdminPwd {String} Artifactory Admin password
   * @param baseUrl {String} Base URL
   * @param name {String} Target name
   */
  function TargetUpdateArtifactory(artifactoryAdminName, artifactoryAdminPwd, baseUrl, name) {
    _classCallCheck(this, TargetUpdateArtifactory);

    TargetUpdateArtifactory.initialize(this, artifactoryAdminName, artifactoryAdminPwd, baseUrl, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateArtifactory, null, [{
    key: "initialize",
    value: function initialize(obj, artifactoryAdminName, artifactoryAdminPwd, baseUrl, name) {
      obj['artifactory-admin-name'] = artifactoryAdminName;
      obj['artifactory-admin-pwd'] = artifactoryAdminPwd;
      obj['base-url'] = baseUrl;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetUpdateArtifactory</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateArtifactory} obj Optional instance to populate.
     * @return {module:model/TargetUpdateArtifactory} The populated <code>TargetUpdateArtifactory</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateArtifactory();

        if (data.hasOwnProperty('artifactory-admin-name')) {
          obj['artifactory-admin-name'] = _ApiClient["default"].convertToType(data['artifactory-admin-name'], 'String');
        }

        if (data.hasOwnProperty('artifactory-admin-pwd')) {
          obj['artifactory-admin-pwd'] = _ApiClient["default"].convertToType(data['artifactory-admin-pwd'], 'String');
        }

        if (data.hasOwnProperty('base-url')) {
          obj['base-url'] = _ApiClient["default"].convertToType(data['base-url'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetUpdateArtifactory;
}();
/**
 * Artifactory Admin Name
 * @member {String} artifactory-admin-name
 */


TargetUpdateArtifactory.prototype['artifactory-admin-name'] = undefined;
/**
 * Artifactory Admin password
 * @member {String} artifactory-admin-pwd
 */

TargetUpdateArtifactory.prototype['artifactory-admin-pwd'] = undefined;
/**
 * Base URL
 * @member {String} base-url
 */

TargetUpdateArtifactory.prototype['base-url'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateArtifactory.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateArtifactory.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateArtifactory.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateArtifactory.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateArtifactory.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateArtifactory.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateArtifactory.prototype['new-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateArtifactory.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateArtifactory.prototype['uid-token'] = undefined;
var _default = TargetUpdateArtifactory;
exports["default"] = _default;
