// akeyless/dist/model/TargetUpdateSectigo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateSectigo model module.
 * @module model/TargetUpdateSectigo
 * @version 4.3.0
 */
var TargetUpdateSectigo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateSectigo</code>.
   * targetUpdateSectigo is a command that updates an existing Sectigo target
   * @alias module:model/TargetUpdateSectigo
   * @param certificateProfileId {Number} Certificate Profile ID in Sectigo account
   * @param customerUri {String} Customer Uri of the Sectigo account
   * @param externalRequester {String} External Requester - a comma separated list of emails
   * @param name {String} Target name
   * @param organizationId {Number} Organization ID in Sectigo account
   * @param password {String} Password of the Sectigo account user
   * @param username {String} Username of the Sectigo account
   */
  function TargetUpdateSectigo(certificateProfileId, customerUri, externalRequester, name, organizationId, password, username) {
    _classCallCheck(this, TargetUpdateSectigo);

    TargetUpdateSectigo.initialize(this, certificateProfileId, customerUri, externalRequester, name, organizationId, password, username);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateSectigo, null, [{
    key: "initialize",
    value: function initialize(obj, certificateProfileId, customerUri, externalRequester, name, organizationId, password, username) {
      obj['certificate-profile-id'] = certificateProfileId;
      obj['customer-uri'] = customerUri;
      obj['external-requester'] = externalRequester;
      obj['name'] = name;
      obj['organization-id'] = organizationId;
      obj['password'] = password;
      obj['username'] = username;
    }
    /**
     * Constructs a <code>TargetUpdateSectigo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateSectigo} obj Optional instance to populate.
     * @return {module:model/TargetUpdateSectigo} The populated <code>TargetUpdateSectigo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateSectigo();

        if (data.hasOwnProperty('certificate-profile-id')) {
          obj['certificate-profile-id'] = _ApiClient["default"].convertToType(data['certificate-profile-id'], 'Number');
        }

        if (data.hasOwnProperty('customer-uri')) {
          obj['customer-uri'] = _ApiClient["default"].convertToType(data['customer-uri'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('external-requester')) {
          obj['external-requester'] = _ApiClient["default"].convertToType(data['external-requester'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('organization-id')) {
          obj['organization-id'] = _ApiClient["default"].convertToType(data['organization-id'], 'Number');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetUpdateSectigo;
}();
/**
 * Certificate Profile ID in Sectigo account
 * @member {Number} certificate-profile-id
 */


TargetUpdateSectigo.prototype['certificate-profile-id'] = undefined;
/**
 * Customer Uri of the Sectigo account
 * @member {String} customer-uri
 */

TargetUpdateSectigo.prototype['customer-uri'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateSectigo.prototype['description'] = undefined;
/**
 * External Requester - a comma separated list of emails
 * @member {String} external-requester
 */

TargetUpdateSectigo.prototype['external-requester'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateSectigo.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateSectigo.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateSectigo.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateSectigo.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateSectigo.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateSectigo.prototype['new-name'] = undefined;
/**
 * Organization ID in Sectigo account
 * @member {Number} organization-id
 */

TargetUpdateSectigo.prototype['organization-id'] = undefined;
/**
 * Password of the Sectigo account user
 * @member {String} password
 */

TargetUpdateSectigo.prototype['password'] = undefined;
/**
 * Timeout waiting for certificate validation in Duration format (1h - 1 Hour, 20m - 20 Minutes, 33m3s - 33 Minutes and 3 Seconds), maximum 1h.
 * @member {String} timeout
 * @default '5m'
 */

TargetUpdateSectigo.prototype['timeout'] = '5m';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateSectigo.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateSectigo.prototype['uid-token'] = undefined;
/**
 * Username of the Sectigo account
 * @member {String} username
 */

TargetUpdateSectigo.prototype['username'] = undefined;
var _default = TargetUpdateSectigo;
exports["default"] = _default;
