// akeyless/dist/model/TargetCreateSsh.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateSsh model module.
 * @module model/TargetCreateSsh
 * @version 4.2.0
 */
var TargetCreateSsh = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateSsh</code>.
   * targetCreateSsh is a command that creates a new ssh target
   * @alias module:model/TargetCreateSsh
   * @param name {String} Target name
   */
  function TargetCreateSsh(name) {
    _classCallCheck(this, TargetCreateSsh);

    TargetCreateSsh.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateSsh, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetCreateSsh</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateSsh} obj Optional instance to populate.
     * @return {module:model/TargetCreateSsh} The populated <code>TargetCreateSsh</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateSsh();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('private-key')) {
          obj['private-key'] = _ApiClient["default"].convertToType(data['private-key'], 'String');
        }

        if (data.hasOwnProperty('private-key-password')) {
          obj['private-key-password'] = _ApiClient["default"].convertToType(data['private-key-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateSsh;
}();
/**
 * Description of the object
 * @member {String} description
 */


TargetCreateSsh.prototype['description'] = undefined;
/**
 * SSH host name
 * @member {String} host
 */

TargetCreateSsh.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateSsh.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateSsh.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateSsh.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateSsh.prototype['name'] = undefined;
/**
 * SSH port
 * @member {String} port
 * @default '22'
 */

TargetCreateSsh.prototype['port'] = '22';
/**
 * SSH private key
 * @member {String} private-key
 */

TargetCreateSsh.prototype['private-key'] = undefined;
/**
 * SSH private key password
 * @member {String} private-key-password
 */

TargetCreateSsh.prototype['private-key-password'] = undefined;
/**
 * SSH password to rotate
 * @member {String} ssh-password
 */

TargetCreateSsh.prototype['ssh-password'] = undefined;
/**
 * SSH username
 * @member {String} ssh-username
 */

TargetCreateSsh.prototype['ssh-username'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateSsh.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateSsh.prototype['uid-token'] = undefined;
var _default = TargetCreateSsh;
exports["default"] = _default;
