// akeyless/dist/model/TargetUpdateSalesforce.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateSalesforce model module.
 * @module model/TargetUpdateSalesforce
 * @version 5.0.1
 */
var TargetUpdateSalesforce = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateSalesforce</code>.
   * targetUpdateSalesforce is a command that updates an existing Salesforce target
   * @alias module:model/TargetUpdateSalesforce
   * @param authFlow {String} type of the auth flow ('jwt' / 'user-password')
   * @param clientId {String} Client ID of the oauth2 app to use for connecting to Salesforce
   * @param email {String} The email of the user attached to the oauth2 app used for connecting to Salesforce
   * @param name {String} Target name
   * @param tenantUrl {String} Url of the Salesforce tenant
   */
  function TargetUpdateSalesforce(authFlow, clientId, email, name, tenantUrl) {
    _classCallCheck(this, TargetUpdateSalesforce);

    TargetUpdateSalesforce.initialize(this, authFlow, clientId, email, name, tenantUrl);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateSalesforce, null, [{
    key: "initialize",
    value: function initialize(obj, authFlow, clientId, email, name, tenantUrl) {
      obj['auth-flow'] = authFlow;
      obj['client-id'] = clientId;
      obj['email'] = email;
      obj['json'] = false;
      obj['name'] = name;
      obj['tenant-url'] = tenantUrl;
    }
    /**
     * Constructs a <code>TargetUpdateSalesforce</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateSalesforce} obj Optional instance to populate.
     * @return {module:model/TargetUpdateSalesforce} The populated <code>TargetUpdateSalesforce</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateSalesforce();

        if (data.hasOwnProperty('app-private-key-data')) {
          obj['app-private-key-data'] = _ApiClient["default"].convertToType(data['app-private-key-data'], 'String');
        }

        if (data.hasOwnProperty('auth-flow')) {
          obj['auth-flow'] = _ApiClient["default"].convertToType(data['auth-flow'], 'String');
        }

        if (data.hasOwnProperty('ca-cert-data')) {
          obj['ca-cert-data'] = _ApiClient["default"].convertToType(data['ca-cert-data'], 'String');
        }

        if (data.hasOwnProperty('ca-cert-name')) {
          obj['ca-cert-name'] = _ApiClient["default"].convertToType(data['ca-cert-name'], 'String');
        }

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('client-secret')) {
          obj['client-secret'] = _ApiClient["default"].convertToType(data['client-secret'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('security-token')) {
          obj['security-token'] = _ApiClient["default"].convertToType(data['security-token'], 'String');
        }

        if (data.hasOwnProperty('tenant-url')) {
          obj['tenant-url'] = _ApiClient["default"].convertToType(data['tenant-url'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TargetUpdateSalesforce</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TargetUpdateSalesforce</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(TargetUpdateSalesforce.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['app-private-key-data'] && !(typeof data['app-private-key-data'] === 'string' || data['app-private-key-data'] instanceof String)) {
        throw new Error("Expected the field `app-private-key-data` to be a primitive type in the JSON string but got " + data['app-private-key-data']);
      } // ensure the json data is a string


      if (data['auth-flow'] && !(typeof data['auth-flow'] === 'string' || data['auth-flow'] instanceof String)) {
        throw new Error("Expected the field `auth-flow` to be a primitive type in the JSON string but got " + data['auth-flow']);
      } // ensure the json data is a string


      if (data['ca-cert-data'] && !(typeof data['ca-cert-data'] === 'string' || data['ca-cert-data'] instanceof String)) {
        throw new Error("Expected the field `ca-cert-data` to be a primitive type in the JSON string but got " + data['ca-cert-data']);
      } // ensure the json data is a string


      if (data['ca-cert-name'] && !(typeof data['ca-cert-name'] === 'string' || data['ca-cert-name'] instanceof String)) {
        throw new Error("Expected the field `ca-cert-name` to be a primitive type in the JSON string but got " + data['ca-cert-name']);
      } // ensure the json data is a string


      if (data['client-id'] && !(typeof data['client-id'] === 'string' || data['client-id'] instanceof String)) {
        throw new Error("Expected the field `client-id` to be a primitive type in the JSON string but got " + data['client-id']);
      } // ensure the json data is a string


      if (data['client-secret'] && !(typeof data['client-secret'] === 'string' || data['client-secret'] instanceof String)) {
        throw new Error("Expected the field `client-secret` to be a primitive type in the JSON string but got " + data['client-secret']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['email'] && !(typeof data['email'] === 'string' || data['email'] instanceof String)) {
        throw new Error("Expected the field `email` to be a primitive type in the JSON string but got " + data['email']);
      } // ensure the json data is a string


      if (data['keep-prev-version'] && !(typeof data['keep-prev-version'] === 'string' || data['keep-prev-version'] instanceof String)) {
        throw new Error("Expected the field `keep-prev-version` to be a primitive type in the JSON string but got " + data['keep-prev-version']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['password'] && !(typeof data['password'] === 'string' || data['password'] instanceof String)) {
        throw new Error("Expected the field `password` to be a primitive type in the JSON string but got " + data['password']);
      } // ensure the json data is a string


      if (data['security-token'] && !(typeof data['security-token'] === 'string' || data['security-token'] instanceof String)) {
        throw new Error("Expected the field `security-token` to be a primitive type in the JSON string but got " + data['security-token']);
      } // ensure the json data is a string


      if (data['tenant-url'] && !(typeof data['tenant-url'] === 'string' || data['tenant-url'] instanceof String)) {
        throw new Error("Expected the field `tenant-url` to be a primitive type in the JSON string but got " + data['tenant-url']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return TargetUpdateSalesforce;
}();

TargetUpdateSalesforce.RequiredProperties = ["auth-flow", "client-id", "email", "name", "tenant-url"];
/**
 * Base64 encoded PEM of the connected app private key (relevant for JWT auth only)
 * @member {String} app-private-key-data
 */

TargetUpdateSalesforce.prototype['app-private-key-data'] = undefined;
/**
 * type of the auth flow ('jwt' / 'user-password')
 * @member {String} auth-flow
 */

TargetUpdateSalesforce.prototype['auth-flow'] = undefined;
/**
 * Base64 encoded PEM cert to use when uploading a new key to Salesforce
 * @member {String} ca-cert-data
 */

TargetUpdateSalesforce.prototype['ca-cert-data'] = undefined;
/**
 * name of the certificate in Salesforce tenant to use when uploading new key
 * @member {String} ca-cert-name
 */

TargetUpdateSalesforce.prototype['ca-cert-name'] = undefined;
/**
 * Client ID of the oauth2 app to use for connecting to Salesforce
 * @member {String} client-id
 */

TargetUpdateSalesforce.prototype['client-id'] = undefined;
/**
 * Client secret of the oauth2 app to use for connecting to Salesforce (required for password flow)
 * @member {String} client-secret
 */

TargetUpdateSalesforce.prototype['client-secret'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateSalesforce.prototype['description'] = undefined;
/**
 * The email of the user attached to the oauth2 app used for connecting to Salesforce
 * @member {String} email
 */

TargetUpdateSalesforce.prototype['email'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateSalesforce.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateSalesforce.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateSalesforce.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateSalesforce.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateSalesforce.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateSalesforce.prototype['new-name'] = undefined;
/**
 * The password of the user attached to the oauth2 app used for connecting to Salesforce (required for user-password flow)
 * @member {String} password
 */

TargetUpdateSalesforce.prototype['password'] = undefined;
/**
 * The security token of the user attached to the oauth2 app used for connecting to Salesforce  (required for user-password flow)
 * @member {String} security-token
 */

TargetUpdateSalesforce.prototype['security-token'] = undefined;
/**
 * Url of the Salesforce tenant
 * @member {String} tenant-url
 */

TargetUpdateSalesforce.prototype['tenant-url'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateSalesforce.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateSalesforce.prototype['uid-token'] = undefined;
var _default = TargetUpdateSalesforce;
exports["default"] = _default;
