// akeyless/dist/model/TargetCreateGke.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateGke model module.
 * @module model/TargetCreateGke
 * @version 4.2.2
 */
var TargetCreateGke = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateGke</code>.
   * @alias module:model/TargetCreateGke
   * @param name {String} Target name
   */
  function TargetCreateGke(name) {
    _classCallCheck(this, TargetCreateGke);

    TargetCreateGke.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateGke, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetCreateGke</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateGke} obj Optional instance to populate.
     * @return {module:model/TargetCreateGke} The populated <code>TargetCreateGke</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateGke();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('gke-account-key')) {
          obj['gke-account-key'] = _ApiClient["default"].convertToType(data['gke-account-key'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-cert')) {
          obj['gke-cluster-cert'] = _ApiClient["default"].convertToType(data['gke-cluster-cert'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-endpoint')) {
          obj['gke-cluster-endpoint'] = _ApiClient["default"].convertToType(data['gke-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-name')) {
          obj['gke-cluster-name'] = _ApiClient["default"].convertToType(data['gke-cluster-name'], 'String');
        }

        if (data.hasOwnProperty('gke-service-account-email')) {
          obj['gke-service-account-email'] = _ApiClient["default"].convertToType(data['gke-service-account-email'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateGke;
}();
/**
 * Description of the object
 * @member {String} description
 */


TargetCreateGke.prototype['description'] = undefined;
/**
 * GKE Service Account key file path
 * @member {String} gke-account-key
 */

TargetCreateGke.prototype['gke-account-key'] = undefined;
/**
 * GKE cluster CA certificate
 * @member {String} gke-cluster-cert
 */

TargetCreateGke.prototype['gke-cluster-cert'] = undefined;
/**
 * GKE cluster URL endpoint
 * @member {String} gke-cluster-endpoint
 */

TargetCreateGke.prototype['gke-cluster-endpoint'] = undefined;
/**
 * GKE cluster name
 * @member {String} gke-cluster-name
 */

TargetCreateGke.prototype['gke-cluster-name'] = undefined;
/**
 * GKE service account email
 * @member {String} gke-service-account-email
 */

TargetCreateGke.prototype['gke-service-account-email'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateGke.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateGke.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateGke.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateGke.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateGke.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateGke.prototype['uid-token'] = undefined;
/**
 * @member {Boolean} use-gw-cloud-identity
 */

TargetCreateGke.prototype['use-gw-cloud-identity'] = undefined;
var _default = TargetCreateGke;
exports["default"] = _default;
