// akeyless/dist/model/TargetCreateRabbitMq.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateRabbitMq model module.
 * @module model/TargetCreateRabbitMq
 * @version 5.0.3
 */
var TargetCreateRabbitMq = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateRabbitMq</code>.
   * targetCreateRabbitMq is a command that creates a new rabbitmq target
   * @alias module:model/TargetCreateRabbitMq
   * @param name {String} Target name
   */
  function TargetCreateRabbitMq(name) {
    _classCallCheck(this, TargetCreateRabbitMq);

    TargetCreateRabbitMq.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateRabbitMq, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['json'] = false;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetCreateRabbitMq</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateRabbitMq} obj Optional instance to populate.
     * @return {module:model/TargetCreateRabbitMq} The populated <code>TargetCreateRabbitMq</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateRabbitMq();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-password')) {
          obj['rabbitmq-server-password'] = _ApiClient["default"].convertToType(data['rabbitmq-server-password'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-uri')) {
          obj['rabbitmq-server-uri'] = _ApiClient["default"].convertToType(data['rabbitmq-server-uri'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-user')) {
          obj['rabbitmq-server-user'] = _ApiClient["default"].convertToType(data['rabbitmq-server-user'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TargetCreateRabbitMq</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TargetCreateRabbitMq</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(TargetCreateRabbitMq.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['rabbitmq-server-password'] && !(typeof data['rabbitmq-server-password'] === 'string' || data['rabbitmq-server-password'] instanceof String)) {
        throw new Error("Expected the field `rabbitmq-server-password` to be a primitive type in the JSON string but got " + data['rabbitmq-server-password']);
      } // ensure the json data is a string


      if (data['rabbitmq-server-uri'] && !(typeof data['rabbitmq-server-uri'] === 'string' || data['rabbitmq-server-uri'] instanceof String)) {
        throw new Error("Expected the field `rabbitmq-server-uri` to be a primitive type in the JSON string but got " + data['rabbitmq-server-uri']);
      } // ensure the json data is a string


      if (data['rabbitmq-server-user'] && !(typeof data['rabbitmq-server-user'] === 'string' || data['rabbitmq-server-user'] instanceof String)) {
        throw new Error("Expected the field `rabbitmq-server-user` to be a primitive type in the JSON string but got " + data['rabbitmq-server-user']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return TargetCreateRabbitMq;
}();

TargetCreateRabbitMq.RequiredProperties = ["name"];
/**
 * Description of the object
 * @member {String} description
 */

TargetCreateRabbitMq.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateRabbitMq.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateRabbitMq.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateRabbitMq.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateRabbitMq.prototype['name'] = undefined;
/**
 * @member {String} rabbitmq-server-password
 */

TargetCreateRabbitMq.prototype['rabbitmq-server-password'] = undefined;
/**
 * @member {String} rabbitmq-server-uri
 */

TargetCreateRabbitMq.prototype['rabbitmq-server-uri'] = undefined;
/**
 * @member {String} rabbitmq-server-user
 */

TargetCreateRabbitMq.prototype['rabbitmq-server-user'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateRabbitMq.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateRabbitMq.prototype['uid-token'] = undefined;
var _default = TargetCreateRabbitMq;
exports["default"] = _default;
