// akeyless/dist/model/TargetCreateSalesforce.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetCreateSalesforce model module.
 * @module model/TargetCreateSalesforce
 * @version 4.2.1
 */
var TargetCreateSalesforce = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetCreateSalesforce</code>.
   * targetCreateSalesforce is a command that creates a new Salesforce target
   * @alias module:model/TargetCreateSalesforce
   * @param authFlow {String} type of the auth flow ('jwt' / 'user-password')
   * @param clientId {String} Client ID of the oauth2 app to use for connecting to Salesforce
   * @param email {String} The email of the user attached to the oauth2 app used for connecting to Salesforce
   * @param name {String} Target name
   * @param tenantUrl {String} Url of the Salesforce tenant
   */
  function TargetCreateSalesforce(authFlow, clientId, email, name, tenantUrl) {
    _classCallCheck(this, TargetCreateSalesforce);

    TargetCreateSalesforce.initialize(this, authFlow, clientId, email, name, tenantUrl);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetCreateSalesforce, null, [{
    key: "initialize",
    value: function initialize(obj, authFlow, clientId, email, name, tenantUrl) {
      obj['auth-flow'] = authFlow;
      obj['client-id'] = clientId;
      obj['email'] = email;
      obj['name'] = name;
      obj['tenant-url'] = tenantUrl;
    }
    /**
     * Constructs a <code>TargetCreateSalesforce</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetCreateSalesforce} obj Optional instance to populate.
     * @return {module:model/TargetCreateSalesforce} The populated <code>TargetCreateSalesforce</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetCreateSalesforce();

        if (data.hasOwnProperty('app-private-key-data')) {
          obj['app-private-key-data'] = _ApiClient["default"].convertToType(data['app-private-key-data'], 'String');
        }

        if (data.hasOwnProperty('auth-flow')) {
          obj['auth-flow'] = _ApiClient["default"].convertToType(data['auth-flow'], 'String');
        }

        if (data.hasOwnProperty('ca-cert-data')) {
          obj['ca-cert-data'] = _ApiClient["default"].convertToType(data['ca-cert-data'], 'String');
        }

        if (data.hasOwnProperty('ca-cert-name')) {
          obj['ca-cert-name'] = _ApiClient["default"].convertToType(data['ca-cert-name'], 'String');
        }

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('client-secret')) {
          obj['client-secret'] = _ApiClient["default"].convertToType(data['client-secret'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('security-token')) {
          obj['security-token'] = _ApiClient["default"].convertToType(data['security-token'], 'String');
        }

        if (data.hasOwnProperty('tenant-url')) {
          obj['tenant-url'] = _ApiClient["default"].convertToType(data['tenant-url'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetCreateSalesforce;
}();
/**
 * Base64 encoded PEM of the connected app private key (relevant for JWT auth only)
 * @member {String} app-private-key-data
 */


TargetCreateSalesforce.prototype['app-private-key-data'] = undefined;
/**
 * type of the auth flow ('jwt' / 'user-password')
 * @member {String} auth-flow
 */

TargetCreateSalesforce.prototype['auth-flow'] = undefined;
/**
 * Base64 encoded PEM cert to use when uploading a new key to Salesforce
 * @member {String} ca-cert-data
 */

TargetCreateSalesforce.prototype['ca-cert-data'] = undefined;
/**
 * name of the certificate in Salesforce tenant to use when uploading new key
 * @member {String} ca-cert-name
 */

TargetCreateSalesforce.prototype['ca-cert-name'] = undefined;
/**
 * Client ID of the oauth2 app to use for connecting to Salesforce
 * @member {String} client-id
 */

TargetCreateSalesforce.prototype['client-id'] = undefined;
/**
 * Client secret of the oauth2 app to use for connecting to Salesforce (required for password flow)
 * @member {String} client-secret
 */

TargetCreateSalesforce.prototype['client-secret'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetCreateSalesforce.prototype['description'] = undefined;
/**
 * The email of the user attached to the oauth2 app used for connecting to Salesforce
 * @member {String} email
 */

TargetCreateSalesforce.prototype['email'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetCreateSalesforce.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetCreateSalesforce.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetCreateSalesforce.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetCreateSalesforce.prototype['name'] = undefined;
/**
 * The password of the user attached to the oauth2 app used for connecting to Salesforce (required for user-password flow)
 * @member {String} password
 */

TargetCreateSalesforce.prototype['password'] = undefined;
/**
 * The security token of the user attached to the oauth2 app used for connecting to Salesforce  (required for user-password flow)
 * @member {String} security-token
 */

TargetCreateSalesforce.prototype['security-token'] = undefined;
/**
 * Url of the Salesforce tenant
 * @member {String} tenant-url
 */

TargetCreateSalesforce.prototype['tenant-url'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetCreateSalesforce.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetCreateSalesforce.prototype['uid-token'] = undefined;
var _default = TargetCreateSalesforce;
exports["default"] = _default;
