// akeyless/dist/model/TargetUpdateAws.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateAws model module.
 * @module model/TargetUpdateAws
 * @version 4.2.2
 */
var TargetUpdateAws = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateAws</code>.
   * @alias module:model/TargetUpdateAws
   * @param accessKey {String} AWS secret access key
   * @param accessKeyId {String} AWS access key ID
   * @param name {String} Target name
   */
  function TargetUpdateAws(accessKey, accessKeyId, name) {
    _classCallCheck(this, TargetUpdateAws);

    TargetUpdateAws.initialize(this, accessKey, accessKeyId, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateAws, null, [{
    key: "initialize",
    value: function initialize(obj, accessKey, accessKeyId, name) {
      obj['access-key'] = accessKey;
      obj['access-key-id'] = accessKeyId;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetUpdateAws</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateAws} obj Optional instance to populate.
     * @return {module:model/TargetUpdateAws} The populated <code>TargetUpdateAws</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateAws();

        if (data.hasOwnProperty('access-key')) {
          obj['access-key'] = _ApiClient["default"].convertToType(data['access-key'], 'String');
        }

        if (data.hasOwnProperty('access-key-id')) {
          obj['access-key-id'] = _ApiClient["default"].convertToType(data['access-key-id'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('session-token')) {
          obj['session-token'] = _ApiClient["default"].convertToType(data['session-token'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return TargetUpdateAws;
}();
/**
 * AWS secret access key
 * @member {String} access-key
 */


TargetUpdateAws.prototype['access-key'] = undefined;
/**
 * AWS access key ID
 * @member {String} access-key-id
 */

TargetUpdateAws.prototype['access-key-id'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateAws.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateAws.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateAws.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateAws.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateAws.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateAws.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateAws.prototype['new-name'] = undefined;
/**
 * AWS region
 * @member {String} region
 * @default 'us-east-2'
 */

TargetUpdateAws.prototype['region'] = 'us-east-2';
/**
 * Required only for temporary security credentials retrieved using STS
 * @member {String} session-token
 */

TargetUpdateAws.prototype['session-token'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateAws.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateAws.prototype['uid-token'] = undefined;
/**
 * Use the GW's Cloud IAM
 * @member {Boolean} use-gw-cloud-identity
 */

TargetUpdateAws.prototype['use-gw-cloud-identity'] = undefined;
var _default = TargetUpdateAws;
exports["default"] = _default;
