// akeyless/dist/model/TargetNameWithHosts.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetNameWithHosts model module.
 * @module model/TargetNameWithHosts
 * @version 5.0.8
 */
var TargetNameWithHosts = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetNameWithHosts</code>.
   * @alias module:model/TargetNameWithHosts
   */
  function TargetNameWithHosts() {
    _classCallCheck(this, TargetNameWithHosts);

    TargetNameWithHosts.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetNameWithHosts, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TargetNameWithHosts</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetNameWithHosts} obj Optional instance to populate.
     * @return {module:model/TargetNameWithHosts} The populated <code>TargetNameWithHosts</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetNameWithHosts();

        if (data.hasOwnProperty('hosts')) {
          obj['hosts'] = _ApiClient["default"].convertToType(data['hosts'], ['String']);
        }

        if (data.hasOwnProperty('target_name')) {
          obj['target_name'] = _ApiClient["default"].convertToType(data['target_name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TargetNameWithHosts</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TargetNameWithHosts</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data['hosts'])) {
        throw new Error("Expected the field `hosts` to be an array in the JSON data but got " + data['hosts']);
      } // ensure the json data is a string


      if (data['target_name'] && !(typeof data['target_name'] === 'string' || data['target_name'] instanceof String)) {
        throw new Error("Expected the field `target_name` to be a primitive type in the JSON string but got " + data['target_name']);
      }

      return true;
    }
  }]);

  return TargetNameWithHosts;
}();
/**
 * @member {Array.<String>} hosts
 */


TargetNameWithHosts.prototype['hosts'] = undefined;
/**
 * @member {String} target_name
 */

TargetNameWithHosts.prototype['target_name'] = undefined;
var _default = TargetNameWithHosts;
exports["default"] = _default;
