// akeyless/dist/model/TargetUpdateRabbitMq.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateRabbitMq model module.
 * @module model/TargetUpdateRabbitMq
 * @version 4.2.0
 */
var TargetUpdateRabbitMq = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateRabbitMq</code>.
   * targetUpdateRabbitMq is a command that updates an existing rabbitmq target
   * @alias module:model/TargetUpdateRabbitMq
   * @param name {String} Target name
   */
  function TargetUpdateRabbitMq(name) {
    _classCallCheck(this, TargetUpdateRabbitMq);

    TargetUpdateRabbitMq.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateRabbitMq, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>TargetUpdateRabbitMq</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateRabbitMq} obj Optional instance to populate.
     * @return {module:model/TargetUpdateRabbitMq} The populated <code>TargetUpdateRabbitMq</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateRabbitMq();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-password')) {
          obj['rabbitmq-server-password'] = _ApiClient["default"].convertToType(data['rabbitmq-server-password'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-uri')) {
          obj['rabbitmq-server-uri'] = _ApiClient["default"].convertToType(data['rabbitmq-server-uri'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-user')) {
          obj['rabbitmq-server-user'] = _ApiClient["default"].convertToType(data['rabbitmq-server-user'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetUpdateRabbitMq;
}();
/**
 * Description of the object
 * @member {String} description
 */


TargetUpdateRabbitMq.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateRabbitMq.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateRabbitMq.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateRabbitMq.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateRabbitMq.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateRabbitMq.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateRabbitMq.prototype['new-name'] = undefined;
/**
 * @member {String} rabbitmq-server-password
 */

TargetUpdateRabbitMq.prototype['rabbitmq-server-password'] = undefined;
/**
 * @member {String} rabbitmq-server-uri
 */

TargetUpdateRabbitMq.prototype['rabbitmq-server-uri'] = undefined;
/**
 * @member {String} rabbitmq-server-user
 */

TargetUpdateRabbitMq.prototype['rabbitmq-server-user'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateRabbitMq.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateRabbitMq.prototype['uid-token'] = undefined;
var _default = TargetUpdateRabbitMq;
exports["default"] = _default;
