// akeyless/dist/model/TargetUpdateWindows.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateWindows model module.
 * @module model/TargetUpdateWindows
 * @version 5.0.14
 */
var TargetUpdateWindows = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateWindows</code>.
   * GatewayUpdateProducerWindowsCmd is a command that updates an existing windows target
   * @alias module:model/TargetUpdateWindows
   * @param hostname {String} Server hostname
   * @param name {String} Target name
   * @param password {String} Privileged user password
   * @param username {String} Privileged username
   */
  function TargetUpdateWindows(hostname, name, password, username) {
    _classCallCheck(this, TargetUpdateWindows);

    TargetUpdateWindows.initialize(this, hostname, name, password, username);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateWindows, null, [{
    key: "initialize",
    value: function initialize(obj, hostname, name, password, username) {
      obj['connection-type'] = 'credentials';
      obj['hostname'] = hostname;
      obj['json'] = false;
      obj['name'] = name;
      obj['password'] = password || 'dummy_value';
      obj['port'] = '5986';
      obj['use-tls'] = 'true';
      obj['username'] = username || 'dummy_value';
    }
    /**
     * Constructs a <code>TargetUpdateWindows</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateWindows} obj Optional instance to populate.
     * @return {module:model/TargetUpdateWindows} The populated <code>TargetUpdateWindows</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateWindows();

        if (data.hasOwnProperty('certificate')) {
          obj['certificate'] = _ApiClient["default"].convertToType(data['certificate'], 'String');
        }

        if (data.hasOwnProperty('connection-type')) {
          obj['connection-type'] = _ApiClient["default"].convertToType(data['connection-type'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('domain')) {
          obj['domain'] = _ApiClient["default"].convertToType(data['domain'], 'String');
        }

        if (data.hasOwnProperty('hostname')) {
          obj['hostname'] = _ApiClient["default"].convertToType(data['hostname'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('parent-target-name')) {
          obj['parent-target-name'] = _ApiClient["default"].convertToType(data['parent-target-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-tls')) {
          obj['use-tls'] = _ApiClient["default"].convertToType(data['use-tls'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TargetUpdateWindows</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TargetUpdateWindows</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(TargetUpdateWindows.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['certificate'] && !(typeof data['certificate'] === 'string' || data['certificate'] instanceof String)) {
        throw new Error("Expected the field `certificate` to be a primitive type in the JSON string but got " + data['certificate']);
      } // ensure the json data is a string


      if (data['connection-type'] && !(typeof data['connection-type'] === 'string' || data['connection-type'] instanceof String)) {
        throw new Error("Expected the field `connection-type` to be a primitive type in the JSON string but got " + data['connection-type']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['domain'] && !(typeof data['domain'] === 'string' || data['domain'] instanceof String)) {
        throw new Error("Expected the field `domain` to be a primitive type in the JSON string but got " + data['domain']);
      } // ensure the json data is a string


      if (data['hostname'] && !(typeof data['hostname'] === 'string' || data['hostname'] instanceof String)) {
        throw new Error("Expected the field `hostname` to be a primitive type in the JSON string but got " + data['hostname']);
      } // ensure the json data is a string


      if (data['keep-prev-version'] && !(typeof data['keep-prev-version'] === 'string' || data['keep-prev-version'] instanceof String)) {
        throw new Error("Expected the field `keep-prev-version` to be a primitive type in the JSON string but got " + data['keep-prev-version']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['parent-target-name'] && !(typeof data['parent-target-name'] === 'string' || data['parent-target-name'] instanceof String)) {
        throw new Error("Expected the field `parent-target-name` to be a primitive type in the JSON string but got " + data['parent-target-name']);
      } // ensure the json data is a string


      if (data['password'] && !(typeof data['password'] === 'string' || data['password'] instanceof String)) {
        throw new Error("Expected the field `password` to be a primitive type in the JSON string but got " + data['password']);
      } // ensure the json data is a string


      if (data['port'] && !(typeof data['port'] === 'string' || data['port'] instanceof String)) {
        throw new Error("Expected the field `port` to be a primitive type in the JSON string but got " + data['port']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['use-tls'] && !(typeof data['use-tls'] === 'string' || data['use-tls'] instanceof String)) {
        throw new Error("Expected the field `use-tls` to be a primitive type in the JSON string but got " + data['use-tls']);
      } // ensure the json data is a string


      if (data['username'] && !(typeof data['username'] === 'string' || data['username'] instanceof String)) {
        throw new Error("Expected the field `username` to be a primitive type in the JSON string but got " + data['username']);
      }

      return true;
    }
  }]);

  return TargetUpdateWindows;
}();

TargetUpdateWindows.RequiredProperties = ["hostname", "name", "password", "username"];
/**
 * SSL CA certificate in base64 encoding generated from a trusted Certificate Authority (CA)
 * @member {String} certificate
 */

TargetUpdateWindows.prototype['certificate'] = undefined;
/**
 * Type of connection to Windows Server [credentials/parent-target]
 * @member {String} connection-type
 * @default 'credentials'
 */

TargetUpdateWindows.prototype['connection-type'] = 'credentials';
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateWindows.prototype['description'] = undefined;
/**
 * User domain name
 * @member {String} domain
 */

TargetUpdateWindows.prototype['domain'] = undefined;
/**
 * Server hostname
 * @member {String} hostname
 */

TargetUpdateWindows.prototype['hostname'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateWindows.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateWindows.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateWindows.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateWindows.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateWindows.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateWindows.prototype['new-name'] = undefined;
/**
 * Name of the parent target, relevant only when connection-type is parent-target
 * @member {String} parent-target-name
 */

TargetUpdateWindows.prototype['parent-target-name'] = undefined;
/**
 * Privileged user password
 * @member {String} password
 * @default 'dummy_value'
 */

TargetUpdateWindows.prototype['password'] = 'dummy_value';
/**
 * Server WinRM port
 * @member {String} port
 * @default '5986'
 */

TargetUpdateWindows.prototype['port'] = '5986';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateWindows.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateWindows.prototype['uid-token'] = undefined;
/**
 * Enable/Disable TLS for WinRM over HTTPS [true/false]
 * @member {String} use-tls
 * @default 'true'
 */

TargetUpdateWindows.prototype['use-tls'] = 'true';
/**
 * Privileged username
 * @member {String} username
 * @default 'dummy_value'
 */

TargetUpdateWindows.prototype['username'] = 'dummy_value';
var _default = TargetUpdateWindows;
exports["default"] = _default;
