// akeyless/dist/model/TargetUpdateZeroSSL.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetUpdateZeroSSL model module.
 * @module model/TargetUpdateZeroSSL
 * @version 5.0.6
 */
var TargetUpdateZeroSSL = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetUpdateZeroSSL</code>.
   * targetUpdateZeroSSL is a command that updates an existing ZeroSSL target
   * @alias module:model/TargetUpdateZeroSSL
   * @param apiKey {String} API Key of the ZeroSSLTarget account
   * @param imapFqdn {String} ImapFQDN of the IMAP service, FQDN or IPv4 address. Must be FQDN if the IMAP is using TLS
   * @param imapPassword {String} ImapPassword to access the IMAP service
   * @param imapUsername {String} ImapUsername to access the IMAP service
   * @param name {String} Target name
   */
  function TargetUpdateZeroSSL(apiKey, imapFqdn, imapPassword, imapUsername, name) {
    _classCallCheck(this, TargetUpdateZeroSSL);

    TargetUpdateZeroSSL.initialize(this, apiKey, imapFqdn, imapPassword, imapUsername, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetUpdateZeroSSL, null, [{
    key: "initialize",
    value: function initialize(obj, apiKey, imapFqdn, imapPassword, imapUsername, name) {
      obj['api-key'] = apiKey;
      obj['imap-fqdn'] = imapFqdn;
      obj['imap-password'] = imapPassword;
      obj['imap-port'] = '993';
      obj['imap-username'] = imapUsername;
      obj['json'] = false;
      obj['name'] = name;
      obj['timeout'] = '5m';
    }
    /**
     * Constructs a <code>TargetUpdateZeroSSL</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetUpdateZeroSSL} obj Optional instance to populate.
     * @return {module:model/TargetUpdateZeroSSL} The populated <code>TargetUpdateZeroSSL</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetUpdateZeroSSL();

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('imap-fqdn')) {
          obj['imap-fqdn'] = _ApiClient["default"].convertToType(data['imap-fqdn'], 'String');
        }

        if (data.hasOwnProperty('imap-password')) {
          obj['imap-password'] = _ApiClient["default"].convertToType(data['imap-password'], 'String');
        }

        if (data.hasOwnProperty('imap-port')) {
          obj['imap-port'] = _ApiClient["default"].convertToType(data['imap-port'], 'String');
        }

        if (data.hasOwnProperty('imap-target-email')) {
          obj['imap-target-email'] = _ApiClient["default"].convertToType(data['imap-target-email'], 'String');
        }

        if (data.hasOwnProperty('imap-username')) {
          obj['imap-username'] = _ApiClient["default"].convertToType(data['imap-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TargetUpdateZeroSSL</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TargetUpdateZeroSSL</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(TargetUpdateZeroSSL.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['api-key'] && !(typeof data['api-key'] === 'string' || data['api-key'] instanceof String)) {
        throw new Error("Expected the field `api-key` to be a primitive type in the JSON string but got " + data['api-key']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['imap-fqdn'] && !(typeof data['imap-fqdn'] === 'string' || data['imap-fqdn'] instanceof String)) {
        throw new Error("Expected the field `imap-fqdn` to be a primitive type in the JSON string but got " + data['imap-fqdn']);
      } // ensure the json data is a string


      if (data['imap-password'] && !(typeof data['imap-password'] === 'string' || data['imap-password'] instanceof String)) {
        throw new Error("Expected the field `imap-password` to be a primitive type in the JSON string but got " + data['imap-password']);
      } // ensure the json data is a string


      if (data['imap-port'] && !(typeof data['imap-port'] === 'string' || data['imap-port'] instanceof String)) {
        throw new Error("Expected the field `imap-port` to be a primitive type in the JSON string but got " + data['imap-port']);
      } // ensure the json data is a string


      if (data['imap-target-email'] && !(typeof data['imap-target-email'] === 'string' || data['imap-target-email'] instanceof String)) {
        throw new Error("Expected the field `imap-target-email` to be a primitive type in the JSON string but got " + data['imap-target-email']);
      } // ensure the json data is a string


      if (data['imap-username'] && !(typeof data['imap-username'] === 'string' || data['imap-username'] instanceof String)) {
        throw new Error("Expected the field `imap-username` to be a primitive type in the JSON string but got " + data['imap-username']);
      } // ensure the json data is a string


      if (data['keep-prev-version'] && !(typeof data['keep-prev-version'] === 'string' || data['keep-prev-version'] instanceof String)) {
        throw new Error("Expected the field `keep-prev-version` to be a primitive type in the JSON string but got " + data['keep-prev-version']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['timeout'] && !(typeof data['timeout'] === 'string' || data['timeout'] instanceof String)) {
        throw new Error("Expected the field `timeout` to be a primitive type in the JSON string but got " + data['timeout']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return TargetUpdateZeroSSL;
}();

TargetUpdateZeroSSL.RequiredProperties = ["api-key", "imap-fqdn", "imap-password", "imap-username", "name"];
/**
 * API Key of the ZeroSSLTarget account
 * @member {String} api-key
 */

TargetUpdateZeroSSL.prototype['api-key'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

TargetUpdateZeroSSL.prototype['description'] = undefined;
/**
 * ImapFQDN of the IMAP service, FQDN or IPv4 address. Must be FQDN if the IMAP is using TLS
 * @member {String} imap-fqdn
 */

TargetUpdateZeroSSL.prototype['imap-fqdn'] = undefined;
/**
 * ImapPassword to access the IMAP service
 * @member {String} imap-password
 */

TargetUpdateZeroSSL.prototype['imap-password'] = undefined;
/**
 * ImapPort of the IMAP service
 * @member {String} imap-port
 * @default '993'
 */

TargetUpdateZeroSSL.prototype['imap-port'] = '993';
/**
 * ImapValidationEmail to use when asking ZeroSSL to send a validation email, if empty will user imap-username
 * @member {String} imap-target-email
 */

TargetUpdateZeroSSL.prototype['imap-target-email'] = undefined;
/**
 * ImapUsername to access the IMAP service
 * @member {String} imap-username
 */

TargetUpdateZeroSSL.prototype['imap-username'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

TargetUpdateZeroSSL.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

TargetUpdateZeroSSL.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

TargetUpdateZeroSSL.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

TargetUpdateZeroSSL.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

TargetUpdateZeroSSL.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

TargetUpdateZeroSSL.prototype['new-name'] = undefined;
/**
 * Timeout waiting for certificate validation in Duration format (1h - 1 Hour, 20m - 20 Minutes, 33m3s - 33 Minutes and 3 Seconds), maximum 1h.
 * @member {String} timeout
 * @default '5m'
 */

TargetUpdateZeroSSL.prototype['timeout'] = '5m';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

TargetUpdateZeroSSL.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

TargetUpdateZeroSSL.prototype['uid-token'] = undefined;
var _default = TargetUpdateZeroSSL;
exports["default"] = _default;
