// akeyless/dist/model/TokenizeOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TokenizeOutput model module.
 * @module model/TokenizeOutput
 * @version 5.0.14
 */
var TokenizeOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TokenizeOutput</code>.
   * @alias module:model/TokenizeOutput
   */
  function TokenizeOutput() {
    _classCallCheck(this, TokenizeOutput);

    TokenizeOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TokenizeOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TokenizeOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TokenizeOutput} obj Optional instance to populate.
     * @return {module:model/TokenizeOutput} The populated <code>TokenizeOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TokenizeOutput();

        if (data.hasOwnProperty('result')) {
          obj['result'] = _ApiClient["default"].convertToType(data['result'], 'String');
        }

        if (data.hasOwnProperty('tweak')) {
          obj['tweak'] = _ApiClient["default"].convertToType(data['tweak'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>TokenizeOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>TokenizeOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['result'] && !(typeof data['result'] === 'string' || data['result'] instanceof String)) {
        throw new Error("Expected the field `result` to be a primitive type in the JSON string but got " + data['result']);
      } // ensure the json data is a string


      if (data['tweak'] && !(typeof data['tweak'] === 'string' || data['tweak'] instanceof String)) {
        throw new Error("Expected the field `tweak` to be a primitive type in the JSON string but got " + data['tweak']);
      }

      return true;
    }
  }]);

  return TokenizeOutput;
}();
/**
 * @member {String} result
 */


TokenizeOutput.prototype['result'] = undefined;
/**
 * @member {String} tweak
 */

TokenizeOutput.prototype['tweak'] = undefined;
var _default = TokenizeOutput;
exports["default"] = _default;
