// akeyless/dist/model/UidListChildren.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UidListChildren model module.
 * @module model/UidListChildren
 * @version 5.0.1
 */
var UidListChildren = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UidListChildren</code>.
   * uidListChildren is a command that lists child token ids of Akeyless Universal Identity.
   * @alias module:model/UidListChildren
   */
  function UidListChildren() {
    _classCallCheck(this, UidListChildren);

    UidListChildren.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UidListChildren, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
    }
    /**
     * Constructs a <code>UidListChildren</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UidListChildren} obj Optional instance to populate.
     * @return {module:model/UidListChildren} The populated <code>UidListChildren</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UidListChildren();

        if (data.hasOwnProperty('auth-method-name')) {
          obj['auth-method-name'] = _ApiClient["default"].convertToType(data['auth-method-name'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UidListChildren</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UidListChildren</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['auth-method-name'] && !(typeof data['auth-method-name'] === 'string' || data['auth-method-name'] instanceof String)) {
        throw new Error("Expected the field `auth-method-name` to be a primitive type in the JSON string but got " + data['auth-method-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return UidListChildren;
}();
/**
 * The universal identity auth method name, required only when uid-token is not provided
 * @member {String} auth-method-name
 */


UidListChildren.prototype['auth-method-name'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UidListChildren.prototype['json'] = false;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UidListChildren.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UidListChildren.prototype['uid-token'] = undefined;
var _default = UidListChildren;
exports["default"] = _default;
