// akeyless/dist/model/UidRevokeToken.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UidRevokeToken model module.
 * @module model/UidRevokeToken
 * @version 5.0.5
 */
var UidRevokeToken = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UidRevokeToken</code>.
   * @alias module:model/UidRevokeToken
   * @param revokeToken {String} the universal identity token/token-id to revoke
   * @param revokeType {String} revokeSelf/revokeAll (delete only this token/this token and his children)
   */
  function UidRevokeToken(revokeToken, revokeType) {
    _classCallCheck(this, UidRevokeToken);

    UidRevokeToken.initialize(this, revokeToken, revokeType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UidRevokeToken, null, [{
    key: "initialize",
    value: function initialize(obj, revokeToken, revokeType) {
      obj['json'] = false;
      obj['revoke-token'] = revokeToken;
      obj['revoke-type'] = revokeType;
    }
    /**
     * Constructs a <code>UidRevokeToken</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UidRevokeToken} obj Optional instance to populate.
     * @return {module:model/UidRevokeToken} The populated <code>UidRevokeToken</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UidRevokeToken();

        if (data.hasOwnProperty('auth-method-name')) {
          obj['auth-method-name'] = _ApiClient["default"].convertToType(data['auth-method-name'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('revoke-token')) {
          obj['revoke-token'] = _ApiClient["default"].convertToType(data['revoke-token'], 'String');
        }

        if (data.hasOwnProperty('revoke-type')) {
          obj['revoke-type'] = _ApiClient["default"].convertToType(data['revoke-type'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UidRevokeToken</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UidRevokeToken</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(UidRevokeToken.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['auth-method-name'] && !(typeof data['auth-method-name'] === 'string' || data['auth-method-name'] instanceof String)) {
        throw new Error("Expected the field `auth-method-name` to be a primitive type in the JSON string but got " + data['auth-method-name']);
      } // ensure the json data is a string


      if (data['revoke-token'] && !(typeof data['revoke-token'] === 'string' || data['revoke-token'] instanceof String)) {
        throw new Error("Expected the field `revoke-token` to be a primitive type in the JSON string but got " + data['revoke-token']);
      } // ensure the json data is a string


      if (data['revoke-type'] && !(typeof data['revoke-type'] === 'string' || data['revoke-type'] instanceof String)) {
        throw new Error("Expected the field `revoke-type` to be a primitive type in the JSON string but got " + data['revoke-type']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return UidRevokeToken;
}();

UidRevokeToken.RequiredProperties = ["revoke-token", "revoke-type"];
/**
 * The universal identity auth method name
 * @member {String} auth-method-name
 */

UidRevokeToken.prototype['auth-method-name'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UidRevokeToken.prototype['json'] = false;
/**
 * the universal identity token/token-id to revoke
 * @member {String} revoke-token
 */

UidRevokeToken.prototype['revoke-token'] = undefined;
/**
 * revokeSelf/revokeAll (delete only this token/this token and his children)
 * @member {String} revoke-type
 */

UidRevokeToken.prototype['revoke-type'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UidRevokeToken.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UidRevokeToken.prototype['uid-token'] = undefined;
var _default = UidRevokeToken;
exports["default"] = _default;
