// akeyless/dist/model/UpdateEKSTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateEKSTarget model module.
 * @module model/UpdateEKSTarget
 * @version 5.0.14
 */
var UpdateEKSTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateEKSTarget</code>.
   * updateEKSTarget is a command that updates an existing target. [Deprecated: Use target-update-eks command]
   * @alias module:model/UpdateEKSTarget
   * @param eksAccessKeyId {String} Access Key ID
   * @param eksClusterCaCert {String} EKS cluster CA certificate
   * @param eksClusterEndpoint {String} EKS cluster URL endpoint
   * @param eksClusterName {String} EKS cluster name
   * @param eksSecretAccessKey {String} Secret Access Key
   * @param name {String} Target name
   */
  function UpdateEKSTarget(eksAccessKeyId, eksClusterCaCert, eksClusterEndpoint, eksClusterName, eksSecretAccessKey, name) {
    _classCallCheck(this, UpdateEKSTarget);

    UpdateEKSTarget.initialize(this, eksAccessKeyId, eksClusterCaCert, eksClusterEndpoint, eksClusterName, eksSecretAccessKey, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateEKSTarget, null, [{
    key: "initialize",
    value: function initialize(obj, eksAccessKeyId, eksClusterCaCert, eksClusterEndpoint, eksClusterName, eksSecretAccessKey, name) {
      obj['eks-access-key-id'] = eksAccessKeyId;
      obj['eks-cluster-ca-cert'] = eksClusterCaCert;
      obj['eks-cluster-endpoint'] = eksClusterEndpoint;
      obj['eks-cluster-name'] = eksClusterName;
      obj['eks-region'] = 'us-east-2';
      obj['eks-secret-access-key'] = eksSecretAccessKey;
      obj['json'] = false;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateEKSTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateEKSTarget} obj Optional instance to populate.
     * @return {module:model/UpdateEKSTarget} The populated <code>UpdateEKSTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateEKSTarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('eks-access-key-id')) {
          obj['eks-access-key-id'] = _ApiClient["default"].convertToType(data['eks-access-key-id'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-ca-cert')) {
          obj['eks-cluster-ca-cert'] = _ApiClient["default"].convertToType(data['eks-cluster-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-endpoint')) {
          obj['eks-cluster-endpoint'] = _ApiClient["default"].convertToType(data['eks-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-name')) {
          obj['eks-cluster-name'] = _ApiClient["default"].convertToType(data['eks-cluster-name'], 'String');
        }

        if (data.hasOwnProperty('eks-region')) {
          obj['eks-region'] = _ApiClient["default"].convertToType(data['eks-region'], 'String');
        }

        if (data.hasOwnProperty('eks-secret-access-key')) {
          obj['eks-secret-access-key'] = _ApiClient["default"].convertToType(data['eks-secret-access-key'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UpdateEKSTarget</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UpdateEKSTarget</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(UpdateEKSTarget.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['comment'] && !(typeof data['comment'] === 'string' || data['comment'] instanceof String)) {
        throw new Error("Expected the field `comment` to be a primitive type in the JSON string but got " + data['comment']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['eks-access-key-id'] && !(typeof data['eks-access-key-id'] === 'string' || data['eks-access-key-id'] instanceof String)) {
        throw new Error("Expected the field `eks-access-key-id` to be a primitive type in the JSON string but got " + data['eks-access-key-id']);
      } // ensure the json data is a string


      if (data['eks-cluster-ca-cert'] && !(typeof data['eks-cluster-ca-cert'] === 'string' || data['eks-cluster-ca-cert'] instanceof String)) {
        throw new Error("Expected the field `eks-cluster-ca-cert` to be a primitive type in the JSON string but got " + data['eks-cluster-ca-cert']);
      } // ensure the json data is a string


      if (data['eks-cluster-endpoint'] && !(typeof data['eks-cluster-endpoint'] === 'string' || data['eks-cluster-endpoint'] instanceof String)) {
        throw new Error("Expected the field `eks-cluster-endpoint` to be a primitive type in the JSON string but got " + data['eks-cluster-endpoint']);
      } // ensure the json data is a string


      if (data['eks-cluster-name'] && !(typeof data['eks-cluster-name'] === 'string' || data['eks-cluster-name'] instanceof String)) {
        throw new Error("Expected the field `eks-cluster-name` to be a primitive type in the JSON string but got " + data['eks-cluster-name']);
      } // ensure the json data is a string


      if (data['eks-region'] && !(typeof data['eks-region'] === 'string' || data['eks-region'] instanceof String)) {
        throw new Error("Expected the field `eks-region` to be a primitive type in the JSON string but got " + data['eks-region']);
      } // ensure the json data is a string


      if (data['eks-secret-access-key'] && !(typeof data['eks-secret-access-key'] === 'string' || data['eks-secret-access-key'] instanceof String)) {
        throw new Error("Expected the field `eks-secret-access-key` to be a primitive type in the JSON string but got " + data['eks-secret-access-key']);
      } // ensure the json data is a string


      if (data['keep-prev-version'] && !(typeof data['keep-prev-version'] === 'string' || data['keep-prev-version'] instanceof String)) {
        throw new Error("Expected the field `keep-prev-version` to be a primitive type in the JSON string but got " + data['keep-prev-version']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return UpdateEKSTarget;
}();

UpdateEKSTarget.RequiredProperties = ["eks-access-key-id", "eks-cluster-ca-cert", "eks-cluster-endpoint", "eks-cluster-name", "eks-secret-access-key", "name"];
/**
 * Deprecated - use description
 * @member {String} comment
 */

UpdateEKSTarget.prototype['comment'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

UpdateEKSTarget.prototype['description'] = undefined;
/**
 * Access Key ID
 * @member {String} eks-access-key-id
 */

UpdateEKSTarget.prototype['eks-access-key-id'] = undefined;
/**
 * EKS cluster CA certificate
 * @member {String} eks-cluster-ca-cert
 */

UpdateEKSTarget.prototype['eks-cluster-ca-cert'] = undefined;
/**
 * EKS cluster URL endpoint
 * @member {String} eks-cluster-endpoint
 */

UpdateEKSTarget.prototype['eks-cluster-endpoint'] = undefined;
/**
 * EKS cluster name
 * @member {String} eks-cluster-name
 */

UpdateEKSTarget.prototype['eks-cluster-name'] = undefined;
/**
 * Region
 * @member {String} eks-region
 * @default 'us-east-2'
 */

UpdateEKSTarget.prototype['eks-region'] = 'us-east-2';
/**
 * Secret Access Key
 * @member {String} eks-secret-access-key
 */

UpdateEKSTarget.prototype['eks-secret-access-key'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateEKSTarget.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

UpdateEKSTarget.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateEKSTarget.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

UpdateEKSTarget.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateEKSTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateEKSTarget.prototype['new-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateEKSTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateEKSTarget.prototype['uid-token'] = undefined;
/**
 * Deprecated
 * @member {Boolean} update-version
 */

UpdateEKSTarget.prototype['update-version'] = undefined;
/**
 * @member {Boolean} use-gw-cloud-identity
 */

UpdateEKSTarget.prototype['use-gw-cloud-identity'] = undefined;
var _default = UpdateEKSTarget;
exports["default"] = _default;
