// akeyless/dist/model/UpdateOidcApp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateOidcApp model module.
 * @module model/UpdateOidcApp
 * @version 5.0.5
 */
var UpdateOidcApp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateOidcApp</code>.
   * @alias module:model/UpdateOidcApp
   * @param name {String} OIDC application name
   */
  function UpdateOidcApp(name) {
    _classCallCheck(this, UpdateOidcApp);

    UpdateOidcApp.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateOidcApp, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['json'] = false;
      obj['name'] = name;
      obj['scopes'] = 'openid';
    }
    /**
     * Constructs a <code>UpdateOidcApp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateOidcApp} obj Optional instance to populate.
     * @return {module:model/UpdateOidcApp} The populated <code>UpdateOidcApp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateOidcApp();

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('permission-assignment')) {
          obj['permission-assignment'] = _ApiClient["default"].convertToType(data['permission-assignment'], 'String');
        }

        if (data.hasOwnProperty('public')) {
          obj['public'] = _ApiClient["default"].convertToType(data['public'], 'Boolean');
        }

        if (data.hasOwnProperty('redirect-uris')) {
          obj['redirect-uris'] = _ApiClient["default"].convertToType(data['redirect-uris'], 'String');
        }

        if (data.hasOwnProperty('scopes')) {
          obj['scopes'] = _ApiClient["default"].convertToType(data['scopes'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UpdateOidcApp</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UpdateOidcApp</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(UpdateOidcApp.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['audience'] && !(typeof data['audience'] === 'string' || data['audience'] instanceof String)) {
        throw new Error("Expected the field `audience` to be a primitive type in the JSON string but got " + data['audience']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['permission-assignment'] && !(typeof data['permission-assignment'] === 'string' || data['permission-assignment'] instanceof String)) {
        throw new Error("Expected the field `permission-assignment` to be a primitive type in the JSON string but got " + data['permission-assignment']);
      } // ensure the json data is a string


      if (data['redirect-uris'] && !(typeof data['redirect-uris'] === 'string' || data['redirect-uris'] instanceof String)) {
        throw new Error("Expected the field `redirect-uris` to be a primitive type in the JSON string but got " + data['redirect-uris']);
      } // ensure the json data is a string


      if (data['scopes'] && !(typeof data['scopes'] === 'string' || data['scopes'] instanceof String)) {
        throw new Error("Expected the field `scopes` to be a primitive type in the JSON string but got " + data['scopes']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return UpdateOidcApp;
}();

UpdateOidcApp.RequiredProperties = ["name"];
/**
 * A comma separated list of allowed audiences
 * @member {String} audience
 */

UpdateOidcApp.prototype['audience'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateOidcApp.prototype['json'] = false;
/**
 * The name of a key that used to encrypt the OIDC application (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateOidcApp.prototype['key'] = undefined;
/**
 * OIDC application name
 * @member {String} name
 */

UpdateOidcApp.prototype['name'] = undefined;
/**
 * A json string defining the access permission assignment for this app
 * @member {String} permission-assignment
 */

UpdateOidcApp.prototype['permission-assignment'] = undefined;
/**
 * Set to true if the app is public (cannot keep secrets)
 * @member {Boolean} public
 */

UpdateOidcApp.prototype['public'] = undefined;
/**
 * A comma separated list of allowed redirect uris
 * @member {String} redirect-uris
 */

UpdateOidcApp.prototype['redirect-uris'] = undefined;
/**
 * A comma separated list of allowed scopes
 * @member {String} scopes
 * @default 'openid'
 */

UpdateOidcApp.prototype['scopes'] = 'openid';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateOidcApp.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateOidcApp.prototype['uid-token'] = undefined;
var _default = UpdateOidcApp;
exports["default"] = _default;
