// akeyless/dist/model/UpdateAuthMethodOCI.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateAuthMethodOCI model module.
 * @module model/UpdateAuthMethodOCI
 * @version 3.6.3
 */
var UpdateAuthMethodOCI = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateAuthMethodOCI</code>.
   * updateAuthMethodOCI is a command that updates an auth method that will be used in the account using OCI principle and groups.
   * @alias module:model/UpdateAuthMethodOCI
   * @param groupOcid {Array.<String>} A list of required groups ocids
   * @param name {String} Auth Method name
   * @param tenantOcid {String} The Oracle Cloud tenant ID
   */
  function UpdateAuthMethodOCI(groupOcid, name, tenantOcid) {
    _classCallCheck(this, UpdateAuthMethodOCI);

    UpdateAuthMethodOCI.initialize(this, groupOcid, name, tenantOcid);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateAuthMethodOCI, null, [{
    key: "initialize",
    value: function initialize(obj, groupOcid, name, tenantOcid) {
      obj['group-ocid'] = groupOcid;
      obj['name'] = name;
      obj['tenant-ocid'] = tenantOcid;
    }
    /**
     * Constructs a <code>UpdateAuthMethodOCI</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateAuthMethodOCI} obj Optional instance to populate.
     * @return {module:model/UpdateAuthMethodOCI} The populated <code>UpdateAuthMethodOCI</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateAuthMethodOCI();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('group-ocid')) {
          obj['group-ocid'] = _ApiClient["default"].convertToType(data['group-ocid'], ['String']);
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('product-type')) {
          obj['product-type'] = _ApiClient["default"].convertToType(data['product-type'], ['String']);
        }

        if (data.hasOwnProperty('tenant-ocid')) {
          obj['tenant-ocid'] = _ApiClient["default"].convertToType(data['tenant-ocid'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateAuthMethodOCI;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


UpdateAuthMethodOCI.prototype['access-expires'] = 0;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

UpdateAuthMethodOCI.prototype['bound-ips'] = undefined;
/**
 * Auth Method description
 * @member {String} description
 */

UpdateAuthMethodOCI.prototype['description'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

UpdateAuthMethodOCI.prototype['force-sub-claims'] = undefined;
/**
 * A list of required groups ocids
 * @member {Array.<String>} group-ocid
 */

UpdateAuthMethodOCI.prototype['group-ocid'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

UpdateAuthMethodOCI.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateAuthMethodOCI.prototype['json'] = false;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 * @default 0
 */

UpdateAuthMethodOCI.prototype['jwt-ttl'] = 0;
/**
 * Auth Method name
 * @member {String} name
 */

UpdateAuthMethodOCI.prototype['name'] = undefined;
/**
 * Auth Method new name
 * @member {String} new-name
 */

UpdateAuthMethodOCI.prototype['new-name'] = undefined;
/**
 * Choose the relevant product type for the auth method [sm, sra, pm, dp, ca]
 * @member {Array.<String>} product-type
 */

UpdateAuthMethodOCI.prototype['product-type'] = undefined;
/**
 * The Oracle Cloud tenant ID
 * @member {String} tenant-ocid
 */

UpdateAuthMethodOCI.prototype['tenant-ocid'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateAuthMethodOCI.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateAuthMethodOCI.prototype['uid-token'] = undefined;
var _default = UpdateAuthMethodOCI;
exports["default"] = _default;
