// akeyless/dist/model/UpdateDBTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateDBTargetDetails model module.
 * @module model/UpdateDBTargetDetails
 * @version 5.0.5
 */
var UpdateDBTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateDBTargetDetails</code>.
   * @alias module:model/UpdateDBTargetDetails
   * @param name {String} Target name
   */
  function UpdateDBTargetDetails(name) {
    _classCallCheck(this, UpdateDBTargetDetails);

    UpdateDBTargetDetails.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateDBTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['json'] = false;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateDBTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateDBTargetDetails} obj Optional instance to populate.
     * @return {module:model/UpdateDBTargetDetails} The populated <code>UpdateDBTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateDBTargetDetails();

        if (data.hasOwnProperty('db-type')) {
          obj['db-type'] = _ApiClient["default"].convertToType(data['db-type'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('mongo-db-name')) {
          obj['mongo-db-name'] = _ApiClient["default"].convertToType(data['mongo-db-name'], 'String');
        }

        if (data.hasOwnProperty('mongo-uri')) {
          obj['mongo-uri'] = _ApiClient["default"].convertToType(data['mongo-uri'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-version')) {
          obj['new-version'] = _ApiClient["default"].convertToType(data['new-version'], 'Boolean');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('pwd')) {
          obj['pwd'] = _ApiClient["default"].convertToType(data['pwd'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-name')) {
          obj['user-name'] = _ApiClient["default"].convertToType(data['user-name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UpdateDBTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UpdateDBTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(UpdateDBTargetDetails.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['db-type'] && !(typeof data['db-type'] === 'string' || data['db-type'] instanceof String)) {
        throw new Error("Expected the field `db-type` to be a primitive type in the JSON string but got " + data['db-type']);
      } // ensure the json data is a string


      if (data['host'] && !(typeof data['host'] === 'string' || data['host'] instanceof String)) {
        throw new Error("Expected the field `host` to be a primitive type in the JSON string but got " + data['host']);
      } // ensure the json data is a string


      if (data['keep-prev-version'] && !(typeof data['keep-prev-version'] === 'string' || data['keep-prev-version'] instanceof String)) {
        throw new Error("Expected the field `keep-prev-version` to be a primitive type in the JSON string but got " + data['keep-prev-version']);
      } // ensure the json data is a string


      if (data['mongo-db-name'] && !(typeof data['mongo-db-name'] === 'string' || data['mongo-db-name'] instanceof String)) {
        throw new Error("Expected the field `mongo-db-name` to be a primitive type in the JSON string but got " + data['mongo-db-name']);
      } // ensure the json data is a string


      if (data['mongo-uri'] && !(typeof data['mongo-uri'] === 'string' || data['mongo-uri'] instanceof String)) {
        throw new Error("Expected the field `mongo-uri` to be a primitive type in the JSON string but got " + data['mongo-uri']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['port'] && !(typeof data['port'] === 'string' || data['port'] instanceof String)) {
        throw new Error("Expected the field `port` to be a primitive type in the JSON string but got " + data['port']);
      } // ensure the json data is a string


      if (data['protection_key'] && !(typeof data['protection_key'] === 'string' || data['protection_key'] instanceof String)) {
        throw new Error("Expected the field `protection_key` to be a primitive type in the JSON string but got " + data['protection_key']);
      } // ensure the json data is a string


      if (data['pwd'] && !(typeof data['pwd'] === 'string' || data['pwd'] instanceof String)) {
        throw new Error("Expected the field `pwd` to be a primitive type in the JSON string but got " + data['pwd']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-name'] && !(typeof data['user-name'] === 'string' || data['user-name'] instanceof String)) {
        throw new Error("Expected the field `user-name` to be a primitive type in the JSON string but got " + data['user-name']);
      }

      return true;
    }
  }]);

  return UpdateDBTargetDetails;
}();

UpdateDBTargetDetails.RequiredProperties = ["name"];
/**
 * @member {String} db-type
 */

UpdateDBTargetDetails.prototype['db-type'] = undefined;
/**
 * @member {String} host
 */

UpdateDBTargetDetails.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateDBTargetDetails.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

UpdateDBTargetDetails.prototype['keep-prev-version'] = undefined;
/**
 * @member {String} mongo-db-name
 */

UpdateDBTargetDetails.prototype['mongo-db-name'] = undefined;
/**
 * @member {String} mongo-uri
 */

UpdateDBTargetDetails.prototype['mongo-uri'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateDBTargetDetails.prototype['name'] = undefined;
/**
 * Deprecated
 * @member {Boolean} new-version
 */

UpdateDBTargetDetails.prototype['new-version'] = undefined;
/**
 * @member {String} port
 */

UpdateDBTargetDetails.prototype['port'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

UpdateDBTargetDetails.prototype['protection_key'] = undefined;
/**
 * @member {String} pwd
 */

UpdateDBTargetDetails.prototype['pwd'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateDBTargetDetails.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateDBTargetDetails.prototype['uid-token'] = undefined;
/**
 * @member {String} user-name
 */

UpdateDBTargetDetails.prototype['user-name'] = undefined;
var _default = UpdateDBTargetDetails;
exports["default"] = _default;
