// akeyless/dist/model/UpdateHashiVaultTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateHashiVaultTarget model module.
 * @module model/UpdateHashiVaultTarget
 * @version 4.2.1
 */
var UpdateHashiVaultTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateHashiVaultTarget</code>.
   * updateHashiVaultTarget is a command that updates an existing target. [Deprecated: Use target-update-hashi-vault command]
   * @alias module:model/UpdateHashiVaultTarget
   * @param name {String} Target name
   */
  function UpdateHashiVaultTarget(name) {
    _classCallCheck(this, UpdateHashiVaultTarget);

    UpdateHashiVaultTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateHashiVaultTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateHashiVaultTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateHashiVaultTarget} obj Optional instance to populate.
     * @return {module:model/UpdateHashiVaultTarget} The populated <code>UpdateHashiVaultTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateHashiVaultTarget();

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('hashi-url')) {
          obj['hashi-url'] = _ApiClient["default"].convertToType(data['hashi-url'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('namespace')) {
          obj['namespace'] = _ApiClient["default"].convertToType(data['namespace'], ['String']);
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('vault-token')) {
          obj['vault-token'] = _ApiClient["default"].convertToType(data['vault-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateHashiVaultTarget;
}();
/**
 * Description of the object
 * @member {String} description
 */


UpdateHashiVaultTarget.prototype['description'] = undefined;
/**
 * HashiCorp Vault API URL, e.g. https://vault-mgr01:8200
 * @member {String} hashi-url
 */

UpdateHashiVaultTarget.prototype['hashi-url'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateHashiVaultTarget.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

UpdateHashiVaultTarget.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateHashiVaultTarget.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

UpdateHashiVaultTarget.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateHashiVaultTarget.prototype['name'] = undefined;
/**
 * Comma-separated list of vault namespaces
 * @member {Array.<String>} namespace
 */

UpdateHashiVaultTarget.prototype['namespace'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateHashiVaultTarget.prototype['new-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateHashiVaultTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateHashiVaultTarget.prototype['uid-token'] = undefined;
/**
 * Deprecated
 * @member {Boolean} update-version
 */

UpdateHashiVaultTarget.prototype['update-version'] = undefined;
/**
 * Vault access token with sufficient permissions
 * @member {String} vault-token
 */

UpdateHashiVaultTarget.prototype['vault-token'] = undefined;
var _default = UpdateHashiVaultTarget;
exports["default"] = _default;
