// akeyless/dist/model/UpdateEventForwarder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateEventForwarder model module.
 * @module model/UpdateEventForwarder
 * @version 5.0.0
 */
var UpdateEventForwarder = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateEventForwarder</code>.
   * updateEventForwarder is a command that updates an event forwarder [Deprecated - please use event-forwarder-update-* command]
   * @alias module:model/UpdateEventForwarder
   * @param name {String} EventForwarder name
   */
  function UpdateEventForwarder(name) {
    _classCallCheck(this, UpdateEventForwarder);

    UpdateEventForwarder.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateEventForwarder, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['auth-type'] = 'user-pass';
      obj['description'] = 'default_comment';
      obj['enable'] = 'true';
      obj['json'] = false;
      obj['name'] = name;
      obj['new-comment'] = 'default_comment';
    }
    /**
     * Constructs a <code>UpdateEventForwarder</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateEventForwarder} obj Optional instance to populate.
     * @return {module:model/UpdateEventForwarder} The populated <code>UpdateEventForwarder</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateEventForwarder();

        if (data.hasOwnProperty('admin-name')) {
          obj['admin-name'] = _ApiClient["default"].convertToType(data['admin-name'], 'String');
        }

        if (data.hasOwnProperty('auth-type')) {
          obj['auth-type'] = _ApiClient["default"].convertToType(data['auth-type'], 'String');
        }

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('email-to')) {
          obj['email-to'] = _ApiClient["default"].convertToType(data['email-to'], 'String');
        }

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'String');
        }

        if (data.hasOwnProperty('event-source-locations')) {
          obj['event-source-locations'] = _ApiClient["default"].convertToType(data['event-source-locations'], ['String']);
        }

        if (data.hasOwnProperty('event-types')) {
          obj['event-types'] = _ApiClient["default"].convertToType(data['event-types'], ['String']);
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-comment')) {
          obj['new-comment'] = _ApiClient["default"].convertToType(data['new-comment'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-email')) {
          obj['user-email'] = _ApiClient["default"].convertToType(data['user-email'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UpdateEventForwarder</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UpdateEventForwarder</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(UpdateEventForwarder.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['admin-name'] && !(typeof data['admin-name'] === 'string' || data['admin-name'] instanceof String)) {
        throw new Error("Expected the field `admin-name` to be a primitive type in the JSON string but got " + data['admin-name']);
      } // ensure the json data is a string


      if (data['auth-type'] && !(typeof data['auth-type'] === 'string' || data['auth-type'] instanceof String)) {
        throw new Error("Expected the field `auth-type` to be a primitive type in the JSON string but got " + data['auth-type']);
      } // ensure the json data is a string


      if (data['client-id'] && !(typeof data['client-id'] === 'string' || data['client-id'] instanceof String)) {
        throw new Error("Expected the field `client-id` to be a primitive type in the JSON string but got " + data['client-id']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['email-to'] && !(typeof data['email-to'] === 'string' || data['email-to'] instanceof String)) {
        throw new Error("Expected the field `email-to` to be a primitive type in the JSON string but got " + data['email-to']);
      } // ensure the json data is a string


      if (data['enable'] && !(typeof data['enable'] === 'string' || data['enable'] instanceof String)) {
        throw new Error("Expected the field `enable` to be a primitive type in the JSON string but got " + data['enable']);
      } // ensure the json data is an array


      if (!Array.isArray(data['event-source-locations'])) {
        throw new Error("Expected the field `event-source-locations` to be an array in the JSON data but got " + data['event-source-locations']);
      } // ensure the json data is an array


      if (!Array.isArray(data['event-types'])) {
        throw new Error("Expected the field `event-types` to be an array in the JSON data but got " + data['event-types']);
      } // ensure the json data is a string


      if (data['host'] && !(typeof data['host'] === 'string' || data['host'] instanceof String)) {
        throw new Error("Expected the field `host` to be a primitive type in the JSON string but got " + data['host']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-comment'] && !(typeof data['new-comment'] === 'string' || data['new-comment'] instanceof String)) {
        throw new Error("Expected the field `new-comment` to be a primitive type in the JSON string but got " + data['new-comment']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['user-email'] && !(typeof data['user-email'] === 'string' || data['user-email'] instanceof String)) {
        throw new Error("Expected the field `user-email` to be a primitive type in the JSON string but got " + data['user-email']);
      }

      return true;
    }
  }]);

  return UpdateEventForwarder;
}();

UpdateEventForwarder.RequiredProperties = ["name"];
/**
 * Workstation Admin Name
 * @member {String} admin-name
 */

UpdateEventForwarder.prototype['admin-name'] = undefined;
/**
 * The authentication type to use when connecting to ServiceNow (user-pass / jwt)
 * @member {String} auth-type
 * @default 'user-pass'
 */

UpdateEventForwarder.prototype['auth-type'] = 'user-pass';
/**
 * The client ID to use when connecting to ServiceNow with jwt authentication
 * @member {String} client-id
 */

UpdateEventForwarder.prototype['client-id'] = undefined;
/**
 * Description of the object
 * @member {String} description
 * @default 'default_comment'
 */

UpdateEventForwarder.prototype['description'] = 'default_comment';
/**
 * A comma seperated list of email addresses to send event to (relevant only for \"email\" Event Forwarder)
 * @member {String} email-to
 */

UpdateEventForwarder.prototype['email-to'] = undefined;
/**
 * Enable/Disable Event Forwarder [true/false]
 * @member {String} enable
 * @default 'true'
 */

UpdateEventForwarder.prototype['enable'] = 'true';
/**
 * Event sources
 * @member {Array.<String>} event-source-locations
 */

UpdateEventForwarder.prototype['event-source-locations'] = undefined;
/**
 * Event types
 * @member {Array.<String>} event-types
 */

UpdateEventForwarder.prototype['event-types'] = undefined;
/**
 * Workstation Host
 * @member {String} host
 */

UpdateEventForwarder.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateEventForwarder.prototype['json'] = false;
/**
 * EventForwarder name
 * @member {String} name
 */

UpdateEventForwarder.prototype['name'] = undefined;
/**
 * Deprecated - use description
 * @member {String} new-comment
 * @default 'default_comment'
 */

UpdateEventForwarder.prototype['new-comment'] = 'default_comment';
/**
 * New EventForwarder name
 * @member {String} new-name
 */

UpdateEventForwarder.prototype['new-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateEventForwarder.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateEventForwarder.prototype['uid-token'] = undefined;
/**
 * The user email to use when connecting to ServiceNow with jwt authentication
 * @member {String} user-email
 */

UpdateEventForwarder.prototype['user-email'] = undefined;
var _default = UpdateEventForwarder;
exports["default"] = _default;
