// akeyless/dist/model/UpdateAWSTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateAWSTarget model module.
 * @module model/UpdateAWSTarget
 * @version 5.0.8
 */
var UpdateAWSTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateAWSTarget</code>.
   * updateAWSTarget is a command that updates a new target. [Deprecated: Use target-update-aws command]
   * @alias module:model/UpdateAWSTarget
   * @param accessKey {String} AWS secret access key
   * @param accessKeyId {String} AWS access key ID
   * @param name {String} Target name
   */
  function UpdateAWSTarget(accessKey, accessKeyId, name) {
    _classCallCheck(this, UpdateAWSTarget);

    UpdateAWSTarget.initialize(this, accessKey, accessKeyId, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateAWSTarget, null, [{
    key: "initialize",
    value: function initialize(obj, accessKey, accessKeyId, name) {
      obj['access-key'] = accessKey;
      obj['access-key-id'] = accessKeyId;
      obj['json'] = false;
      obj['name'] = name;
      obj['region'] = 'us-east-2';
    }
    /**
     * Constructs a <code>UpdateAWSTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateAWSTarget} obj Optional instance to populate.
     * @return {module:model/UpdateAWSTarget} The populated <code>UpdateAWSTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateAWSTarget();

        if (data.hasOwnProperty('access-key')) {
          obj['access-key'] = _ApiClient["default"].convertToType(data['access-key'], 'String');
        }

        if (data.hasOwnProperty('access-key-id')) {
          obj['access-key-id'] = _ApiClient["default"].convertToType(data['access-key-id'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('generate-external-id')) {
          obj['generate-external-id'] = _ApiClient["default"].convertToType(data['generate-external-id'], 'Boolean');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('max-versions')) {
          obj['max-versions'] = _ApiClient["default"].convertToType(data['max-versions'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('role-arn')) {
          obj['role-arn'] = _ApiClient["default"].convertToType(data['role-arn'], 'String');
        }

        if (data.hasOwnProperty('session-token')) {
          obj['session-token'] = _ApiClient["default"].convertToType(data['session-token'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UpdateAWSTarget</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UpdateAWSTarget</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(UpdateAWSTarget.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['access-key'] && !(typeof data['access-key'] === 'string' || data['access-key'] instanceof String)) {
        throw new Error("Expected the field `access-key` to be a primitive type in the JSON string but got " + data['access-key']);
      } // ensure the json data is a string


      if (data['access-key-id'] && !(typeof data['access-key-id'] === 'string' || data['access-key-id'] instanceof String)) {
        throw new Error("Expected the field `access-key-id` to be a primitive type in the JSON string but got " + data['access-key-id']);
      } // ensure the json data is a string


      if (data['comment'] && !(typeof data['comment'] === 'string' || data['comment'] instanceof String)) {
        throw new Error("Expected the field `comment` to be a primitive type in the JSON string but got " + data['comment']);
      } // ensure the json data is a string


      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['keep-prev-version'] && !(typeof data['keep-prev-version'] === 'string' || data['keep-prev-version'] instanceof String)) {
        throw new Error("Expected the field `keep-prev-version` to be a primitive type in the JSON string but got " + data['keep-prev-version']);
      } // ensure the json data is a string


      if (data['key'] && !(typeof data['key'] === 'string' || data['key'] instanceof String)) {
        throw new Error("Expected the field `key` to be a primitive type in the JSON string but got " + data['key']);
      } // ensure the json data is a string


      if (data['max-versions'] && !(typeof data['max-versions'] === 'string' || data['max-versions'] instanceof String)) {
        throw new Error("Expected the field `max-versions` to be a primitive type in the JSON string but got " + data['max-versions']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['new-name'] && !(typeof data['new-name'] === 'string' || data['new-name'] instanceof String)) {
        throw new Error("Expected the field `new-name` to be a primitive type in the JSON string but got " + data['new-name']);
      } // ensure the json data is a string


      if (data['region'] && !(typeof data['region'] === 'string' || data['region'] instanceof String)) {
        throw new Error("Expected the field `region` to be a primitive type in the JSON string but got " + data['region']);
      } // ensure the json data is a string


      if (data['role-arn'] && !(typeof data['role-arn'] === 'string' || data['role-arn'] instanceof String)) {
        throw new Error("Expected the field `role-arn` to be a primitive type in the JSON string but got " + data['role-arn']);
      } // ensure the json data is a string


      if (data['session-token'] && !(typeof data['session-token'] === 'string' || data['session-token'] instanceof String)) {
        throw new Error("Expected the field `session-token` to be a primitive type in the JSON string but got " + data['session-token']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      }

      return true;
    }
  }]);

  return UpdateAWSTarget;
}();

UpdateAWSTarget.RequiredProperties = ["access-key", "access-key-id", "name"];
/**
 * AWS secret access key
 * @member {String} access-key
 */

UpdateAWSTarget.prototype['access-key'] = undefined;
/**
 * AWS access key ID
 * @member {String} access-key-id
 */

UpdateAWSTarget.prototype['access-key-id'] = undefined;
/**
 * Deprecated - use description
 * @member {String} comment
 */

UpdateAWSTarget.prototype['comment'] = undefined;
/**
 * Description of the object
 * @member {String} description
 */

UpdateAWSTarget.prototype['description'] = undefined;
/**
 * A unique auto-generated value used in your AWS account when configuring your AWS IAM role to securely delegate access to Akeyless. Relevant only when using GW cloud ID
 * @member {Boolean} generate-external-id
 */

UpdateAWSTarget.prototype['generate-external-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UpdateAWSTarget.prototype['json'] = false;
/**
 * Whether to keep previous version [true/false]. If not set, use default according to account settings
 * @member {String} keep-prev-version
 */

UpdateAWSTarget.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateAWSTarget.prototype['key'] = undefined;
/**
 * Set the maximum number of versions, limited by the account settings defaults.
 * @member {String} max-versions
 */

UpdateAWSTarget.prototype['max-versions'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateAWSTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateAWSTarget.prototype['new-name'] = undefined;
/**
 * AWS region
 * @member {String} region
 * @default 'us-east-2'
 */

UpdateAWSTarget.prototype['region'] = 'us-east-2';
/**
 * AWS IAM role identifier that Gateway will assume in your AWS account, relevant only when using external ID
 * @member {String} role-arn
 */

UpdateAWSTarget.prototype['role-arn'] = undefined;
/**
 * Required only for temporary security credentials retrieved using STS
 * @member {String} session-token
 */

UpdateAWSTarget.prototype['session-token'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateAWSTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateAWSTarget.prototype['uid-token'] = undefined;
/**
 * Deprecated
 * @member {Boolean} update-version
 */

UpdateAWSTarget.prototype['update-version'] = undefined;
/**
 * Use the GW's Cloud IAM
 * @member {Boolean} use-gw-cloud-identity
 */

UpdateAWSTarget.prototype['use-gw-cloud-identity'] = undefined;
var _default = UpdateAWSTarget;
exports["default"] = _default;
