// akeyless/dist/model/Update.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Update model module.
 * @module model/Update
 * @version 5.0.6
 */
var Update = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Update</code>.
   * @alias module:model/Update
   */
  function Update() {
    _classCallCheck(this, Update);

    Update.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Update, null, [{
    key: "initialize",
    value: function initialize(obj) {
      obj['json'] = false;
      obj['version'] = 'latest';
    }
    /**
     * Constructs a <code>Update</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Update} obj Optional instance to populate.
     * @return {module:model/Update} The populated <code>Update</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Update();

        if (data.hasOwnProperty('artifact-repository')) {
          obj['artifact-repository'] = _ApiClient["default"].convertToType(data['artifact-repository'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('show-changelog')) {
          obj['show-changelog'] = _ApiClient["default"].convertToType(data['show-changelog'], 'Boolean');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>Update</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Update</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['artifact-repository'] && !(typeof data['artifact-repository'] === 'string' || data['artifact-repository'] instanceof String)) {
        throw new Error("Expected the field `artifact-repository` to be a primitive type in the JSON string but got " + data['artifact-repository']);
      } // ensure the json data is a string


      if (data['version'] && !(typeof data['version'] === 'string' || data['version'] instanceof String)) {
        throw new Error("Expected the field `version` to be a primitive type in the JSON string but got " + data['version']);
      }

      return true;
    }
  }]);

  return Update;
}();
/**
 * Alternative CLI repository url. e.g. https://artifacts.site2.akeyless.io
 * @member {String} artifact-repository
 */


Update.prototype['artifact-repository'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

Update.prototype['json'] = false;
/**
 * Show the changelog between the current version and the latest one and exit (update will not be performed)
 * @member {Boolean} show-changelog
 */

Update.prototype['show-changelog'] = undefined;
/**
 * The CLI version
 * @member {String} version
 * @default 'latest'
 */

Update.prototype['version'] = 'latest';
var _default = Update;
exports["default"] = _default;
