// akeyless/dist/model/UpdateGroupOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateGroupOutput model module.
 * @module model/UpdateGroupOutput
 * @version 5.0.1
 */
var UpdateGroupOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateGroupOutput</code>.
   * @alias module:model/UpdateGroupOutput
   */
  function UpdateGroupOutput() {
    _classCallCheck(this, UpdateGroupOutput);

    UpdateGroupOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateGroupOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UpdateGroupOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateGroupOutput} obj Optional instance to populate.
     * @return {module:model/UpdateGroupOutput} The populated <code>UpdateGroupOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateGroupOutput();

        if (data.hasOwnProperty('group_name')) {
          obj['group_name'] = _ApiClient["default"].convertToType(data['group_name'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UpdateGroupOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UpdateGroupOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['group_name'] && !(typeof data['group_name'] === 'string' || data['group_name'] instanceof String)) {
        throw new Error("Expected the field `group_name` to be a primitive type in the JSON string but got " + data['group_name']);
      } // ensure the json data is a string


      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }

      return true;
    }
  }]);

  return UpdateGroupOutput;
}();
/**
 * @member {String} group_name
 */


UpdateGroupOutput.prototype['group_name'] = undefined;
/**
 * @member {String} id
 */

UpdateGroupOutput.prototype['id'] = undefined;
/**
 * @member {String} name
 */

UpdateGroupOutput.prototype['name'] = undefined;
var _default = UpdateGroupOutput;
exports["default"] = _default;
