// akeyless/dist/model/UsageEventSetting.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UsageEventSetting model module.
 * @module model/UsageEventSetting
 * @version 4.2.2
 */
var UsageEventSetting = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UsageEventSetting</code>.
   * @alias module:model/UsageEventSetting
   */
  function UsageEventSetting() {
    _classCallCheck(this, UsageEventSetting);

    UsageEventSetting.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UsageEventSetting, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UsageEventSetting</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UsageEventSetting} obj Optional instance to populate.
     * @return {module:model/UsageEventSetting} The populated <code>UsageEventSetting</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UsageEventSetting();

        if (data.hasOwnProperty('enable')) {
          obj['enable'] = _ApiClient["default"].convertToType(data['enable'], 'Boolean');
        }

        if (data.hasOwnProperty('enable_time')) {
          obj['enable_time'] = _ApiClient["default"].convertToType(data['enable_time'], 'Date');
        }

        if (data.hasOwnProperty('interval_by_days')) {
          obj['interval_by_days'] = _ApiClient["default"].convertToType(data['interval_by_days'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return UsageEventSetting;
}();
/**
 * @member {Boolean} enable
 */


UsageEventSetting.prototype['enable'] = undefined;
/**
 * @member {Date} enable_time
 */

UsageEventSetting.prototype['enable_time'] = undefined;
/**
 * @member {Number} interval_by_days
 */

UsageEventSetting.prototype['interval_by_days'] = undefined;
var _default = UsageEventSetting;
exports["default"] = _default;
