// akeyless/dist/model/UsageReportSummary.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UsageReportSummary model module.
 * @module model/UsageReportSummary
 * @version 5.0.8
 */
var UsageReportSummary = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UsageReportSummary</code>.
   * @alias module:model/UsageReportSummary
   */
  function UsageReportSummary() {
    _classCallCheck(this, UsageReportSummary);

    UsageReportSummary.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UsageReportSummary, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UsageReportSummary</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UsageReportSummary} obj Optional instance to populate.
     * @return {module:model/UsageReportSummary} The populated <code>UsageReportSummary</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UsageReportSummary();

        if (data.hasOwnProperty('clients_by_auth_method_types')) {
          obj['clients_by_auth_method_types'] = _ApiClient["default"].convertToType(data['clients_by_auth_method_types'], {
            'String': 'Number'
          });
        }

        if (data.hasOwnProperty('product')) {
          obj['product'] = _ApiClient["default"].convertToType(data['product'], 'String');
        }

        if (data.hasOwnProperty('secrets_by_types')) {
          obj['secrets_by_types'] = _ApiClient["default"].convertToType(data['secrets_by_types'], {
            'String': 'Number'
          });
        }

        if (data.hasOwnProperty('time')) {
          obj['time'] = _ApiClient["default"].convertToType(data['time'], 'Number');
        }

        if (data.hasOwnProperty('total_clients')) {
          obj['total_clients'] = _ApiClient["default"].convertToType(data['total_clients'], 'Number');
        }

        if (data.hasOwnProperty('total_secrets')) {
          obj['total_secrets'] = _ApiClient["default"].convertToType(data['total_secrets'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UsageReportSummary</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UsageReportSummary</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['product'] && !(typeof data['product'] === 'string' || data['product'] instanceof String)) {
        throw new Error("Expected the field `product` to be a primitive type in the JSON string but got " + data['product']);
      }

      return true;
    }
  }]);

  return UsageReportSummary;
}();
/**
 * @member {Object.<String, Number>} clients_by_auth_method_types
 */


UsageReportSummary.prototype['clients_by_auth_method_types'] = undefined;
/**
 * @member {String} product
 */

UsageReportSummary.prototype['product'] = undefined;
/**
 * @member {Object.<String, Number>} secrets_by_types
 */

UsageReportSummary.prototype['secrets_by_types'] = undefined;
/**
 * @member {Number} time
 */

UsageReportSummary.prototype['time'] = undefined;
/**
 * @member {Number} total_clients
 */

UsageReportSummary.prototype['total_clients'] = undefined;
/**
 * @member {Number} total_secrets
 */

UsageReportSummary.prototype['total_secrets'] = undefined;
var _default = UsageReportSummary;
exports["default"] = _default;
