// akeyless/dist/model/UscCreate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UscCreate model module.
 * @module model/UscCreate
 * @version 3.6.3
 */
var UscCreate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UscCreate</code>.
   * uscCreate is a command that creates a new secret in a Universal Secrets Connector
   * @alias module:model/UscCreate
   * @param secretName {String} Name for the new universal secrets
   * @param uscName {String} Name of the Universal Secrets Connector item
   * @param value {String} Value of the universal secrets item, either text or base64 encoded binary
   */
  function UscCreate(secretName, uscName, value) {
    _classCallCheck(this, UscCreate);

    UscCreate.initialize(this, secretName, uscName, value);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UscCreate, null, [{
    key: "initialize",
    value: function initialize(obj, secretName, uscName, value) {
      obj['secret-name'] = secretName;
      obj['usc-name'] = uscName;
      obj['value'] = value;
    }
    /**
     * Constructs a <code>UscCreate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UscCreate} obj Optional instance to populate.
     * @return {module:model/UscCreate} The populated <code>UscCreate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UscCreate();

        if (data.hasOwnProperty('binary-value')) {
          obj['binary-value'] = _ApiClient["default"].convertToType(data['binary-value'], 'Boolean');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('secret-name')) {
          obj['secret-name'] = _ApiClient["default"].convertToType(data['secret-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('usc-name')) {
          obj['usc-name'] = _ApiClient["default"].convertToType(data['usc-name'], 'String');
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UscCreate;
}();
/**
 * Use this option if the universal secrets value is a base64 encoded binary
 * @member {Boolean} binary-value
 */


UscCreate.prototype['binary-value'] = undefined;
/**
 * Description of the universal secrets
 * @member {String} description
 */

UscCreate.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UscCreate.prototype['json'] = false;
/**
 * Name for the new universal secrets
 * @member {String} secret-name
 */

UscCreate.prototype['secret-name'] = undefined;
/**
 * Tags for the universal secrets
 * @member {Object.<String, String>} tags
 */

UscCreate.prototype['tags'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UscCreate.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UscCreate.prototype['uid-token'] = undefined;
/**
 * Name of the Universal Secrets Connector item
 * @member {String} usc-name
 */

UscCreate.prototype['usc-name'] = undefined;
/**
 * Value of the universal secrets item, either text or base64 encoded binary
 * @member {String} value
 */

UscCreate.prototype['value'] = undefined;
var _default = UscCreate;
exports["default"] = _default;
