// akeyless/dist/model/UscCreateSecretOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UscCreateSecretOutput model module.
 * @module model/UscCreateSecretOutput
 * @version 5.0.6
 */
var UscCreateSecretOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UscCreateSecretOutput</code>.
   * @alias module:model/UscCreateSecretOutput
   */
  function UscCreateSecretOutput() {
    _classCallCheck(this, UscCreateSecretOutput);

    UscCreateSecretOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UscCreateSecretOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UscCreateSecretOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UscCreateSecretOutput} obj Optional instance to populate.
     * @return {module:model/UscCreateSecretOutput} The populated <code>UscCreateSecretOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UscCreateSecretOutput();

        if (data.hasOwnProperty('secret_id')) {
          obj['secret_id'] = _ApiClient["default"].convertToType(data['secret_id'], 'String');
        }

        if (data.hasOwnProperty('version_id')) {
          obj['version_id'] = _ApiClient["default"].convertToType(data['version_id'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UscCreateSecretOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UscCreateSecretOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['secret_id'] && !(typeof data['secret_id'] === 'string' || data['secret_id'] instanceof String)) {
        throw new Error("Expected the field `secret_id` to be a primitive type in the JSON string but got " + data['secret_id']);
      } // ensure the json data is a string


      if (data['version_id'] && !(typeof data['version_id'] === 'string' || data['version_id'] instanceof String)) {
        throw new Error("Expected the field `version_id` to be a primitive type in the JSON string but got " + data['version_id']);
      }

      return true;
    }
  }]);

  return UscCreateSecretOutput;
}();
/**
 * @member {String} secret_id
 */


UscCreateSecretOutput.prototype['secret_id'] = undefined;
/**
 * @member {String} version_id
 */

UscCreateSecretOutput.prototype['version_id'] = undefined;
var _default = UscCreateSecretOutput;
exports["default"] = _default;
