// akeyless/dist/model/UscDelete.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UscDelete model module.
 * @module model/UscDelete
 * @version 4.1.0
 */
var UscDelete = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UscDelete</code>.
   * uscDelete is a command that deletes a secret from a Universal Secrets Connector
   * @alias module:model/UscDelete
   * @param secretId {String} The universal secrets id (or name, for AWS, Azure, K8s or Hashi vault targets) to delete
   * @param uscName {String} Name of the Universal Secrets Connector item
   */
  function UscDelete(secretId, uscName) {
    _classCallCheck(this, UscDelete);

    UscDelete.initialize(this, secretId, uscName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UscDelete, null, [{
    key: "initialize",
    value: function initialize(obj, secretId, uscName) {
      obj['secret-id'] = secretId;
      obj['usc-name'] = uscName;
    }
    /**
     * Constructs a <code>UscDelete</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UscDelete} obj Optional instance to populate.
     * @return {module:model/UscDelete} The populated <code>UscDelete</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UscDelete();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('namespace')) {
          obj['namespace'] = _ApiClient["default"].convertToType(data['namespace'], 'String');
        }

        if (data.hasOwnProperty('secret-id')) {
          obj['secret-id'] = _ApiClient["default"].convertToType(data['secret-id'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('usc-name')) {
          obj['usc-name'] = _ApiClient["default"].convertToType(data['usc-name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UscDelete;
}();
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */


UscDelete.prototype['json'] = false;
/**
 * The namespace (relevant for Hashi vault target)
 * @member {String} namespace
 */

UscDelete.prototype['namespace'] = undefined;
/**
 * The universal secrets id (or name, for AWS, Azure, K8s or Hashi vault targets) to delete
 * @member {String} secret-id
 */

UscDelete.prototype['secret-id'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UscDelete.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UscDelete.prototype['uid-token'] = undefined;
/**
 * Name of the Universal Secrets Connector item
 * @member {String} usc-name
 */

UscDelete.prototype['usc-name'] = undefined;
var _default = UscDelete;
exports["default"] = _default;
