// akeyless/dist/model/UscGetSecretOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UscGetSecretOutput model module.
 * @module model/UscGetSecretOutput
 * @version 5.0.3
 */
var UscGetSecretOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UscGetSecretOutput</code>.
   * @alias module:model/UscGetSecretOutput
   */
  function UscGetSecretOutput() {
    _classCallCheck(this, UscGetSecretOutput);

    UscGetSecretOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UscGetSecretOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UscGetSecretOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UscGetSecretOutput} obj Optional instance to populate.
     * @return {module:model/UscGetSecretOutput} The populated <code>UscGetSecretOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UscGetSecretOutput();

        if (data.hasOwnProperty('binary_value')) {
          obj['binary_value'] = _ApiClient["default"].convertToType(data['binary_value'], 'Boolean');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UscGetSecretOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UscGetSecretOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      } // ensure the json data is a string


      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      } // ensure the json data is a string


      if (data['value'] && !(typeof data['value'] === 'string' || data['value'] instanceof String)) {
        throw new Error("Expected the field `value` to be a primitive type in the JSON string but got " + data['value']);
      }

      return true;
    }
  }]);

  return UscGetSecretOutput;
}();
/**
 * @member {Boolean} binary_value
 */


UscGetSecretOutput.prototype['binary_value'] = undefined;
/**
 * @member {String} id
 */

UscGetSecretOutput.prototype['id'] = undefined;
/**
 * @member {Object} metadata
 */

UscGetSecretOutput.prototype['metadata'] = undefined;
/**
 * @member {String} name
 */

UscGetSecretOutput.prototype['name'] = undefined;
/**
 * @member {String} value
 */

UscGetSecretOutput.prototype['value'] = undefined;
var _default = UscGetSecretOutput;
exports["default"] = _default;
