// akeyless/dist/model/UscSyncInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UscSyncInfo model module.
 * @module model/UscSyncInfo
 * @version 5.0.8
 */
var UscSyncInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UscSyncInfo</code>.
   * @alias module:model/UscSyncInfo
   */
  function UscSyncInfo() {
    _classCallCheck(this, UscSyncInfo);

    UscSyncInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UscSyncInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UscSyncInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UscSyncInfo} obj Optional instance to populate.
     * @return {module:model/UscSyncInfo} The populated <code>UscSyncInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UscSyncInfo();

        if (data.hasOwnProperty('jq_secret_filter')) {
          obj['jq_secret_filter'] = _ApiClient["default"].convertToType(data['jq_secret_filter'], 'String');
        }

        if (data.hasOwnProperty('last_error')) {
          obj['last_error'] = _ApiClient["default"].convertToType(data['last_error'], 'String');
        }

        if (data.hasOwnProperty('namespace')) {
          obj['namespace'] = _ApiClient["default"].convertToType(data['namespace'], 'String');
        }

        if (data.hasOwnProperty('secret_id')) {
          obj['secret_id'] = _ApiClient["default"].convertToType(data['secret_id'], 'String');
        }

        if (data.hasOwnProperty('secret_name')) {
          obj['secret_name'] = _ApiClient["default"].convertToType(data['secret_name'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UscSyncInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UscSyncInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['jq_secret_filter'] && !(typeof data['jq_secret_filter'] === 'string' || data['jq_secret_filter'] instanceof String)) {
        throw new Error("Expected the field `jq_secret_filter` to be a primitive type in the JSON string but got " + data['jq_secret_filter']);
      } // ensure the json data is a string


      if (data['last_error'] && !(typeof data['last_error'] === 'string' || data['last_error'] instanceof String)) {
        throw new Error("Expected the field `last_error` to be a primitive type in the JSON string but got " + data['last_error']);
      } // ensure the json data is a string


      if (data['namespace'] && !(typeof data['namespace'] === 'string' || data['namespace'] instanceof String)) {
        throw new Error("Expected the field `namespace` to be a primitive type in the JSON string but got " + data['namespace']);
      } // ensure the json data is a string


      if (data['secret_id'] && !(typeof data['secret_id'] === 'string' || data['secret_id'] instanceof String)) {
        throw new Error("Expected the field `secret_id` to be a primitive type in the JSON string but got " + data['secret_id']);
      } // ensure the json data is a string


      if (data['secret_name'] && !(typeof data['secret_name'] === 'string' || data['secret_name'] instanceof String)) {
        throw new Error("Expected the field `secret_name` to be a primitive type in the JSON string but got " + data['secret_name']);
      }

      return true;
    }
  }]);

  return UscSyncInfo;
}();
/**
 * @member {String} jq_secret_filter
 */


UscSyncInfo.prototype['jq_secret_filter'] = undefined;
/**
 * @member {String} last_error
 */

UscSyncInfo.prototype['last_error'] = undefined;
/**
 * @member {String} namespace
 */

UscSyncInfo.prototype['namespace'] = undefined;
/**
 * @member {String} secret_id
 */

UscSyncInfo.prototype['secret_id'] = undefined;
/**
 * @member {String} secret_name
 */

UscSyncInfo.prototype['secret_name'] = undefined;
var _default = UscSyncInfo;
exports["default"] = _default;
