// akeyless/dist/model/UscUpdate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UscUpdate model module.
 * @module model/UscUpdate
 * @version 5.0.3
 */
var UscUpdate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UscUpdate</code>.
   * uscUpdate is a command that updates a secret in a Universal Secrets Connector
   * @alias module:model/UscUpdate
   * @param secretId {String} The universal secrets id (or name, for AWS, Azure, K8s or Hashi vault targets) to update
   * @param uscName {String} Name of the Universal Secrets Connector item
   * @param value {String} Value of the universal secrets item, either text or base64 encoded binary
   */
  function UscUpdate(secretId, uscName, value) {
    _classCallCheck(this, UscUpdate);

    UscUpdate.initialize(this, secretId, uscName, value);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UscUpdate, null, [{
    key: "initialize",
    value: function initialize(obj, secretId, uscName, value) {
      obj['json'] = false;
      obj['secret-id'] = secretId;
      obj['usc-name'] = uscName;
      obj['value'] = value;
    }
    /**
     * Constructs a <code>UscUpdate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UscUpdate} obj Optional instance to populate.
     * @return {module:model/UscUpdate} The populated <code>UscUpdate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UscUpdate();

        if (data.hasOwnProperty('binary-value')) {
          obj['binary-value'] = _ApiClient["default"].convertToType(data['binary-value'], 'Boolean');
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('namespace')) {
          obj['namespace'] = _ApiClient["default"].convertToType(data['namespace'], 'String');
        }

        if (data.hasOwnProperty('secret-id')) {
          obj['secret-id'] = _ApiClient["default"].convertToType(data['secret-id'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('usc-name')) {
          obj['usc-name'] = _ApiClient["default"].convertToType(data['usc-name'], 'String');
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>UscUpdate</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UscUpdate</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(UscUpdate.RequiredProperties),
          _step;

      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;

          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        } // ensure the json data is a string

      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }

      if (data['description'] && !(typeof data['description'] === 'string' || data['description'] instanceof String)) {
        throw new Error("Expected the field `description` to be a primitive type in the JSON string but got " + data['description']);
      } // ensure the json data is a string


      if (data['namespace'] && !(typeof data['namespace'] === 'string' || data['namespace'] instanceof String)) {
        throw new Error("Expected the field `namespace` to be a primitive type in the JSON string but got " + data['namespace']);
      } // ensure the json data is a string


      if (data['secret-id'] && !(typeof data['secret-id'] === 'string' || data['secret-id'] instanceof String)) {
        throw new Error("Expected the field `secret-id` to be a primitive type in the JSON string but got " + data['secret-id']);
      } // ensure the json data is a string


      if (data['token'] && !(typeof data['token'] === 'string' || data['token'] instanceof String)) {
        throw new Error("Expected the field `token` to be a primitive type in the JSON string but got " + data['token']);
      } // ensure the json data is a string


      if (data['uid-token'] && !(typeof data['uid-token'] === 'string' || data['uid-token'] instanceof String)) {
        throw new Error("Expected the field `uid-token` to be a primitive type in the JSON string but got " + data['uid-token']);
      } // ensure the json data is a string


      if (data['usc-name'] && !(typeof data['usc-name'] === 'string' || data['usc-name'] instanceof String)) {
        throw new Error("Expected the field `usc-name` to be a primitive type in the JSON string but got " + data['usc-name']);
      } // ensure the json data is a string


      if (data['value'] && !(typeof data['value'] === 'string' || data['value'] instanceof String)) {
        throw new Error("Expected the field `value` to be a primitive type in the JSON string but got " + data['value']);
      }

      return true;
    }
  }]);

  return UscUpdate;
}();

UscUpdate.RequiredProperties = ["secret-id", "usc-name", "value"];
/**
 * Use this option if the universal secrets value is a base64 encoded binary
 * @member {Boolean} binary-value
 */

UscUpdate.prototype['binary-value'] = undefined;
/**
 * Description of the universal secrets
 * @member {String} description
 */

UscUpdate.prototype['description'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 * @default false
 */

UscUpdate.prototype['json'] = false;
/**
 * The namespace (relevant for Hashi vault target)
 * @member {String} namespace
 */

UscUpdate.prototype['namespace'] = undefined;
/**
 * The universal secrets id (or name, for AWS, Azure, K8s or Hashi vault targets) to update
 * @member {String} secret-id
 */

UscUpdate.prototype['secret-id'] = undefined;
/**
 * Tags for the universal secrets
 * @member {Object.<String, String>} tags
 */

UscUpdate.prototype['tags'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UscUpdate.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UscUpdate.prototype['uid-token'] = undefined;
/**
 * Name of the Universal Secrets Connector item
 * @member {String} usc-name
 */

UscUpdate.prototype['usc-name'] = undefined;
/**
 * Value of the universal secrets item, either text or base64 encoded binary
 * @member {String} value
 */

UscUpdate.prototype['value'] = undefined;
var _default = UscUpdate;
exports["default"] = _default;
