// akeyless/dist/model/ValidateTokenOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ValidateTokenOutput model module.
 * @module model/ValidateTokenOutput
 * @version 5.0.5
 */
var ValidateTokenOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ValidateTokenOutput</code>.
   * @alias module:model/ValidateTokenOutput
   */
  function ValidateTokenOutput() {
    _classCallCheck(this, ValidateTokenOutput);

    ValidateTokenOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ValidateTokenOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ValidateTokenOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ValidateTokenOutput} obj Optional instance to populate.
     * @return {module:model/ValidateTokenOutput} The populated <code>ValidateTokenOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ValidateTokenOutput();

        if (data.hasOwnProperty('expiration')) {
          obj['expiration'] = _ApiClient["default"].convertToType(data['expiration'], 'String');
        }

        if (data.hasOwnProperty('is_valid')) {
          obj['is_valid'] = _ApiClient["default"].convertToType(data['is_valid'], 'Boolean');
        }

        if (data.hasOwnProperty('last_rotate')) {
          obj['last_rotate'] = _ApiClient["default"].convertToType(data['last_rotate'], 'String');
        }

        if (data.hasOwnProperty('reason')) {
          obj['reason'] = _ApiClient["default"].convertToType(data['reason'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>ValidateTokenOutput</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ValidateTokenOutput</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['expiration'] && !(typeof data['expiration'] === 'string' || data['expiration'] instanceof String)) {
        throw new Error("Expected the field `expiration` to be a primitive type in the JSON string but got " + data['expiration']);
      } // ensure the json data is a string


      if (data['last_rotate'] && !(typeof data['last_rotate'] === 'string' || data['last_rotate'] instanceof String)) {
        throw new Error("Expected the field `last_rotate` to be a primitive type in the JSON string but got " + data['last_rotate']);
      } // ensure the json data is a string


      if (data['reason'] && !(typeof data['reason'] === 'string' || data['reason'] instanceof String)) {
        throw new Error("Expected the field `reason` to be a primitive type in the JSON string but got " + data['reason']);
      }

      return true;
    }
  }]);

  return ValidateTokenOutput;
}();
/**
 * @member {String} expiration
 */


ValidateTokenOutput.prototype['expiration'] = undefined;
/**
 * @member {Boolean} is_valid
 */

ValidateTokenOutput.prototype['is_valid'] = undefined;
/**
 * @member {String} last_rotate
 */

ValidateTokenOutput.prototype['last_rotate'] = undefined;
/**
 * @member {String} reason
 */

ValidateTokenOutput.prototype['reason'] = undefined;
/**
 * @member {Number} ttl
 */

ValidateTokenOutput.prototype['ttl'] = undefined;
var _default = ValidateTokenOutput;
exports["default"] = _default;
