// akeyless/dist/model/VaultlessTokenizerInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _EmailTokenizerInfo = _interopRequireDefault(require("./EmailTokenizerInfo"));

var _RegexpTokenizerInfo = _interopRequireDefault(require("./RegexpTokenizerInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The VaultlessTokenizerInfo model module.
 * @module model/VaultlessTokenizerInfo
 * @version 5.0.8
 */
var VaultlessTokenizerInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VaultlessTokenizerInfo</code>.
   * @alias module:model/VaultlessTokenizerInfo
   */
  function VaultlessTokenizerInfo() {
    _classCallCheck(this, VaultlessTokenizerInfo);

    VaultlessTokenizerInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(VaultlessTokenizerInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>VaultlessTokenizerInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VaultlessTokenizerInfo} obj Optional instance to populate.
     * @return {module:model/VaultlessTokenizerInfo} The populated <code>VaultlessTokenizerInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VaultlessTokenizerInfo();

        if (data.hasOwnProperty('email_tokenizer_info')) {
          obj['email_tokenizer_info'] = _EmailTokenizerInfo["default"].constructFromObject(data['email_tokenizer_info']);
        }

        if (data.hasOwnProperty('key_name')) {
          obj['key_name'] = _ApiClient["default"].convertToType(data['key_name'], 'String');
        }

        if (data.hasOwnProperty('regexp_tokenizer_info')) {
          obj['regexp_tokenizer_info'] = _RegexpTokenizerInfo["default"].constructFromObject(data['regexp_tokenizer_info']);
        }

        if (data.hasOwnProperty('template_type')) {
          obj['template_type'] = _ApiClient["default"].convertToType(data['template_type'], 'String');
        }

        if (data.hasOwnProperty('tweak')) {
          obj['tweak'] = _ApiClient["default"].convertToType(data['tweak'], 'String');
        }

        if (data.hasOwnProperty('tweak_type')) {
          obj['tweak_type'] = _ApiClient["default"].convertToType(data['tweak_type'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>VaultlessTokenizerInfo</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>VaultlessTokenizerInfo</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `email_tokenizer_info`
      if (data['email_tokenizer_info']) {
        // data not null
        _EmailTokenizerInfo["default"].validateJSON(data['email_tokenizer_info']);
      } // ensure the json data is a string


      if (data['key_name'] && !(typeof data['key_name'] === 'string' || data['key_name'] instanceof String)) {
        throw new Error("Expected the field `key_name` to be a primitive type in the JSON string but got " + data['key_name']);
      } // validate the optional field `regexp_tokenizer_info`


      if (data['regexp_tokenizer_info']) {
        // data not null
        _RegexpTokenizerInfo["default"].validateJSON(data['regexp_tokenizer_info']);
      } // ensure the json data is a string


      if (data['template_type'] && !(typeof data['template_type'] === 'string' || data['template_type'] instanceof String)) {
        throw new Error("Expected the field `template_type` to be a primitive type in the JSON string but got " + data['template_type']);
      } // ensure the json data is a string


      if (data['tweak'] && !(typeof data['tweak'] === 'string' || data['tweak'] instanceof String)) {
        throw new Error("Expected the field `tweak` to be a primitive type in the JSON string but got " + data['tweak']);
      } // ensure the json data is a string


      if (data['tweak_type'] && !(typeof data['tweak_type'] === 'string' || data['tweak_type'] instanceof String)) {
        throw new Error("Expected the field `tweak_type` to be a primitive type in the JSON string but got " + data['tweak_type']);
      }

      return true;
    }
  }]);

  return VaultlessTokenizerInfo;
}();
/**
 * @member {module:model/EmailTokenizerInfo} email_tokenizer_info
 */


VaultlessTokenizerInfo.prototype['email_tokenizer_info'] = undefined;
/**
 * @member {String} key_name
 */

VaultlessTokenizerInfo.prototype['key_name'] = undefined;
/**
 * @member {module:model/RegexpTokenizerInfo} regexp_tokenizer_info
 */

VaultlessTokenizerInfo.prototype['regexp_tokenizer_info'] = undefined;
/**
 * @member {String} template_type
 */

VaultlessTokenizerInfo.prototype['template_type'] = undefined;
/**
 * Tweak used in the case of internal tweak type
 * @member {String} tweak
 */

VaultlessTokenizerInfo.prototype['tweak'] = undefined;
/**
 * @member {String} tweak_type
 */

VaultlessTokenizerInfo.prototype['tweak_type'] = undefined;
var _default = VaultlessTokenizerInfo;
exports["default"] = _default;
