// akeyless/dist/model/WalletDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WalletDetails model module.
 * @module model/WalletDetails
 * @version 5.0.1
 */
var WalletDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WalletDetails</code>.
   * @alias module:model/WalletDetails
   */
  function WalletDetails() {
    _classCallCheck(this, WalletDetails);

    WalletDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WalletDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WalletDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WalletDetails} obj Optional instance to populate.
     * @return {module:model/WalletDetails} The populated <code>WalletDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WalletDetails();

        if (data.hasOwnProperty('login_type')) {
          obj['login_type'] = _ApiClient["default"].convertToType(data['login_type'], 'String');
        }

        if (data.hasOwnProperty('p12_data_base64')) {
          obj['p12_data_base64'] = _ApiClient["default"].convertToType(data['p12_data_base64'], 'String');
        }

        if (data.hasOwnProperty('sso_data_base64')) {
          obj['sso_data_base64'] = _ApiClient["default"].convertToType(data['sso_data_base64'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>WalletDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>WalletDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['login_type'] && !(typeof data['login_type'] === 'string' || data['login_type'] instanceof String)) {
        throw new Error("Expected the field `login_type` to be a primitive type in the JSON string but got " + data['login_type']);
      } // ensure the json data is a string


      if (data['p12_data_base64'] && !(typeof data['p12_data_base64'] === 'string' || data['p12_data_base64'] instanceof String)) {
        throw new Error("Expected the field `p12_data_base64` to be a primitive type in the JSON string but got " + data['p12_data_base64']);
      } // ensure the json data is a string


      if (data['sso_data_base64'] && !(typeof data['sso_data_base64'] === 'string' || data['sso_data_base64'] instanceof String)) {
        throw new Error("Expected the field `sso_data_base64` to be a primitive type in the JSON string but got " + data['sso_data_base64']);
      }

      return true;
    }
  }]);

  return WalletDetails;
}();
/**
 * @member {String} login_type
 */


WalletDetails.prototype['login_type'] = undefined;
/**
 * @member {String} p12_data_base64
 */

WalletDetails.prototype['p12_data_base64'] = undefined;
/**
 * @member {String} sso_data_base64
 */

WalletDetails.prototype['sso_data_base64'] = undefined;
var _default = WalletDetails;
exports["default"] = _default;
