// akeyless/dist/model/WebBastionRdpRecord.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AwsStorage = _interopRequireDefault(require("./AwsStorage"));

var _AzureStorage = _interopRequireDefault(require("./AzureStorage"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WebBastionRdpRecord model module.
 * @module model/WebBastionRdpRecord
 * @version 5.0.14
 */
var WebBastionRdpRecord = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebBastionRdpRecord</code>.
   * @alias module:model/WebBastionRdpRecord
   */
  function WebBastionRdpRecord() {
    _classCallCheck(this, WebBastionRdpRecord);

    WebBastionRdpRecord.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WebBastionRdpRecord, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WebBastionRdpRecord</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebBastionRdpRecord} obj Optional instance to populate.
     * @return {module:model/WebBastionRdpRecord} The populated <code>WebBastionRdpRecord</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebBastionRdpRecord();

        if (data.hasOwnProperty('aws')) {
          obj['aws'] = _AwsStorage["default"].constructFromObject(data['aws']);
        }

        if (data.hasOwnProperty('azure')) {
          obj['azure'] = _AzureStorage["default"].constructFromObject(data['azure']);
        }

        if (data.hasOwnProperty('compress')) {
          obj['compress'] = _ApiClient["default"].convertToType(data['compress'], 'Boolean');
        }

        if (data.hasOwnProperty('encryption_key')) {
          obj['encryption_key'] = _ApiClient["default"].convertToType(data['encryption_key'], 'String');
        }

        if (data.hasOwnProperty('recording_quality')) {
          obj['recording_quality'] = _ApiClient["default"].convertToType(data['recording_quality'], 'String');
        }

        if (data.hasOwnProperty('storage_type')) {
          obj['storage_type'] = _ApiClient["default"].convertToType(data['storage_type'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>WebBastionRdpRecord</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>WebBastionRdpRecord</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `aws`
      if (data['aws']) {
        // data not null
        _AwsStorage["default"].validateJSON(data['aws']);
      } // validate the optional field `azure`


      if (data['azure']) {
        // data not null
        _AzureStorage["default"].validateJSON(data['azure']);
      } // ensure the json data is a string


      if (data['encryption_key'] && !(typeof data['encryption_key'] === 'string' || data['encryption_key'] instanceof String)) {
        throw new Error("Expected the field `encryption_key` to be a primitive type in the JSON string but got " + data['encryption_key']);
      } // ensure the json data is a string


      if (data['recording_quality'] && !(typeof data['recording_quality'] === 'string' || data['recording_quality'] instanceof String)) {
        throw new Error("Expected the field `recording_quality` to be a primitive type in the JSON string but got " + data['recording_quality']);
      } // ensure the json data is a string


      if (data['storage_type'] && !(typeof data['storage_type'] === 'string' || data['storage_type'] instanceof String)) {
        throw new Error("Expected the field `storage_type` to be a primitive type in the JSON string but got " + data['storage_type']);
      }

      return true;
    }
  }]);

  return WebBastionRdpRecord;
}();
/**
 * @member {module:model/AwsStorage} aws
 */


WebBastionRdpRecord.prototype['aws'] = undefined;
/**
 * @member {module:model/AzureStorage} azure
 */

WebBastionRdpRecord.prototype['azure'] = undefined;
/**
 * @member {Boolean} compress
 */

WebBastionRdpRecord.prototype['compress'] = undefined;
/**
 * @member {String} encryption_key
 */

WebBastionRdpRecord.prototype['encryption_key'] = undefined;
/**
 * @member {String} recording_quality
 */

WebBastionRdpRecord.prototype['recording_quality'] = undefined;
/**
 * @member {String} storage_type
 */

WebBastionRdpRecord.prototype['storage_type'] = undefined;
var _default = WebBastionRdpRecord;
exports["default"] = _default;
