// akeyless/dist/model/WebBastionConf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _WebBastionGuacamole = _interopRequireDefault(require("./WebBastionGuacamole"));

var _WebBastionRdpRecord = _interopRequireDefault(require("./WebBastionRdpRecord"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WebBastionConf model module.
 * @module model/WebBastionConf
 * @version 5.0.5
 */
var WebBastionConf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebBastionConf</code>.
   * @alias module:model/WebBastionConf
   */
  function WebBastionConf() {
    _classCallCheck(this, WebBastionConf);

    WebBastionConf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WebBastionConf, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WebBastionConf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebBastionConf} obj Optional instance to populate.
     * @return {module:model/WebBastionConf} The populated <code>WebBastionConf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebBastionConf();

        if (data.hasOwnProperty('guacamole')) {
          obj['guacamole'] = _WebBastionGuacamole["default"].constructFromObject(data['guacamole']);
        }

        if (data.hasOwnProperty('rdp_record')) {
          obj['rdp_record'] = _WebBastionRdpRecord["default"].constructFromObject(data['rdp_record']);
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>WebBastionConf</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>WebBastionConf</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `guacamole`
      if (data['guacamole']) {
        // data not null
        _WebBastionGuacamole["default"].validateJSON(data['guacamole']);
      } // validate the optional field `rdp_record`


      if (data['rdp_record']) {
        // data not null
        _WebBastionRdpRecord["default"].validateJSON(data['rdp_record']);
      }

      return true;
    }
  }]);

  return WebBastionConf;
}();
/**
 * @member {module:model/WebBastionGuacamole} guacamole
 */


WebBastionConf.prototype['guacamole'] = undefined;
/**
 * @member {module:model/WebBastionRdpRecord} rdp_record
 */

WebBastionConf.prototype['rdp_record'] = undefined;
var _default = WebBastionConf;
exports["default"] = _default;
