// akeyless/dist/model/WebHookNotiForwarderPublicDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WebHookNotiForwarderPublicDetails model module.
 * @module model/WebHookNotiForwarderPublicDetails
 * @version 4.1.0
 */
var WebHookNotiForwarderPublicDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WebHookNotiForwarderPublicDetails</code>.
   * @alias module:model/WebHookNotiForwarderPublicDetails
   */
  function WebHookNotiForwarderPublicDetails() {
    _classCallCheck(this, WebHookNotiForwarderPublicDetails);

    WebHookNotiForwarderPublicDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WebHookNotiForwarderPublicDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WebHookNotiForwarderPublicDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WebHookNotiForwarderPublicDetails} obj Optional instance to populate.
     * @return {module:model/WebHookNotiForwarderPublicDetails} The populated <code>WebHookNotiForwarderPublicDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebHookNotiForwarderPublicDetails();

        if (data.hasOwnProperty('auth_type')) {
          obj['auth_type'] = _ApiClient["default"].convertToType(data['auth_type'], 'String');
        }

        if (data.hasOwnProperty('endpoint_url')) {
          obj['endpoint_url'] = _ApiClient["default"].convertToType(data['endpoint_url'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return WebHookNotiForwarderPublicDetails;
}();
/**
 * @member {String} auth_type
 */


WebHookNotiForwarderPublicDetails.prototype['auth_type'] = undefined;
/**
 * @member {String} endpoint_url
 */

WebHookNotiForwarderPublicDetails.prototype['endpoint_url'] = undefined;
/**
 * Auth type - User Password
 * @member {String} username
 */

WebHookNotiForwarderPublicDetails.prototype['username'] = undefined;
var _default = WebHookNotiForwarderPublicDetails;
exports["default"] = _default;
