// akeyless/dist/model/WindowsTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WindowsTargetDetails model module.
 * @module model/WindowsTargetDetails
 * @version 5.0.1
 */
var WindowsTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WindowsTargetDetails</code>.
   * WindowsTargetDetails
   * @alias module:model/WindowsTargetDetails
   */
  function WindowsTargetDetails() {
    _classCallCheck(this, WindowsTargetDetails);

    WindowsTargetDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WindowsTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WindowsTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WindowsTargetDetails} obj Optional instance to populate.
     * @return {module:model/WindowsTargetDetails} The populated <code>WindowsTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WindowsTargetDetails();

        if (data.hasOwnProperty('certificate')) {
          obj['certificate'] = _ApiClient["default"].convertToType(data['certificate'], 'String');
        }

        if (data.hasOwnProperty('connection_type')) {
          obj['connection_type'] = _ApiClient["default"].convertToType(data['connection_type'], 'String');
        }

        if (data.hasOwnProperty('domain_name')) {
          obj['domain_name'] = _ApiClient["default"].convertToType(data['domain_name'], 'String');
        }

        if (data.hasOwnProperty('hostname')) {
          obj['hostname'] = _ApiClient["default"].convertToType(data['hostname'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('use_tls')) {
          obj['use_tls'] = _ApiClient["default"].convertToType(data['use_tls'], 'Boolean');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>WindowsTargetDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>WindowsTargetDetails</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['certificate'] && !(typeof data['certificate'] === 'string' || data['certificate'] instanceof String)) {
        throw new Error("Expected the field `certificate` to be a primitive type in the JSON string but got " + data['certificate']);
      } // ensure the json data is a string


      if (data['connection_type'] && !(typeof data['connection_type'] === 'string' || data['connection_type'] instanceof String)) {
        throw new Error("Expected the field `connection_type` to be a primitive type in the JSON string but got " + data['connection_type']);
      } // ensure the json data is a string


      if (data['domain_name'] && !(typeof data['domain_name'] === 'string' || data['domain_name'] instanceof String)) {
        throw new Error("Expected the field `domain_name` to be a primitive type in the JSON string but got " + data['domain_name']);
      } // ensure the json data is a string


      if (data['hostname'] && !(typeof data['hostname'] === 'string' || data['hostname'] instanceof String)) {
        throw new Error("Expected the field `hostname` to be a primitive type in the JSON string but got " + data['hostname']);
      } // ensure the json data is a string


      if (data['password'] && !(typeof data['password'] === 'string' || data['password'] instanceof String)) {
        throw new Error("Expected the field `password` to be a primitive type in the JSON string but got " + data['password']);
      } // ensure the json data is a string


      if (data['port'] && !(typeof data['port'] === 'string' || data['port'] instanceof String)) {
        throw new Error("Expected the field `port` to be a primitive type in the JSON string but got " + data['port']);
      } // ensure the json data is a string


      if (data['username'] && !(typeof data['username'] === 'string' || data['username'] instanceof String)) {
        throw new Error("Expected the field `username` to be a primitive type in the JSON string but got " + data['username']);
      }

      return true;
    }
  }]);

  return WindowsTargetDetails;
}();
/**
 * @member {String} certificate
 */


WindowsTargetDetails.prototype['certificate'] = undefined;
/**
 * @member {String} connection_type
 */

WindowsTargetDetails.prototype['connection_type'] = undefined;
/**
 * @member {String} domain_name
 */

WindowsTargetDetails.prototype['domain_name'] = undefined;
/**
 * @member {String} hostname
 */

WindowsTargetDetails.prototype['hostname'] = undefined;
/**
 * @member {String} password
 */

WindowsTargetDetails.prototype['password'] = undefined;
/**
 * @member {String} port
 */

WindowsTargetDetails.prototype['port'] = undefined;
/**
 * @member {Boolean} use_tls
 */

WindowsTargetDetails.prototype['use_tls'] = undefined;
/**
 * @member {String} username
 */

WindowsTargetDetails.prototype['username'] = undefined;
var _default = WindowsTargetDetails;
exports["default"] = _default;
