// akeyless/dist/model/WindowsService.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _WindowsServiceAttributes = _interopRequireDefault(require("./WindowsServiceAttributes"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WindowsService model module.
 * @module model/WindowsService
 * @version 4.2.2
 */
var WindowsService = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WindowsService</code>.
   * @alias module:model/WindowsService
   */
  function WindowsService() {
    _classCallCheck(this, WindowsService);

    WindowsService.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WindowsService, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WindowsService</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WindowsService} obj Optional instance to populate.
     * @return {module:model/WindowsService} The populated <code>WindowsService</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WindowsService();

        if (data.hasOwnProperty('attributes')) {
          obj['attributes'] = _WindowsServiceAttributes["default"].constructFromObject(data['attributes']);
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return WindowsService;
}();
/**
 * @member {module:model/WindowsServiceAttributes} attributes
 */


WindowsService.prototype['attributes'] = undefined;
/**
 * @member {String} host
 */

WindowsService.prototype['host'] = undefined;
/**
 * @member {String} name
 */

WindowsService.prototype['name'] = undefined;
var _default = WindowsService;
exports["default"] = _default;
