// akeyless/dist/model/WindowsServiceAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The WindowsServiceAttributes model module.
 * @module model/WindowsServiceAttributes
 * @version 5.0.1
 */
var WindowsServiceAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WindowsServiceAttributes</code>.
   * @alias module:model/WindowsServiceAttributes
   */
  function WindowsServiceAttributes() {
    _classCallCheck(this, WindowsServiceAttributes);

    WindowsServiceAttributes.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(WindowsServiceAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>WindowsServiceAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WindowsServiceAttributes} obj Optional instance to populate.
     * @return {module:model/WindowsServiceAttributes} The populated <code>WindowsServiceAttributes</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WindowsServiceAttributes();

        if (data.hasOwnProperty('connection_type')) {
          obj['connection_type'] = _ApiClient["default"].convertToType(data['connection_type'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('use_tls')) {
          obj['use_tls'] = _ApiClient["default"].convertToType(data['use_tls'], 'Boolean');
        }
      }

      return obj;
    }
    /**
     * Validates the JSON data with respect to <code>WindowsServiceAttributes</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>WindowsServiceAttributes</code>.
     */

  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['connection_type'] && !(typeof data['connection_type'] === 'string' || data['connection_type'] instanceof String)) {
        throw new Error("Expected the field `connection_type` to be a primitive type in the JSON string but got " + data['connection_type']);
      } // ensure the json data is a string


      if (data['port'] && !(typeof data['port'] === 'string' || data['port'] instanceof String)) {
        throw new Error("Expected the field `port` to be a primitive type in the JSON string but got " + data['port']);
      }

      return true;
    }
  }]);

  return WindowsServiceAttributes;
}();
/**
 * @member {String} connection_type
 */


WindowsServiceAttributes.prototype['connection_type'] = undefined;
/**
 * @member {String} port
 */

WindowsServiceAttributes.prototype['port'] = undefined;
/**
 * @member {Boolean} use_tls
 */

WindowsServiceAttributes.prototype['use_tls'] = undefined;
var _default = WindowsServiceAttributes;
exports["default"] = _default;
